{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def load_gene_names():\n",
    "    ret = set()\n",
    "    syns = {}\n",
    "\n",
    "    with open('genenames.txt') as f:\n",
    "        c = 0\n",
    "        for l in f:\n",
    "            c += 1\n",
    "            if c == 1:\n",
    "                header = l.split('\\t')\n",
    "                syn_index = header.index('Synonyms')\n",
    "                symb_index = header.index('Previous Symbols')\n",
    "                continue\n",
    "                \n",
    "            ls = l.split('\\t')\n",
    "            gene = ls[1]\n",
    "            ret.add(gene)\n",
    "            this_syn = [x.strip() for x in ls[syn_index].split(',')]\n",
    "            for x in this_syn:\n",
    "                syns[x] = gene\n",
    "            \n",
    "            symbols = [x.strip() for x in ls[symb_index].split(',')]\n",
    "            for x in symbols:\n",
    "                syns[x] = gene\n",
    "            \n",
    "            \n",
    "    return {x: None for x in ret}, syns\n",
    "\n",
    "stop = {\n",
    "    '2': None,\n",
    "    'T': None,\n",
    "    'B': None,\n",
    "}\n",
    "\n",
    "def check_gene_name(x):\n",
    "    \n",
    "    if x in stop:\n",
    "        return None\n",
    "    \n",
    "    if x in genenames:\n",
    "        if not 'realnames' in stats:\n",
    "            stats['realnames'] = 0\n",
    "        stats['realnames'] += 1\n",
    "        \n",
    "        return x\n",
    "    \n",
    "    if x in genesyns:\n",
    "        \n",
    "        if not 'syns' in stats:\n",
    "            stats['syns'] = 0\n",
    "        stats['syns'] += 1\n",
    "        \n",
    "        return genesyns[x]\n",
    "    \n",
    "    extra_synonyms = {\n",
    "        # Found : Real\n",
    "        'FXII': 'F12',\n",
    "    }\n",
    "    \n",
    "    if x in extra_synonyms:\n",
    "        return extra_synonyms[x]\n",
    "    \n",
    "    if x.lower() in ['mitochondrial', 'mtdna']:\n",
    "        \n",
    "        if not 'mitochondrial' in stats:\n",
    "            stats['mitochondrial'] = 0\n",
    "        stats['mitochondrial'] += 1\n",
    "        \n",
    "        return 'mtdna'\n",
    "    \n",
    "    return None\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "genenames, genesyns = load_gene_names()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "genenames['BRCA2']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'HPRT'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-10-511b85dce355>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mgenesyns\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'HPRT'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m: 'HPRT'"
     ]
    }
   ],
   "source": [
    "genesyns['HPRT']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Record: 1\n",
      "Record: 2\n",
      "Record: 3\n",
      "Record: 4\n",
      "Record: 5\n",
      "Record: 6\n",
      "Record: 7\n",
      "Record: 8\n",
      "Record: 9\n",
      "Record: 10\n",
      "Record: 11\n",
      "Record: 12\n",
      "Record: 13\n",
      "Record: 14\n",
      "Record: 15\n",
      "Record: 16\n",
      "Record: 17\n",
      "Record: 18\n",
      "Record: 19\n",
      "Record: 20\n",
      "Record: 21\n",
      "Record: 22\n",
      "Record: 23\n",
      "Record: 24\n",
      "Record: 25\n",
      "Record: 26\n",
      "Record: 27\n",
      "Record: 28\n",
      "Record: 29\n",
      "Record: 30\n",
      "Record: 31\n",
      "Record: 32\n",
      "Record: 33\n",
      "Record: 34\n",
      "Record: 35\n",
      "Record: 36\n",
      "Record: 37\n",
      "Record: 38\n",
      "Record: 39\n",
      "Record: 40\n",
      "Record: 41\n",
      "Record: 42\n",
      "Record: 43\n",
      "Record: 44\n",
      "Record: 45\n",
      "Record: 46\n",
      "Record: 47\n",
      "Record: 48\n",
      "Record: 49\n",
      "Record: 50\n",
      "Record: 51\n",
      "Record: 52\n",
      "Record: 53\n",
      "Record: 54\n",
      "Record: 55\n",
      "Record: 56\n",
      "Record: 57\n",
      "Record: 58\n",
      "Record: 59\n",
      "Record: 60\n",
      "Record: 61\n",
      "Record: 62\n",
      "Record: 63\n",
      "Record: 64\n",
      "Record: 65\n",
      "Record: 66\n",
      "Record: 67\n",
      "('Abstract counter:', 100)\n",
      "Record: 68\n",
      "Record: 69\n",
      "Record: 70\n",
      "Record: 71\n",
      "Record: 72\n",
      "Record: 73\n",
      "Record: 74\n",
      "Record: 75\n",
      "Record: 76\n",
      "Record: 77\n",
      "Record: 78\n",
      "Record: 79\n",
      "Record: 80\n",
      "Record: 81\n",
      "Record: 82\n",
      "Record: 83\n",
      "Record: 84\n",
      "Record: 85\n",
      "Record: 86\n",
      "Record: 87\n",
      "Record: 88\n",
      "Record: 89\n",
      "Record: 90\n",
      "Record: 91\n",
      "Record: 92\n",
      "Record: 93\n",
      "Record: 94\n",
      "Record: 95\n",
      "Record: 96\n",
      "Record: 97\n",
      "Record: 98\n",
      "Record: 99\n",
      "('Abstract counter:', 200)\n",
      "('HGVS FOUND:', 100)\n",
      "Record: 100\n",
      "Generated: parsed_abstracts.json\n",
      "Generated: stats.json\n"
     ]
    }
   ],
   "source": [
    "import re\n",
    "import json\n",
    "import codecs\n",
    "\n",
    "stats = {}\n",
    "\n",
    "def gen_abstracts(filename, per=100):\n",
    "    abstract_counter = 0\n",
    "    hgvs_counter = 0\n",
    "    with open(filename) as f:\n",
    "        for l in f:\n",
    "                        \n",
    "            data = json.loads(l)\n",
    "            for result in data['resultList']['result']:\n",
    "                \n",
    "                if not 'pubmedcounter' in stats:\n",
    "                    stats['pubmedcounter'] = 0\n",
    "                stats['pubmedcounter'] += 1\n",
    "                \n",
    "                if 'abstractText'in result:\n",
    "                    abstract = result['abstractText']\n",
    "                    abstract_counter += 1\n",
    "                    if abstract_counter % per == 0:\n",
    "                        print ('Abstract counter:', abstract_counter)\n",
    "                        \n",
    "                    if not 'abstractcounter' in stats:\n",
    "                        stats['abstractcounter'] = 0\n",
    "                    stats['abstractcounter'] += 1\n",
    "                        \n",
    "                    #if 'NM_' in abstract:\n",
    "                    #    print (abstract)\n",
    "                    #    assert False\n",
    "                    yield abstract\n",
    "                else:\n",
    "                    if not 'noabstract' in stats:\n",
    "                        stats['noabstract'] = 0\n",
    "                    stats['noabstract'] += 1\n",
    "\n",
    "def gen_hgvs(filename):\n",
    "    hgvs_counter = 0\n",
    "    \n",
    "    for abstract in gen_abstracts(filename):\n",
    "        \n",
    "        transcripts = get_transcripts(abstract)\n",
    "        \n",
    "        for hgvs in gen_hgvs_2(abstract):\n",
    "            hgvs_counter += 1\n",
    "\n",
    "            if hgvs_counter % 100 == 0:\n",
    "                print ('HGVS FOUND:', hgvs_counter)\n",
    "            \n",
    "            gene = get_gene_names(abstract, hgvs[1])\n",
    "            \n",
    "            yield abstract, gene, ''.join(hgvs), transcripts\n",
    "\n",
    "def get_gene_names(abstract, hgvs):\n",
    "    pos = abstract.index(hgvs)\n",
    "    \n",
    "    before = abstract[:pos]\n",
    "    after = abstract[pos+len(hgvs):]\n",
    "    \n",
    "    #Check before\n",
    "    before = before.split()[::-1]\n",
    "    before = [noparenthesis(x) for x in before]\n",
    "    \n",
    "    for word in before:\n",
    "        name = check_gene_name(word)\n",
    "        if name:\n",
    "            return name\n",
    "    \n",
    "    #Check after\n",
    "    after = [noparenthesis(x) for x in after.split()]\n",
    "    for word in after:\n",
    "        name = check_gene_name(word)\n",
    "        if name:\n",
    "            return name\n",
    "    \n",
    "    if not 'no_gene' in stats:\n",
    "        stats['no_gene'] = 0\n",
    "        \n",
    "    stats['no_gene'] += 1\n",
    "    \n",
    "    #print abstract\n",
    "    #print hgvs\n",
    "    \n",
    "    #if stats['no_gene'] > 1:\n",
    "    #    assert False\n",
    "    return None\n",
    "\n",
    "def gen_hgvs_2(abstract):\n",
    "    #245 A>\n",
    "    #found = re.findall(r'(.{1,100})([\\d]+[\\s]*[ACGT]+[\\s]*>)(.{1,100})',abstract, re.UNICODE)\n",
    "    found = re.findall(r'[\\d]+[\\s]*[ACGT]+[\\s]*>',abstract, re.UNICODE)\n",
    "    \n",
    "    if not found:\n",
    "        if not 'nohgvs' in stats:\n",
    "            stats['nohgvs'] = 0\n",
    "        stats['nohgvs'] += 1\n",
    "    \n",
    "    for f in found:\n",
    "        start = abstract.index(f)\n",
    "        before = abstract[:start]\n",
    "        after = abstract[start+len(f):]\n",
    "        \n",
    "        # We do not want any HGVS before and after the one that we found\n",
    "        before = before.replace('>','')\n",
    "        after = after.replace('>', '')\n",
    "\n",
    "        yield before, f, after\n",
    "    \n",
    "\n",
    "def get_transcripts(text):\n",
    "    return re.findall(r'[A-Z][A-Z]_[\\d]{4,15}\\.?[\\d]*', text, re.UNICODE)\n",
    "        \n",
    "def nowhite(p):\n",
    "    return re.sub(r'[\\s]', '', p, flags=re.UNICODE)\n",
    "\n",
    "def noparenthesis(p):\n",
    "    return re.sub(r'[\\(\\),\\.\\:]', '', p, flags=re.UNICODE)\n",
    "\n",
    "def post_process(hgvs):\n",
    "\n",
    "    s = None\n",
    "    \n",
    "    if not s:\n",
    "        # c.6348+1G>A\n",
    "        s = re.search(r'[CcGgpP][\\s]*\\.[\\s]*[\\d]+[\\s]*[\\+\\-][\\s]*[\\d]+[\\s]*[ACGT]+[\\s]*>[\\s]*[ACGT]+', hgvs, re.UNICODE)\n",
    "    \n",
    "    if not s:\n",
    "        # c.969C > A\n",
    "        s = re.search(r'[CcGgpPm][\\s]*\\.[\\s]*[\\d]+[\\s]*[ACGT][\\s]*>[\\s]*[ACGTS]+', hgvs, re.UNICODE)\n",
    "    \n",
    "    if not s:\n",
    "        # 236G > A\n",
    "        s = re.search(r'[\\d]+[\\s]*[ACGT]+[\\s]*>[\\s]*[ACGT]+', hgvs, re.UNICODE)\n",
    "    \n",
    "    if not s:\n",
    "        # c.743C > del \n",
    "        s = re.search(r'[cCgGpP]\\.[\\d]+[ACGT]+[\\s]*>[\\s]*del', hgvs, re.UNICODE)\n",
    "    \n",
    "    #print ('=====')\n",
    "    if s:\n",
    "        result = nowhite(s.group(0))\n",
    "        #print (hgvs)\n",
    "        #print (result)\n",
    "        return result\n",
    "    else:\n",
    "        #print (hgvs)\n",
    "        #assert False\n",
    "        \n",
    "        if not 'noparsehgvs' in stats:\n",
    "            stats['noparsehgvs'] = 0\n",
    "        stats['noparsehgvs'] += 1\n",
    "        \n",
    "        return None\n",
    "        \n",
    "        \n",
    "def do_1():\n",
    "    output_fn = 'parsed_abstracts.json'\n",
    "    output_f = codecs.open(output_fn, 'w', 'utf-8')\n",
    "    records = 0\n",
    "    for abstract, gene, hgvs, transcripts in gen_hgvs('abstracts.json'):\n",
    "        hgvs = post_process(hgvs)\n",
    "        if hgvs is None:\n",
    "            continue\n",
    "\n",
    "        \n",
    "        records += 1\n",
    "        \n",
    "        #if records <8:\n",
    "        #    continue\n",
    "        \n",
    "        if records > 100:\n",
    "            break\n",
    "        \n",
    "        to_save = {\n",
    "            'abstract': abstract,\n",
    "            'gene': gene,\n",
    "            'hgvs': hgvs,\n",
    "            'transcripts': transcripts,\n",
    "            'record': records\n",
    "        }\n",
    "        \n",
    "        #print gene, hgvs, transcripts\n",
    "        if False:\n",
    "            output_f.write('=========== {} ================\\n'.format(records))\n",
    "            output_f.write(abstract + '\\n')\n",
    "            output_f.write('GENE:' + gene + '\\n')\n",
    "            output_f.write('HGVS:'+ hgvs + '\\n')\n",
    "            output_f.write('Transcripts:' + str(transcripts) + '\\n')\n",
    "        if True:\n",
    "            print 'Record:', records\n",
    "            json.dump(to_save, output_f)\n",
    "            output_f.write('\\n')\n",
    "            output_f.flush()\n",
    "               \n",
    "    output_f.close()\n",
    "    print 'Generated:', output_fn\n",
    "    \n",
    "    with open('stats.json', 'w') as f:\n",
    "        json.dump(stats, f)\n",
    "    print 'Generated:', 'stats.json'\n",
    "\n",
    "def do_2():\n",
    "    counter={}\n",
    "    for abstract in gen_abstracts('abstracts.json', per=10000):\n",
    "        for tr in ['NT_', 'NG_', 'NC_']:\n",
    "            if tr in abstract:\n",
    "                #print '=====  {}  ======'.format(tr)\n",
    "                if not tr in counter:\n",
    "                    counter[tr] = []\n",
    "                    \n",
    "                counter[tr].append(abstract)\n",
    "                #print abstract\n",
    "                #assert False\n",
    "                \n",
    "    #print (counter)\n",
    "    return counter\n",
    "        \n",
    "do_1()\n",
    "#do_2()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input\ttranscript\tgene\tstrand\tcoordinates(gDNA/cDNA/protein)\tregion\tinfo\r\n",
      "SERPINF1:c.1067\tCCDS11012 (protein_coding)\tSERPINF1\t+\tchr17:g.1680550T/c.1067T/.\tinside_[cds_in_exon_7]\tsource=CCDS\r\n"
     ]
    }
   ],
   "source": [
    "!/home/user/mutationinfo/localpython/bin/transvar canno --ccds -i 'SERPINF1:c.1067' --refversion hg19"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'LIS1'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-36-6a6307ca66fe>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mgenenames\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'LIS1'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m: 'LIS1'"
     ]
    }
   ],
   "source": [
    "genenames['LIS1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('Abstract counter:', 10000)\n",
      "('Abstract counter:', 20000)\n",
      "('Abstract counter:', 30000)\n",
      "('Abstract counter:', 40000)\n",
      "('Abstract counter:', 50000)\n",
      "('Abstract counter:', 60000)\n",
      "('Abstract counter:', 70000)\n",
      "('Abstract counter:', 80000)\n",
      "('Abstract counter:', 90000)\n",
      "('Abstract counter:', 100000)\n",
      "('Abstract counter:', 110000)\n",
      "('Abstract counter:', 120000)\n",
      "('Abstract counter:', 130000)\n",
      "('Abstract counter:', 140000)\n",
      "('Abstract counter:', 150000)\n",
      "('Abstract counter:', 160000)\n",
      "('Abstract counter:', 170000)\n",
      "('Abstract counter:', 180000)\n",
      "('Abstract counter:', 190000)\n",
      "('Abstract counter:', 200000)\n",
      "('Abstract counter:', 210000)\n",
      "('Abstract counter:', 220000)\n",
      "('Abstract counter:', 230000)\n",
      "('Abstract counter:', 240000)\n",
      "('Abstract counter:', 250000)\n",
      "('Abstract counter:', 260000)\n",
      "('Abstract counter:', 270000)\n",
      "('Abstract counter:', 280000)\n",
      "('Abstract counter:', 290000)\n",
      "('Abstract counter:', 300000)\n",
      "('Abstract counter:', 310000)\n",
      "('Abstract counter:', 320000)\n",
      "('Abstract counter:', 330000)\n",
      "('Abstract counter:', 340000)\n",
      "('Abstract counter:', 350000)\n",
      "('Abstract counter:', 360000)\n",
      "('Abstract counter:', 370000)\n",
      "('Abstract counter:', 380000)\n",
      "('Abstract counter:', 390000)\n",
      "('Abstract counter:', 400000)\n",
      "('Abstract counter:', 410000)\n",
      "('Abstract counter:', 420000)\n",
      "('Abstract counter:', 430000)\n",
      "('Abstract counter:', 440000)\n",
      "('Abstract counter:', 450000)\n",
      "('Abstract counter:', 460000)\n",
      "('Abstract counter:', 470000)\n",
      "('Abstract counter:', 480000)\n",
      "('Abstract counter:', 490000)\n",
      "('Abstract counter:', 500000)\n",
      "('Abstract counter:', 510000)\n",
      "('Abstract counter:', 520000)\n",
      "('Abstract counter:', 530000)\n",
      "('Abstract counter:', 540000)\n",
      "('Abstract counter:', 550000)\n",
      "('Abstract counter:', 560000)\n",
      "('Abstract counter:', 570000)\n",
      "('Abstract counter:', 580000)\n",
      "('Abstract counter:', 590000)\n",
      "('Abstract counter:', 600000)\n",
      "('Abstract counter:', 610000)\n",
      "('Abstract counter:', 620000)\n",
      "('Abstract counter:', 630000)\n",
      "('Abstract counter:', 640000)\n",
      "('Abstract counter:', 650000)\n",
      "('Abstract counter:', 660000)\n",
      "('Abstract counter:', 670000)\n",
      "('Abstract counter:', 680000)\n",
      "('Abstract counter:', 690000)\n",
      "('Abstract counter:', 700000)\n",
      "('Abstract counter:', 710000)\n",
      "('Abstract counter:', 720000)\n",
      "('Abstract counter:', 730000)\n",
      "('Abstract counter:', 740000)\n",
      "('Abstract counter:', 750000)\n",
      "('Abstract counter:', 760000)\n",
      "('Abstract counter:', 770000)\n",
      "('Abstract counter:', 780000)\n",
      "('Abstract counter:', 790000)\n",
      "('Abstract counter:', 800000)\n",
      "('Abstract counter:', 810000)\n",
      "('Abstract counter:', 820000)\n",
      "('Abstract counter:', 830000)\n",
      "('Abstract counter:', 840000)\n",
      "('Abstract counter:', 850000)\n",
      "('Abstract counter:', 860000)\n",
      "('Abstract counter:', 870000)\n",
      "('Abstract counter:', 880000)\n",
      "('Abstract counter:', 890000)\n",
      "('Abstract counter:', 900000)\n",
      "('Abstract counter:', 910000)\n"
     ]
    }
   ],
   "source": [
    "counter = do_2()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('nt_ng_abstracts.json', 'w') as f:\n",
    "    json.dump(counter, f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NG_ 52\n",
      "NT_ 24\n",
      "NC_ 68\n"
     ]
    }
   ],
   "source": [
    "for x,y in counter.iteritems():\n",
    "    print x, len(y)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/user/mutationinfo/localpython/lib/python2.7/site-packages/psycopg2-2.7.4-py2.7-linux-x86_64.egg/psycopg2/__init__.py:144: UserWarning: The psycopg2 wheel package will be renamed from release 2.8; in order to keep installing from binary please use \"pip install psycopg2-binary\" instead. For details see: <http://initd.org/psycopg/docs/install.html#binary-install-from-pypi>.\n",
      "  \"\"\")\n"
     ]
    }
   ],
   "source": [
    "from MutationInfo import MutationInfo as MI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING dbfile.open_index: Falling back to hash index: unable to import bsddb\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING:pygr-log:Falling back to hash index: unable to import bsddb\n"
     ]
    }
   ],
   "source": [
    "mi = MI()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# RUN_1\n",
    "* Run with:  ```/home/user/mutationinfo/localpython/bin/python run_1.py```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Overwriting run_1.py\n"
     ]
    }
   ],
   "source": [
    "%%writefile run_1.py\n",
    "# -*- coding: utf-8 -*-\n",
    "\n",
    "def load_gene_names():\n",
    "    ret = set()\n",
    "    syns = {}\n",
    "\n",
    "    with open('genenames.txt') as f:\n",
    "        c = 0\n",
    "        for l in f:\n",
    "            c += 1\n",
    "            if c == 1:\n",
    "                header = l.split('\\t')\n",
    "                syn_index = header.index('Synonyms')\n",
    "                symb_index = header.index('Previous Symbols')\n",
    "                continue\n",
    "                \n",
    "            ls = l.split('\\t')\n",
    "            gene = ls[1]\n",
    "            ret.add(gene)\n",
    "            this_syn = [x.strip() for x in ls[syn_index].split(',')]\n",
    "            for x in this_syn:\n",
    "                syns[x] = gene\n",
    "            \n",
    "            symbols = [x.strip() for x in ls[symb_index].split(',')]\n",
    "            for x in symbols:\n",
    "                syns[x] = gene\n",
    "            \n",
    "            \n",
    "    return {x: None for x in ret}, syns\n",
    "\n",
    "stop = {\n",
    "    '2': None,\n",
    "    'T': None,\n",
    "    'B': None,\n",
    "}\n",
    "\n",
    "def check_gene_name(x):\n",
    "    \n",
    "    if x in stop:\n",
    "        return None\n",
    "    \n",
    "    if x in genenames:\n",
    "        if not 'realnames' in stats:\n",
    "            stats['realnames'] = 0\n",
    "        stats['realnames'] += 1\n",
    "        \n",
    "        return x\n",
    "    \n",
    "    if x in genesyns:\n",
    "        \n",
    "        if not 'syns' in stats:\n",
    "            stats['syns'] = 0\n",
    "        stats['syns'] += 1\n",
    "        \n",
    "        return genesyns[x]\n",
    "    \n",
    "    extra_synonyms = {\n",
    "        # Found : Real\n",
    "        'FXII': 'F12',\n",
    "    }\n",
    "    \n",
    "    if x in extra_synonyms:\n",
    "        return extra_synonyms[x]\n",
    "    \n",
    "    if x.lower() in ['mitochondrial', 'mtdna']:\n",
    "        \n",
    "        if not 'mitochondrial' in stats:\n",
    "            stats['mitochondrial'] = 0\n",
    "        stats['mitochondrial'] += 1\n",
    "        \n",
    "        return 'mtdna'\n",
    "    \n",
    "    return None\n",
    "\n",
    "###################################\n",
    "genenames, genesyns = load_gene_names()\n",
    "\n",
    "import re\n",
    "import gzip\n",
    "import json\n",
    "import codecs\n",
    "\n",
    "stats = {}\n",
    "\n",
    "def gen_abstracts(filename, per=100):\n",
    "    abstract_counter = 0\n",
    "    hgvs_counter = 0\n",
    "    with gzip.open(filename, 'rt') as f:\n",
    "        for l in f:\n",
    "                        \n",
    "            data = json.loads(l)\n",
    "            for result in data['resultList']['result']:\n",
    "                \n",
    "                if not 'pubmedcounter' in stats:\n",
    "                    stats['pubmedcounter'] = 0\n",
    "                stats['pubmedcounter'] += 1\n",
    "                \n",
    "                if 'abstractText'in result:\n",
    "                    abstract = result['abstractText']\n",
    "                    abstract_counter += 1\n",
    "                    if abstract_counter % per == 0:\n",
    "                        print ('Abstract counter:', abstract_counter)\n",
    "                        \n",
    "                    if not 'abstractcounter' in stats:\n",
    "                        stats['abstractcounter'] = 0\n",
    "                    stats['abstractcounter'] += 1\n",
    "                        \n",
    "                    #if 'NM_' in abstract:\n",
    "                    #    print (abstract)\n",
    "                    #    assert False\n",
    "                    yield abstract\n",
    "                else:\n",
    "                    if not 'noabstract' in stats:\n",
    "                        stats['noabstract'] = 0\n",
    "                    stats['noabstract'] += 1\n",
    "\n",
    "def gen_hgvs(filename):\n",
    "    hgvs_counter = 0\n",
    "    \n",
    "    for abstract in gen_abstracts(filename):\n",
    "        \n",
    "        transcripts = get_transcripts(abstract)\n",
    "        \n",
    "        for hgvs in gen_hgvs_2(abstract):\n",
    "            hgvs_counter += 1\n",
    "\n",
    "            if hgvs_counter % 100 == 0:\n",
    "                print ('HGVS FOUND:', hgvs_counter)\n",
    "            \n",
    "            gene = get_gene_names(abstract, hgvs[1])\n",
    "            \n",
    "            yield abstract, gene, ''.join(hgvs), transcripts\n",
    "\n",
    "def get_gene_names(abstract, hgvs):\n",
    "    pos = abstract.index(hgvs)\n",
    "    \n",
    "    before = abstract[:pos]\n",
    "    after = abstract[pos+len(hgvs):]\n",
    "    \n",
    "    #Check before\n",
    "    before = before.split()[::-1]\n",
    "    before = [noparenthesis(x) for x in before]\n",
    "    \n",
    "    for word in before:\n",
    "        name = check_gene_name(word)\n",
    "        if name:\n",
    "            return name\n",
    "    \n",
    "    #Check after\n",
    "    after = [noparenthesis(x) for x in after.split()]\n",
    "    for word in after:\n",
    "        name = check_gene_name(word)\n",
    "        if name:\n",
    "            return name\n",
    "    \n",
    "    if not 'no_gene' in stats:\n",
    "        stats['no_gene'] = 0\n",
    "        \n",
    "    stats['no_gene'] += 1\n",
    "    \n",
    "    #print abstract\n",
    "    #print hgvs\n",
    "    \n",
    "    #if stats['no_gene'] > 1:\n",
    "    #    assert False\n",
    "    return None\n",
    "\n",
    "def gen_hgvs_2(abstract):\n",
    "    #245 A>\n",
    "    #found = re.findall(r'(.{1,100})([\\d]+[\\s]*[ACGT]+[\\s]*>)(.{1,100})',abstract, re.UNICODE)\n",
    "    found = re.findall(r'[\\d]+[\\s]*[ACGT]+[\\s]*>',abstract, re.UNICODE)\n",
    "    \n",
    "    if not found:\n",
    "        if not 'nohgvs' in stats:\n",
    "            stats['nohgvs'] = 0\n",
    "        stats['nohgvs'] += 1\n",
    "    \n",
    "    for f in found:\n",
    "        start = abstract.index(f)\n",
    "        before = abstract[:start]\n",
    "        after = abstract[start+len(f):]\n",
    "        \n",
    "        # We do not want any HGVS before and after the one that we found\n",
    "        before = before.replace('>','')\n",
    "        after = after.replace('>', '')\n",
    "\n",
    "        yield before, f, after\n",
    "    \n",
    "\n",
    "def get_transcripts(text):\n",
    "    return re.findall(r'[A-Z][A-Z]_[\\d]{4,15}\\.?[\\d]*', text, re.UNICODE)\n",
    "        \n",
    "def nowhite(p):\n",
    "    return re.sub(r'[\\s]', '', p, flags=re.UNICODE)\n",
    "\n",
    "def noparenthesis(p):\n",
    "    return re.sub(r'[\\(\\),\\.\\:]', '', p, flags=re.UNICODE)\n",
    "\n",
    "def post_process(hgvs):\n",
    "\n",
    "    s = None\n",
    "    \n",
    "    if not s:\n",
    "        # c.6348+1G>A\n",
    "        s = re.search(r'[CcGgpP][\\s]*\\.[\\s]*[\\d]+[\\s]*[\\+\\-][\\s]*[\\d]+[\\s]*[ACGT]+[\\s]*>[\\s]*[ACGT]+', hgvs, re.UNICODE)\n",
    "    \n",
    "    if not s:\n",
    "        # c.969C > A\n",
    "        s = re.search(r'[CcGgpPm][\\s]*\\.[\\s]*[\\d]+[\\s]*[ACGT][\\s]*>[\\s]*[ACGTS]+', hgvs, re.UNICODE)\n",
    "    \n",
    "    if not s:\n",
    "        # 236G > A\n",
    "        s = re.search(r'[\\d]+[\\s]*[ACGT]+[\\s]*>[\\s]*[ACGT]+', hgvs, re.UNICODE)\n",
    "    \n",
    "    if not s:\n",
    "        # c.743C > del \n",
    "        s = re.search(r'[cCgGpP]\\.[\\d]+[ACGT]+[\\s]*>[\\s]*del', hgvs, re.UNICODE)\n",
    "    \n",
    "    #print ('=====')\n",
    "    if s:\n",
    "        result = nowhite(s.group(0))\n",
    "        #print (hgvs)\n",
    "        #print (result)\n",
    "        return result\n",
    "    else:\n",
    "        #print (hgvs)\n",
    "        #assert False\n",
    "        \n",
    "        if not 'noparsehgvs' in stats:\n",
    "            stats['noparsehgvs'] = 0\n",
    "        stats['noparsehgvs'] += 1\n",
    "        \n",
    "        return None\n",
    "        \n",
    "        \n",
    "def do_1():\n",
    "    output_fn = 'parsed_abstracts.json'\n",
    "    output_f = codecs.open(output_fn, 'w', 'utf-8')\n",
    "    records = 0\n",
    "    for abstract, gene, hgvs, transcripts in gen_hgvs('abstracts.json.gz'):\n",
    "        hgvs = post_process(hgvs)\n",
    "        if hgvs is None:\n",
    "            continue\n",
    "\n",
    "        \n",
    "        records += 1\n",
    "        if records % 100 == 0:\n",
    "            print 'Records:', records\n",
    "        \n",
    "        #if records <8:\n",
    "        #    continue\n",
    "        \n",
    "        #if records > 100:\n",
    "        #    break\n",
    "        \n",
    "        to_save = {\n",
    "            'abstract': abstract,\n",
    "            'gene': gene,\n",
    "            'hgvs': hgvs,\n",
    "            'transcripts': transcripts,\n",
    "            'record': records\n",
    "        }\n",
    "        \n",
    "        #print gene, hgvs, transcripts\n",
    "        if False:\n",
    "            output_f.write('=========== {} ================\\n'.format(records))\n",
    "            output_f.write(abstract + '\\n')\n",
    "            output_f.write('GENE:' + gene + '\\n')\n",
    "            output_f.write('HGVS:'+ hgvs + '\\n')\n",
    "            output_f.write('Transcripts:' + str(transcripts) + '\\n')\n",
    "        if True:\n",
    "            json.dump(to_save, output_f)\n",
    "            output_f.write('\\n')\n",
    "            output_f.flush()\n",
    "               \n",
    "    output_f.close()\n",
    "    print 'Generated:', output_fn\n",
    "    \n",
    "    with open('stats.json', 'w') as f:\n",
    "        json.dump(stats, f)\n",
    "    print 'Generated:', 'stats.json'\n",
    "\n",
    "def do_2():\n",
    "    counter={}\n",
    "    for abstract in gen_abstracts('abstracts.json.gz', per=10000):\n",
    "        for tr in ['NT_', 'NG_', 'NC_']:\n",
    "            if tr in abstract:\n",
    "                #print '=====  {}  ======'.format(tr)\n",
    "                if not tr in counter:\n",
    "                    counter[tr] = []\n",
    "                    \n",
    "                counter[tr].append(abstract)\n",
    "                #print abstract\n",
    "                #assert False\n",
    "                \n",
    "    #print (counter)\n",
    "    return counter\n",
    "        \n",
    "do_1()\n",
    "#do_2()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# RUN_2\n",
    "\n",
    "* Run with ```/home/user/mutationinfo/localpython/bin/python run_2.py```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Overwriting run_2.py\n"
     ]
    }
   ],
   "source": [
    "%%writefile run_2.py\n",
    "\n",
    "#assert False # THIS IS ON PURPOSE. DO NOT RUN THIS\n",
    "\n",
    "import re\n",
    "import json\n",
    "import subprocess\n",
    "\n",
    "#import pandas as pd\n",
    "\n",
    "from StringIO import StringIO\n",
    "\n",
    "def execute(command):\n",
    "    process = subprocess.Popen(command, shell=True,\n",
    "                           stdout=subprocess.PIPE, \n",
    "                           stderr=subprocess.PIPE)\n",
    "\n",
    "    # wait for the process to terminate\n",
    "    out, err = process.communicate()\n",
    "    errcode = process.returncode\n",
    "    \n",
    "    out = out.decode('utf-8')\n",
    "    err = err.decode('utf-8')\n",
    "    \n",
    "    #print ('STDOUT:')\n",
    "    #print (out)\n",
    "    #print ('ERR:')\n",
    "    #print (err)\n",
    "    #print ('RETURN CODE:', errcode)\n",
    "    \n",
    "    return out\n",
    "\n",
    "def inverse(s):\n",
    "    \n",
    "    return {\n",
    "        'A': 'T',\n",
    "        'T': 'A',\n",
    "        'C': 'G',\n",
    "        'G': 'C',\n",
    "    }[s]\n",
    "\n",
    "stats_2 = {}\n",
    "\n",
    "def get_transvar_reference(coord, strand):\n",
    "    s = re.search(r'chr[\\dXY]+:g.[\\d]+([ACGT]+)/', coord)\n",
    "    if s:\n",
    "        reference = s.group(1)\n",
    "    else:\n",
    "        return None\n",
    "\n",
    "    if strand == '-':\n",
    "        return inverse(reference)\n",
    "    elif strand == '+':\n",
    "        return reference\n",
    "    else:\n",
    "        assert False\n",
    "\n",
    "\n",
    "def parse_transvar_output(output,command=None, hgvs=None, abstract=None):\n",
    "    ls = output.split('\\n')\n",
    "    ls = [x.split('\\t') for x in ls if x.strip()] \n",
    "    strand_i = ls[0].index('strand')\n",
    "    coord_i = ls[0].index('coordinates(gDNA/cDNA/protein)')\n",
    "    transcript_i = ls[0].index('transcript')\n",
    "    \n",
    "    ret = {l[transcript_i]:get_transvar_reference(l[coord_i], l[strand_i])  for l in ls[1:]}\n",
    "    return ret    \n",
    "\n",
    "def load_parsed_abstracts():\n",
    "    with open('parsed_abstracts.json') as f:\n",
    "        for l in f:\n",
    "            data = json.loads(l)\n",
    "            yield data\n",
    "            \n",
    "def apply_transvar():\n",
    "    \n",
    "    transvar_cmd_p = \"/home/user/mutationinfo/localpython/bin/transvar canno --ccds -i '{hgvs}' --refversion hg19\"\n",
    "    fn = 'parsed_abstracts_transvar.json'\n",
    "    f = open(fn, 'w')\n",
    "    counter = 0\n",
    "    input_transvar_fn = 'genes_hgvs.txt'\n",
    "    input_transvar_f = open(input_transvar_fn, 'w')\n",
    "    for parsed in load_parsed_abstracts():\n",
    "        counter += 1\n",
    "        \n",
    "        if counter < 0:\n",
    "            continue\n",
    "            \n",
    "        #if counter > 100:\n",
    "        #    break\n",
    "        \n",
    "        print 'Counter:', counter\n",
    "        \n",
    "        abstract = parsed['abstract']\n",
    "        \n",
    "        if 'dog' in abstract:\n",
    "            stats_2['nothumnan'] = stats_2.get('nothumnan', 0) + 1\n",
    "            continue\n",
    "\n",
    "        \n",
    "        #print parsed\n",
    "        hgvs = parsed['hgvs']\n",
    "        \n",
    "        #Get hgvs reference\n",
    "        s = re.search(r'([ACGT]+)>', hgvs)\n",
    "        if not s:\n",
    "            continue\n",
    "            #assert False\n",
    "            \n",
    "        hgvs_reference = s.group(1)\n",
    "                \n",
    "        #If it starts with a number assuming coding position\n",
    "        if re.search(r'^[\\d]+', hgvs, re.UNICODE):\n",
    "            hgvs = 'c.' + hgvs\n",
    "            \n",
    "            \n",
    "        #Get everything except the reference\n",
    "        s = re.search(r'^([^ACGT]+)[ACGT]+>', hgvs)\n",
    "        if not s:\n",
    "            continue\n",
    "            #assert False\n",
    "        except_reference = s.group(1)\n",
    "\n",
    "        \n",
    "        gene = parsed['gene']\n",
    "        if not gene:\n",
    "            continue\n",
    "            #assert False\n",
    "\n",
    "        transcripts = parsed['transcripts']\n",
    "        if transcripts:\n",
    "            pass\n",
    "            #assert False\n",
    "        \n",
    "        hgvs = gene + ':' + hgvs\n",
    "        hgvs_no_reference = gene + ':' + except_reference\n",
    "        #transvar_cmd = transvar_cmd_p.format(hgvs=hgvs)\n",
    "        transvar_cmd = transvar_cmd_p.format(hgvs=hgvs_no_reference)\n",
    "        #print transvar_cmd\n",
    "        #print abstract\n",
    "        #transvar_output = execute(transvar_cmd) # DO NOT EXECUTE IT\n",
    "        input_transvar_f.write('{}\\t{}\\n'.format(gene, except_reference))\n",
    "        continue\n",
    "        \n",
    "        parsed['transvar'] = transvar_output\n",
    "        f.write(json.dumps(parsed) + '\\n')\n",
    "        continue\n",
    "        \n",
    "        #print transvar_output\n",
    "        #transvar_output_f = StringIO(transvar_output)\n",
    "        #df = pd.read_csv(transvar_output_f, sep='\\t')\n",
    "        #print df\n",
    "        \n",
    "        transvar_references = parse_transvar_output(transvar_output, transvar_cmd, hgvs, abstract)\n",
    "        if len(transvar_references) == 0:\n",
    "            assert False\n",
    "            \n",
    "        if len(transvar_references) == 1:\n",
    "            if transvar_references.values()[0] == hgvs_reference:\n",
    "                stats_2['transvarok'] = stats_2.get('transvarok', 0) + 1\n",
    "            else:                \n",
    "                print transvar_cmd\n",
    "                print 'HGVS:', hgvs\n",
    "                print 'transvar_reference:', transvar_references.values()[0], 'Hgvs_reference:', hgvs_reference\n",
    "                assert transvar_references.values()[0]\n",
    "                assert hgvs_reference\n",
    "                stats_2['1_reference_not_compatible'] = stats_2.get('1_reference_not_compatible', 0) + 1\n",
    "        else:\n",
    "            # Take all references\n",
    "            all_references = list(set(transvar_references.values()))\n",
    "            if len(all_references) == 1:\n",
    "                if all_references[0] == hgvs_reference:\n",
    "                    stats_2['1_reference_many_transcripts'] = stats_2.get('1_reference_many_transcripts', 0) + 1\n",
    "                else:\n",
    "                    assert False\n",
    "            else:\n",
    "                assert hgvs_reference # Not null\n",
    "                assert [x for x in transvar_references.values() if x] # Not empty\n",
    "                if hgvs_reference in all_references:\n",
    "                    stats_2['many_references_exists'] = stats_2.get('many_references_exists', 0) + 1\n",
    "                else:\n",
    "                    stats_2['many_references_NOT_exists'] = stats_2.get('many_references_NOT_exists', 0) + 1\n",
    "\n",
    "    f.close()\n",
    "    input_transvar_f.close()\n",
    "    print 'Created_file:', input_transvar_fn\n",
    "    with open('stats_2.json', 'w') as f:\n",
    "        json.dump(stats_2, f)\n",
    "    \n",
    "    print 'created file:', fn\n",
    "    print 'Created file: stats_2.json'\n",
    "\n",
    "def do_3():\n",
    "    apply_transvar()\n",
    "    \n",
    "do_3()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### RUN:\n",
    "\n",
    "```\n",
    "/home/user/mutationinfo/localpython/bin/transvar canno --ccds -l genes_hgvs.txt -g 1 -m 2 --refversion hg19 --oneline > transvar_output.txt\n",
    "```\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input\ttranscript\tgene\tstrand\tcoordinates(gDNA/cDNA/protein)\tregion\tinfo\r\n",
      "RHAG:c.236G>A\tCCDS4927 (protein_coding)\tRHAG\t-\tchr6:g.49586997C>T/c.236G>A/p.S79N\tinside_[cds_in_exon_2]\tCSQN=Missense;reference_codon=AGT;alternative_codon=AAT;source=CCDS\r\n"
     ]
    }
   ],
   "source": [
    "!/home/user/mutationinfo/localpython/bin/transvar canno --ccds -i 'RHAG:c.236G>A' --refversion hg19"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input\ttranscript\tgene\tstrand\tcoordinates(gDNA/cDNA/protein)\tregion\tinfo\n",
      "EXT2:c.969C>A\t.\t.\t.\t././.\t.\tno_valid_transcript_found\n"
     ]
    }
   ],
   "source": [
    "!/home/user/mutationinfo/localpython/bin/transvar canno --ccds -i 'EXT2:c.969C>A' --refversion hg19"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input\ttranscript\tgene\tstrand\tcoordinates(gDNA/cDNA/protein)\tregion\tinfo\n",
      "EGFR:c.2576T>G\t.\t.\t.\t././.\t.\tno_valid_transcript_found\n"
     ]
    }
   ],
   "source": [
    "!/home/user/mutationinfo/localpython/bin/transvar canno --ccds -i 'EGFR:c.2576T>G' --refversion hg19"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input\ttranscript\tgene\tstrand\tcoordinates(gDNA/cDNA/protein)\tregion\tinfo\n",
      "EGFR:c.2576\tCCDS5514 (protein_coding)\tEGFR\t+\tchr7:g.55259518C/c.2576C/.\tinside_[cds_in_exon_21]\tsource=CCDS\n"
     ]
    }
   ],
   "source": [
    "!/home/user/mutationinfo/localpython/bin/transvar canno --ccds -i 'EGFR:c.2576' --refversion hg19"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "3 cases of exon 21 2576T>G mutation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input\ttranscript\tgene\tstrand\tcoordinates(gDNA/cDNA/protein)\tregion\tinfo\n",
      "EGFR:c.2576\tCCDS5514 (protein_coding)\tEGFR\t+\tchr7:g.55259518C/c.2576C/.\tinside_[cds_in_exon_21]\tsource=CCDS\n"
     ]
    }
   ],
   "source": [
    "!/home/user/mutationinfo/localpython/bin/transvar canno --ccds -i 'EGFR:c.2576' --refversion hg19"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input\ttranscript\tgene\tstrand\tcoordinates(gDNA/cDNA/protein)\tregion\tinfo\n",
      "BRCA1:c.5251\tCCDS11453 (protein_coding)\tBRCA1\t-\tchr17:g.41209095G/c.5251C/.\tinside_[cds_in_exon_18]\tsource=CCDS\n",
      "BRCA1:c.5251\tCCDS11456 (protein_coding)\tBRCA1\t-\tchr17:g.41215355T/c.5251A/.\tinside_[cds_in_exon_18]\tsource=CCDS\n",
      "BRCA1:c.5251\tCCDS11459 (protein_coding)\tBRCA1\t-\tchr17:g.41201152A/c.5251T/.\tinside_[cds_in_exon_18]\tsource=CCDS\n"
     ]
    }
   ],
   "source": [
    "!/home/user/mutationinfo/localpython/bin/transvar canno --ccds -i 'BRCA1:c.5251' --refversion hg19"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input\ttranscript\tgene\tstrand\tcoordinates(gDNA/cDNA/protein)\tregion\tinfo\n",
      "BRCA1:c.5251C>T\tCCDS11453 (protein_coding)\tBRCA1\t-\tchr17:g.41209095G>A/c.5251C>T/p.R1751*\tinside_[cds_in_exon_18]\tCSQN=Nonsense;reference_codon=CGA;alternative_codon=TGA;source=CCDS\n"
     ]
    }
   ],
   "source": [
    "!/home/user/mutationinfo/localpython/bin/transvar canno --ccds -i 'BRCA1:c.5251C>T' --refversion hg19"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input\ttranscript\tgene\tstrand\tcoordinates(gDNA/cDNA/protein)\tregion\tinfo\n",
      "BRCA1:c.5251A>T\tCCDS11456 (protein_coding)\tBRCA1\t-\tchr17:g.41215355T>A/c.5251A>T/p.N1751Y\tinside_[cds_in_exon_18]\tCSQN=Missense;reference_codon=AAT;alternative_codon=TAT;source=CCDS\n"
     ]
    }
   ],
   "source": [
    "!/home/user/mutationinfo/localpython/bin/transvar canno --ccds -i 'BRCA1:c.5251A>T' --refversion hg19"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input\ttranscript\tgene\tstrand\tcoordinates(gDNA/cDNA/protein)\tregion\tinfo\r\n",
      "IGSF1:c.2066\tCCDS14629 (protein_coding)\tIGSF1\t-\tchrX:g.130412084G/c.2066C/.\tinside_[cds_in_exon_12]\tsource=CCDS\r\n",
      "IGSF1:c.2066\tCCDS55490 (protein_coding)\tIGSF1\t-\tchrX:g.130412057T/c.2066A/.\tinside_[cds_in_exon_11]\tsource=CCDS\r\n",
      "IGSF1:c.2066\tCCDS55491 (protein_coding)\tIGSF1\t-\tchrX:g.130412099G/c.2066C/.\tinside_[cds_in_exon_12]\tsource=CCDS\r\n"
     ]
    }
   ],
   "source": [
    "!/home/user/mutationinfo/localpython/bin/transvar canno --ccds -i 'IGSF1:c.2066' --refversion hg19"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input\ttranscript\tgene\tstrand\tcoordinates(gDNA/cDNA/protein)\tregion\tinfo\n",
      "IGSF1:c.2065+1G>A\t.\t.\t.\t././.\t.\tno_valid_transcript_found\n"
     ]
    }
   ],
   "source": [
    "!/home/user/mutationinfo/localpython/bin/transvar canno --ccds -i 'IGSF1:c.2065+1G>A' --refversion hg19"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input\ttranscript\tgene\tstrand\tcoordinates(gDNA/cDNA/protein)\tregion\tinfo\r\n",
      "FAH:c.1062+5\tCCDS10314 (protein_coding)\tFAH\t+\tchr15:g.80472572G/c.1062+5G/.\tinside_[intron_between_exon_12_and_13]\tsource=CCDS\r\n"
     ]
    }
   ],
   "source": [
    "!/home/user/mutationinfo/localpython/bin/transvar canno --ccds -i 'FAH:c.1062+5' --refversion hg19"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input\ttranscript\tgene\tstrand\tcoordinates(gDNA/cDNA/protein)\tregion\tinfo\n",
      "[wrap_exception] warning: invalid_gene_PCD\n",
      "PCD:c.710\t.\t.\t.\t././.\t.\tError_invalid_gene_PCD\n"
     ]
    }
   ],
   "source": [
    "!/home/user/mutationinfo/localpython/bin/transvar canno --ccds -i 'PCD:c.710' --refversion hg19 "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Run 3\n",
    "Run with ```/home/user/mutationinfo/localpython/bin/python run_3.py```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing run_3.py\n"
     ]
    }
   ],
   "source": [
    "%%writefile run_3.py\n",
    "import re\n",
    "import json\n",
    "import time\n",
    "import requests\n",
    "\n",
    "def load_parsed_abstracts():\n",
    "    with open('parsed_abstracts.json') as f:\n",
    "        for l in f:\n",
    "            data = json.loads(l)\n",
    "            yield data\n",
    "            \n",
    "def hgvs_from_record(parsed):\n",
    "        \n",
    "    abstract = parsed['abstract']\n",
    "\n",
    "    if 'dog' in abstract:\n",
    "        #stats_2['nothumnan'] = stats_2.get('nothumnan', 0) + 1\n",
    "        return None\n",
    "\n",
    "\n",
    "    #print parsed\n",
    "    hgvs = parsed['hgvs']\n",
    "\n",
    "    #Get hgvs reference\n",
    "    s = re.search(r'([ACGT]+)>', hgvs)\n",
    "    if not s:\n",
    "        return None\n",
    "        #assert False\n",
    "\n",
    "    hgvs_reference = s.group(1)\n",
    "\n",
    "    #If it starts with a number assuming coding position\n",
    "    if re.search(r'^[\\d]+', hgvs, re.UNICODE):\n",
    "        hgvs = 'c.' + hgvs\n",
    "\n",
    "\n",
    "    #Get everything except the reference\n",
    "    s = re.search(r'^([^ACGT]+)[ACGT]+>', hgvs)\n",
    "    if not s:\n",
    "        return None\n",
    "        #continue\n",
    "        #assert False\n",
    "    except_reference = s.group(1)\n",
    "\n",
    "\n",
    "    gene = parsed['gene']\n",
    "    if not gene:\n",
    "        return None\n",
    "        #assert False\n",
    "\n",
    "    transcripts = parsed['transcripts']\n",
    "    if transcripts:\n",
    "        pass\n",
    "        #assert False\n",
    "\n",
    "    hgvs = gene + ':' + hgvs\n",
    "    hgvs_no_reference = gene + ':' + except_reference\n",
    "    #transvar_cmd = transvar_cmd_p.format(hgvs=hgvs)\n",
    "    return hgvs\n",
    "\n",
    "def vep_parse(hgvs):\n",
    "    headers={ \"Content-Type\" : \"application/json\"}\n",
    "    vep_url = \"https://rest.ensembl.org/vep/human/hgvs/{var}?\"\n",
    "\n",
    "    url = vep_url.format(var=hgvs)\n",
    "    responce = requests.get(url, headers=headers)\n",
    "    if responce.ok:\n",
    "        return responce.json()\n",
    "    else:\n",
    "        print ('Error:')\n",
    "        print (responce.text)\n",
    "\n",
    "def vep_parse_post(hgvs):\n",
    "\n",
    " \n",
    "    server = \"https://rest.ensembl.org\"\n",
    "    ext = \"/vep/human/hgvs\"\n",
    "    headers={ \"Content-Type\" : \"application/json\", \"Accept\" : \"application/json\"}\n",
    "    \n",
    "    assert type(hgvs) is list\n",
    "    #data = '[' + ','.join([rept(x) for x in hgvs]) + ']'\n",
    "    data = json.dumps({\"hgvs_notations\": hgvs})\n",
    "    print data\n",
    "    \n",
    "    r = requests.post(server+ext, headers=headers, data=data)\n",
    " \n",
    "    if not r.ok:\n",
    "        print ('ERROR:')\n",
    "        print (r.text)\n",
    "        return None\n",
    " \n",
    "    decoded = r.json()\n",
    "    return decoded\n",
    "\n",
    "def vep_run():\n",
    "    \n",
    "    count = 0\n",
    "    \n",
    "    f = open('vep_results.json', 'w')\n",
    "    \n",
    "    all_hgvs = set()\n",
    "    for parsed in load_parsed_abstracts():\n",
    "        count += 1\n",
    "        \n",
    "        print ('count:', count)\n",
    "        \n",
    "        if count < 1:\n",
    "            continue\n",
    "            \n",
    "        #if count > 10:\n",
    "        #    break\n",
    "        \n",
    "        hgvs = hgvs_from_record(parsed)\n",
    "        print hgvs\n",
    "        \n",
    "        if hgvs is None:\n",
    "            continue\n",
    "        \n",
    "        all_hgvs.add(hgvs)\n",
    "        if len(all_hgvs) <200:\n",
    "            continue\n",
    "        \n",
    "        # Accessing VEP!\n",
    "        vep = vep_parse_post(list(all_hgvs))\n",
    "        \n",
    "        # Saving\n",
    "        f.write(json.dumps(vep) + '\\n')\n",
    "        \n",
    "        #Emptying set\n",
    "        all_hgvs = set()\n",
    "        time.sleep(10) # Be nice\n",
    "        \n",
    "        #vep = vep_parse(hgvs)\n",
    "        #vep = vep_parse_post([hgvs])\n",
    "        #print vep\n",
    "        #assert False\n",
    "\n",
    "    # Last batch\n",
    "    vep = vep_parse_post(list(all_hgvs))    \n",
    "    f.write(json.dumps(vep) + '\\n')\n",
    "\n",
    "        \n",
    "    f.close()\n",
    "    print ('Created: vep_results.json')\n",
    "        \n",
    "vep_run()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/user/mutationinfo/localpython/lib/python2.7/site-packages/psycopg2-2.7.4-py2.7-linux-x86_64.egg/psycopg2/__init__.py:144: UserWarning: The psycopg2 wheel package will be renamed from release 2.8; in order to keep installing from binary please use \"pip install psycopg2-binary\" instead. For details see: <http://initd.org/psycopg/docs/install.html#binary-install-from-pypi>.\n",
      "  \"\"\")\n"
     ]
    }
   ],
   "source": [
    "import hgvs.parser\n",
    "import hgvs.dataproviders.uta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "hdp = hgvs.dataproviders.uta.connect()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "hgvs_parser = hgvs.parser.Parser().parse_hgvs_variant"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    " p = hgvs_parser('RHAG:c.236G>A')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "import hgvs.variantmapper\n",
    "import hgvs.assemblymapper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "am = hgvs.assemblymapper.AssemblyMapper(hdp, alt_aln_method=u'genewise',)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "ename": "HGVSDataNotAvailableError",
     "evalue": "No alignments for RHAG in GRCh38 using genewise",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mHGVSDataNotAvailableError\u001b[0m                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-19-6f5829b3df9d>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mam\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mc_to_g\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mp\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/home/user/mutationinfo/localpython/lib/python2.7/site-packages/hgvs-1.1.1-py2.7.egg/hgvs/assemblymapper.pyc\u001b[0m in \u001b[0;36mc_to_g\u001b[0;34m(self, var_c)\u001b[0m\n\u001b[1;32m     92\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     93\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mc_to_g\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvar_c\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 94\u001b[0;31m         \u001b[0malt_ac\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_alt_ac_for_tx_ac\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvar_c\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mac\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     95\u001b[0m         \u001b[0mvar_out\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msuper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mAssemblyMapper\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mc_to_g\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvar_c\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0malt_ac\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0malt_aln_method\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0malt_aln_method\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     96\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_maybe_normalize\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvar_out\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/home/user/mutationinfo/localpython/lib/python2.7/site-packages/hgvs-1.1.1-py2.7.egg/hgvs/assemblymapper.pyc\u001b[0m in \u001b[0;36m_alt_ac_for_tx_ac\u001b[0;34m(self, tx_ac)\u001b[0m\n\u001b[1;32m    138\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0malt_acs\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    139\u001b[0m             raise HGVSDataNotAvailableError(\"No alignments for {tx_ac} in {an} using {am}\".format(\n\u001b[0;32m--> 140\u001b[0;31m                 tx_ac=tx_ac, an=self.assembly_name, am=self.alt_aln_method))\n\u001b[0m\u001b[1;32m    141\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    142\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0malt_acs\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mHGVSDataNotAvailableError\u001b[0m: No alignments for RHAG in GRCh38 using genewise"
     ]
    }
   ],
   "source": [
    "am.c_to_g(p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'variantmapper' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-12-ce4f3ed11a66>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mvariantmapper\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'variantmapper' is not defined"
     ]
    }
   ],
   "source": [
    "variantmapper"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# VEP but for hg19\n",
    "Run with ```/home/user/mutationinfo/localpython/bin/python run_4.py```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Overwriting run_4.py\n"
     ]
    }
   ],
   "source": [
    "%%writefile run_4.py\n",
    "import re\n",
    "import json\n",
    "import time\n",
    "import requests\n",
    "\n",
    "def load_parsed_abstracts():\n",
    "    with open('parsed_abstracts.json') as f:\n",
    "        for l in f:\n",
    "            data = json.loads(l)\n",
    "            yield data\n",
    "            \n",
    "def hgvs_from_record(parsed):\n",
    "        \n",
    "    abstract = parsed['abstract']\n",
    "\n",
    "    if 'dog' in abstract:\n",
    "        #stats_2['nothumnan'] = stats_2.get('nothumnan', 0) + 1\n",
    "        return None\n",
    "\n",
    "\n",
    "    #print parsed\n",
    "    hgvs = parsed['hgvs']\n",
    "\n",
    "    #Get hgvs reference\n",
    "    s = re.search(r'([ACGT]+)>', hgvs)\n",
    "    if not s:\n",
    "        return None\n",
    "        #assert False\n",
    "\n",
    "    hgvs_reference = s.group(1)\n",
    "\n",
    "    #If it starts with a number assuming coding position\n",
    "    if re.search(r'^[\\d]+', hgvs, re.UNICODE):\n",
    "        hgvs = 'c.' + hgvs\n",
    "\n",
    "\n",
    "    #Get everything except the reference\n",
    "    s = re.search(r'^([^ACGT]+)[ACGT]+>', hgvs)\n",
    "    if not s:\n",
    "        return None\n",
    "        #continue\n",
    "        #assert False\n",
    "    except_reference = s.group(1)\n",
    "\n",
    "\n",
    "    gene = parsed['gene']\n",
    "    if not gene:\n",
    "        return None\n",
    "        #assert False\n",
    "\n",
    "    transcripts = parsed['transcripts']\n",
    "    if transcripts:\n",
    "        pass\n",
    "        #assert False\n",
    "\n",
    "    hgvs = gene + ':' + hgvs\n",
    "    hgvs_no_reference = gene + ':' + except_reference\n",
    "    #transvar_cmd = transvar_cmd_p.format(hgvs=hgvs)\n",
    "    return hgvs\n",
    "\n",
    "def vep_parse(hgvs):\n",
    "    headers={ \"Content-Type\" : \"application/json\"}\n",
    "    vep_url = \"https://grch37.rest.ensembl.org/vep/human/hgvs/{var}?\"\n",
    "\n",
    "    url = vep_url.format(var=hgvs)\n",
    "    responce = requests.get(url, headers=headers)\n",
    "    if responce.ok:\n",
    "        return responce.json()\n",
    "    else:\n",
    "        print ('Error:')\n",
    "        print (responce.text)\n",
    "\n",
    "def vep_parse_post(hgvs):\n",
    "\n",
    " \n",
    "    server = \"https://grch37.rest.ensembl.org\"\n",
    "    ext = \"/vep/human/hgvs\"\n",
    "    headers={ \"Content-Type\" : \"application/json\", \"Accept\" : \"application/json\"}\n",
    "    \n",
    "    assert type(hgvs) is list\n",
    "    #data = '[' + ','.join([rept(x) for x in hgvs]) + ']'\n",
    "    data = json.dumps({\"hgvs_notations\": hgvs})\n",
    "    print data\n",
    "    \n",
    "    r = requests.post(server+ext, headers=headers, data=data)\n",
    " \n",
    "    if not r.ok:\n",
    "        print ('ERROR:')\n",
    "        print (r.text)\n",
    "        return None\n",
    " \n",
    "    decoded = r.json()\n",
    "    return decoded\n",
    "\n",
    "def vep_run():\n",
    "    \n",
    "    count = 0\n",
    "    \n",
    "    f = open('vep_results_hg19.json', 'w')\n",
    "    \n",
    "    all_hgvs = set()\n",
    "    for parsed in load_parsed_abstracts():\n",
    "        count += 1\n",
    "        \n",
    "        print ('count:', count)\n",
    "        \n",
    "        if count < 1:\n",
    "            continue\n",
    "            \n",
    "        #if count > 10:\n",
    "        #    break\n",
    "        \n",
    "        hgvs = hgvs_from_record(parsed)\n",
    "        print hgvs\n",
    "        \n",
    "        if hgvs is None:\n",
    "            continue\n",
    "        \n",
    "        all_hgvs.add(hgvs)\n",
    "        if len(all_hgvs) <200:\n",
    "            continue\n",
    "        \n",
    "        # Accessing VEP!\n",
    "        vep = vep_parse_post(list(all_hgvs))\n",
    "        \n",
    "        #print vep\n",
    "        #assert False\n",
    "        \n",
    "        # Saving\n",
    "        f.write(json.dumps(vep) + '\\n')\n",
    "        f.flush()\n",
    "        \n",
    "        #Emptying set\n",
    "        all_hgvs = set()\n",
    "        time.sleep(10) # Be nice\n",
    "        \n",
    "        #vep = vep_parse(hgvs)\n",
    "        #vep = vep_parse_post([hgvs])\n",
    "        #print vep\n",
    "        #assert False\n",
    "\n",
    "    # Last batch\n",
    "    vep = vep_parse_post(list(all_hgvs))    \n",
    "    f.write(json.dumps(vep) + '\\n')\n",
    "\n",
    "        \n",
    "    f.close()\n",
    "    print ('Created: vep_results_hg19.json')\n",
    "        \n",
    "vep_run()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create stats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Lines: 158\n",
      "Record: 15075\n"
     ]
    }
   ],
   "source": [
    "import json\n",
    "\n",
    "def vep_info(vep):\n",
    "    # Take only the essential info\n",
    "    \n",
    "    #print json.dumps(vep, indent=4)\n",
    "    #transcripts = [x['transcript_id'] for x in vep['transcript_consequences']]\n",
    "    transcripts = [x['gene_id'] for x in vep['transcript_consequences']]\n",
    "    transcripts = set(transcripts)\n",
    "    #print transcripts\n",
    "    #assert False\n",
    "    \n",
    "    ret = {\n",
    "        'chr': vep['seq_region_name'],\n",
    "        'start' : vep['start'],\n",
    "        'end' : vep['end'],\n",
    "        'allele_string' : vep['allele_string'].split('/'),\n",
    "        'strand' : vep['strand'],\n",
    "        'tr': len(transcripts)\n",
    "    }\n",
    "   \n",
    "    return ret\n",
    "\n",
    "def load_vep_results(fn):    \n",
    "    ret = {}\n",
    "    \n",
    "    with open(fn) as f:\n",
    "        c = 0\n",
    "        for l in f:\n",
    "            c += 1\n",
    "            data = json.loads(l)\n",
    "\n",
    "            #print data[0]\n",
    "            #assert False\n",
    "            \n",
    "            for variant in data:\n",
    "                #if not 'BRCA1' in variant['id']:\n",
    "                #    continue\n",
    "                ret[variant['id']] = vep_info(variant)\n",
    "            \n",
    "    print 'Lines:', c\n",
    "    print 'Record:', len(ret)\n",
    "    return ret\n",
    "\n",
    "#vep = load_vep_results('vep_results.json') # Replace this with 'vep_results_hg19.json'\n",
    "vep = load_vep_results('vep_results_hg19.json') # Replace this with 'vep_results_hg19.json'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'allele_string': [u'T', u'G'],\n",
       " 'chr': u'12',\n",
       " 'end': 56118130,\n",
       " 'start': 56118130,\n",
       " 'strand': 1}"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vep['RDH5:c.758T>G']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def load_transvar_results(fn):\n",
    "    '''\n",
    "    transvar_output.txt\n",
    "    '''\n",
    "\n",
    "    ret = {}\n",
    "    with open(fn) as f:\n",
    "        c=0\n",
    "        for l in f:\n",
    "            c+=1\n",
    "            ls = l.replace('\\n', '').split('\\t')\n",
    "            if c == 1:\n",
    "                header = ls\n",
    "                header_str = l\n",
    "                continue\n",
    "                \n",
    "            ret[ls[0]] = '\\t'.join(header[1:]) + '\\n' + '\\t'.join(ls[1:]).replace('|||\\t', '\\n')\n",
    "            #break\n",
    "        print ('Records:', c)\n",
    "    return ret"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('Records:', 42115)\n"
     ]
    }
   ],
   "source": [
    "transvar=load_transvar_results('transvar_output.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['RHAG|c.236']"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "transvar.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "transcript\tgene\tstrand\tcoordinates(gDNA/cDNA/protein)\tregion\tinfo\n",
      "CCDS10690 (protein_coding)\tPHKG2\t+\tchr16:g.30764875C/c.553C/.\tinside_[cds_in_exon_5]\tsource=CCDS\t\n",
      "CCDS54002 (protein_coding)\tPHKG2\t+\tchr16:g.30764875C/c.553C/.\tinside_[cds_in_exon_5]\tsource=CCDS\n"
     ]
    }
   ],
   "source": [
    "print transvar['PHKG2|c.553']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FLNA:c.1621G>T\n",
      "G\n",
      "FLNA|c.1621\n",
      "[u'NM_001456.3']\n",
      "NM_001456.3:c.1621G>T\n",
      "NM_001456.3:c.1621G>T\n",
      "153593574\n",
      "L2HGDH:c.178G>A\n",
      "G\n",
      "L2HGDH|c.178\n",
      "[u'NM_024884.2']\n",
      "NM_024884.2:c.178G>A\n",
      "NM_024884.2:c.178G>A\n",
      "50769698\n",
      "AP4M1:c.991C>T\n",
      "C\n",
      "AP4M1|c.991\n",
      "[u'NM_006594.4']\n",
      "NM_006594.4:c.991C>T\n",
      "NM_006594.4:c.991C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "HSF4:c.190A>G\n",
      "A\n",
      "HSF4|c.190\n",
      "[u'NM_001040667.2']\n",
      "NM_001040667.2:c.190A>G\n",
      "NM_001040667.2:c.190A>G\n",
      "67199491\n",
      "DMC1:c.106G>A\n",
      "G\n",
      "DMC1|c.106\n",
      "[u'NM_007068.3']\n",
      "NM_007068.3:c.106G>A\n",
      "NM_007068.3:c.106G>A\n",
      "38962732\n",
      "WHRN:c.1814T>C\n",
      "T\n",
      "WHRN|c.1814\n",
      "[u'NM_138691']\n",
      "NM_138691:c.1814T>C\n",
      "NM_138691:c.1814T>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "NOG:c.611G>A\n",
      "G\n",
      "NOG|c.611\n",
      "[u'NM_005450.4']\n",
      "NM_005450.4:c.611G>A\n",
      "NM_005450.4:c.611G>A\n",
      "54672195\n",
      "NR2F1:c.286A>G\n",
      "A\n",
      "NR2F1|c.286\n",
      "[u'NM_005654']\n",
      "NM_005654:c.286A>G\n",
      "NM_005654:c.286A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "NBAS:c.3596G>A\n",
      "G\n",
      "NBAS|c.3596\n",
      "[u'NM_015909.3']\n",
      "NM_015909.3:c.3596G>A\n",
      "NM_015909.3:c.3596G>A\n",
      "15514839\n",
      "NBAS:c.209+1G>A\n",
      "G\n",
      "NBAS|c.209+1\n",
      "[u'NM_015909.3']\n",
      "NM_015909.3:c.209+1G>A\n",
      "NM_015909.3:c.209+1G>A\n",
      "15696906\n",
      "GNB1:c.230G>T\n",
      "G\n",
      "GNB1|c.230\n",
      "[u'NM_001282539.1']\n",
      "NM_001282539.1:c.230G>T\n",
      "NM_001282539.1:c.230G>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "FLCN:c.649C>T\n",
      "C\n",
      "FLCN|c.649\n",
      "[u'NM_144997']\n",
      "NM_144997:c.649C>T\n",
      "NM_144997:c.649C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "ETFDH:c.250G>A\n",
      "G\n",
      "ETFDH|c.250\n",
      "[u'NM_004453']\n",
      "NM_004453:c.250G>A\n",
      "NM_004453:c.250G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "ETFDH:c.920C>G\n",
      "C\n",
      "ETFDH|c.920\n",
      "[u'NM_004453']\n",
      "NM_004453:c.920C>G\n",
      "NM_004453:c.920C>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "EDNRB:c.1280C>T\n",
      "C\n",
      "EDNRB|c.1280\n",
      "[u'NM_031857']\n",
      "NM_031857:c.1280C>T\n",
      "NM_031857:c.1280C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "EDNRB:c.1425C>G\n",
      "C\n",
      "EDNRB|c.1425\n",
      "[u'NM_031857']\n",
      "NM_031857:c.1425C>G\n",
      "NM_031857:c.1425C>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "IGF2:c.101G>A\n",
      "G\n",
      "IGF2|c.101\n",
      "[u'NM_000612']\n",
      "NM_000612:c.101G>A\n",
      "NM_000612:c.101G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "SMOC1:c.367T>C\n",
      "T\n",
      "SMOC1|c.367\n",
      "[u'NM_001034852.2']\n",
      "NM_001034852.2:c.367T>C\n",
      "NM_001034852.2:c.367T>C\n",
      "70420238\n",
      "ZDHHC9:c.286C>T\n",
      "C\n",
      "ZDHHC9|c.286\n",
      "[u'NM_001008222.2']\n",
      "NM_001008222.2:c.286C>T\n",
      "NM_001008222.2:c.286C>T\n",
      "128962999\n",
      "TMEM67:c.725A>G\n",
      "A\n",
      "TMEM67|c.725\n",
      "[u'NM_153704.5']\n",
      "NM_153704.5:c.725A>G\n",
      "NM_153704.5:c.725A>G\n",
      "94792831\n",
      "ERCC8:c.1122G>C\n",
      "G\n",
      "ERCC8|c.1122\n",
      "[u'NM_000082']\n",
      "NM_000082:c.1122G>C\n",
      "NM_000082:c.1122G>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PAX6:c.1183+1G>A\n",
      "G\n",
      "PAX6|c.1183+1\n",
      "[u'NG_008679.1']\n",
      "NG_008679.1:c.1183+1G>A\n",
      "NG_008679.1:c.1183+1G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "VARS2:c.3650G>A\n",
      "G\n",
      "VARS2|c.3650\n",
      "[u'NM_006295']\n",
      "NM_006295:c.3650G>A\n",
      "NM_006295:c.3650G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "VARS2:c.3650G>A\n",
      "G\n",
      "VARS2|c.3650\n",
      "[u'NM_006295']\n",
      "NM_006295:c.3650G>A\n",
      "NM_006295:c.3650G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "BBS2:c.1677C>A\n",
      "C\n",
      "BBS2|c.1677\n",
      "[u'NM_024685.3']\n",
      "NM_024685.3:c.1677C>A\n",
      "NM_024685.3:c.1677C>A\n",
      "76740088\n",
      "BBS2:c.1974T>G\n",
      "T\n",
      "BBS2|c.1974\n",
      "[u'NM_024685.3']\n",
      "NM_024685.3:c.1974T>G\n",
      "NM_024685.3:c.1974T>G\n",
      "76739791\n",
      "{'end': 56519587, 'tr': 1, 'allele_string': [u'T', u'G'], 'start': 56519587, 'chr': u'16', 'strand': -1}\n",
      "We report a 10-year-old girl with Bardet-Biedl syndrome caused by a novel mutation in the Bardet-Biedl syndrome 10 (BBS10) gene. She had multiple malformations, including a dysmorphic face, postaxial polydactyly, polycystic kidney and amblyopia. She presented with typical BBS features, including intellectual disability with emotional outbursts and mild obesity. Whole-exome sequencing identified compound heterozygous mutations with NM_024685.3:c.1677C>A [p.(Tyr559*)] and c.1974T>G [p.(Tyr658*)]. To our knowledge, the latter mutation has never been reported previously.\n",
      "KCNQ2:c.601C>T\n",
      "C\n",
      "KCNQ2|c.601\n",
      "[u'NM_172107.2']\n",
      "NM_172107.2:c.601C>T\n",
      "NM_172107.2:c.601C>T\n",
      "62076101\n",
      "MAT1A:c.1081G>T\n",
      "G\n",
      "MAT1A|c.1081\n",
      "[u'NM_000429.2']\n",
      "NM_000429.2:c.1081G>T\n",
      "NM_000429.2:c.1081G>T\n",
      "82034280\n",
      "CD151:c.351+2T>C\n",
      "T\n",
      "CD151|c.351+2\n",
      "[u'NM_139029']\n",
      "NM_139029:c.351+2T>C\n",
      "NM_139029:c.351+2T>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "AR:c.2450-42G>A\n",
      "G\n",
      "AR|c.2450-42\n",
      "[u'NM_000044.4']\n",
      "NM_000044.4:c.2450-42G>A\n",
      "NM_000044.4:c.2450-42G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "B3GAT3:c.667G>A\n",
      "G\n",
      "B3GAT3|c.667\n",
      "[u'NM_012200']\n",
      "NM_012200:c.667G>A\n",
      "NM_012200:c.667G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "COQ8B:c.521A>G\n",
      "A\n",
      "COQ8B|c.521\n",
      "[u'NM_024876.3']\n",
      "NM_024876.3:c.521A>G\n",
      "NM_024876.3:c.521A>G\n",
      "41211056\n",
      "NADK2:g.36241900A>G\n",
      "A\n",
      "NADK2|g.36241900\n",
      "[u'NM_001085411.1']\n",
      "NM_001085411.1:g.36241900A>G\n",
      "NM_001085411.1:g.36241900A>G\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "ZZZ3:c.122G>T\n",
      "G\n",
      "ZZZ3|c.122\n",
      "[u'NM_170707.3']\n",
      "NM_170707.3:c.122G>T\n",
      "NM_170707.3:c.122G>T\n",
      "156084831\n",
      "PEX10:c.530T>G\n",
      "T\n",
      "PEX10|c.530\n",
      "[u'NM_153818.1']\n",
      "NM_153818.1:c.530T>G\n",
      "NM_153818.1:c.530T>G\n",
      "2339961\n",
      "FBXO7:c.1408G>T\n",
      "G\n",
      "FBXO7|c.1408\n",
      "[u'NM_012179.3']\n",
      "NM_012179.3:c.1408G>T\n",
      "NM_012179.3:c.1408G>T\n",
      "32894356\n",
      "FBXO7:c.152A>G\n",
      "A\n",
      "FBXO7|c.152\n",
      "[u'NM_012179.3']\n",
      "NM_012179.3:c.152A>G\n",
      "NM_012179.3:c.152A>G\n",
      "32874997\n",
      "PRF1:c.1120T>G\n",
      "T\n",
      "PRF1|c.1120\n",
      "[u'NM_001083116']\n",
      "NM_001083116:c.1120T>G\n",
      "NM_001083116:c.1120T>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PAH:c.2A>G\n",
      "A\n",
      "PAH|c.2\n",
      "[u'NM_001456']\n",
      "NM_001456:c.2A>G\n",
      "NM_001456:c.2A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PAH:c.2A>G\n",
      "A\n",
      "PAH|c.2\n",
      "[u'NM_001456']\n",
      "NM_001456:c.2A>G\n",
      "NM_001456:c.2A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "STAG3:c.1573+5G>A\n",
      "G\n",
      "STAG3|c.1573+5\n",
      "[u'NM_012447.2']\n",
      "NM_012447.2:c.1573+5G>A\n",
      "NM_012447.2:c.1573+5G>A\n",
      "99796995\n",
      "FH:c.1G>C\n",
      "G\n",
      "FH|c.1\n",
      "[u'NM_000143.3']\n",
      "NM_000143.3:c.1G>C\n",
      "NM_000143.3:c.1G>C\n",
      "241683022\n",
      "FH:c.1G>C\n",
      "G\n",
      "FH|c.1\n",
      "[u'NM_000143.3']\n",
      "NM_000143.3:c.1G>C\n",
      "NM_000143.3:c.1G>C\n",
      "241683022\n",
      "MRE11:c.1090C>T\n",
      "C\n",
      "MRE11|c.1090\n",
      "[u'NM_005590']\n",
      "NM_005590:c.1090C>T\n",
      "NM_005590:c.1090C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "SLC12A3:c.1276A>T\n",
      "A\n",
      "SLC12A3|c.1276\n",
      "[u'NM_000339.2']\n",
      "NM_000339.2:c.1276A>T\n",
      "NM_000339.2:c.1276A>T\n",
      "56913080\n",
      "PRKD1:c.979G>A\n",
      "G\n",
      "PRKD1|c.979\n",
      "[u'NM_000744']\n",
      "NM_000744:c.979G>A\n",
      "NM_000744:c.979G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PITX2:c.272G>A\n",
      "G\n",
      "PITX2|c.272\n",
      "[u'NM_153427.2']\n",
      "NM_153427.2:c.272G>A\n",
      "NM_153427.2:c.272G>A\n",
      "111539825\n",
      "{'end': 111542459, 'tr': 1, 'allele_string': [u'G', u'A'], 'start': 111542459, 'chr': u'4', 'strand': -1}\n",
      "Axenfeld-Rieger syndrome is a rare autosomal dominant condition. Anomalies include anterior segment dysgenesis of the eye, dental anomalies, maxillary hypoplasia, periumbilical anomalies, and congenital heart defects. We report a patient with Peters anomaly, dysmorphic features, congenital heart defect, umbilical hernia, short stature, and developmental delay. Diagnostic sequencing of 23 genes known to be causally related to the condition was performed on the patient, parents, and maternal grandparents. A variant of uncertain significance in PITX2 was identified. The mother had the same mutation and the father did not. The mother had decreased vision, congenitally missing teeth, and required jaw surgery as a child. Her asymptomatic parents elected to be tested and were negative for the mutation. The mutation, NM_153427.2:c.272G>A (p.Arg91Gln), is predicted to be damaging by PolyPhen-2 (score of 0.997), identified as a missense mutation with an allele frequency of 1.648e-05 by the Exome Aggregation Consortium, and has been reported in ClinVar once, by the laboratory that analyzed our patient's sample. Due to the in silico predictions and the results of family studies, it is suggested that this variant can be classified as pathogenic according to the American College of Medical Genetics and Genomics 2015 rule Pathogenic(iii)(b), specifically rules PS2, PM2, PM5, PP1, and PP3.\n",
      "CAMTA2:c.6G>A\n",
      "G\n",
      "CAMTA2|c.6\n",
      "[u'NM_001171166.1']\n",
      "NM_001171166.1:c.6G>A\n",
      "NM_001171166.1:c.6G>A\n",
      "4890919\n",
      "FBP1:c.825+1G>A\n",
      "G\n",
      "FBP1|c.825+1\n",
      "[u'NM_000507.3']\n",
      "NM_000507.3:c.825+1G>A\n",
      "NM_000507.3:c.825+1G>A\n",
      "97367738\n",
      "FBP1:c.355G>A\n",
      "G\n",
      "FBP1|c.355\n",
      "[u'NM_000507.3']\n",
      "NM_000507.3:c.355G>A\n",
      "NM_000507.3:c.355G>A\n",
      "97380121\n",
      "FBP1:c.490G>A\n",
      "G\n",
      "FBP1|c.490\n",
      "[u'NM_000507.3']\n",
      "NM_000507.3:c.490G>A\n",
      "NM_000507.3:c.490G>A\n",
      "97372280\n",
      "POLD1:c.3209T>A\n",
      "T\n",
      "POLD1|c.3209\n",
      "[u'NM_002691.3']\n",
      "NM_002691.3:c.3209T>A\n",
      "NM_002691.3:c.3209T>A\n",
      "50920517\n",
      "HP:c.190+1G>C\n",
      "G\n",
      "HP|c.190+1\n",
      "[u'NM_001126102.1']\n",
      "NM_001126102.1:c.190+1G>C\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NM_001126102.1:c.190+1G>C\n",
      "72090531\n",
      "NEDD4L:c.2617G>A\n",
      "G\n",
      "NEDD4L|c.2617\n",
      "[u'NM_015277']\n",
      "NM_015277:c.2617G>A\n",
      "NM_015277:c.2617G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "TGFBR2:c.1381T>C\n",
      "T\n",
      "TGFBR2|c.1381\n",
      "[u'NM_003242.5']\n",
      "NM_003242.5:c.1381T>C\n",
      "NM_003242.5:c.1381T>C\n",
      "30715723\n",
      "{'end': 30715648, 'tr': 1, 'allele_string': [u'T', u'C'], 'start': 30715648, 'chr': u'3', 'strand': 1}\n",
      "Loeys-Dietz syndrome (LDS) is an autosomal dominant connective tissue disorder characterized mainly by cardiovascular, craniofacial and skeletal features. We report on a patient with LDS, whose prenatal examination was compatible with the diagnosis of arthrogryposis multiplex congenita. Neonatal assessment showed craniofacial and cardiovascular findings suggestive of LDS whose diagnosis was confirmed by the detection of a novel mutation (HGVN: NM_003242.5 (TGFBR2): c.1381T > C (p.(Cys461Arg))) in the TGFBR2 gene. Few prenatal and neonatal cases of LDS have been reported in the literature. We reviewed all cases reported to date with perinatal onset to delineate the clinical manifestations that allow us to prompt diagnosis of this syndrome at an early stage to prevent fatal cardiovascular complications. Furthermore we discuss the multidisciplinary follow up required in these patients.\n",
      "ATRX:c.7156C>T\n",
      "C\n",
      "ATRX|c.7156\n",
      "[u'NM_000489.4']\n",
      "NM_000489.4:c.7156C>T\n",
      "NM_000489.4:c.7156C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "FAP:c.325G>C\n",
      "G\n",
      "FAP|c.325\n",
      "[u'NM_000371.3']\n",
      "NM_000371.3:c.325G>C\n",
      "NM_000371.3:c.325G>C\n",
      "29175207\n",
      "FAP:c.325G>C\n",
      "G\n",
      "FAP|c.325\n",
      "[u'NM_000371.3']\n",
      "NM_000371.3:c.325G>C\n",
      "NM_000371.3:c.325G>C\n",
      "29175207\n",
      "DHX15:c.664C>G\n",
      "C\n",
      "DHX15|c.664\n",
      "[u'NM_001358']\n",
      "NM_001358:c.664C>G\n",
      "NM_001358:c.664C>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "ALDH18A1:c.377G>A\n",
      "G\n",
      "ALDH18A1|c.377\n",
      "[u'NM_002860.3']\n",
      "NM_002860.3:c.377G>A\n",
      "NM_002860.3:c.377G>A\n",
      "97397120\n",
      "EMC1:c.19564510C>T\n",
      "C\n",
      "EMC1|c.19564510\n",
      "[u'NM_015047.2']\n",
      "NM_015047.2:c.19564510C>T\n",
      "NM_015047.2:c.19564510C>T\n",
      "Could not parse: [\"NC_000001.10:g.-15731G>A\"]\n",
      "None\n",
      "EMC1:c.1212+1G>A\n",
      "G\n",
      "EMC1|c.1212+1\n",
      "[u'NM_015047.2']\n",
      "NM_015047.2:c.1212+1G>A\n",
      "NM_015047.2:c.1212+1G>A\n",
      "19564510\n",
      "POFUT1:c.485C>T\n",
      "C\n",
      "POFUT1|c.485\n",
      "[u'NM_015352.1']\n",
      "NM_015352.1:c.485C>T\n",
      "NM_015352.1:c.485C>T\n",
      "30804467\n",
      "FGFR1:c.1115G>A\n",
      "G\n",
      "FGFR1|c.1115\n",
      "[u'NM_001174067']\n",
      "NM_001174067:c.1115G>A\n",
      "NM_001174067:c.1115G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PSEN1:c.1164C>G\n",
      "C\n",
      "PSEN1|c.1164\n",
      "[u'NM_000021.3']\n",
      "NM_000021.3:c.1164C>G\n",
      "NM_000021.3:c.1164C>G\n",
      "73683868\n",
      "COL4A5:c.2245-40A>G\n",
      "A\n",
      "COL4A5|c.2245-40\n",
      "[u'NM_000091.4']\n",
      "NM_000091.4:c.2245-40A>G\n",
      "NM_000091.4:c.2245-40A>G\n",
      "228145137\n",
      "{'end': 107849932, 'tr': 1, 'allele_string': [u'A', u'G'], 'start': 107849932, 'chr': u'X', 'strand': 1}\n",
      "Alport syndrome (AS) is an inherited progressive renal disease caused by mutations in COL4A3, COL4A4, and COL4A5 genes. Despite simultaneous screening of these genes being widely available, mutation detection still remains incomplete in a non-marginal portion of patients. Here, we applied whole-exome sequencing (WES) in 3 Italian families negative after candidate-gene analyses. In Family 1, we identified a novel heterozygous intronic variant (c.2245-40A>G) -outside the conventionally screened candidate region for diagnosis- potentially disrupting COL4A5 exon29 splicing. Using a minigene-based approach in HEK293 cells we demonstrated that this variant abolishes exon29 branch site, causing exon skipping. Moreover, skewed X-inactivation of the c.2245-40A>G allele correlated with disease severity in heterozygous females. In Family 2, WES highlighted a novel COL4A5 hemizygous missense mutation (p.Gly491Asp), which segregates with the phenotype and impacts on a highly-conserved residue. Finally, in Family 3, we detected a homozygous 24-bp in-frame deletion in COL4A3 exon1 (NM_000091.4:c.30_53del:p.Val11_Leu18del or c.40_63del24:p.Leu14_Leu21del), which is ambiguously annotated in databases, although it corresponds to a recurrent AS mutation. Functional analyses showed that this deletion disrupts COL4A3 signal peptide, possibly altering protein secretion. In conclusion, WES -together with functional studies- was fundamental for molecular diagnosis in 3 AS families, highlighting pathogenic variants that escaped previous screenings.\n",
      "COL4A5:c.2245-40A>G\n",
      "A\n",
      "COL4A5|c.2245-40\n",
      "[u'NM_000091.4']\n",
      "NM_000091.4:c.2245-40A>G\n",
      "NM_000091.4:c.2245-40A>G\n",
      "228145137\n",
      "{'end': 107849932, 'tr': 1, 'allele_string': [u'A', u'G'], 'start': 107849932, 'chr': u'X', 'strand': 1}\n",
      "Alport syndrome (AS) is an inherited progressive renal disease caused by mutations in COL4A3, COL4A4, and COL4A5 genes. Despite simultaneous screening of these genes being widely available, mutation detection still remains incomplete in a non-marginal portion of patients. Here, we applied whole-exome sequencing (WES) in 3 Italian families negative after candidate-gene analyses. In Family 1, we identified a novel heterozygous intronic variant (c.2245-40A>G) -outside the conventionally screened candidate region for diagnosis- potentially disrupting COL4A5 exon29 splicing. Using a minigene-based approach in HEK293 cells we demonstrated that this variant abolishes exon29 branch site, causing exon skipping. Moreover, skewed X-inactivation of the c.2245-40A>G allele correlated with disease severity in heterozygous females. In Family 2, WES highlighted a novel COL4A5 hemizygous missense mutation (p.Gly491Asp), which segregates with the phenotype and impacts on a highly-conserved residue. Finally, in Family 3, we detected a homozygous 24-bp in-frame deletion in COL4A3 exon1 (NM_000091.4:c.30_53del:p.Val11_Leu18del or c.40_63del24:p.Leu14_Leu21del), which is ambiguously annotated in databases, although it corresponds to a recurrent AS mutation. Functional analyses showed that this deletion disrupts COL4A3 signal peptide, possibly altering protein secretion. In conclusion, WES -together with functional studies- was fundamental for molecular diagnosis in 3 AS families, highlighting pathogenic variants that escaped previous screenings.\n",
      "NR3C2:c.1781G>A\n",
      "G\n",
      "NR3C2|c.1781\n",
      "[u'NM_021625.4']\n",
      "NM_021625.4:c.1781G>A\n",
      "NM_021625.4:c.1781G>A\n",
      "110230500\n",
      "CYREN:c.481C>T\n",
      "C\n",
      "CYREN|c.481\n",
      "[u'NM_000551.3']\n",
      "NM_000551.3:c.481C>T\n",
      "NM_000551.3:c.481C>T\n",
      "10191488\n",
      "ALG9:c.860A>G\n",
      "A\n",
      "ALG9|c.860\n",
      "[u'NM_1234567890']\n",
      "NM_1234567890:c.860A>G\n",
      "NM_1234567890:c.860A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "JAG1:c.205C>T\n",
      "C\n",
      "JAG1|c.205\n",
      "[u'NM_032193.3']\n",
      "NM_032193.3:c.205C>T\n",
      "NM_032193.3:c.205C>T\n",
      "65487856\n",
      "DNMT3A:c.2323-2A>T\n",
      "A\n",
      "DNMT3A|c.2323-2\n",
      "[u'NM_022552.4']\n",
      "NM_022552.4:c.2323-2A>T\n",
      "NM_022552.4:c.2323-2A>T\n",
      "25462086\n",
      "AARS2:c.578T>G\n",
      "T\n",
      "AARS2|c.578\n",
      "[u'NM_020745.3']\n",
      "NM_020745.3:c.578T>G\n",
      "NM_020745.3:c.578T>G\n",
      "44279130\n",
      "AARS2:c.595C>T\n",
      "C\n",
      "AARS2|c.595\n",
      "[u'NM_020745.3']\n",
      "NM_020745.3:c.595C>T\n",
      "NM_020745.3:c.595C>T\n",
      "44278885\n",
      "CUL4B:c.2698G>T\n",
      "G\n",
      "CUL4B|c.2698\n",
      "[u'NM_003588.3']\n",
      "NM_003588.3:c.2698G>T\n",
      "NM_003588.3:c.2698G>T\n",
      "119660660\n",
      "OCLN:c.1037+5G>C\n",
      "G\n",
      "OCLN|c.1037+5\n",
      "[u'NM_001205254']\n",
      "NM_001205254:c.1037+5G>C\n",
      "NM_001205254:c.1037+5G>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "OCLN:c.1169C>G\n",
      "C\n",
      "OCLN|c.1169\n",
      "[u'NM_001205254']\n",
      "NM_001205254:c.1169C>G\n",
      "NM_001205254:c.1169C>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PGAP3:c.851A>G\n",
      "A\n",
      "PGAP3|c.851\n",
      "[u'NM_033419.3']\n",
      "NM_033419.3:c.851A>G\n",
      "NM_033419.3:c.851A>G\n",
      "37829352\n",
      "KAT6B:c.4943C>G\n",
      "C\n",
      "KAT6B|c.4943\n",
      "[u'NM_001256468.1']\n",
      "NM_001256468.1:c.4943C>G\n",
      "NM_001256468.1:c.4943C>G\n",
      "76790074\n",
      "{'end': 76789525, 'tr': 2, 'allele_string': [u'C', u'G'], 'start': 76789525, 'chr': u'10', 'strand': 1}\n",
      "Say-Barber/Biesecker/Young-Simpson syndrome (SBBYSS; OMIM 603736) is a rare syndrome with multiple congenital anomalies/malformations. The clinical diagnosis is usually based on a phenotype with a mask-like face and severe blepharophimosis and ptosis as well as other distinctive facial traits. We present a girl with dysmorphic features, an atrial septal defect, and developmental delay. Previous genetic testing (array-CGH, 22q11 deletion, PTPN11 and MLL2 mutation analysis) gave normal results. We performed whole-exome sequencing (WES) and identified a heterozygous nonsense mutation in the KAT6B gene, NM_001256468.1: c.4943C>G (p.S1648*). The mutation led to a premature stop codon and occurred de novo. KAT6B sequence variants have previously been identified in patients with SBBYSS, and the phenotype of the girl is similar to other patients diagnosed with SBBYSS. This case report provides additional evidence for the correlation between the KAT6B mutation and SBBYSS. If a patient is suspected of having a blepharophimosis syndrome or SBBYSS, we recommend sequencing the KAT6B gene. This is a further example showing that WES can assist diagnosis.\n",
      "GNB4:c.169A>G\n",
      "A\n",
      "GNB4|c.169\n",
      "[u'NM_021629.3']\n",
      "NM_021629.3:c.169A>G\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NM_021629.3:c.169A>G\n",
      "179137221\n",
      "COL5A2:c.682G>A\n",
      "G\n",
      "COL5A2|c.682\n",
      "[u'NM_000393.3']\n",
      "NM_000393.3:c.682G>A\n",
      "NM_000393.3:c.682G>A\n",
      "189951460\n",
      "TRAPPC11:c.1893+3A>G\n",
      "A\n",
      "TRAPPC11|c.1893+3\n",
      "[u'NM_021942.5']\n",
      "NM_021942.5:c.1893+3A>G\n",
      "NM_021942.5:c.1893+3A>G\n",
      "184607904\n",
      "TRAPPC11:c.904A>G\n",
      "A\n",
      "TRAPPC11|c.904\n",
      "[u'NM_021942.5']\n",
      "NM_021942.5:c.904A>G\n",
      "NM_021942.5:c.904A>G\n",
      "184600578\n",
      "PIEZO2:c.1384C>T\n",
      "C\n",
      "PIEZO2|c.1384\n",
      "[u'NM_022068']\n",
      "NM_022068:c.1384C>T\n",
      "NM_022068:c.1384C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "KRT3:c.174C>T\n",
      "C\n",
      "KRT3|c.174\n",
      "[u'NM_000280']\n",
      "NM_000280:c.174C>T\n",
      "NM_000280:c.174C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "SLC26A4:c.4567A>G\n",
      "A\n",
      "SLC26A4|c.4567\n",
      "[u'NM_022124']\n",
      "NM_022124:c.4567A>G\n",
      "NM_022124:c.4567A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "SLC26A4:c.5101G>A\n",
      "G\n",
      "SLC26A4|c.5101\n",
      "[u'NM_022124']\n",
      "NM_022124:c.5101G>A\n",
      "NM_022124:c.5101G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "SLC26A4:c.4567A>G\n",
      "A\n",
      "SLC26A4|c.4567\n",
      "[u'NM_022124']\n",
      "NM_022124:c.4567A>G\n",
      "NM_022124:c.4567A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "SLC26A4:c.5101G>A\n",
      "G\n",
      "SLC26A4|c.5101\n",
      "[u'NM_022124']\n",
      "NM_022124:c.5101G>A\n",
      "NM_022124:c.5101G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "SLC26A4:c.4567A>G\n",
      "A\n",
      "SLC26A4|c.4567\n",
      "[u'NM_022124']\n",
      "NM_022124:c.4567A>G\n",
      "NM_022124:c.4567A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "SLC26A4:c.5101G>A\n",
      "G\n",
      "SLC26A4|c.5101\n",
      "[u'NM_022124']\n",
      "NM_022124:c.5101G>A\n",
      "NM_022124:c.5101G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "ZZZ3:c.323C>T\n",
      "C\n",
      "ZZZ3|c.323\n",
      "[u'NM_004614.4']\n",
      "NM_004614.4:c.323C>T\n",
      "NM_004614.4:c.323C>T\n",
      "66565335\n",
      "IDS:c.85C>T\n",
      "C\n",
      "IDS|c.85\n",
      "[u'NM_000202.7']\n",
      "NM_000202.7:c.85C>T\n",
      "NM_000202.7:c.85C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PSMB8:c.212C>T\n",
      "C\n",
      "PSMB8|c.212\n",
      "[u'NM_004159.4']\n",
      "NM_004159.4:c.212C>T\n",
      "NM_004159.4:c.212C>T\n",
      "32810790\n",
      "IDS:c.85C>T\n",
      "C\n",
      "IDS|c.85\n",
      "[u'NM_000202.7']\n",
      "NM_000202.7:c.85C>T\n",
      "NM_000202.7:c.85C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "IDS:c.182C>T\n",
      "C\n",
      "IDS|c.182\n",
      "[u'NM_000202.7']\n",
      "NM_000202.7:c.182C>T\n",
      "NM_000202.7:c.182C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CTU2:c.873G>A\n",
      "G\n",
      "CTU2|c.873\n",
      "[u'NM_001012762.1']\n",
      "NM_001012762.1:c.873G>A\n",
      "NM_001012762.1:c.873G>A\n",
      "88779855\n",
      "DIAPH1:c.3637C>T\n",
      "C\n",
      "DIAPH1|c.3637\n",
      "[u'NM_005219']\n",
      "NM_005219:c.3637C>T\n",
      "NM_005219:c.3637C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "MEFV:c.2082G>A\n",
      "G\n",
      "MEFV|c.2082\n",
      "[u'NM_000243.2']\n",
      "NM_000243.2:c.2082G>A\n",
      "NM_000243.2:c.2082G>A\n",
      "3293405\n",
      "AHSG:c.950G>A\n",
      "G\n",
      "AHSG|c.950\n",
      "[u'NM_001622']\n",
      "NM_001622:c.950G>A\n",
      "NM_001622:c.950G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "HSPE1:c.217C>T\n",
      "C\n",
      "HSPE1|c.217\n",
      "[u'NM_002157.2']\n",
      "NM_002157.2:c.217C>T\n",
      "NM_002157.2:c.217C>T\n",
      "198367811\n",
      "MTO1:c.1510C>T\n",
      "C\n",
      "MTO1|c.1510\n",
      "[u'NM_001123226']\n",
      "NM_001123226:c.1510C>T\n",
      "NM_001123226:c.1510C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "MTO1:c.1669G>A\n",
      "G\n",
      "MTO1|c.1669\n",
      "[u'NM_001123226']\n",
      "NM_001123226:c.1669G>A\n",
      "NM_001123226:c.1669G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "ZZZ3:c.484T>C\n",
      "T\n",
      "ZZZ3|c.484\n",
      "[u'NM_006941.3']\n",
      "NM_006941.3:c.484T>C\n",
      "NM_006941.3:c.484T>C\n",
      "38374087\n",
      "SOX10:c.163A>T\n",
      "A\n",
      "SOX10|c.163\n",
      "[u'NM_006941.3']\n",
      "NM_006941.3:c.163A>T\n",
      "NM_006941.3:c.163A>T\n",
      "38379629\n",
      "GC:c.2494G>A\n",
      "G\n",
      "GC|c.2494\n",
      "[u'NM_004360.4']\n",
      "NM_004360.4:c.2494G>A\n",
      "NM_004360.4:c.2494G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "BSCL2:c.985C>T\n",
      "C\n",
      "BSCL2|c.985\n",
      "[u'NM_001122955.3']\n",
      "NM_001122955.3:c.985C>T\n",
      "NM_001122955.3:c.985C>T\n",
      "62458772\n",
      "NARFL:c.482G>T\n",
      "G\n",
      "NARFL|c.482\n",
      "[u'NM_022493']\n",
      "NM_022493:c.482G>T\n",
      "NM_022493:c.482G>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "RNF216:c.2061G>A\n",
      "G\n",
      "RNF216|c.2061\n",
      "[u'NM_207111.3']\n",
      "NM_207111.3:c.2061G>A\n",
      "NM_207111.3:c.2061G>A\n",
      "5751392\n",
      "TNNT3:c.1956150G>A\n",
      "G\n",
      "TNNT3|c.1956150\n",
      "[u'NM_006757.3']\n",
      "NM_006757.3:c.1956150G>A\n",
      "NM_006757.3:c.1956150G>A\n",
      "3915095\n",
      "TNNT3:c.681+1G>A\n",
      "G\n",
      "TNNT3|c.681+1\n",
      "[u'NM_006757.3']\n",
      "NM_006757.3:c.681+1G>A\n",
      "NM_006757.3:c.681+1G>A\n",
      "1956150\n",
      "SLC26A4:c.2118C>A\n",
      "C\n",
      "SLC26A4|c.2118\n",
      "[u'NP_000432.1']\n",
      "NP_000432.1:c.2118C>A\n",
      "NP_000432.1:c.2118C>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "SLC26A4:c.919-2A>G\n",
      "A\n",
      "SLC26A4|c.919-2\n",
      "[u'NP_000432.1']\n",
      "NP_000432.1:c.919-2A>G\n",
      "NP_000432.1:c.919-2A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "SLC26A4:c.919-2A>G\n",
      "A\n",
      "SLC26A4|c.919-2\n",
      "[u'NP_000432.1']\n",
      "NP_000432.1:c.919-2A>G\n",
      "NP_000432.1:c.919-2A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "SLC26A4:c.2118C>A\n",
      "C\n",
      "SLC26A4|c.2118\n",
      "[u'NP_000432.1']\n",
      "NP_000432.1:c.2118C>A\n",
      "NP_000432.1:c.2118C>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "SLC26A4:c.2118C>A\n",
      "C\n",
      "SLC26A4|c.2118\n",
      "[u'NP_000432.1']\n",
      "NP_000432.1:c.2118C>A\n",
      "NP_000432.1:c.2118C>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "SLC26A4:c.2118C>A\n",
      "C\n",
      "SLC26A4|c.2118\n",
      "[u'NP_000432.1']\n",
      "NP_000432.1:c.2118C>A\n",
      "NP_000432.1:c.2118C>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "SLC26A4:c.2118C>A\n",
      "C\n",
      "SLC26A4|c.2118\n",
      "[u'NP_000432.1']\n",
      "NP_000432.1:c.2118C>A\n",
      "NP_000432.1:c.2118C>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "SLC26A4:c.2118C>A\n",
      "C\n",
      "SLC26A4|c.2118\n",
      "[u'NP_000432.1']\n",
      "NP_000432.1:c.2118C>A\n",
      "NP_000432.1:c.2118C>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "SLC26A4:c.919-2A>G\n",
      "A\n",
      "SLC26A4|c.919-2\n",
      "[u'NP_000432.1']\n",
      "NP_000432.1:c.919-2A>G\n",
      "NP_000432.1:c.919-2A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "ABCA3:c.604G>A\n",
      "G\n",
      "ABCA3|c.604\n",
      "[u'NM_001089']\n",
      "NM_001089:c.604G>A\n",
      "NM_001089:c.604G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "WDR45:c.830+2T>C\n",
      "T\n",
      "WDR45|c.830+2\n",
      "[u'NM_007075.3']\n",
      "NM_007075.3:c.830+2T>C\n",
      "NM_007075.3:c.830+2T>C\n",
      "48933021\n",
      "TTN:c.88880A>C\n",
      "A\n",
      "TTN|c.88880\n",
      "[u'NM_001256850.1']\n",
      "NM_001256850.1:c.88880A>C\n",
      "NM_001256850.1:c.88880A>C\n",
      "179412550\n",
      "FOXF1:c.86544406C>A\n",
      "C\n",
      "FOXF1|c.86544406\n",
      "[u'NM_001451']\n",
      "NM_001451:c.86544406C>A\n",
      "NM_001451:c.86544406C>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "IL33:c.487-1G>C\n",
      "G\n",
      "IL33|c.487-1\n",
      "[u'NM_001199640']\n",
      "NM_001199640:c.487-1G>C\n",
      "NM_001199640:c.487-1G>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "TCTN1:c.342-2A>G\n",
      "A\n",
      "TCTN1|c.342-2\n",
      "[u'NM_001082538.2']\n",
      "NM_001082538.2:c.342-2A>G\n",
      "NM_001082538.2:c.342-2A>G\n",
      "111064165\n",
      "COL1A2:c.2123G>A\n",
      "G\n",
      "COL1A2|c.2123\n",
      "[u'NM_000089.3']\n",
      "NM_000089.3:c.2123G>A\n",
      "NM_000089.3:c.2123G>A\n",
      "94049588\n",
      "MEGF10:c.2981-2A>G\n",
      "A\n",
      "MEGF10|c.2981-2\n",
      "[u'NM_032446.2']\n",
      "NM_032446.2:c.2981-2A>G\n",
      "NM_032446.2:c.2981-2A>G\n",
      "126790256\n",
      "HYLS1:c.900A>C\n",
      "A\n",
      "HYLS1|c.900\n",
      "[u'NM_145014.2']\n",
      "NM_145014.2:c.900A>C\n",
      "NM_145014.2:c.900A>C\n",
      "125770163\n",
      "CPS1:c.603G>C\n",
      "G\n",
      "CPS1|c.603\n",
      "[u'NM_153006.2']\n",
      "NM_153006.2:c.603G>C\n",
      "NM_153006.2:c.603G>C\n",
      "42083181\n",
      "CPS1:c.603G>C\n",
      "G\n",
      "CPS1|c.603\n",
      "[u'NM_153006.2']\n",
      "NM_153006.2:c.603G>C\n",
      "NM_153006.2:c.603G>C\n",
      "42083181\n",
      "ATP8A2:c.1287G>T\n",
      "G\n",
      "ATP8A2|c.1287\n",
      "[u'NM_016529']\n",
      "NM_016529:c.1287G>T\n",
      "NM_016529:c.1287G>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "ATP8A2:c.1630G>C\n",
      "G\n",
      "ATP8A2|c.1630\n",
      "[u'NM_016529']\n",
      "NM_016529:c.1630G>C\n",
      "NM_016529:c.1630G>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "ATP8A2:c.1873C>T\n",
      "C\n",
      "ATP8A2|c.1873\n",
      "[u'NM_016529']\n",
      "NM_016529:c.1873C>T\n",
      "NM_016529:c.1873C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "ATP8A2:c.1128C>G\n",
      "C\n",
      "ATP8A2|c.1128\n",
      "[u'NM_016529']\n",
      "NM_016529:c.1128C>G\n",
      "NM_016529:c.1128C>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "MSX1:c.4861985T>G\n",
      "T\n",
      "MSX1|c.4861985\n",
      "[u'NM_002448']\n",
      "NM_002448:c.4861985T>G\n",
      "NM_002448:c.4861985T>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "MSX1:c.359T>G\n",
      "T\n",
      "MSX1|c.359\n",
      "[u'NM_002448']\n",
      "NM_002448:c.359T>G\n",
      "NM_002448:c.359T>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "RARS:c.5A>G\n",
      "A\n",
      "RARS|c.5\n",
      "[u'NM_002887.3']\n",
      "NM_002887.3:c.5A>G\n",
      "NM_002887.3:c.5A>G\n",
      "167913508\n",
      "RARS:c.1367C>T\n",
      "C\n",
      "RARS|c.1367\n",
      "[u'NM_002887.3']\n",
      "NM_002887.3:c.1367C>T\n",
      "NM_002887.3:c.1367C>T\n",
      "167937606\n",
      "SCN5A:c.5284G>T\n",
      "G\n",
      "SCN5A|c.5284\n",
      "[u'NM_000335']\n",
      "NM_000335:c.5284G>T\n",
      "NM_000335:c.5284G>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "BAZ1A:c.4043T>G\n",
      "T\n",
      "BAZ1A|c.4043\n",
      "[u'NM_182648.2']\n",
      "NM_182648.2:c.4043T>G\n",
      "NM_182648.2:c.4043T>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PTEN:c.138C>A\n",
      "C\n",
      "PTEN|c.138\n",
      "[u'NM_000314']\n",
      "NM_000314:c.138C>A\n",
      "NM_000314:c.138C>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "MTHFR:c.1530A>G\n",
      "A\n",
      "MTHFR|c.1530\n",
      "[u'NM_005957.4']\n",
      "NM_005957.4:c.1530A>G\n",
      "NM_005957.4:c.1530A>G\n",
      "11853964\n",
      "ZZZ3:c.1129C>T\n",
      "C\n",
      "ZZZ3|c.1129\n",
      "[u'NM_005957.4']\n",
      "NM_005957.4:c.1129C>T\n",
      "NM_005957.4:c.1129C>T\n",
      "11854823\n",
      "BBS2:c.4428+3A>G\n",
      "A\n",
      "BBS2|c.4428+3\n",
      "[u'NM_015662.2']\n",
      "NM_015662.2:c.4428+3A>G\n",
      "NM_015662.2:c.4428+3A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PIK3R1:c.1425+1G>A\n",
      "G\n",
      "PIK3R1|c.1425+1\n",
      "[u'NM_181523.2']\n",
      "NM_181523.2:c.1425+1G>A\n",
      "NM_181523.2:c.1425+1G>A\n",
      "67589663\n",
      "TP63:c.1063G>C\n",
      "G\n",
      "TP63|c.1063\n",
      "[u'NM_003722.4']\n",
      "NM_003722.4:c.1063G>C\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NM_003722.4:c.1063G>C\n",
      "189586439\n",
      "GNB4:c.659T>C\n",
      "T\n",
      "GNB4|c.659\n",
      "[u'NM_021629.3']\n",
      "NM_021629.3:c.659T>C\n",
      "NM_021629.3:c.659T>C\n",
      "179131240\n",
      "COL11A1:c.3168+5G>A\n",
      "G\n",
      "COL11A1|c.3168+5\n",
      "[u'NM_001854.3']\n",
      "NM_001854.3:c.3168+5G>A\n",
      "NM_001854.3:c.3168+5G>A\n",
      "103427417\n",
      "KIF1A:c.2909G>A\n",
      "G\n",
      "KIF1A|c.2909\n",
      "[u'NM_001244008.1']\n",
      "NM_001244008.1:c.2909G>A\n",
      "NM_001244008.1:c.2909G>A\n",
      "241689914\n",
      "GPI:c.1640G>A\n",
      "G\n",
      "GPI|c.1640\n",
      "[u'NM_001127178.1']\n",
      "NM_001127178.1:c.1640G>A\n",
      "NM_001127178.1:c.1640G>A\n",
      "517273\n",
      "KNL1:c.6673-19T>A\n",
      "T\n",
      "KNL1|c.6673-19\n",
      "[u'NM_170589.4']\n",
      "NM_170589.4:c.6673-19T>A\n",
      "NM_170589.4:c.6673-19T>A\n",
      "40949534\n",
      "C1QA:c.622C>T\n",
      "C\n",
      "C1QA|c.622\n",
      "[u'NM_015991']\n",
      "NM_015991:c.622C>T\n",
      "NM_015991:c.622C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "C1QA:c.164-2A>C\n",
      "A\n",
      "C1QA|c.164-2\n",
      "[u'NM_015991']\n",
      "NM_015991:c.164-2A>C\n",
      "NM_015991:c.164-2A>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PIK3CA:c.335T>A\n",
      "T\n",
      "PIK3CA|c.335\n",
      "[u'NM_006218.2']\n",
      "NM_006218.2:c.335T>A\n",
      "NM_006218.2:c.335T>A\n",
      "178916948\n",
      "PLP1:c.454-9T>G\n",
      "T\n",
      "PLP1|c.454-9\n",
      "[u'NM_000533.4']\n",
      "NM_000533.4:c.454-9T>G\n",
      "NM_000533.4:c.454-9T>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "ALX3:c.661-1G>C\n",
      "G\n",
      "ALX3|c.661-1\n",
      "[u'NM_006982.2']\n",
      "NM_006982.2:c.661-1G>C\n",
      "NM_006982.2:c.661-1G>C\n",
      "85694932\n",
      "{'end': 110604120, 'tr': 3, 'allele_string': [u'G', u'C'], 'start': 110604120, 'chr': u'1', 'strand': -1}\n",
      "Frontonasal dysplasia (FND) is a heterogeneous group of disorders characterized by hypertelorism, telecanthus, broad nasal root, wide prominent nasal bridge, short and wide nasal ridge, broad columella and smooth philtrum. To date one X-linked and three autosomal recessive forms of FND have been reported in different ethnic groups. We sought to identify the gene responsible for FND in a consanguineous Pakistani family segregating the disorder in autosomal recessive pattern. Genome-wide homozygosity mapping using 250KNsp array revealed five homozygous regions in the selected affected individuals. Exome sequencing found a novel splice acceptor site variant (c.661-1G>C: NM_006982.2) in ALX1. Sanger sequencing confirmed the correct segregation of the pathogenic variant in the whole family. Our study concludes that the splice site variant identified in the ALX1 gene causes mild form of FND.\n",
      "HEXA:c.1510C>T\n",
      "C\n",
      "HEXA|c.1510\n",
      "[u'NM_000520.4']\n",
      "NM_000520.4:c.1510C>T\n",
      "NM_000520.4:c.1510C>T\n",
      "72637803\n",
      "CDK5RAP2:c.4114C>T\n",
      "C\n",
      "CDK5RAP2|c.4114\n",
      "[u'XM_011518861.1']\n",
      "XM_011518861.1:c.4114C>T\n",
      "XM_011518861.1:c.4114C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "LGI1:c.1013T>C\n",
      "T\n",
      "LGI1|c.1013\n",
      "[u'NM_005097.2']\n",
      "NM_005097.2:c.1013T>C\n",
      "NM_005097.2:c.1013T>C\n",
      "95556899\n",
      "MYBPC1:c.556G>A\n",
      "G\n",
      "MYBPC1|c.556\n",
      "[u'NM_002465']\n",
      "NM_002465:c.556G>A\n",
      "NM_002465:c.556G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PAX6:c.2T>A\n",
      "T\n",
      "PAX6|c.2\n",
      "[u'NG_008679.1']\n",
      "NG_008679.1:c.2T>A\n",
      "NG_008679.1:c.2T>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PAX6:c.2T>A\n",
      "T\n",
      "PAX6|c.2\n",
      "[u'NG_008679.1']\n",
      "NG_008679.1:c.2T>A\n",
      "NG_008679.1:c.2T>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PTH1R:c.611T>A\n",
      "T\n",
      "PTH1R|c.611\n",
      "[u'NM_000316']\n",
      "NM_000316:c.611T>A\n",
      "NM_000316:c.611T>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "REEP1:c.7GTAATAT>AC\n",
      "GTAATAT\n",
      "REEP1|c.7\n",
      "[u'NM_022912']\n",
      "NM_022912:c.7GTAATAT>AC\n",
      "NM_022912:c.7GTAATAT>AC\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "FXN:g.41022153G>A\n",
      "G\n",
      "FXN|g.41022153\n",
      "[u'NM_002875']\n",
      "NM_002875:g.41022153G>A\n",
      "NM_002875:g.41022153G>A\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "GJA1:c.124G>C\n",
      "G\n",
      "GJA1|c.124\n",
      "[u'NM_000165.3']\n",
      "NM_000165.3:c.124G>C\n",
      "NM_000165.3:c.124G>C\n",
      "121768117\n",
      "C9orf72:g.120962C>T\n",
      "C\n",
      "C9orf72|g.120962\n",
      "[u'NG_007398.1']\n",
      "NG_007398.1:g.120962C>T\n",
      "NG_007398.1:g.120962C>T\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "MAPT:c.750C>A\n",
      "C\n",
      "MAPT|c.750\n",
      "[u'NG_007398.1']\n",
      "NG_007398.1:c.750C>A\n",
      "NG_007398.1:c.750C>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CHCHD10:c.63C>T\n",
      "C\n",
      "CHCHD10|c.63\n",
      "[u'NG_007398.1']\n",
      "NG_007398.1:c.63C>T\n",
      "NG_007398.1:c.63C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "TEAD1:c.71G>A\n",
      "G\n",
      "TEAD1|c.71\n",
      "[u'NG_007398.1']\n",
      "NG_007398.1:c.71G>A\n",
      "NG_007398.1:c.71G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PHKG2:c.1963G>A\n",
      "G\n",
      "PHKG2|c.1963\n",
      "[u'NM_000292.2']\n",
      "NM_000292.2:c.1963G>A\n",
      "NM_000292.2:c.1963G>A\n",
      "18938150\n",
      "{'end': 30769160, 'tr': 3, 'allele_string': [u'G', u'A'], 'start': 30769160, 'chr': u'16', 'strand': 1}\n",
      "Glucogenosis type IX is caused by pathogenic variants of the PHKA2 gene. Herein, we report a patient with clinical symptoms compatible with Glycogen Storage Disease type IXa. PYGL, PHKA1, PHKA2, PHKB and PHKG2 genes were analyzed by Next Generation Sequencing (NGS). We identified the previously undescribed hemizygous missense variant NM_000292.2(PHKA2):c.1963G > A, p.(Glu655Lys) in PHKA2 exon 18. In silico analyses showed two possible pathogenic consequences: it affects a highly conserved amino acid and disrupts the exon 18 canonical splice donor site. The variant was found as a \"de novo\" event.\n",
      "mtdna:c.323C>T\n",
      "C\n",
      "mtdna|c.323\n",
      "[u'NM_004614.4']\n",
      "NM_004614.4:c.323C>T\n",
      "NM_004614.4:c.323C>T\n",
      "66565335\n",
      "OXSM:c.2750-1G>T\n",
      "G\n",
      "OXSM|c.2750-1\n",
      "[u'NM_024757.4']\n",
      "NM_024757.4:c.2750-1G>T\n",
      "NM_024757.4:c.2750-1G>T\n",
      "140705949\n",
      "mtdna:m.4087A>G\n",
      "A\n",
      "mtdna|m.4087\n",
      "[u'NC_012920.1']\n",
      "NC_012920.1:m.4087A>G\n",
      "NC_012920.1:m.4087A>G\n",
      "Could not parse: [\"NC_012920.1(TRNF_v001):n.*3440A>G\", \"NC_012920.1(RNR1_v001):n.*2486A>G\", \"NC_012920.1(TRNV_v001):n.*2417A>G\", \"NC_012920.1(RNR2_v001):n.*858A>G\", \"NC_012920.1(TRNL1_v001):n.*783A>G\", \"NC_012920.1(ND1_v001):c.781A>G\", \"NC_012920.1(TRNI_v001):n.-176A>G\", \"NC_012920.1(TRNQ_v001):n.*242T>C\", \"NC_012920.1(TRNM_v001):n.-315A>G\", \"NC_012920.1(ND2_v001):c.-383A>G\", \"NC_012920.1(TRNW_v001):n.-1425A>G\", \"NC_012920.1(TRNA_v001):n.*1500T>C\", \"NC_012920.1(TRNN_v001):n.*1570T>C\", \"NC_012920.1(TRNC_v001):n.*1674T>C\", \"NC_012920.1(TRNY_v001):n.*1739T>C\", \"NC_012920.1(COX1_v001):c.-1817A>G\", \"NC_012920.1(TRNS1_v001):n.*3359T>C\", \"NC_012920.1(TRND_v001):n.-3431A>G\", \"NC_012920.1(COX2_v001):c.-3499A>G\", \"NC_012920.1(TRNK_v001):n.-4208A>G\", \"NC_012920.1(ATP8_v001):c.-4279A>G\", \"NC_012920.1(ATP6_v001):c.-4440A>G\"]\n",
      "None\n",
      "mtdna:m.4087A>G\n",
      "A\n",
      "mtdna|m.4087\n",
      "[u'NC_012920.1']\n",
      "NC_012920.1:m.4087A>G\n",
      "NC_012920.1:m.4087A>G\n",
      "Could not parse: [\"NC_012920.1(TRNF_v001):n.*3440A>G\", \"NC_012920.1(RNR1_v001):n.*2486A>G\", \"NC_012920.1(TRNV_v001):n.*2417A>G\", \"NC_012920.1(RNR2_v001):n.*858A>G\", \"NC_012920.1(TRNL1_v001):n.*783A>G\", \"NC_012920.1(ND1_v001):c.781A>G\", \"NC_012920.1(TRNI_v001):n.-176A>G\", \"NC_012920.1(TRNQ_v001):n.*242T>C\", \"NC_012920.1(TRNM_v001):n.-315A>G\", \"NC_012920.1(ND2_v001):c.-383A>G\", \"NC_012920.1(TRNW_v001):n.-1425A>G\", \"NC_012920.1(TRNA_v001):n.*1500T>C\", \"NC_012920.1(TRNN_v001):n.*1570T>C\", \"NC_012920.1(TRNC_v001):n.*1674T>C\", \"NC_012920.1(TRNY_v001):n.*1739T>C\", \"NC_012920.1(COX1_v001):c.-1817A>G\", \"NC_012920.1(TRNS1_v001):n.*3359T>C\", \"NC_012920.1(TRND_v001):n.-3431A>G\", \"NC_012920.1(COX2_v001):c.-3499A>G\", \"NC_012920.1(TRNK_v001):n.-4208A>G\", \"NC_012920.1(ATP8_v001):c.-4279A>G\", \"NC_012920.1(ATP6_v001):c.-4440A>G\"]\n",
      "None\n",
      "mtdna:m.4087A>G\n",
      "A\n",
      "mtdna|m.4087\n",
      "[u'NC_012920.1']\n",
      "NC_012920.1:m.4087A>G\n",
      "NC_012920.1:m.4087A>G\n",
      "Could not parse: [\"NC_012920.1(TRNF_v001):n.*3440A>G\", \"NC_012920.1(RNR1_v001):n.*2486A>G\", \"NC_012920.1(TRNV_v001):n.*2417A>G\", \"NC_012920.1(RNR2_v001):n.*858A>G\", \"NC_012920.1(TRNL1_v001):n.*783A>G\", \"NC_012920.1(ND1_v001):c.781A>G\", \"NC_012920.1(TRNI_v001):n.-176A>G\", \"NC_012920.1(TRNQ_v001):n.*242T>C\", \"NC_012920.1(TRNM_v001):n.-315A>G\", \"NC_012920.1(ND2_v001):c.-383A>G\", \"NC_012920.1(TRNW_v001):n.-1425A>G\", \"NC_012920.1(TRNA_v001):n.*1500T>C\", \"NC_012920.1(TRNN_v001):n.*1570T>C\", \"NC_012920.1(TRNC_v001):n.*1674T>C\", \"NC_012920.1(TRNY_v001):n.*1739T>C\", \"NC_012920.1(COX1_v001):c.-1817A>G\", \"NC_012920.1(TRNS1_v001):n.*3359T>C\", \"NC_012920.1(TRND_v001):n.-3431A>G\", \"NC_012920.1(COX2_v001):c.-3499A>G\", \"NC_012920.1(TRNK_v001):n.-4208A>G\", \"NC_012920.1(ATP8_v001):c.-4279A>G\", \"NC_012920.1(ATP6_v001):c.-4440A>G\"]\n",
      "None\n",
      "MT-ND1:c.781A>G\n",
      "A\n",
      "MT-ND1|c.781\n",
      "[u'NC_012920.1']\n",
      "NC_012920.1:c.781A>G\n",
      "NC_012920.1:c.781A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "POLG:c.925C>T\n",
      "C\n",
      "POLG|c.925\n",
      "[u'NM_002693.2']\n",
      "NM_002693.2:c.925C>T\n",
      "NM_002693.2:c.925C>T\n",
      "89872272\n",
      "POLG:c.2244G>T\n",
      "G\n",
      "POLG|c.2244\n",
      "[u'NM_002693.2']\n",
      "NM_002693.2:c.2244G>T\n",
      "NM_002693.2:c.2244G>T\n",
      "89866656\n",
      "NDUFAF6:c.96046914T>C\n",
      "T\n",
      "NDUFAF6|c.96046914\n",
      "[u'NM_152416.3']\n",
      "NM_152416.3:c.96046914T>C\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NM_152416.3:c.96046914T>C\n",
      "192116077\n",
      "NDUFAF6:c.298-768T>C\n",
      "T\n",
      "NDUFAF6|c.298-768\n",
      "[u'NM_152416.3']\n",
      "NM_152416.3:c.298-768T>C\n",
      "NM_152416.3:c.298-768T>C\n",
      "96046914\n",
      "NDUFAF6:c.96046951A>G\n",
      "A\n",
      "NDUFAF6|c.96046951\n",
      "[u'NM_152416.3']\n",
      "NM_152416.3:c.96046951A>G\n",
      "NM_152416.3:c.96046951A>G\n",
      "192116114\n",
      "NDUFAF6:c.298-731A>G\n",
      "A\n",
      "NDUFAF6|c.298-731\n",
      "[u'NM_152416.3']\n",
      "NM_152416.3:c.298-731A>G\n",
      "NM_152416.3:c.298-731A>G\n",
      "96046951\n",
      "ELANE:c.401A>C\n",
      "A\n",
      "ELANE|c.401\n",
      "[u'NM_001972.2']\n",
      "NM_001972.2:c.401A>C\n",
      "NM_001972.2:c.401A>C\n",
      "855598\n",
      "AVP:c.61T>C\n",
      "T\n",
      "AVP|c.61\n",
      "[u'NM_000490.4']\n",
      "NM_000490.4:c.61T>C\n",
      "NM_000490.4:c.61T>C\n",
      "3065260\n",
      "GJB2:c.109G>A\n",
      "G\n",
      "GJB2|c.109\n",
      "[u'NM_004004']\n",
      "NM_004004:c.109G>A\n",
      "NM_004004:c.109G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "GJB2:c.109G>A\n",
      "G\n",
      "GJB2|c.109\n",
      "[u'NM_004004']\n",
      "NM_004004:c.109G>A\n",
      "NM_004004:c.109G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "GJB2:c.109G>A\n",
      "G\n",
      "GJB2|c.109\n",
      "[u'NM_004004']\n",
      "NM_004004:c.109G>A\n",
      "NM_004004:c.109G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "TMEM231:c.784G>A\n",
      "G\n",
      "TMEM231|c.784\n",
      "[u'NM_001077416.2']\n",
      "NM_001077416.2:c.784G>A\n",
      "NM_001077416.2:c.784G>A\n",
      "75576539\n",
      "TMEM231:c.406T>G\n",
      "T\n",
      "TMEM231|c.406\n",
      "[u'NM_001077416.2']\n",
      "NM_001077416.2:c.406T>G\n",
      "NM_001077416.2:c.406T>G\n",
      "75589764\n",
      "TMTC1:c.460C>G\n",
      "C\n",
      "TMTC1|c.460\n",
      "[u'NM_001200.3']\n",
      "NM_001200.3:c.460C>G\n",
      "NM_001200.3:c.460C>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "TMTC1:c.584G>T\n",
      "G\n",
      "TMTC1|c.584\n",
      "[u'NM_001200.3']\n",
      "NM_001200.3:c.584G>T\n",
      "NM_001200.3:c.584G>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "NBAS:c.680A>C\n",
      "A\n",
      "NBAS|c.680\n",
      "[u'NM_015909.3']\n",
      "NM_015909.3:c.680A>C\n",
      "NM_015909.3:c.680A>C\n",
      "15674733\n",
      "NBAS:c.1749G>A\n",
      "G\n",
      "NBAS|c.1749\n",
      "[u'NM_015909.3']\n",
      "NM_015909.3:c.1749G>A\n",
      "NM_015909.3:c.1749G>A\n",
      "15608634\n",
      "COL5A1:c.4610G>T\n",
      "G\n",
      "COL5A1|c.4610\n",
      "[u'NM_000093.4']\n",
      "NM_000093.4:c.4610G>T\n",
      "NM_000093.4:c.4610G>T\n",
      "137714845\n",
      "GATA3:c.856A>G\n",
      "A\n",
      "GATA3|c.856\n",
      "[u'NM_001002295.1']\n",
      "NM_001002295.1:c.856A>G\n",
      "NM_001002295.1:c.856A>G\n",
      "8106033\n",
      "GATA3:c.1017C>G\n",
      "C\n",
      "GATA3|c.1017\n",
      "[u'NM_001002295.1']\n",
      "NM_001002295.1:c.1017C>G\n",
      "NM_001002295.1:c.1017C>G\n",
      "8111528\n",
      "GATA3:c.896G>A\n",
      "G\n",
      "GATA3|c.896\n",
      "[u'NM_001002295.1']\n",
      "NM_001002295.1:c.896G>A\n",
      "NM_001002295.1:c.896G>A\n",
      "8106073\n",
      "GATA3:c.1051-2A>G\n",
      "A\n",
      "GATA3|c.1051-2\n",
      "[u'NM_001002295.1']\n",
      "NM_001002295.1:c.1051-2A>G\n",
      "NM_001002295.1:c.1051-2A>G\n",
      "8115700\n",
      "XYLT2:c.2188C>T\n",
      "C\n",
      "XYLT2|c.2188\n",
      "[u'NM_022167.3']\n",
      "NM_022167.3:c.2188C>T\n",
      "NM_022167.3:c.2188C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "mtdna:g.2885C>A\n",
      "C\n",
      "mtdna|g.2885\n",
      "[u'AC_000176']\n",
      "AC_000176:g.2885C>A\n",
      "AC_000176:g.2885C>A\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "FLT4:c.3704C>G\n",
      "C\n",
      "FLT4|c.3704\n",
      "[u'NM_182925.4']\n",
      "NM_182925.4:c.3704C>G\n",
      "NM_182925.4:c.3704C>G\n",
      "180037008\n",
      "mtdna:c.1442T>C\n",
      "T\n",
      "mtdna|c.1442\n",
      "[u'NM_012160.4']\n",
      "NM_012160.4:c.1442T>C\n",
      "NM_012160.4:c.1442T>C\n",
      "99323551\n",
      "mtdna:c.1442T>C\n",
      "T\n",
      "mtdna|c.1442\n",
      "[u'NM_012160.4']\n",
      "NM_012160.4:c.1442T>C\n",
      "NM_012160.4:c.1442T>C\n",
      "99323551\n",
      "mtdna:c.1442T>C\n",
      "T\n",
      "mtdna|c.1442\n",
      "[u'NM_012160.4']\n",
      "NM_012160.4:c.1442T>C\n",
      "NM_012160.4:c.1442T>C\n",
      "99323551\n",
      "CHKB:c.810T>A\n",
      "T\n",
      "CHKB|c.810\n",
      "[u'NM_005198.4']\n",
      "NM_005198.4:c.810T>A\n",
      "NM_005198.4:c.810T>A\n",
      "51018627\n",
      "GCNT2:c.85C>T\n",
      "C\n",
      "GCNT2|c.85\n",
      "[u'NM_000202.7']\n",
      "NM_000202.7:c.85C>T\n",
      "NM_000202.7:c.85C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "ATP7B:c.2335T>G\n",
      "T\n",
      "ATP7B|c.2335\n",
      "[u'NG_008806.1']\n",
      "NG_008806.1:c.2335T>G\n",
      "NG_008806.1:c.2335T>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "ATP7B:g.58164T>G\n",
      "T\n",
      "ATP7B|g.58164\n",
      "[u'NG_008806.1']\n",
      "NG_008806.1:g.58164T>G\n",
      "NG_008806.1:g.58164T>G\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "ATP7B:c.2335T>G\n",
      "T\n",
      "ATP7B|c.2335\n",
      "[u'NG_008806.1']\n",
      "NG_008806.1:c.2335T>G\n",
      "NG_008806.1:c.2335T>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "ATP7B:g.47964C>T\n",
      "C\n",
      "ATP7B|g.47964\n",
      "[u'NG_008806.1']\n",
      "NG_008806.1:g.47964C>T\n",
      "NG_008806.1:g.47964C>T\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "ATP7B:g.51482G>A\n",
      "G\n",
      "ATP7B|g.51482\n",
      "[u'NG_008806.1']\n",
      "NG_008806.1:g.51482G>A\n",
      "NG_008806.1:g.51482G>A\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "ATP7B:g.54622A>G\n",
      "A\n",
      "ATP7B|g.54622\n",
      "[u'NG_008806.1']\n",
      "NG_008806.1:g.54622A>G\n",
      "NG_008806.1:g.54622A>G\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "ATP7B:g.56255G>A\n",
      "G\n",
      "ATP7B|g.56255\n",
      "[u'NG_008806.1']\n",
      "NG_008806.1:g.56255G>A\n",
      "NG_008806.1:g.56255G>A\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "ATP7B:g.59042G>T\n",
      "G\n",
      "ATP7B|g.59042\n",
      "[u'NG_008806.1']\n",
      "NG_008806.1:g.59042G>T\n",
      "NG_008806.1:g.59042G>T\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "ATP7B:g.66363G>A\n",
      "G\n",
      "ATP7B|g.66363\n",
      "[u'NG_008806.1']\n",
      "NG_008806.1:g.66363G>A\n",
      "NG_008806.1:g.66363G>A\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "ATP7B:g.70004G>C\n",
      "G\n",
      "ATP7B|g.70004\n",
      "[u'NG_008806.1']\n",
      "NG_008806.1:g.70004G>C\n",
      "NG_008806.1:g.70004G>C\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "ATP7B:g.72244A>G\n",
      "A\n",
      "ATP7B|g.72244\n",
      "[u'NG_008806.1']\n",
      "NG_008806.1:g.72244A>G\n",
      "NG_008806.1:g.72244A>G\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "SLC6A9:c.1219A>G\n",
      "A\n",
      "SLC6A9|c.1219\n",
      "[u'NM_201649.3']\n",
      "NM_201649.3:c.1219A>G\n",
      "NM_201649.3:c.1219A>G\n",
      "44467262\n",
      "WWOX:c.606-1G>A\n",
      "G\n",
      "WWOX|c.606-1\n",
      "[u'NM_016373.3']\n",
      "NM_016373.3:c.606-1G>A\n",
      "NM_016373.3:c.606-1G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "Records: 10000\n",
      "MYH8:c.2021G>A\n",
      "G\n",
      "MYH8|c.2021\n",
      "[u'NM_002472.2']\n",
      "NM_002472.2:c.2021G>A\n",
      "NM_002472.2:c.2021G>A\n",
      "10310241\n",
      "GYS2:c.1720T>C\n",
      "T\n",
      "GYS2|c.1720\n",
      "[u'NM_021957.3']\n",
      "NM_021957.3:c.1720T>C\n",
      "NM_021957.3:c.1720T>C\n",
      "21693433\n",
      "NF1:c.1466A>G\n",
      "A\n",
      "NF1|c.1466\n",
      "[u'NM_000267.3']\n",
      "NM_000267.3:c.1466A>G\n",
      "NM_000267.3:c.1466A>G\n",
      "29541542\n",
      "NF1:c.1466A>G\n",
      "A\n",
      "NF1|c.1466\n",
      "[u'NM_000267.3']\n",
      "NM_000267.3:c.1466A>G\n",
      "NM_000267.3:c.1466A>G\n",
      "29541542\n",
      "GJB2:c.2410T>C\n",
      "T\n",
      "GJB2|c.2410\n",
      "[u'NM_004004.5']\n",
      "NM_004004.5:c.2410T>C\n",
      "NM_004004.5:c.2410T>C\n",
      "20761311\n",
      "GJB2:c.792T>C\n",
      "T\n",
      "GJB2|c.792\n",
      "[u'NM_004004.5']\n",
      "NM_004004.5:c.792T>C\n",
      "NM_004004.5:c.792T>C\n",
      "20762929\n",
      "PIGP:c.74T>C\n",
      "T\n",
      "PIGP|c.74\n",
      "[u'NM_153681.2']\n",
      "NM_153681.2:c.74T>C\n",
      "NM_153681.2:c.74T>C\n",
      "38444814\n",
      "AIP:c.910C>T\n",
      "C\n",
      "AIP|c.910\n",
      "[u'NM_003977.3']\n",
      "NM_003977.3:c.910C>T\n",
      "NM_003977.3:c.910C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CTSC:c.899G>A\n",
      "G\n",
      "CTSC|c.899\n",
      "[u'NM_001814.4']\n",
      "NM_001814.4:c.899G>A\n",
      "NM_001814.4:c.899G>A\n",
      "88027667\n",
      "TAS2R38:c.3413G>A\n",
      "G\n",
      "TAS2R38|c.3413\n",
      "[u'NM_024757.4']\n",
      "NM_024757.4:c.3413G>A\n",
      "NM_024757.4:c.3413G>A\n",
      "140711929\n",
      "ASNS:c.970C>T\n",
      "C\n",
      "ASNS|c.970\n",
      "[u'NM_001178076.1']\n",
      "NM_001178076.1:c.970C>T\n",
      "NM_001178076.1:c.970C>T\n",
      "97483911\n",
      "{'end': 97486062, 'tr': 1, 'allele_string': [u'C', u'T'], 'start': 97486062, 'chr': u'7', 'strand': -1}\n",
      "Asparagine synthetase deficiency (OMIM# 615574) is a very rare newly described neurometabolic disorder characterized by congenital microcephaly and severe global developmental delay, associated with intractable seizures or hyperekplexia. Brain MRI typically shows cerebral atrophy with simplified gyral pattern and delayed myelination. Only 12 cases have been described to date. The disease is caused by homozygous or compound heterozygous mutations in the ASNS gene on chromosome 7q21.Family 1 is a multiplex consanguineous family with five affected members, while Family 2 is simplex. One affected from each family was available for detailed phenotyping. Both patients (Patients 1 and 2) presented at birth with microcephaly and severe hyperekplexia, and were found to have gross brain malformation characterized by simplified gyral pattern, and hypoplastic cerebellum and pons. EEG showed no epileptiform discharge in Patient 2 but multifocal discharges in patient 1. Patient 2 is currently four years old with severe neurodevelopmental delay, quadriplegia and cortical blindness. Whole exome sequencing (WES) revealed a novel homozygous mutation in ASNS (NM_001178076.1) in each patient (c.970C > T:p.(Arg324*) and c.944A > G:p.(Tyr315Cys)).Our results expand the mutational spectrum of the recently described asparagine synthetase deficiency and show a remarkable clinical homogeneity among affected individuals, which should facilitate its recognition and molecular confirmation for pertinent and timely genetic counseling.\n",
      "ASNS:c.944A>G\n",
      "A\n",
      "ASNS|c.944\n",
      "[u'NM_001178076.1']\n",
      "NM_001178076.1:c.944A>G\n",
      "NM_001178076.1:c.944A>G\n",
      "97483937\n",
      "MT-ATP6:c.2539G>A\n",
      "G\n",
      "MT-ATP6|c.2539\n",
      "[u'NM_000188.2']\n",
      "NM_000188.2:c.2539G>A\n",
      "NM_000188.2:c.2539G>A\n",
      "71158514\n",
      "LMBR1:g.156584153A>G\n",
      "A\n",
      "LMBR1|g.156584153\n",
      "[u'NC_000007.14']\n",
      "NC_000007.14:g.156584153A>G\n",
      "NC_000007.14:g.156584153A>G\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "P4HB:c.1178A>G\n",
      "A\n",
      "P4HB|c.1178\n",
      "[u'NM_000918.3']\n",
      "NM_000918.3:c.1178A>G\n",
      "NM_000918.3:c.1178A>G\n",
      "79803618\n",
      "HSD17B3:c.576G>A\n",
      "G\n",
      "HSD17B3|c.576\n",
      "[u'NM_000197.1']\n",
      "NM_000197.1:c.576G>A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NM_000197.1:c.576G>A\n",
      "99007657\n",
      "SCN9A:c.5218G>C\n",
      "G\n",
      "SCN9A|c.5218\n",
      "[u'NM_002977']\n",
      "NM_002977:c.5218G>C\n",
      "NM_002977:c.5218G>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "YARS:c.638C>T\n",
      "C\n",
      "YARS|c.638\n",
      "[u'NM_003680.3']\n",
      "NM_003680.3:c.638C>T\n",
      "NM_003680.3:c.638C>T\n",
      "33256809\n",
      "YARS:c.1573G>A\n",
      "G\n",
      "YARS|c.1573\n",
      "[u'NM_003680.3']\n",
      "NM_003680.3:c.1573G>A\n",
      "NM_003680.3:c.1573G>A\n",
      "33241596\n",
      "SH2B2:c.6G>A\n",
      "G\n",
      "SH2B2|c.6\n",
      "[u'NM_000383.2']\n",
      "NM_000383.2:c.6G>A\n",
      "NM_000383.2:c.6G>A\n",
      "45705895\n",
      "ALS2:c.3512+1G>A\n",
      "G\n",
      "ALS2|c.3512+1\n",
      "[u'NM_020919.3']\n",
      "NM_020919.3:c.3512+1G>A\n",
      "NM_020919.3:c.3512+1G>A\n",
      "202589017\n",
      "NF1:c.3113+1G>A\n",
      "G\n",
      "NF1|c.3113+1\n",
      "[u'NM_000267.3']\n",
      "NM_000267.3:c.3113+1G>A\n",
      "NM_000267.3:c.3113+1G>A\n",
      "29557401\n",
      "ARX:c.1614G>T\n",
      "G\n",
      "ARX|c.1614\n",
      "[u'NM_139058.2']\n",
      "NM_139058.2:c.1614G>T\n",
      "NM_139058.2:c.1614G>T\n",
      "25022862\n",
      "FGFR3:c.1637C>A\n",
      "C\n",
      "FGFR3|c.1637\n",
      "[u'NM_000142.4']\n",
      "NM_000142.4:c.1637C>A\n",
      "NM_000142.4:c.1637C>A\n",
      "1807388\n",
      "{'end': 1807382, 'tr': 1, 'allele_string': [u'C', u'A'], 'start': 1807382, 'chr': u'4', 'strand': 1}\n",
      "Most reported mutations in the FGFR3 gene are dominant activating mutations that cause a variety of short-limbed bone dysplasias including achondroplasia and syndromic craniosynostosis. We report the phenotype and underlying molecular abnormality in two brothers, born to first cousin parents. The clinical picture is characterized by tall stature and severe skeletal abnormalities leading to inability to walk, with camptodactyly, arachnodactyly, and scoliosis. Whole exome sequencing revealed a homozygous novel missense mutation in the FGFR3 gene in exon 12 (NM_000142.4:c.1637C>A: p.(Thr546Lys)). The variant is found in the kinase domain of the protein and is predicted to be pathogenic. It is located near a known hotspot for hypochondroplasia. This is the first report of a homozygous loss-of-function mutation in FGFR3 in human that results in a skeletal overgrowth syndrome.\n",
      "MYOD1:c.188C>A\n",
      "C\n",
      "MYOD1|c.188\n",
      "[u'NM_002478.4']\n",
      "NM_002478.4:c.188C>A\n",
      "NM_002478.4:c.188C>A\n",
      "17741517\n",
      "ROGDI:c.117+1G>T\n",
      "G\n",
      "ROGDI|c.117+1\n",
      "[u'NM_024589.1']\n",
      "NM_024589.1:c.117+1G>T\n",
      "NM_024589.1:c.117+1G>T\n",
      "4852382\n",
      "ROGDI:g.4852382C>A\n",
      "C\n",
      "ROGDI|g.4852382\n",
      "[u'NM_024589.1']\n",
      "NM_024589.1:g.4852382C>A\n",
      "NM_024589.1:g.4852382C>A\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "PCBD1:c.2609G>A\n",
      "G\n",
      "PCBD1|c.2609\n",
      "[u'NM_001243342.1']\n",
      "NM_001243342.1:c.2609G>A\n",
      "NM_001243342.1:c.2609G>A\n",
      "78061565\n",
      "SLC12A3:c.1095+5G>A\n",
      "G\n",
      "SLC12A3|c.1095+5\n",
      "[u'NM_000339.2']\n",
      "NM_000339.2:c.1095+5G>A\n",
      "NM_000339.2:c.1095+5G>A\n",
      "56906703\n",
      "POLE:c.1089C>A\n",
      "C\n",
      "POLE|c.1089\n",
      "[u'NM_006231.2']\n",
      "NM_006231.2:c.1089C>A\n",
      "NM_006231.2:c.1089C>A\n",
      "133252338\n",
      "L1CAM:c.604G>A\n",
      "G\n",
      "L1CAM|c.604\n",
      "[u'NM_000425.3']\n",
      "NM_000425.3:c.604G>A\n",
      "NM_000425.3:c.604G>A\n",
      "153136335\n",
      "CCDC151:c.925G>T\n",
      "G\n",
      "CCDC151|c.925\n",
      "[u'NM_145045.4']\n",
      "NM_145045.4:c.925G>T\n",
      "NM_145045.4:c.925G>T\n",
      "11537002\n",
      "CDKL5:c.533G>A\n",
      "G\n",
      "CDKL5|c.533\n",
      "[u'NM_003159.2']\n",
      "NM_003159.2:c.533G>A\n",
      "NM_003159.2:c.533G>A\n",
      "18602452\n",
      "CDKL5:c.1A>G\n",
      "A\n",
      "CDKL5|c.1\n",
      "[u'NM_003159.2']\n",
      "NM_003159.2:c.1A>G\n",
      "NM_003159.2:c.1A>G\n",
      "18525217\n",
      "CDKL5:c.1A>G\n",
      "A\n",
      "CDKL5|c.1\n",
      "[u'NM_003159.2']\n",
      "NM_003159.2:c.1A>G\n",
      "NM_003159.2:c.1A>G\n",
      "18525217\n",
      "CDKL5:c.1375C>T\n",
      "C\n",
      "CDKL5|c.1375\n",
      "[u'NM_003159.2']\n",
      "NM_003159.2:c.1375C>T\n",
      "NM_003159.2:c.1375C>T\n",
      "18622419\n",
      "WISP3:c.156C>A\n",
      "C\n",
      "WISP3|c.156\n",
      "[u'NP_003871.1']\n",
      "NP_003871.1:c.156C>A\n",
      "NP_003871.1:c.156C>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "WISP3:c.643+1G>A\n",
      "G\n",
      "WISP3|c.643+1\n",
      "[u'NP_003871.1']\n",
      "NP_003871.1:c.643+1G>A\n",
      "NP_003871.1:c.643+1G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "KCNJ13:c.359T>C\n",
      "T\n",
      "KCNJ13|c.359\n",
      "[u'NM_002242.4']\n",
      "NM_002242.4:c.359T>C\n",
      "NM_002242.4:c.359T>C\n",
      "233635714\n",
      "CAPN3:c.1524+1G>T\n",
      "G\n",
      "CAPN3|c.1524+1\n",
      "[u'NM_000070']\n",
      "NM_000070:c.1524+1G>T\n",
      "NM_000070:c.1524+1G>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CAPN3:c.1524+1G>T\n",
      "G\n",
      "CAPN3|c.1524+1\n",
      "[u'NM_000070']\n",
      "NM_000070:c.1524+1G>T\n",
      "NM_000070:c.1524+1G>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CAPN3:c.2384C>T\n",
      "C\n",
      "CAPN3|c.2384\n",
      "[u'NM_000070']\n",
      "NM_000070:c.2384C>T\n",
      "NM_000070:c.2384C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "COL4A3:c.3266G>A\n",
      "G\n",
      "COL4A3|c.3266\n",
      "[u'NM_000091.4']\n",
      "NM_000091.4:c.3266G>A\n",
      "NM_000091.4:c.3266G>A\n",
      "228157962\n",
      "STIM1:c.910C>T\n",
      "C\n",
      "STIM1|c.910\n",
      "[u'NM_003156.3']\n",
      "NM_003156.3:c.910C>T\n",
      "NM_003156.3:c.910C>T\n",
      "4095850\n",
      "STIM1:c.910C>T\n",
      "C\n",
      "STIM1|c.910\n",
      "[u'NM_003156.3']\n",
      "NM_003156.3:c.910C>T\n",
      "NM_003156.3:c.910C>T\n",
      "4095850\n",
      "DMP1:c.1A>G\n",
      "A\n",
      "DMP1|c.1\n",
      "[u'NM_001079911.2']\n",
      "NM_001079911.2:c.1A>G\n",
      "NM_001079911.2:c.1A>G\n",
      "88577645\n",
      "B4GALT7:c.808C>T\n",
      "C\n",
      "B4GALT7|c.808\n",
      "[u'NM_007255.2']\n",
      "NM_007255.2:c.808C>T\n",
      "NM_007255.2:c.808C>T\n",
      "177035995\n",
      "MPV17:c.278A>C\n",
      "A\n",
      "MPV17|c.278\n",
      "[u'NM_002437.4']\n",
      "NM_002437.4:c.278A>C\n",
      "NM_002437.4:c.278A>C\n",
      "27535548\n",
      "GNAS:c.682C>T\n",
      "C\n",
      "GNAS|c.682\n",
      "[u'NM_000516.4']\n",
      "NM_000516.4:c.682C>T\n",
      "NM_000516.4:c.682C>T\n",
      "57484598\n",
      "{'end': 57429002, 'tr': 2, 'allele_string': [u'C', u'T'], 'start': 57429002, 'chr': u'20', 'strand': 1}\n",
      "Pseudohypoparathyroidism is a rare genetic disorder characterised by end-organ resistance to parathyroid hormone due to a defect of the guanine nucleotide-binding protein alpha that simulates activity of the polypeptide 1 (GNAS) gene. Patients with type 1a pseudohypoparathyroidism display different features of Albright's hereditary osteodystrophy as well as multi-hormone resistance. We describe a Chinese woman and her son, who presented with different symptoms of pseudohypoparathyroidism and clinically manifested different degree of Albright's hereditary osteodystrophy. Genetic study detected a mutation [NM_000516.4(GNAS):c682C>T (p.Arg228Cys)] in the GNAS gene.\n",
      "FGF23:c.471C>A\n",
      "C\n",
      "FGF23|c.471\n",
      "[u'NM_020638.2']\n",
      "NM_020638.2:c.471C>A\n",
      "NM_020638.2:c.471C>A\n",
      "4479794\n",
      "RS1:c.304C>T\n",
      "C\n",
      "RS1|c.304\n",
      "[u'NM_000330.3']\n",
      "NM_000330.3:c.304C>T\n",
      "NM_000330.3:c.304C>T\n",
      "18665333\n",
      "MCDR3:c.139G>A\n",
      "G\n",
      "MCDR3|c.139\n",
      "[u'NM_001278634']\n",
      "NM_001278634:c.139G>A\n",
      "NM_001278634:c.139G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "HFE:c.847C>T\n",
      "C\n",
      "HFE|c.847\n",
      "[u'NM_000410.3']\n",
      "NM_000410.3:c.847C>T\n",
      "NM_000410.3:c.847C>T\n",
      "26093143\n",
      "ADSL:c.71C>T\n",
      "C\n",
      "ADSL|c.71\n",
      "[u'NM_000026']\n",
      "NM_000026:c.71C>T\n",
      "NM_000026:c.71C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "ADSL:c.134G>A\n",
      "G\n",
      "ADSL|c.134\n",
      "[u'NM_000026']\n",
      "NM_000026:c.134G>A\n",
      "NM_000026:c.134G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "DTNBP1:c.307C>T\n",
      "C\n",
      "DTNBP1|c.307\n",
      "[u'NM_032122.4']\n",
      "NM_032122.4:c.307C>T\n",
      "NM_032122.4:c.307C>T\n",
      "15627622\n",
      "NEU1:c.699C>A\n",
      "C\n",
      "NEU1|c.699\n",
      "[u'NM_000434.3']\n",
      "NM_000434.3:c.699C>A\n",
      "NM_000434.3:c.699C>A\n",
      "31828315\n",
      "NEU1:c.803A>G\n",
      "A\n",
      "NEU1|c.803\n",
      "[u'NM_000434.3']\n",
      "NM_000434.3:c.803A>G\n",
      "NM_000434.3:c.803A>G\n",
      "31828037\n",
      "CHKB:c.810T>A\n",
      "T\n",
      "CHKB|c.810\n",
      "[u'NM_005198.4']\n",
      "NM_005198.4:c.810T>A\n",
      "NM_005198.4:c.810T>A\n",
      "51018627\n",
      "ZNF644:c.1201A>G\n",
      "A\n",
      "ZNF644|c.1201\n",
      "[u'NM_201269']\n",
      "NM_201269:c.1201A>G\n",
      "NM_201269:c.1201A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "ZNF644:c.2867C>G\n",
      "C\n",
      "ZNF644|c.2867\n",
      "[u'NM_201269']\n",
      "NM_201269:c.2867C>G\n",
      "NM_201269:c.2867C>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "ZNF644:c.3833A>G\n",
      "A\n",
      "ZNF644|c.3833\n",
      "[u'NM_201269']\n",
      "NM_201269:c.3833A>G\n",
      "NM_201269:c.3833A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "ZNF644:c.2565A>G\n",
      "A\n",
      "ZNF644|c.2565\n",
      "[u'NM_201269']\n",
      "NM_201269:c.2565A>G\n",
      "NM_201269:c.2565A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "ZNF644:c.219C>A\n",
      "C\n",
      "ZNF644|c.219\n",
      "[u'NM_201269']\n",
      "NM_201269:c.219C>A\n",
      "NM_201269:c.219C>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "ZNF644:c.3833A>G\n",
      "A\n",
      "ZNF644|c.3833\n",
      "[u'NM_201269']\n",
      "NM_201269:c.3833A>G\n",
      "NM_201269:c.3833A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PAX9:c.87G>C\n",
      "G\n",
      "PAX9|c.87\n",
      "[u'NC_000014']\n",
      "NC_000014:c.87G>C\n",
      "NC_000014:c.87G>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PAX9:c.87G>C\n",
      "G\n",
      "PAX9|c.87\n",
      "[u'NC_000014']\n",
      "NC_000014:c.87G>C\n",
      "NC_000014:c.87G>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "RINT1:c.343C>T\n",
      "C\n",
      "RINT1|c.343\n",
      "[u'NM_021930.4']\n",
      "NM_021930.4:c.343C>T\n",
      "NM_021930.4:c.343C>T\n",
      "105182924\n",
      "RINT1:c.1207G>T\n",
      "G\n",
      "RINT1|c.1207\n",
      "[u'NM_021930.4']\n",
      "NM_021930.4:c.1207G>T\n",
      "NM_021930.4:c.1207G>T\n",
      "105190807\n",
      "SLC25A13:c.615+5G>A\n",
      "G\n",
      "SLC25A13|c.615+5\n",
      "[u'NM_138459.3']\n",
      "NM_138459.3:c.615+5G>A\n",
      "NM_138459.3:c.615+5G>A\n",
      "118015272\n",
      "{'end': 95822344, 'tr': 1, 'allele_string': [u'G', u'A'], 'start': 95822344, 'chr': u'7', 'strand': -1}\n",
      "To investigate the differences in the mutation spectra of the SLC25A13 gene mutations from specific regions of China.Genetic analyses of SLC25A13 mutations were performed in 535 patients with neonatal intrahepatic cholestasis from our center over eight years. Unrelated infants with at least one mutant allele were enrolled to calculate the proportion of SLC25A13 mutations in different regions of China. The boundary between northern and southern China was drawn at the historical border of the Yangtze River.A total of 63 unrelated patients (about 11% of cases with intrahepatic cholestasis) from 16 provinces or municipalities in China had mutations in the SLC25A13 gene, of these 16 (25%) were homozygotes, 28 (44%) were compound heterozygotes and 19 (30%) were heterozygotes. In addition to four well described common mutations (c.851_854del, c.1638_1660dup23, c.615+5G>A and c.1750+72_1751-4dup17insNM_138459.3:2667 also known as IVS16ins3kb), 13 other mutation types were identified, including three novel mutations: c.985_986insT, c.287T>C and c.1349A>G. According to the geographical division criteria, 60 mutant alleles were identified in patients from the southern areas of China, 43 alleles were identified in patients from the border, and 4 alleles were identified in patients from the northern areas of China. The proportion of four common mutations was higher in south region (56/60, 93%) than that in the border region (34/43, 79%, χ(2) = 4.621, P = 0.032) and the northern region (2/4, 50%, χ(2) = 8.288, P = 0.041).The SLC25A13 mutation spectra among the three regions of China were different, providing a basis for the improvement of diagnostic strategies and interpretation of genetic diagnosis.\n",
      "SLC25A13:c.287T>C\n",
      "T\n",
      "SLC25A13|c.287\n",
      "[u'NM_138459.3']\n",
      "NM_138459.3:c.287T>C\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NM_138459.3:c.287T>C\n",
      "117997120\n",
      "{'end': 95864155, 'tr': 1, 'allele_string': [u'T', u'C'], 'start': 95864155, 'chr': u'7', 'strand': -1}\n",
      "To investigate the differences in the mutation spectra of the SLC25A13 gene mutations from specific regions of China.Genetic analyses of SLC25A13 mutations were performed in 535 patients with neonatal intrahepatic cholestasis from our center over eight years. Unrelated infants with at least one mutant allele were enrolled to calculate the proportion of SLC25A13 mutations in different regions of China. The boundary between northern and southern China was drawn at the historical border of the Yangtze River.A total of 63 unrelated patients (about 11% of cases with intrahepatic cholestasis) from 16 provinces or municipalities in China had mutations in the SLC25A13 gene, of these 16 (25%) were homozygotes, 28 (44%) were compound heterozygotes and 19 (30%) were heterozygotes. In addition to four well described common mutations (c.851_854del, c.1638_1660dup23, c.615+5G>A and c.1750+72_1751-4dup17insNM_138459.3:2667 also known as IVS16ins3kb), 13 other mutation types were identified, including three novel mutations: c.985_986insT, c.287T>C and c.1349A>G. According to the geographical division criteria, 60 mutant alleles were identified in patients from the southern areas of China, 43 alleles were identified in patients from the border, and 4 alleles were identified in patients from the northern areas of China. The proportion of four common mutations was higher in south region (56/60, 93%) than that in the border region (34/43, 79%, χ(2) = 4.621, P = 0.032) and the northern region (2/4, 50%, χ(2) = 8.288, P = 0.041).The SLC25A13 mutation spectra among the three regions of China were different, providing a basis for the improvement of diagnostic strategies and interpretation of genetic diagnosis.\n",
      "SLC25A13:c.1349A>G\n",
      "A\n",
      "SLC25A13|c.1349\n",
      "[u'NM_138459.3']\n",
      "NM_138459.3:c.1349A>G\n",
      "NM_138459.3:c.1349A>G\n",
      "118028645\n",
      "MRE11:c.140C>T\n",
      "C\n",
      "MRE11|c.140\n",
      "[u'NM_005590']\n",
      "NM_005590:c.140C>T\n",
      "NM_005590:c.140C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "ACVR1:c.617G>A\n",
      "G\n",
      "ACVR1|c.617\n",
      "[u'NM_001105.4']\n",
      "NM_001105.4:c.617G>A\n",
      "NM_001105.4:c.617G>A\n",
      "158630626\n",
      "mtdna:c.665G>C\n",
      "G\n",
      "mtdna|c.665\n",
      "[u'NM_001079855']\n",
      "NM_001079855:c.665G>C\n",
      "NM_001079855:c.665G>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "BBS2:c.173T>G\n",
      "T\n",
      "BBS2|c.173\n",
      "[u'NM_001195306']\n",
      "NM_001195306:c.173T>G\n",
      "NM_001195306:c.173T>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "MED12:c.1862G>A\n",
      "G\n",
      "MED12|c.1862\n",
      "[u'NM_005120.2']\n",
      "NM_005120.2:c.1862G>A\n",
      "NM_005120.2:c.1862G>A\n",
      "70344126\n",
      "TGM6:c.1550T>G\n",
      "T\n",
      "TGM6|c.1550\n",
      "[u'NM_198994.2']\n",
      "NM_198994.2:c.1550T>G\n",
      "NM_198994.2:c.1550T>G\n",
      "2398091\n",
      "MTSS1:c.532A>T\n",
      "A\n",
      "MTSS1|c.532\n",
      "[u'NM_005159.4']\n",
      "NM_005159.4:c.532A>T\n",
      "NM_005159.4:c.532A>T\n",
      "35084693\n",
      "G6PC3:c.974T>G\n",
      "T\n",
      "G6PC3|c.974\n",
      "[u'NM_138387.3']\n",
      "NM_138387.3:c.974T>G\n",
      "NM_138387.3:c.974T>G\n",
      "42153344\n",
      "VLDLR:c.665C>T\n",
      "C\n",
      "VLDLR|c.665\n",
      "[u'NM_178153.2']\n",
      "NM_178153.2:c.665C>T\n",
      "NM_178153.2:c.665C>T\n",
      "110644258\n",
      "{'end': 2643376, 'tr': 2, 'allele_string': [u'C', u'T'], 'start': 2643376, 'chr': u'9', 'strand': 1}\n",
      "Malformations of cortical development include a wide range of brain developmental anomalies that commonly lead to developmental delay and epilepsy. Lissencephaly and subcortical band heterotopia are major malformations of cortical development due to abnormal neuronal migration and several genes have been identified including ARX, DCX, LIS1, RELN, TUBA1A, and VLDLR. Traditionally, genetic testing for lissencephaly and subcortical band heterotopia has been done in the order of the probability of detection of mutation according to the radiologic features, but the success rate could be variable with this time-consuming approach. In this study we used whole-exome sequencing to identify mutations in a 5-year-old girl with lissencephaly spectrum with subcortical band heterotopia. After excluding lissencephaly-related genes, one deleterious mutation (NM_178153.2:c.665C > T, p.Thr222Ile) in the DCX gene was identified. Further Sanger sequencing validated the variant in the patient but not in both parents indicating a de novo mutation. The present report demonstrates that whole-exome sequencing may be a useful tool for the identification of mutations in patients with lissencephaly and subcortical band heterotopias as well as malformations of cortical development.\n",
      "XPC:g.18246G>A\n",
      "G\n",
      "XPC|g.18246\n",
      "[u'NG_011763.1']\n",
      "NG_011763.1:g.18246G>A\n",
      "NG_011763.1:g.18246G>A\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "XPC:g.18810G>T\n",
      "G\n",
      "XPC|g.18810\n",
      "[u'NG_011763.1']\n",
      "NG_011763.1:g.18810G>T\n",
      "NG_011763.1:g.18810G>T\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "NKX2-1:c.516G>T\n",
      "G\n",
      "NKX2-1|c.516\n",
      "[u'NM_003317.3']\n",
      "NM_003317.3:c.516G>T\n",
      "NM_003317.3:c.516G>T\n",
      "36987083\n",
      "NKX2-1:c.623G>C\n",
      "G\n",
      "NKX2-1|c.623\n",
      "[u'NM_003317.3']\n",
      "NM_003317.3:c.623G>C\n",
      "NM_003317.3:c.623G>C\n",
      "36986976\n",
      "{'end': 36987066, 'tr': 4, 'allele_string': [u'G', u'C'], 'start': 36987066, 'chr': u'14', 'strand': -1}\n",
      "The thyroid transcription factor 1 (TTF-1) is encoded, on chromosome 14q13, by the gene termed TITF-1/NKX2.1. Mutations in this gene have been associated with chorea, hypothyroidism, and lung disease, all included in the \"brain-thyroid-lung syndrome.\" We here describe two cases of novel missense mutations [NM_003317.3:c.516G>T and c.623G>C resulting in p.(Gln172His) and p.(Trp208Ser), respectively] in TITF-1/NKX2-1 in non-consanguineous patients. We provide a functional study of the role of the two mutations on the TTF-1 ability to bind DNA and to trans-activate both thyroid and lung specific gene promoters. Our results confirm the difficulty to correlate the TTF-1 activity with the clinical phenotype of affected patients and highlight the need to increase the limited knowledge we have on the activity of TTF-1 in neuronal cells.\n",
      "YAP1:c.370C>T\n",
      "C\n",
      "YAP1|c.370\n",
      "[u'NM_001130145']\n",
      "NM_001130145:c.370C>T\n",
      "NM_001130145:c.370C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "YAP1:c.1066G>T\n",
      "G\n",
      "YAP1|c.1066\n",
      "[u'NM_001130145']\n",
      "NM_001130145:c.1066G>T\n",
      "NM_001130145:c.1066G>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "YAP1:c.370C>T\n",
      "C\n",
      "YAP1|c.370\n",
      "[u'NM_001130145']\n",
      "NM_001130145:c.370C>T\n",
      "NM_001130145:c.370C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "YAP1:c.1066G>T\n",
      "G\n",
      "YAP1|c.1066\n",
      "[u'NM_001130145']\n",
      "NM_001130145:c.1066G>T\n",
      "NM_001130145:c.1066G>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "SHH:g.106737G>T\n",
      "G\n",
      "SHH|g.106737\n",
      "[u'NG_009240.1']\n",
      "NG_009240.1:g.106737G>T\n",
      "NG_009240.1:g.106737G>T\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "LMBR1:c.404G>T\n",
      "G\n",
      "LMBR1|c.404\n",
      "[u'NG_009240.1']\n",
      "NG_009240.1:c.404G>T\n",
      "NG_009240.1:c.404G>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "ALDH5A1:c.901A>G\n",
      "A\n",
      "ALDH5A1|c.901\n",
      "[u'NM_001080']\n",
      "NM_001080:c.901A>G\n",
      "NM_001080:c.901A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "KALRN:c.3644C>A\n",
      "C\n",
      "KALRN|c.3644\n",
      "[u'NM_003947.4']\n",
      "NM_003947.4:c.3644C>A\n",
      "NM_003947.4:c.3644C>A\n",
      "124174121\n",
      "SRY:c.341A>G\n",
      "A\n",
      "SRY|c.341\n",
      "[u'NM_005634']\n",
      "NM_005634:c.341A>G\n",
      "NM_005634:c.341A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "SMS:c.443A>G\n",
      "A\n",
      "SMS|c.443\n",
      "[u'NM_004595.4']\n",
      "NM_004595.4:c.443A>G\n",
      "NM_004595.4:c.443A>G\n",
      "21995292\n",
      "CD4:g.70A>G\n",
      "A\n",
      "CD4|g.70\n",
      "[u'NR_003051']\n",
      "NR_003051:g.70A>G\n",
      "NR_003051:g.70A>G\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "MOGS:c.65C>A\n",
      "C\n",
      "MOGS|c.65\n",
      "[u'NM_006302.2']\n",
      "NM_006302.2:c.65C>A\n",
      "NM_006302.2:c.65C>A\n",
      "74692310\n",
      "MOGS:c.329G>A\n",
      "G\n",
      "MOGS|c.329\n",
      "[u'NM_006302.2']\n",
      "NM_006302.2:c.329G>A\n",
      "NM_006302.2:c.329G>A\n",
      "74692046\n",
      "MOGS:c.370C>T\n",
      "C\n",
      "MOGS|c.370\n",
      "[u'NM_006302.2']\n",
      "NM_006302.2:c.370C>T\n",
      "NM_006302.2:c.370C>T\n",
      "74691832\n",
      "HSP90AA1:g.1209A>G\n",
      "A\n",
      "HSP90AA1|g.1209\n",
      "[u'AC_000178.1']\n",
      "AC_000178.1:g.1209A>G\n",
      "AC_000178.1:g.1209A>G\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "HSP90AA1:g.1209A>G\n",
      "A\n",
      "HSP90AA1|g.1209\n",
      "[u'AC_000178.1']\n",
      "AC_000178.1:g.1209A>G\n",
      "AC_000178.1:g.1209A>G\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "TTN:c.90263G>T\n",
      "G\n",
      "TTN|c.90263\n",
      "[u'NM_001256850']\n",
      "NM_001256850:c.90263G>T\n",
      "NM_001256850:c.90263G>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "EHMT1:c.2712+1G>A\n",
      "G\n",
      "EHMT1|c.2712+1\n",
      "[u'NM_024757.4']\n",
      "NM_024757.4:c.2712+1G>A\n",
      "NM_024757.4:c.2712+1G>A\n",
      "140695437\n",
      "EHMT1:c.2712+1G>A\n",
      "G\n",
      "EHMT1|c.2712+1\n",
      "[u'NM_024757.4']\n",
      "NM_024757.4:c.2712+1G>A\n",
      "NM_024757.4:c.2712+1G>A\n",
      "140695437\n",
      "CAPN1:g.98533962C>G\n",
      "C\n",
      "CAPN1|g.98533962\n",
      "[u'NM_174003']\n",
      "NM_174003:g.98533962C>G\n",
      "NM_174003:g.98533962C>G\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "CAPN1:g.98535683A>G\n",
      "A\n",
      "CAPN1|g.98535683\n",
      "[u'NM_174003']\n",
      "NM_174003:g.98535683A>G\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NM_174003:g.98535683A>G\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "CAPN1:g.98545188T>A\n",
      "T\n",
      "CAPN1|g.98545188\n",
      "[u'NM_174003']\n",
      "NM_174003:g.98545188T>A\n",
      "NM_174003:g.98545188T>A\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "CAPN1:g.98535683A>G\n",
      "A\n",
      "CAPN1|g.98535683\n",
      "[u'NM_174003']\n",
      "NM_174003:g.98535683A>G\n",
      "NM_174003:g.98535683A>G\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "PSMC3IP:c.337+33A>G\n",
      "A\n",
      "PSMC3IP|c.337+33\n",
      "[u'NM_016556.3']\n",
      "NM_016556.3:c.337+33A>G\n",
      "NM_016556.3:c.337+33A>G\n",
      "40726084\n",
      "GLDC:c.2296G>T\n",
      "G\n",
      "GLDC|c.2296\n",
      "[u'NM_000170.2']\n",
      "NM_000170.2:c.2296G>T\n",
      "NM_000170.2:c.2296G>T\n",
      "6554688\n",
      "SLC17A5:c.526-2A>G\n",
      "A\n",
      "SLC17A5|c.526-2\n",
      "[u'NM_012434']\n",
      "NM_012434:c.526-2A>G\n",
      "NM_012434:c.526-2A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PIGT:c.1342C>T\n",
      "C\n",
      "PIGT|c.1342\n",
      "[u'NM_015937.5']\n",
      "NM_015937.5:c.1342C>T\n",
      "NM_015937.5:c.1342C>T\n",
      "44052963\n",
      "ERCC6:c.2008C>T\n",
      "C\n",
      "ERCC6|c.2008\n",
      "[u'NM_000124.2']\n",
      "NM_000124.2:c.2008C>T\n",
      "NM_000124.2:c.2008C>T\n",
      "50690894\n",
      "HOXD3:c.543-199G>T\n",
      "G\n",
      "HOXD3|c.543-199\n",
      "[u'NP_008829.3']\n",
      "NP_008829.3:c.543-199G>T\n",
      "NP_008829.3:c.543-199G>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "HOXD3:c.543-34G>A\n",
      "G\n",
      "HOXD3|c.543-34\n",
      "[u'NP_008829.3']\n",
      "NP_008829.3:c.543-34G>A\n",
      "NP_008829.3:c.543-34G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PITX2:c.45+76C>T\n",
      "C\n",
      "PITX2|c.45+76\n",
      "[u'NP_008829.3']\n",
      "NP_008829.3:c.45+76C>T\n",
      "NP_008829.3:c.45+76C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "NF2:g.74409T>A\n",
      "T\n",
      "NF2|g.74409\n",
      "[u'NG_009057.1']\n",
      "NG_009057.1:g.74409T>A\n",
      "NG_009057.1:g.74409T>A\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "LZTFL1:c.260T>C\n",
      "T\n",
      "LZTFL1|c.260\n",
      "[u'NM_020347.2']\n",
      "NM_020347.2:c.260T>C\n",
      "NM_020347.2:c.260T>C\n",
      "45877145\n",
      "LZTFL1:c.778G>T\n",
      "G\n",
      "LZTFL1|c.778\n",
      "[u'NM_020347.2']\n",
      "NM_020347.2:c.778G>T\n",
      "NM_020347.2:c.778G>T\n",
      "45868951\n",
      "CAPN1:g.98533962C>G\n",
      "C\n",
      "CAPN1|g.98533962\n",
      "[u'NM_174003']\n",
      "NM_174003:g.98533962C>G\n",
      "NM_174003:g.98533962C>G\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "CAPN1:g.98535683A>G\n",
      "A\n",
      "CAPN1|g.98535683\n",
      "[u'NM_174003']\n",
      "NM_174003:g.98535683A>G\n",
      "NM_174003:g.98535683A>G\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "CAPN1:g.98545188T>A\n",
      "T\n",
      "CAPN1|g.98545188\n",
      "[u'NM_174003']\n",
      "NM_174003:g.98545188T>A\n",
      "NM_174003:g.98545188T>A\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "CAPN1:g.98535683A>G\n",
      "A\n",
      "CAPN1|g.98535683\n",
      "[u'NM_174003']\n",
      "NM_174003:g.98535683A>G\n",
      "NM_174003:g.98535683A>G\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "PAEP:c.1448T>C\n",
      "T\n",
      "PAEP|c.1448\n",
      "[u'NM_000157.3']\n",
      "NM_000157.3:c.1448T>C\n",
      "NM_000157.3:c.1448T>C\n",
      "155205043\n",
      "RECQL4:c.2059-1G>C\n",
      "G\n",
      "RECQL4|c.2059-1\n",
      "[u'NM_004260']\n",
      "NM_004260:c.2059-1G>C\n",
      "NM_004260:c.2059-1G>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "SMARCB1:c.359T>C\n",
      "T\n",
      "SMARCB1|c.359\n",
      "[u'NM_001040108.1']\n",
      "NM_001040108.1:c.359T>C\n",
      "NM_001040108.1:c.359T>C\n",
      "75516000\n",
      "{'end': 24135872, 'tr': 1, 'allele_string': [u'T', u'C'], 'start': 24135872, 'chr': u'22', 'strand': 1}\n",
      "A 4-month-old male infant presented with severe developmental delay, cerebellar, brainstem, and cutaneous hemangiomas, bilateral tumors (vestibular, hypoglossal, cervical, and lumbar spinal), and few café-au-lait macules. Cerebellar and lumbar tumor biopsies revealed venous telangiectasia and intraneural perineuroma, respectively. Sequencing NF1, NF2, and RASA1 (blood), and NF2 and SMARCB1 (lumbar biopsy) was negative for pathogenic mutations. Clinical exome sequencing (CES), requested for tumor syndrome diagnosis, revealed two heterozygous missense variants, c.359T>C;p.Phe120Ser and c.3344G>A;p.Arg1115Gln, in MLH3 (NM_001040108.1), a DNA mismatch repair (MMR) gene, Polyphen-predicted as probably damaging, and benign, respectively. Sanger sequencing confirmed both variants in the proband, and their absence in the mother; biological father unavailable. Both biopsied tissues were negative for microsatellite instability, and expressed MLH1, MSH2, PMS2, MSH6, and MLH3 immunohistochemically. Chromosomal microarray showed a 133 kb segment copy number duplication of 14q12 region encompassing FOXG1, possibly explaining the developmental delay, but not the tumors. The presence of MLH3 variants with multiple benign neural and vascular tumors was intriguing for their possible role in the pathogenesis of these neoplasms, which were suspicious for, but not diagnostic of, constitutional MMR deficiency. However, functional assays of non-neoplastic patient-derived cells showed intact base-base MMR function. Also, no previous FOXG1-aberrant patient was reported with tumors. We now report a 3-year-old FOXG1-duplicated patient with a yet undescribed tumor syndrome with clinical features of neurofibromatosis types I and II, where several validation studies could not ascertain the significance of CES findings; further studies may elucidate precise mechanisms and diagnosis for clinical management, including tumor surveillance.\n",
      "SMARCB1:c.3344G>A\n",
      "G\n",
      "SMARCB1|c.3344\n",
      "[u'NM_001040108.1']\n",
      "NM_001040108.1:c.3344G>A\n",
      "NM_001040108.1:c.3344G>A\n",
      "75509117\n",
      "LMBR1:g.106954C>T\n",
      "C\n",
      "LMBR1|g.106954\n",
      "[u'NG_009240.1']\n",
      "NG_009240.1:g.106954C>T\n",
      "NG_009240.1:g.106954C>T\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "LMBR1:c.619C>T\n",
      "C\n",
      "LMBR1|c.619\n",
      "[u'NG_009240.1']\n",
      "NG_009240.1:c.619C>T\n",
      "NG_009240.1:c.619C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "STUB1:c.621C>G\n",
      "C\n",
      "STUB1|c.621\n",
      "[u'NM_005861']\n",
      "NM_005861:c.621C>G\n",
      "NM_005861:c.621C>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "TPO:c.2669G>A\n",
      "G\n",
      "TPO|c.2669\n",
      "[u'NM_000547.5']\n",
      "NM_000547.5:c.2669G>A\n",
      "NM_000547.5:c.2669G>A\n",
      "1544416\n",
      "OTOF:c.5332G>T\n",
      "G\n",
      "OTOF|c.5332\n",
      "[u'NM_194248.2']\n",
      "NM_194248.2:c.5332G>T\n",
      "NM_194248.2:c.5332G>T\n",
      "26684765\n",
      "OTOF:c.5332G>T\n",
      "G\n",
      "OTOF|c.5332\n",
      "[u'NM_194248.2']\n",
      "NM_194248.2:c.5332G>T\n",
      "NM_194248.2:c.5332G>T\n",
      "26684765\n",
      "PRIMPOL:c.265T>G\n",
      "T\n",
      "PRIMPOL|c.265\n",
      "[u'NM_152683.2']\n",
      "NM_152683.2:c.265T>G\n",
      "NM_152683.2:c.265T>G\n",
      "185580578\n",
      "MFN2:c.1066A>G\n",
      "A\n",
      "MFN2|c.1066\n",
      "[u'NM_014874.3']\n",
      "NM_014874.3:c.1066A>G\n",
      "NM_014874.3:c.1066A>G\n",
      "12062066\n",
      "KAT6A:c.4108G>T\n",
      "G\n",
      "KAT6A|c.4108\n",
      "[u'NM_001099412.1']\n",
      "NM_001099412.1:c.4108G>T\n",
      "NM_001099412.1:c.4108G>T\n",
      "41791630\n",
      "Records: 20000\n",
      "BMP15:g.1773T>C\n",
      "T\n",
      "BMP15|g.1773\n",
      "[u'NC_006091.3']\n",
      "NC_006091.3:g.1773T>C\n",
      "NC_006091.3:g.1773T>C\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "GCK:c.878T>C\n",
      "T\n",
      "GCK|c.878\n",
      "[u'NM_000162.3']\n",
      "NM_000162.3:c.878T>C\n",
      "NM_000162.3:c.878T>C\n",
      "44186203\n",
      "{'end': 44186206, 'tr': 1, 'allele_string': [u'T', u'C'], 'start': 44186206, 'chr': u'7', 'strand': -1}\n",
      "Maturity-onset diabetes of the young type 2 (MODY2) is an autosomal dominant inherited disease caused by heterozygous inactivating mutations in the glucokinase (GCK) gene and is characterized by mild noninsulin-dependent fasting hyperglycemia. It is treated with diet only, and complications are extremely rare. We present a report of a family with MODY2 caused by a novel NM_000162.3:c.878T>C mutation in exon 8 of the GCK gene. Testing for MODY2 and reporting all novel mutations are important to avoid difficulties in the interpretation of genetic test results and to provide fast and definitive diagnosis for all patients with this disease.\n",
      "DMD:c.10141C>T\n",
      "C\n",
      "DMD|c.10141\n",
      "[u'NM_004006.1']\n",
      "NM_004006.1:c.10141C>T\n",
      "NM_004006.1:c.10141C>T\n",
      "31196868\n",
      "SLC25A1:c.593G>A\n",
      "G\n",
      "SLC25A1|c.593\n",
      "[u'NM_005984.4']\n",
      "NM_005984.4:c.593G>A\n",
      "NM_005984.4:c.593G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "ERCC8:c.618-2A>G\n",
      "A\n",
      "ERCC8|c.618-2\n",
      "[u'NM_000082']\n",
      "NM_000082:c.618-2A>G\n",
      "NM_000082:c.618-2A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "GCNT2:g.8662T>C\n",
      "T\n",
      "GCNT2|g.8662\n",
      "[u'NG_011595.1']\n",
      "NG_011595.1:g.8662T>C\n",
      "NG_011595.1:g.8662T>C\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "GCNT2:g.8662T>C\n",
      "T\n",
      "GCNT2|g.8662\n",
      "[u'NG_011595.1']\n",
      "NG_011595.1:g.8662T>C\n",
      "NG_011595.1:g.8662T>C\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "ARV1:c.294+1G>A\n",
      "G\n",
      "ARV1|c.294+1\n",
      "[u'NP_073623.1']\n",
      "NP_073623.1:c.294+1G>A\n",
      "NP_073623.1:c.294+1G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "COL5A1:c.2A>G\n",
      "A\n",
      "COL5A1|c.2\n",
      "[u'NM_000093.3']\n",
      "NM_000093.3:c.2A>G\n",
      "NM_000093.3:c.2A>G\n",
      "137534035\n",
      "COL5A1:c.2A>G\n",
      "A\n",
      "COL5A1|c.2\n",
      "[u'NM_000093.3']\n",
      "NM_000093.3:c.2A>G\n",
      "NM_000093.3:c.2A>G\n",
      "137534035\n",
      "PAH:c.6T>A\n",
      "T\n",
      "PAH|c.6\n",
      "[u'NM_000277']\n",
      "NM_000277:c.6T>A\n",
      "NM_000277:c.6T>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PAH:c.2A>G\n",
      "A\n",
      "PAH|c.2\n",
      "[u'NM_000277']\n",
      "NM_000277:c.2A>G\n",
      "NM_000277:c.2A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PAH:c.1G>A\n",
      "G\n",
      "PAH|c.1\n",
      "[u'NM_000277']\n",
      "NM_000277:c.1G>A\n",
      "NM_000277:c.1G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PAH:c.5G>C\n",
      "G\n",
      "PAH|c.5\n",
      "[u'NM_000277']\n",
      "NM_000277:c.5G>C\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NM_000277:c.5G>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "SMARCAD1:c.378+1G>T\n",
      "G\n",
      "SMARCAD1|c.378+1\n",
      "[u'NG_031945.1']\n",
      "NG_031945.1:c.378+1G>T\n",
      "NG_031945.1:c.378+1G>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PGAP2:c.46C>T\n",
      "C\n",
      "PGAP2|c.46\n",
      "[u'NP_001243169.1']\n",
      "NP_001243169.1:c.46C>T\n",
      "NP_001243169.1:c.46C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PGAP2:c.380T>C\n",
      "T\n",
      "PGAP2|c.380\n",
      "[u'NP_001243169.1']\n",
      "NP_001243169.1:c.380T>C\n",
      "NP_001243169.1:c.380T>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PGAP2:c.479C>T\n",
      "C\n",
      "PGAP2|c.479\n",
      "[u'NP_001243169.1']\n",
      "NP_001243169.1:c.479C>T\n",
      "NP_001243169.1:c.479C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "MLH1:c.27C>A\n",
      "C\n",
      "MLH1|c.27\n",
      "[u'NM_000249.2']\n",
      "NM_000249.2:c.27C>A\n",
      "NM_000249.2:c.27C>A\n",
      "37035065\n",
      "MLH1:c.85G>T\n",
      "G\n",
      "MLH1|c.85\n",
      "[u'NM_000249.2']\n",
      "NM_000249.2:c.85G>T\n",
      "NM_000249.2:c.85G>T\n",
      "37035123\n",
      "MLH1:c.27C>A\n",
      "C\n",
      "MLH1|c.27\n",
      "[u'NM_000249.2']\n",
      "NM_000249.2:c.27C>A\n",
      "NM_000249.2:c.27C>A\n",
      "37035065\n",
      "MLH1:c.85G>T\n",
      "G\n",
      "MLH1|c.85\n",
      "[u'NM_000249.2']\n",
      "NM_000249.2:c.85G>T\n",
      "NM_000249.2:c.85G>T\n",
      "37035123\n",
      "SERPING1:c.1459C>T\n",
      "C\n",
      "SERPING1|c.1459\n",
      "[u'NG_009625']\n",
      "NG_009625:c.1459C>T\n",
      "NG_009625:c.1459C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "SERPING1:c.1174C>T\n",
      "C\n",
      "SERPING1|c.1174\n",
      "[u'NG_009625']\n",
      "NG_009625:c.1174C>T\n",
      "NG_009625:c.1174C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "LDLR:c.2140+5G>A\n",
      "G\n",
      "LDLR|c.2140+5\n",
      "[u'NM_000527']\n",
      "NM_000527:c.2140+5G>A\n",
      "NM_000527:c.2140+5G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "SIX6:c.2015T>C\n",
      "T\n",
      "SIX6|c.2015\n",
      "[u'NC_007308']\n",
      "NC_007308:c.2015T>C\n",
      "NC_007308:c.2015T>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "GRM1:c.2660+2T>G\n",
      "T\n",
      "GRM1|c.2660+2\n",
      "[u'NM_000838.3']\n",
      "NM_000838.3:c.2660+2T>G\n",
      "NM_000838.3:c.2660+2T>G\n",
      "146720837\n",
      "PLAC8:c.925G>C\n",
      "G\n",
      "PLAC8|c.925\n",
      "[u'NG_000007.3']\n",
      "NG_000007.3:c.925G>C\n",
      "NG_000007.3:c.925G>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PLAC8:c.92G>T\n",
      "G\n",
      "PLAC8|c.92\n",
      "[u'NG_000007.3']\n",
      "NG_000007.3:c.92G>T\n",
      "NG_000007.3:c.92G>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PLAC8:c.47G>A\n",
      "G\n",
      "PLAC8|c.47\n",
      "[u'NG_000007.3']\n",
      "NG_000007.3:c.47G>A\n",
      "NG_000007.3:c.47G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PLAC8:c.93G>C\n",
      "G\n",
      "PLAC8|c.93\n",
      "[u'NG_000007.3']\n",
      "NG_000007.3:c.93G>C\n",
      "NG_000007.3:c.93G>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "MTHFR:c.677C>T\n",
      "C\n",
      "MTHFR|c.677\n",
      "[u'NM_005957.4']\n",
      "NM_005957.4:c.677C>T\n",
      "NM_005957.4:c.677C>T\n",
      "11856366\n",
      "MTHFR:c.665C>T\n",
      "C\n",
      "MTHFR|c.665\n",
      "[u'NM_005957.4']\n",
      "NM_005957.4:c.665C>T\n",
      "NM_005957.4:c.665C>T\n",
      "11856378\n",
      "MTHFR:c.677C>T\n",
      "C\n",
      "MTHFR|c.677\n",
      "[u'NM_005957.4']\n",
      "NM_005957.4:c.677C>T\n",
      "NM_005957.4:c.677C>T\n",
      "11856366\n",
      "MTHFR:c.665C>T\n",
      "C\n",
      "MTHFR|c.665\n",
      "[u'NM_005957.4']\n",
      "NM_005957.4:c.665C>T\n",
      "NM_005957.4:c.665C>T\n",
      "11856378\n",
      "HBB:c.35A>G\n",
      "A\n",
      "HBB|c.35\n",
      "[u'NG_000007.3']\n",
      "NG_000007.3:c.35A>G\n",
      "NG_000007.3:c.35A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "HBB:c.142G>A\n",
      "G\n",
      "HBB|c.142\n",
      "[u'NG_000007.3']\n",
      "NG_000007.3:c.142G>A\n",
      "NG_000007.3:c.142G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "HBB:c.182G>A\n",
      "G\n",
      "HBB|c.182\n",
      "[u'NG_000007.3']\n",
      "NG_000007.3:c.182G>A\n",
      "NG_000007.3:c.182G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "LARS2:c.899C>T\n",
      "C\n",
      "LARS2|c.899\n",
      "[u'NM_015340.3']\n",
      "NM_015340.3:c.899C>T\n",
      "NM_015340.3:c.899C>T\n",
      "45518000\n",
      "LARS2:c.1912G>A\n",
      "G\n",
      "LARS2|c.1912\n",
      "[u'NM_015340.3']\n",
      "NM_015340.3:c.1912G>A\n",
      "NM_015340.3:c.1912G>A\n",
      "45557636\n",
      "KARS:c.1312C>T\n",
      "C\n",
      "KARS|c.1312\n",
      "[u'NM_005548.2']\n",
      "NM_005548.2:c.1312C>T\n",
      "NM_005548.2:c.1312C>T\n",
      "75665092\n",
      "KARS:c.1573G>A\n",
      "G\n",
      "KARS|c.1573\n",
      "[u'NM_005548.2']\n",
      "NM_005548.2:c.1573G>A\n",
      "NM_005548.2:c.1573G>A\n",
      "75662589\n",
      "PROS1:c.1907A>G\n",
      "A\n",
      "PROS1|c.1907\n",
      "[u'NM_000313.3']\n",
      "NM_000313.3:c.1907A>G\n",
      "NM_000313.3:c.1907A>G\n",
      "93593213\n",
      "PCBD1:c.449A>G\n",
      "A\n",
      "PCBD1|c.449\n",
      "[u'NM_031427.3']\n",
      "NM_031427.3:c.449A>G\n",
      "NM_031427.3:c.449A>G\n",
      "74156135\n",
      "ASNS:c.866G>C\n",
      "G\n",
      "ASNS|c.866\n",
      "[u'NM_183356.3']\n",
      "NM_183356.3:c.866G>C\n",
      "NM_183356.3:c.866G>C\n",
      "97487627\n",
      "ASNS:c.1010C>T\n",
      "C\n",
      "ASNS|c.1010\n",
      "[u'NM_183356.3']\n",
      "NM_183356.3:c.1010C>T\n",
      "NM_183356.3:c.1010C>T\n",
      "97486022\n",
      "EDA:c.866G>T\n",
      "G\n",
      "EDA|c.866\n",
      "[u'NM_001399.4']\n",
      "NM_001399.4:c.866G>T\n",
      "NM_001399.4:c.866G>T\n",
      "69253320\n",
      "EDA:c.1135T>G\n",
      "T\n",
      "EDA|c.1135\n",
      "[u'NM_001399.4']\n",
      "NM_001399.4:c.1135T>G\n",
      "NM_001399.4:c.1135T>G\n",
      "69255418\n",
      "PRPS1:c.337G>T\n",
      "G\n",
      "PRPS1|c.337\n",
      "[u'NM_002764.3']\n",
      "NM_002764.3:c.337G>T\n",
      "NM_002764.3:c.337G>T\n",
      "106884162\n",
      "PRPS1:c.343A>G\n",
      "A\n",
      "PRPS1|c.343\n",
      "[u'NM_002764.3']\n",
      "NM_002764.3:c.343A>G\n",
      "NM_002764.3:c.343A>G\n",
      "106884168\n",
      "PRPS1:c.925G>T\n",
      "G\n",
      "PRPS1|c.925\n",
      "[u'NM_002764.3']\n",
      "NM_002764.3:c.925G>T\n",
      "NM_002764.3:c.925G>T\n",
      "106893230\n",
      "PRPS1:c.343A>G\n",
      "A\n",
      "PRPS1|c.343\n",
      "[u'NM_002764.3']\n",
      "NM_002764.3:c.343A>G\n",
      "NM_002764.3:c.343A>G\n",
      "106884168\n",
      "PRPS1:c.925G>T\n",
      "G\n",
      "PRPS1|c.925\n",
      "[u'NM_002764.3']\n",
      "NM_002764.3:c.925G>T\n",
      "NM_002764.3:c.925G>T\n",
      "106893230\n",
      "VHL:c.463+1G>T\n",
      "G\n",
      "VHL|c.463+1\n",
      "[u'NM_000551.2']\n",
      "NM_000551.2:c.463+1G>T\n",
      "NM_000551.2:c.463+1G>T\n",
      "10188321\n",
      "VHL:c.233A>G\n",
      "A\n",
      "VHL|c.233\n",
      "[u'NM_000551.2']\n",
      "NM_000551.2:c.233A>G\n",
      "NM_000551.2:c.233A>G\n",
      "10183764\n",
      "VHL:c.239G>T\n",
      "G\n",
      "VHL|c.239\n",
      "[u'NM_000551.2']\n",
      "NM_000551.2:c.239G>T\n",
      "NM_000551.2:c.239G>T\n",
      "10183770\n",
      "VHL:c.319C>G\n",
      "C\n",
      "VHL|c.319\n",
      "[u'NM_000551.2']\n",
      "NM_000551.2:c.319C>G\n",
      "NM_000551.2:c.319C>G\n",
      "10183850\n",
      "VHL:c.481C>T\n",
      "C\n",
      "VHL|c.481\n",
      "[u'NM_000551.2']\n",
      "NM_000551.2:c.481C>T\n",
      "NM_000551.2:c.481C>T\n",
      "10191488\n",
      "VHL:c.482G>A\n",
      "G\n",
      "VHL|c.482\n",
      "[u'NM_000551.2']\n",
      "NM_000551.2:c.482G>A\n",
      "NM_000551.2:c.482G>A\n",
      "10191489\n",
      "VHL:c.499C>T\n",
      "C\n",
      "VHL|c.499\n",
      "[u'NM_000551.2']\n",
      "NM_000551.2:c.499C>T\n",
      "NM_000551.2:c.499C>T\n",
      "10191506\n",
      "FH:c.1413A>G\n",
      "A\n",
      "FH|c.1413\n",
      "[u'NM_001195798.1']\n",
      "NM_001195798.1:c.1413A>G\n",
      "NM_001195798.1:c.1413A>G\n",
      "11224265\n",
      "KIDINS220:c.92+5G>C\n",
      "G\n",
      "KIDINS220|c.92+5\n",
      "[u'NG_000007.3']\n",
      "NG_000007.3:c.92+5G>C\n",
      "NG_000007.3:c.92+5G>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "KIDINS220:c.138C>G\n",
      "C\n",
      "KIDINS220|c.138\n",
      "[u'NG_000007.3']\n",
      "NG_000007.3:c.138C>G\n",
      "NG_000007.3:c.138C>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "KIDINS220:c.140C>T\n",
      "C\n",
      "KIDINS220|c.140\n",
      "[u'NG_000007.3']\n",
      "NG_000007.3:c.140C>T\n",
      "NG_000007.3:c.140C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "SMARCA4:c.3277C>T\n",
      "C\n",
      "SMARCA4|c.3277\n",
      "[u'NG_011556.1']\n",
      "NG_011556.1:c.3277C>T\n",
      "NG_011556.1:c.3277C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "SMARCA4:c.3760G>T\n",
      "G\n",
      "SMARCA4|c.3760\n",
      "[u'NG_011556.1']\n",
      "NG_011556.1:c.3760G>T\n",
      "NG_011556.1:c.3760G>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PSEN1:c.839A>C\n",
      "A\n",
      "PSEN1|c.839\n",
      "[u'NM_000021.3']\n",
      "NM_000021.3:c.839A>C\n",
      "NM_000021.3:c.839A>C\n",
      "73664808\n",
      "CACNA1A:c.4110T>G\n",
      "T\n",
      "CACNA1A|c.4110\n",
      "[u'NM_000068.3']\n",
      "NM_000068.3:c.4110T>G\n",
      "NM_000068.3:c.4110T>G\n",
      "13372416\n",
      "GPAT4:g.152G>C\n",
      "G\n",
      "GPAT4|g.152\n",
      "[u'NC_007328.3']\n",
      "NC_007328.3:g.152G>C\n",
      "NC_007328.3:g.152G>C\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "GPAT4:c.8124G>A\n",
      "G\n",
      "GPAT4|c.8124\n",
      "[u'NC_007328.3']\n",
      "NC_007328.3:c.8124G>A\n",
      "NC_007328.3:c.8124G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "GPAT4:c.9263C>G\n",
      "C\n",
      "GPAT4|c.9263\n",
      "[u'NC_007328.3']\n",
      "NC_007328.3:c.9263C>G\n",
      "NC_007328.3:c.9263C>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "GPAT4:c.16436G>A\n",
      "G\n",
      "GPAT4|c.16436\n",
      "[u'NC_007328.3']\n",
      "NC_007328.3:c.16436G>A\n",
      "NC_007328.3:c.16436G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "GPAT4:c.9263C>G\n",
      "C\n",
      "GPAT4|c.9263\n",
      "[u'NC_007328.3']\n",
      "NC_007328.3:c.9263C>G\n",
      "NC_007328.3:c.9263C>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "GPAT4:c.9263C>G\n",
      "C\n",
      "GPAT4|c.9263\n",
      "[u'NC_007328.3']\n",
      "NC_007328.3:c.9263C>G\n",
      "NC_007328.3:c.9263C>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "SSB:c.552-2A>G\n",
      "A\n",
      "SSB|c.552-2\n",
      "[u'NM_000194.2']\n",
      "NM_000194.2:c.552-2A>G\n",
      "NM_000194.2:c.552-2A>G\n",
      "133632655\n",
      "mtdna:m.13094T>C\n",
      "T\n",
      "mtdna|m.13094\n",
      "[u'NC_012920.1']\n",
      "NC_012920.1:m.13094T>C\n",
      "NC_012920.1:m.13094T>C\n",
      "Could not parse: [\"NC_012920.1(COX2_v001):c.*4825T>C\", \"NC_012920.1(TRNK_v001):n.*4730T>C\", \"NC_012920.1(ATP8_v001):c.*4522T>C\", \"NC_012920.1(ATP6_v001):c.*3887T>C\", \"NC_012920.1(COX3_v001):c.*3104T>C\", \"NC_012920.1(TRNG_v001):n.*3036T>C\", \"NC_012920.1(ND3_v001):c.*2690T>C\", \"NC_012920.1(TRNR_v001):n.*2625T>C\", \"NC_012920.1(ND4L_v001):c.*2328T>C\", \"NC_012920.1(ND4_v001):c.*957T>C\", \"NC_012920.1(TRNH_v001):n.*888T>C\", \"NC_012920.1(TRNS2_v001):n.*829T>C\", \"NC_012920.1(TRNL2_v001):n.*758T>C\", \"NC_012920.1(ND5_v001):c.758T>C\", \"NC_012920.1(ND6_v001):c.*1055A>G\", \"NC_012920.1(TRNE_v001):n.*1580A>G\", \"NC_012920.1(CYTB_v001):c.-1653T>C\", \"NC_012920.1(TRNT_v001):n.-2794T>C\", \"NC_012920.1(TRNP_v001):n.*2862A>G\"]\n",
      "None\n",
      "NDUFV1:c.1162+4A>C\n",
      "A\n",
      "NDUFV1|c.1162+4\n",
      "[u'NM_007103.3']\n",
      "NM_007103.3:c.1162+4A>C\n",
      "NM_007103.3:c.1162+4A>C\n",
      "67379453\n",
      "NDUFV1:c.640G>A\n",
      "G\n",
      "NDUFV1|c.640\n",
      "[u'NM_007103.3']\n",
      "NM_007103.3:c.640G>A\n",
      "NM_007103.3:c.640G>A\n",
      "67377981\n",
      "GALNT12:c.907G>A\n",
      "G\n",
      "GALNT12|c.907\n",
      "[u'NM_024642.4']\n",
      "NM_024642.4:c.907G>A\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NM_024642.4:c.907G>A\n",
      "101594229\n",
      "GALNT12:c.907G>A\n",
      "G\n",
      "GALNT12|c.907\n",
      "[u'NM_024642.4']\n",
      "NM_024642.4:c.907G>A\n",
      "NM_024642.4:c.907G>A\n",
      "101594229\n",
      "GALNT12:c.907G>A\n",
      "G\n",
      "GALNT12|c.907\n",
      "[u'NM_024642.4']\n",
      "NM_024642.4:c.907G>A\n",
      "NM_024642.4:c.907G>A\n",
      "101594229\n",
      "SPARC:c.497G>A\n",
      "G\n",
      "SPARC|c.497\n",
      "[u'NM_003118.3']\n",
      "NM_003118.3:c.497G>A\n",
      "NM_003118.3:c.497G>A\n",
      "151047116\n",
      "SPARC:c.787G>A\n",
      "G\n",
      "SPARC|c.787\n",
      "[u'NM_003118.3']\n",
      "NM_003118.3:c.787G>A\n",
      "NM_003118.3:c.787G>A\n",
      "151043744\n",
      "COG6:c.1167-24A>G\n",
      "A\n",
      "COG6|c.1167-24\n",
      "[u'NM_020751.2']\n",
      "NM_020751.2:c.1167-24A>G\n",
      "NM_020751.2:c.1167-24A>G\n",
      "40273614\n",
      "OCA2:g.1124477G>A\n",
      "G\n",
      "OCA2|g.1124477\n",
      "[u'NW_001494061']\n",
      "NW_001494061:g.1124477G>A\n",
      "NW_001494061:g.1124477G>A\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "OCA2:g.1118561T>C\n",
      "T\n",
      "OCA2|g.1118561\n",
      "[u'NW_001494061']\n",
      "NW_001494061:g.1118561T>C\n",
      "NW_001494061:g.1118561T>C\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "AMBN:g.13081786G>A\n",
      "G\n",
      "AMBN|g.13081786\n",
      "[u'NM_014208.3']\n",
      "NM_014208.3:g.13081786G>A\n",
      "NM_014208.3:g.13081786G>A\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "AMBN:c.727G>A\n",
      "G\n",
      "AMBN|c.727\n",
      "[u'NM_014208.3']\n",
      "NM_014208.3:c.727G>A\n",
      "NM_014208.3:c.727G>A\n",
      "88534065\n",
      "PITX2:g.18117T>C\n",
      "T\n",
      "PITX2|g.18117\n",
      "[u'AC_000163']\n",
      "AC_000163:g.18117T>C\n",
      "AC_000163:g.18117T>C\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "PITX2:g.18161C>G\n",
      "C\n",
      "PITX2|g.18161\n",
      "[u'AC_000163']\n",
      "AC_000163:g.18161C>G\n",
      "AC_000163:g.18161C>G\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "PITX2:g.18322C>A\n",
      "C\n",
      "PITX2|g.18322\n",
      "[u'AC_000163']\n",
      "AC_000163:g.18322C>A\n",
      "AC_000163:g.18322C>A\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "PITX2:g.18353T>C\n",
      "T\n",
      "PITX2|g.18353\n",
      "[u'AC_000163']\n",
      "AC_000163:g.18353T>C\n",
      "AC_000163:g.18353T>C\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "PITX2:c.79C>A\n",
      "C\n",
      "PITX2|c.79\n",
      "[u'AC_000163']\n",
      "AC_000163:c.79C>A\n",
      "AC_000163:c.79C>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PITX2:c.110T>C\n",
      "T\n",
      "PITX2|c.110\n",
      "[u'AC_000163']\n",
      "AC_000163:c.110T>C\n",
      "AC_000163:c.110T>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "HSF4:c.1213C>T\n",
      "C\n",
      "HSF4|c.1213\n",
      "[u'NM_001040667']\n",
      "NM_001040667:c.1213C>T\n",
      "NM_001040667:c.1213C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PRDM16:g.577G>T\n",
      "G\n",
      "PRDM16|g.577\n",
      "[u'NC_007314.3']\n",
      "NC_007314.3:g.577G>T\n",
      "NC_007314.3:g.577G>T\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "PRDM16:c.614T>C\n",
      "T\n",
      "PRDM16|c.614\n",
      "[u'NC_007314.3']\n",
      "NC_007314.3:c.614T>C\n",
      "NC_007314.3:c.614T>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PRDM16:c.212237T>C\n",
      "T\n",
      "PRDM16|c.212237\n",
      "[u'NC_007314.3']\n",
      "NC_007314.3:c.212237T>C\n",
      "NC_007314.3:c.212237T>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "LMNA:c.1774G>A\n",
      "G\n",
      "LMNA|c.1774\n",
      "[u'NM_170707.3']\n",
      "NM_170707.3:c.1774G>A\n",
      "NM_170707.3:c.1774G>A\n",
      "156108354\n",
      "EDAR:c.361C>T\n",
      "C\n",
      "EDAR|c.361\n",
      "[u'NM_080738']\n",
      "NM_080738:c.361C>T\n",
      "NM_080738:c.361C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PSMB5:g.417C>G\n",
      "C\n",
      "PSMB5|g.417\n",
      "[u'NM_000546.5']\n",
      "NM_000546.5:g.417C>G\n",
      "NM_000546.5:g.417C>G\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "CD55:g.12345T>C\n",
      "T\n",
      "CD55|g.12345\n",
      "[u'NC_007300']\n",
      "NC_007300:g.12345T>C\n",
      "NC_007300:g.12345T>C\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "MTHFR:c.677C>T\n",
      "C\n",
      "MTHFR|c.677\n",
      "[u'NM_005957']\n",
      "NM_005957:c.677C>T\n",
      "NM_005957:c.677C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "MTHFR:c.677C>T\n",
      "C\n",
      "MTHFR|c.677\n",
      "[u'NM_005957']\n",
      "NM_005957:c.677C>T\n",
      "NM_005957:c.677C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "Records: 30000\n",
      "GJB1:c.459C>T\n",
      "C\n",
      "GJB1|c.459\n",
      "[u'NM_000166']\n",
      "NM_000166:c.459C>T\n",
      "NM_000166:c.459C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "GJB1:c.713G>A\n",
      "G\n",
      "GJB1|c.713\n",
      "[u'NM_000166']\n",
      "NM_000166:c.713G>A\n",
      "NM_000166:c.713G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "GJB1:c.459C>T\n",
      "C\n",
      "GJB1|c.459\n",
      "[u'NM_000166']\n",
      "NM_000166:c.459C>T\n",
      "NM_000166:c.459C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "GJB1:c.459C>T\n",
      "C\n",
      "GJB1|c.459\n",
      "[u'NM_000166']\n",
      "NM_000166:c.459C>T\n",
      "NM_000166:c.459C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "HFE:c.845G>A\n",
      "G\n",
      "HFE|c.845\n",
      "[u'NC_000006.10']\n",
      "NC_000006.10:c.845G>A\n",
      "NC_000006.10:c.845G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "HFE:c.845G>A\n",
      "G\n",
      "HFE|c.845\n",
      "[u'NC_000006.10']\n",
      "NC_000006.10:c.845G>A\n",
      "NC_000006.10:c.845G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "HFE:c.187C>G\n",
      "C\n",
      "HFE|c.187\n",
      "[u'NC_000006.10']\n",
      "NC_000006.10:c.187C>G\n",
      "NC_000006.10:c.187C>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "HFE:c.128G>A\n",
      "G\n",
      "HFE|c.128\n",
      "[u'NC_000006.10']\n",
      "NC_000006.10:c.128G>A\n",
      "NC_000006.10:c.128G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "HFE:c.187C>G\n",
      "C\n",
      "HFE|c.187\n",
      "[u'NC_000006.10']\n",
      "NC_000006.10:c.187C>G\n",
      "NC_000006.10:c.187C>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "HFE:c.128G>A\n",
      "G\n",
      "HFE|c.128\n",
      "[u'NC_000006.10']\n",
      "NC_000006.10:c.128G>A\n",
      "NC_000006.10:c.128G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "HFE:c.187C>G\n",
      "C\n",
      "HFE|c.187\n",
      "[u'NC_000006.10']\n",
      "NC_000006.10:c.187C>G\n",
      "NC_000006.10:c.187C>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "HFE:c.845G>A\n",
      "G\n",
      "HFE|c.845\n",
      "[u'NC_000006.10']\n",
      "NC_000006.10:c.845G>A\n",
      "NC_000006.10:c.845G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PTEN:c.545T>C\n",
      "T\n",
      "PTEN|c.545\n",
      "[u'NM_000314.4']\n",
      "NM_000314.4:c.545T>C\n",
      "NM_000314.4:c.545T>C\n",
      "89711927\n",
      "SACS:c.3491T>A\n",
      "T\n",
      "SACS|c.3491\n",
      "[u'NM_014363.3']\n",
      "NM_014363.3:c.3491T>A\n",
      "NM_014363.3:c.3491T>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "ALG9:c.1173+2T>A\n",
      "T\n",
      "ALG9|c.1173+2\n",
      "[u'NM_024740.2']\n",
      "NM_024740.2:c.1173+2T>A\n",
      "NM_024740.2:c.1173+2T>A\n",
      "111711377\n",
      "{'end': 111711376, 'tr': 2, 'allele_string': [u'T', u'A'], 'start': 111711376, 'chr': u'11', 'strand': -1}\n",
      "A rare lethal autosomal recessive syndrome with skeletal dysplasia, polycystic kidneys and multiple malformations was first described by Gillessen-Kaesbach et al and subsequently by Nishimura et al. The skeletal features uniformly comprise a round pelvis, mesomelic shortening of the upper limbs and defective ossification of the cervical spine. We studied two unrelated families including three affected fetuses with Gillessen-Kaesbach-Nishimura syndrome using whole-exome and Sanger sequencing, comparative genome hybridization and homozygosity mapping. All affected patients were shown to have a novel homozygous splice variant NM_024740.2: c.1173+2T>A in the ALG9 gene, encoding alpha-1,2-mannosyltransferase, involved in the formation of the lipid-linked oligosaccharide precursor of N-glycosylation. RNA analysis demonstrated skipping of exon 10, leading to shorter RNA. Mass spectrometric analysis showed an increase in monoglycosylated transferrin as compared with control tissues, confirming that this is a congenital disorder of glycosylation (CDG). Only three liveborn children with ALG9-CDG have been previously reported, all with missense variants. All three suffered from intellectual disability, muscular hypotonia, microcephaly and renal cysts, but none had skeletal dysplasia. Our study shows that some pathogenic variants in ALG9 can present as a lethal skeletal dysplasia with visceral malformations as the most severe phenotype. The skeletal features overlap with that previously reported for ALG3- and ALG12-CDG, suggesting that this subset of glycosylation disorders constitutes a new diagnostic group of skeletal dysplasias.\n",
      "SERPINE1:g.566G>A\n",
      "G\n",
      "SERPINE1|g.566\n",
      "[u'NM_213910']\n",
      "NM_213910:g.566G>A\n",
      "NM_213910:g.566G>A\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "SERPINE1:c.612A>G\n",
      "A\n",
      "SERPINE1|c.612\n",
      "[u'NM_213910']\n",
      "NM_213910:c.612A>G\n",
      "NM_213910:c.612A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "mtdna:m.6267G>A\n",
      "G\n",
      "mtdna|m.6267\n",
      "[u'NP_536845.1']\n",
      "NP_536845.1:m.6267G>A\n",
      "NP_536845.1:m.6267G>A\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "mtdna:m.6267G>A\n",
      "G\n",
      "mtdna|m.6267\n",
      "[u'NP_536845.1']\n",
      "NP_536845.1:m.6267G>A\n",
      "NP_536845.1:m.6267G>A\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "mtdna:m.6267G>A\n",
      "G\n",
      "mtdna|m.6267\n",
      "[u'NP_536845.1']\n",
      "NP_536845.1:m.6267G>A\n",
      "NP_536845.1:m.6267G>A\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "mtdna:m.6267G>A\n",
      "G\n",
      "mtdna|m.6267\n",
      "[u'NP_536845.1']\n",
      "NP_536845.1:m.6267G>A\n",
      "NP_536845.1:m.6267G>A\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "mtdna:m.6267G>A\n",
      "G\n",
      "mtdna|m.6267\n",
      "[u'NP_536845.1']\n",
      "NP_536845.1:m.6267G>A\n",
      "NP_536845.1:m.6267G>A\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "MALT1:c.1019-2A>G\n",
      "A\n",
      "MALT1|c.1019-2\n",
      "[u'NM_006785.3']\n",
      "NM_006785.3:c.1019-2A>G\n",
      "NM_006785.3:c.1019-2A>G\n",
      "56390278\n",
      "CFTR:c.3718-2477C>T\n",
      "C\n",
      "CFTR|c.3718-2477\n",
      "[u'NM_000492.2']\n",
      "NM_000492.2:c.3718-2477C>T\n",
      "NM_000492.2:c.3718-2477C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CFTR:c.1558G>T\n",
      "G\n",
      "CFTR|c.1558\n",
      "[u'NM_000492.2']\n",
      "NM_000492.2:c.1558G>T\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NM_000492.2:c.1558G>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CFTR:c.1364C>A\n",
      "C\n",
      "CFTR|c.1364\n",
      "[u'NM_000492.2']\n",
      "NM_000492.2:c.1364C>A\n",
      "NM_000492.2:c.1364C>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CFTR:c.1040G>C\n",
      "G\n",
      "CFTR|c.1040\n",
      "[u'NM_000492.2']\n",
      "NM_000492.2:c.1040G>C\n",
      "NM_000492.2:c.1040G>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CFTR:c.3846G>A\n",
      "G\n",
      "CFTR|c.3846\n",
      "[u'NM_000492.2']\n",
      "NM_000492.2:c.3846G>A\n",
      "NM_000492.2:c.3846G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CFTR:c.350G>A\n",
      "G\n",
      "CFTR|c.350\n",
      "[u'NM_000492.2']\n",
      "NM_000492.2:c.350G>A\n",
      "NM_000492.2:c.350G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CFTR:c.254G>A\n",
      "G\n",
      "CFTR|c.254\n",
      "[u'NM_000492.2']\n",
      "NM_000492.2:c.254G>A\n",
      "NM_000492.2:c.254G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CFTR:c.1G>T\n",
      "G\n",
      "CFTR|c.1\n",
      "[u'NM_000492.2']\n",
      "NM_000492.2:c.1G>T\n",
      "NM_000492.2:c.1G>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CFTR:c.1G>T\n",
      "G\n",
      "CFTR|c.1\n",
      "[u'NM_000492.2']\n",
      "NM_000492.2:c.1G>T\n",
      "NM_000492.2:c.1G>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CFTR:c.5G>A\n",
      "G\n",
      "CFTR|c.5\n",
      "[u'NM_000492.2']\n",
      "NM_000492.2:c.5G>A\n",
      "NM_000492.2:c.5G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CFTR:c.5G>A\n",
      "G\n",
      "CFTR|c.5\n",
      "[u'NM_000492.2']\n",
      "NM_000492.2:c.5G>A\n",
      "NM_000492.2:c.5G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "NR3C2:c.1A>T\n",
      "A\n",
      "NR3C2|c.1\n",
      "[u'NM_006517']\n",
      "NM_006517:c.1A>T\n",
      "NM_006517:c.1A>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "NR3C2:c.1673G>A\n",
      "G\n",
      "NR3C2|c.1673\n",
      "[u'NM_006517']\n",
      "NM_006517:c.1673G>A\n",
      "NM_006517:c.1673G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "NR3C2:c.1673G>A\n",
      "G\n",
      "NR3C2|c.1673\n",
      "[u'NM_006517']\n",
      "NM_006517:c.1673G>A\n",
      "NM_006517:c.1673G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "NR3C2:c.1A>T\n",
      "A\n",
      "NR3C2|c.1\n",
      "[u'NM_006517']\n",
      "NM_006517:c.1A>T\n",
      "NM_006517:c.1A>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "NR3C2:c.1A>T\n",
      "A\n",
      "NR3C2|c.1\n",
      "[u'NM_006517']\n",
      "NM_006517:c.1A>T\n",
      "NM_006517:c.1A>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "MCF2L:c.2066A>G\n",
      "A\n",
      "MCF2L|c.2066\n",
      "[u'NM_001112732.1']\n",
      "NM_001112732.1:c.2066A>G\n",
      "NM_001112732.1:c.2066A>G\n",
      "113736760\n",
      "{'end': 113739210, 'tr': 1, 'allele_string': [u'A', u'G'], 'start': 113739210, 'chr': u'13', 'strand': 1}\n",
      "Cardiovascular disease (CVD) is a major cause of death in Western societies. CVD risk is largely genetically determined. The molecular pathology is, however, not elucidated in a large number of families suffering from CVD. We applied exclusion linkage analysis and next-generation sequencing to elucidate the molecular defect underlying premature CVD in a small pedigree, comprising two generations of which six members suffered from premature CVD. A total of three variants showed co-segregation with the disease status in the family. Two of these variants were excluded from further analysis based on the prevalence in replication cohorts, whereas a non-synonymous variant in MCF.2 Cell Line Derived Transforming Sequence-like protein (MCF2L, c.2066A>G; p.(Asp689Gly); NM_001112732.1), located in the DH domain, was only present in the studied family. MCF2L is a guanine exchange factor that potentially links pathways that signal through Rac1 and RhoA. Indeed, in HeLa cells, MCF2L689Gly failed to activate Rac1 as well as RhoA, resulting in impaired stress fiber formation. Moreover, MCF2L protein was found in human atherosclerotic lesions but not in healthy tissue segments. In conclusion, a rare functional variant in MCF2L, leading to impaired DH function, was identified in a small pedigree with premature CVD. The presence of MCF2L in human atherosclerotic plaque specimen lends further support to its potential role in atherosclerosis.\n",
      "MAOA:g.80340C>T\n",
      "C\n",
      "MAOA|g.80340\n",
      "[u'NC_007331.3']\n",
      "NC_007331.3:g.80340C>T\n",
      "NC_007331.3:g.80340C>T\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "FGFR2:c.890G>C\n",
      "G\n",
      "FGFR2|c.890\n",
      "[u'NM_000141']\n",
      "NM_000141:c.890G>C\n",
      "NM_000141:c.890G>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "SH2D1A:c.47G>A\n",
      "G\n",
      "SH2D1A|c.47\n",
      "[u'NP_002342']\n",
      "NP_002342:c.47G>A\n",
      "NP_002342:c.47G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "GPR143:g.25288G>A\n",
      "G\n",
      "GPR143|g.25288\n",
      "[u'NC_000023.8']\n",
      "NC_000023.8:g.25288G>A\n",
      "NC_000023.8:g.25288G>A\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "CS:c.34G>A\n",
      "G\n",
      "CS|c.34\n",
      "[u'NM_005343.2']\n",
      "NM_005343.2:c.34G>A\n",
      "NM_005343.2:c.34G>A\n",
      "534289\n",
      "{'end': 56693952, 'tr': 3, 'allele_string': [u'G', u'A'], 'start': 56693952, 'chr': u'12', 'strand': -1}\n",
      "Costello syndrome (CS) is a rare congenital condition caused by heterozygous de novo missense mutations affecting the codon for glycine 12 or 13 of the HRAS gene. We have identified 39 CS patients harboring the p.Gly12Ser mutation (NM_005343.2:c.34 G > A), two patients with c.35G > C mutations resulting in p.Gly12Ala substitutions, and one patient carrying the p.Gly13Cys substitution (c.37G > A). We analyzed the region flanking the mutated sites in 42 probands and 59 parents, and used four polymorphic markers to trace the parental origin of the germline mutations: one highly polymorphic hexanucleotide (GGGCCT) repeat region, defining three alleles with different numbers of repeat units (two, three, or four), and three SNPs. One of the SNPs, rs12628 (c.81T > C), was found in strong linkage disequilibrium with the hexanucleotide repeat region. Out of a total of 24 probands with polymorphic markers, 16 informative families were tested and the paternal origin of the germline mutation was found in 14 CS probands; a distribution that is neither consistent with an equal likelihood of mutations arising in either parent (P = 0.0018), nor with exclusive paternal origin.\n",
      "CS:c.35G>C\n",
      "G\n",
      "CS|c.35\n",
      "[u'NM_005343.2']\n",
      "NM_005343.2:c.35G>C\n",
      "NM_005343.2:c.35G>C\n",
      "534288\n",
      "{'end': 56693951, 'tr': 3, 'allele_string': [u'G', u'C'], 'start': 56693951, 'chr': u'12', 'strand': -1}\n",
      "Costello syndrome (CS) is a rare congenital condition caused by heterozygous de novo missense mutations affecting the codon for glycine 12 or 13 of the HRAS gene. We have identified 39 CS patients harboring the p.Gly12Ser mutation (NM_005343.2:c.34 G > A), two patients with c.35G > C mutations resulting in p.Gly12Ala substitutions, and one patient carrying the p.Gly13Cys substitution (c.37G > A). We analyzed the region flanking the mutated sites in 42 probands and 59 parents, and used four polymorphic markers to trace the parental origin of the germline mutations: one highly polymorphic hexanucleotide (GGGCCT) repeat region, defining three alleles with different numbers of repeat units (two, three, or four), and three SNPs. One of the SNPs, rs12628 (c.81T > C), was found in strong linkage disequilibrium with the hexanucleotide repeat region. Out of a total of 24 probands with polymorphic markers, 16 informative families were tested and the paternal origin of the germline mutation was found in 14 CS probands; a distribution that is neither consistent with an equal likelihood of mutations arising in either parent (P = 0.0018), nor with exclusive paternal origin.\n",
      "CS:c.37G>A\n",
      "G\n",
      "CS|c.37\n",
      "[u'NM_005343.2']\n",
      "NM_005343.2:c.37G>A\n",
      "NM_005343.2:c.37G>A\n",
      "534286\n",
      "CS:c.81T>C\n",
      "T\n",
      "CS|c.81\n",
      "[u'NM_005343.2']\n",
      "NM_005343.2:c.81T>C\n",
      "NM_005343.2:c.81T>C\n",
      "534242\n",
      "{'end': 56680391, 'tr': 2, 'allele_string': [u'T', u'C'], 'start': 56680391, 'chr': u'12', 'strand': -1}\n",
      "Costello syndrome (CS) is a rare congenital condition caused by heterozygous de novo missense mutations affecting the codon for glycine 12 or 13 of the HRAS gene. We have identified 39 CS patients harboring the p.Gly12Ser mutation (NM_005343.2:c.34 G > A), two patients with c.35G > C mutations resulting in p.Gly12Ala substitutions, and one patient carrying the p.Gly13Cys substitution (c.37G > A). We analyzed the region flanking the mutated sites in 42 probands and 59 parents, and used four polymorphic markers to trace the parental origin of the germline mutations: one highly polymorphic hexanucleotide (GGGCCT) repeat region, defining three alleles with different numbers of repeat units (two, three, or four), and three SNPs. One of the SNPs, rs12628 (c.81T > C), was found in strong linkage disequilibrium with the hexanucleotide repeat region. Out of a total of 24 probands with polymorphic markers, 16 informative families were tested and the paternal origin of the germline mutation was found in 14 CS probands; a distribution that is neither consistent with an equal likelihood of mutations arising in either parent (P = 0.0018), nor with exclusive paternal origin.\n",
      "SEPT9:c.278C>T\n",
      "C\n",
      "SEPT9|c.278\n",
      "[u'NM_006640.3']\n",
      "NM_006640.3:c.278C>T\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NM_006640.3:c.278C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "SEPT9:c.262C>T\n",
      "C\n",
      "SEPT9|c.262\n",
      "[u'NM_006640.3']\n",
      "NM_006640.3:c.262C>T\n",
      "NM_006640.3:c.262C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PCYT1A:c.12A>G\n",
      "A\n",
      "PCYT1A|c.12\n",
      "[u'NM_001046020']\n",
      "NM_001046020:c.12A>G\n",
      "NM_001046020:c.12A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PCYT1A:c.276C>T\n",
      "C\n",
      "PCYT1A|c.276\n",
      "[u'NM_001046020']\n",
      "NM_001046020:c.276C>T\n",
      "NM_001046020:c.276C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PCYT1A:c.321A>G\n",
      "A\n",
      "PCYT1A|c.321\n",
      "[u'NM_001046020']\n",
      "NM_001046020:c.321A>G\n",
      "NM_001046020:c.321A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PCYT1A:c.396C>T\n",
      "C\n",
      "PCYT1A|c.396\n",
      "[u'NM_001046020']\n",
      "NM_001046020:c.396C>T\n",
      "NM_001046020:c.396C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PCYT1A:c.12A>G\n",
      "A\n",
      "PCYT1A|c.12\n",
      "[u'NM_001046020']\n",
      "NM_001046020:c.12A>G\n",
      "NM_001046020:c.12A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PCYT1A:c.321A>G\n",
      "A\n",
      "PCYT1A|c.321\n",
      "[u'NM_001046020']\n",
      "NM_001046020:c.321A>G\n",
      "NM_001046020:c.321A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PCYT1A:c.396C>T\n",
      "C\n",
      "PCYT1A|c.396\n",
      "[u'NM_001046020']\n",
      "NM_001046020:c.396C>T\n",
      "NM_001046020:c.396C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PCYT1A:c.276C>T\n",
      "C\n",
      "PCYT1A|c.276\n",
      "[u'NM_001046020']\n",
      "NM_001046020:c.276C>T\n",
      "NM_001046020:c.276C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PCYT1A:c.276C>T\n",
      "C\n",
      "PCYT1A|c.276\n",
      "[u'NM_001046020']\n",
      "NM_001046020:c.276C>T\n",
      "NM_001046020:c.276C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "PCYT1A:c.276C>T\n",
      "C\n",
      "PCYT1A|c.276\n",
      "[u'NM_001046020']\n",
      "NM_001046020:c.276C>T\n",
      "NM_001046020:c.276C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "GHRHR:g.5203C>T\n",
      "C\n",
      "GHRHR|g.5203\n",
      "[u'NC_007302']\n",
      "NC_007302:g.5203C>T\n",
      "NC_007302:g.5203C>T\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "GHRHR:c.7307C>G\n",
      "C\n",
      "GHRHR|c.7307\n",
      "[u'NC_007302']\n",
      "NC_007302:c.7307C>G\n",
      "NC_007302:c.7307C>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "GHRHR:c.9583G>A\n",
      "G\n",
      "GHRHR|c.9583\n",
      "[u'NC_007302']\n",
      "NC_007302:c.9583G>A\n",
      "NC_007302:c.9583G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "GHRHR:c.9668A>C\n",
      "A\n",
      "GHRHR|c.9668\n",
      "[u'NC_007302']\n",
      "NC_007302:c.9668A>C\n",
      "NC_007302:c.9668A>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "GHRHR:c.9583G>A\n",
      "G\n",
      "GHRHR|c.9583\n",
      "[u'NC_007302']\n",
      "NC_007302:c.9583G>A\n",
      "NC_007302:c.9583G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "GHRHR:c.9668A>C\n",
      "A\n",
      "GHRHR|c.9668\n",
      "[u'NC_007302']\n",
      "NC_007302:c.9668A>C\n",
      "NC_007302:c.9668A>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CHD4:c.3380G>A\n",
      "G\n",
      "CHD4|c.3380\n",
      "[u'NM_001273.3']\n",
      "NM_001273.3:c.3380G>A\n",
      "NM_001273.3:c.3380G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CHD4:c.3443G>T\n",
      "G\n",
      "CHD4|c.3443\n",
      "[u'NM_001273.3']\n",
      "NM_001273.3:c.3443G>T\n",
      "NM_001273.3:c.3443G>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CHD4:c.3518G>T\n",
      "G\n",
      "CHD4|c.3518\n",
      "[u'NM_001273.3']\n",
      "NM_001273.3:c.3518G>T\n",
      "NM_001273.3:c.3518G>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CHD4:c.3008G>A\n",
      "G\n",
      "CHD4|c.3008\n",
      "[u'NM_001273.3']\n",
      "NM_001273.3:c.3008G>A\n",
      "NM_001273.3:c.3008G>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "NRP2:c.1938-21T>C\n",
      "T\n",
      "NRP2|c.1938-21\n",
      "[u'XM_371590']\n",
      "XM_371590:c.1938-21T>C\n",
      "XM_371590:c.1938-21T>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "NRP2:c.1938-21T>C\n",
      "T\n",
      "NRP2|c.1938-21\n",
      "[u'XM_371590']\n",
      "XM_371590:c.1938-21T>C\n",
      "XM_371590:c.1938-21T>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "IDH1:c.933-28C>T\n",
      "C\n",
      "IDH1|c.933-28\n",
      "[u'XM_371590']\n",
      "XM_371590:c.933-28C>T\n",
      "XM_371590:c.933-28C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "Records: 40000\n",
      "ALOXE3:c.814C>T\n",
      "C\n",
      "ALOXE3|c.814\n",
      "[u'NM_001165960.1']\n",
      "NM_001165960.1:c.814C>T\n",
      "NM_001165960.1:c.814C>T\n",
      "8018941\n",
      "CTSD:g.569834T>C\n",
      "T\n",
      "CTSD|g.569834\n",
      "[u'NT_009237.17']\n",
      "NT_009237.17:g.569834T>C\n",
      "NT_009237.17:g.569834T>C\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "MGMT:c.56C>T\n",
      "C\n",
      "MGMT|c.56\n",
      "[u'NM_002412.2']\n",
      "NM_002412.2:c.56C>T\n",
      "NM_002412.2:c.56C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "MGMT:c.56C>T\n",
      "C\n",
      "MGMT|c.56\n",
      "[u'NM_002412.2']\n",
      "NM_002412.2:c.56C>T\n",
      "NM_002412.2:c.56C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "MGMT:c.56C>T\n",
      "C\n",
      "MGMT|c.56\n",
      "[u'NM_002412.2']\n",
      "NM_002412.2:c.56C>T\n",
      "NM_002412.2:c.56C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "GIF:g.68A>G\n",
      "A\n",
      "GIF|g.68\n",
      "[u'NM_005142']\n",
      "NM_005142:g.68A>G\n",
      "NM_005142:g.68A>G\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "LMLN:g.67C>G\n",
      "C\n",
      "LMLN|g.67\n",
      "[u'NM_005142']\n",
      "NM_005142:g.67C>G\n",
      "NM_005142:g.67C>G\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "GIF:g.68A>G\n",
      "A\n",
      "GIF|g.68\n",
      "[u'NM_005142']\n",
      "NM_005142:g.68A>G\n",
      "NM_005142:g.68A>G\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "GIF:g.68A>G\n",
      "A\n",
      "GIF|g.68\n",
      "[u'NM_005142']\n",
      "NM_005142:g.68A>G\n",
      "NM_005142:g.68A>G\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "GIF:g.68A>G\n",
      "A\n",
      "GIF|g.68\n",
      "[u'NM_005142']\n",
      "NM_005142:g.68A>G\n",
      "NM_005142:g.68A>G\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "LMLN:g.67C>G\n",
      "C\n",
      "LMLN|g.67\n",
      "[u'NM_005142']\n",
      "NM_005142:g.67C>G\n",
      "NM_005142:g.67C>G\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "GIF:g.68A>G\n",
      "A\n",
      "GIF|g.68\n",
      "[u'NM_005142']\n",
      "NM_005142:g.68A>G\n",
      "NM_005142:g.68A>G\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "GIF:g.68A>G\n",
      "A\n",
      "GIF|g.68\n",
      "[u'NM_005142']\n",
      "NM_005142:g.68A>G\n",
      "NM_005142:g.68A>G\n",
      "DATA RETURNED: None\n",
      "TEXT RETURNED: null\n",
      "None\n",
      "CYP3A5:c.13108T>C\n",
      "T\n",
      "CYP3A5|c.13108\n",
      "[u'NG_000004.2']\n",
      "NG_000004.2:c.13108T>C\n",
      "NG_000004.2:c.13108T>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CYP3A5:c.16025A>G\n",
      "A\n",
      "CYP3A5|c.16025\n",
      "[u'NG_000004.2']\n",
      "NG_000004.2:c.16025A>G\n",
      "NG_000004.2:c.16025A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CYP3A5:c.16903A>G\n",
      "A\n",
      "CYP3A5|c.16903\n",
      "[u'NG_000004.2']\n",
      "NG_000004.2:c.16903A>G\n",
      "NG_000004.2:c.16903A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CYP3A5:c.16993C>G\n",
      "C\n",
      "CYP3A5|c.16993\n",
      "[u'NG_000004.2']\n",
      "NG_000004.2:c.16993C>G\n",
      "NG_000004.2:c.16993C>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CYP3A5:c.27448C>A\n",
      "C\n",
      "CYP3A5|c.27448\n",
      "[u'NG_000004.2']\n",
      "NG_000004.2:c.27448C>A\n",
      "NG_000004.2:c.27448C>A\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CYP3A5:c.29782A>G\n",
      "A\n",
      "CYP3A5|c.29782\n",
      "[u'NG_000004.2']\n",
      "NG_000004.2:c.29782A>G\n",
      "NG_000004.2:c.29782A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CYP3A5:c.31551T>C\n",
      "T\n",
      "CYP3A5|c.31551\n",
      "[u'NG_000004.2']\n",
      "NG_000004.2:c.31551T>C\n",
      "NG_000004.2:c.31551T>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CYP3A5:c.6986A>G\n",
      "A\n",
      "CYP3A5|c.6986\n",
      "[u'NG_000004.2']\n",
      "NG_000004.2:c.6986A>G\n",
      "NG_000004.2:c.6986A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CYP3A5:c.29782A>G\n",
      "A\n",
      "CYP3A5|c.29782\n",
      "[u'NG_000004.2']\n",
      "NG_000004.2:c.29782A>G\n",
      "NG_000004.2:c.29782A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CYP3A5:c.31551T>C\n",
      "T\n",
      "CYP3A5|c.31551\n",
      "[u'NG_000004.2']\n",
      "NG_000004.2:c.31551T>C\n",
      "NG_000004.2:c.31551T>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CYP3A5:c.12952T>C\n",
      "T\n",
      "CYP3A5|c.12952\n",
      "[u'NG_000004.2']\n",
      "NG_000004.2:c.12952T>C\n",
      "NG_000004.2:c.12952T>C\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CYP3A5:c.6986A>G\n",
      "A\n",
      "CYP3A5|c.6986\n",
      "[u'NG_000004.2']\n",
      "NG_000004.2:c.6986A>G\n",
      "NG_000004.2:c.6986A>G\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "CYP3A5:c.31611C>T\n",
      "C\n",
      "CYP3A5|c.31611\n",
      "[u'NG_000004.2']\n",
      "NG_000004.2:c.31611C>T\n",
      "NG_000004.2:c.31611C>T\n",
      "DATA RETURNED: [None]\n",
      "TEXT RETURNED: [null]\n",
      "None\n",
      "Created problematic_abstracts.json\n",
      "{\n",
      "    \"TRANSCRIPT_COUNTER_6\": 194, \n",
      "    \"TRANSCRIPT_COUNTER_7\": 103, \n",
      "    \"TRANSCRIPT_COUNTER_4\": 955, \n",
      "    \"TRANSCRIPT_COUNTER_5\": 255, \n",
      "    \"TRANSCRIPT_COUNTER_2\": 6026, \n",
      "    \"TRANSCRIPT_COUNTER_3\": 2630, \n",
      "    \"YES_VEP\": 22573, \n",
      "    \"reference_found_only\": 19048, \n",
      "    \"reference_found\": 23041, \n",
      "    \"reference_not_found\": 9226, \n",
      "    \"vep_concordant\": 21278, \n",
      "    \"TRANSCRIPT_COUNTER_8\": 90, \n",
      "    \"TRANSCRIPT_COUNTER_9\": 31, \n",
      "    \"not_reference_found_only\": 9226, \n",
      "    \"NO_VEP\": 19541, \n",
      "    \"NT__ TRANSCRIPTS WITHOUT HGVS\": 42, \n",
      "    \"NO_VEP_NO_TR\": 8777, \n",
      "    \"YES_VEP_YES_TR\": 21503, \n",
      "    \"NT__ TRANSCRIPTS MORE THAN ONE\": 320, \n",
      "    \"TRANSCRIPT_COUNTER_1\": 21896, \n",
      "    \"vep_discordant\": 3652, \n",
      "    \"hgvs_reference_SAME_VEP\": 22572, \n",
      "    \"NT__ TRANSCRIPTS WITH HGVS\": 880, \n",
      "    \"reference_found_and_reference_not_found\": 3993, \n",
      "    \"transvar_found_something\": 32267, \n",
      "    \"more_than_one\": 10371, \n",
      "    \"not_vep_concordant_and_discordant\": 225, \n",
      "    \"COULD NOT FIND HGVS\": 2384, \n",
      "    \"vep_concordant_and_not_discordant\": 17851, \n",
      "    \"TRANSCRIPT_COUNTER_13\": 6, \n",
      "    \"NT__ TRANSCRIPTS ALL PERFECT AND VERSION\": 318, \n",
      "    \"ALL_PERFECT\": 17850, \n",
      "    \"NT__ TRANSCRIPTS WITH HGVS AND VERSION\": 652, \n",
      "    \"MUTALYZER FAILED\": 307, \n",
      "    \"NT__ TRANSCRIPTS ALL PERFECT\": 410, \n",
      "    \"YES_TR\": 32267, \n",
      "    \"TRANSCRIPT_COUNTER_19\": 3, \n",
      "    \"MUTALYZER MISMATCHED POSITION\": 22, \n",
      "    \"TRANSCRIPT_COUNTER_15\": 4, \n",
      "    \"TRANSCRIPT_COUNTER_16\": 7, \n",
      "    \"NO_TR\": 9847, \n",
      "    \"TRANSCRIPT_COUNTER_10\": 22, \n",
      "    \"TRANSCRIPT_COUNTER_11\": 26, \n",
      "    \"TRANSCRIPT_COUNTER_12\": 15, \n",
      "    \"YES_VEP_NO_TR\": 1070, \n",
      "    \"MUTALYZER_OK\": 177, \n",
      "    \"NO_VEP_YES_TR\": 10764, \n",
      "    \"TRANSCRIPT_COUNTER_14\": 4, \n",
      "    \"hgvs_reference_NOT_PRESENT_IN_VEP\": 1, \n",
      "    \"vep_concordant_and_discordant\": 3427\n",
      "}\n",
      "Length of vep_tr_transcripts: 21503\n"
     ]
    }
   ],
   "source": [
    "import re\n",
    "import codecs\n",
    "import json\n",
    "import requests\n",
    "\n",
    "def hgvs_from_record(parsed):\n",
    "        \n",
    "    abstract = parsed['abstract']\n",
    "\n",
    "    if 'dog' in abstract:\n",
    "        #stats_2['nothumnan'] = stats_2.get('nothumnan', 0) + 1\n",
    "        return None\n",
    "\n",
    "\n",
    "    #print parsed\n",
    "    hgvs = parsed['hgvs']\n",
    "\n",
    "    #Get hgvs reference\n",
    "    s = re.search(r'([ACGT]+)>', hgvs)\n",
    "    if not s:\n",
    "        return None\n",
    "        #assert False\n",
    "\n",
    "    hgvs_reference = s.group(1)\n",
    "\n",
    "    #If it starts with a number assuming coding position\n",
    "    if re.search(r'^[\\d]+', hgvs, re.UNICODE):\n",
    "        hgvs = 'c.' + hgvs\n",
    "\n",
    "\n",
    "    #Get everything except the reference\n",
    "    s = re.search(r'^([^ACGT]+)[ACGT]+>', hgvs)\n",
    "    if not s:\n",
    "        return None\n",
    "        #continue\n",
    "        #assert False\n",
    "    except_reference = s.group(1)\n",
    "\n",
    "\n",
    "    gene = parsed['gene']\n",
    "    if not gene:\n",
    "        return None\n",
    "        #assert False\n",
    "\n",
    "    transcripts = parsed['transcripts']\n",
    "    if transcripts:\n",
    "        pass\n",
    "        #assert False\n",
    "\n",
    "    hgvs = gene + ':' + hgvs\n",
    "    hgvs_no_reference = gene + ':' + except_reference\n",
    "    transvar_key = gene + '|' + except_reference\n",
    "    #transvar_cmd = transvar_cmd_p.format(hgvs=hgvs)\n",
    "    return hgvs, hgvs_reference, transvar_key\n",
    "\n",
    "def mutalyzer(var):\n",
    "    r = requests.get('https://mutalyzer.nl/json/numberConversion' ,\n",
    "    {\n",
    "    'build' : 'hg19' ,\n",
    "    'variant' : var,\n",
    "    })\n",
    "    print var\n",
    "    \n",
    "    if not r.ok:\n",
    "        print r.text\n",
    "        assert False\n",
    "    \n",
    "    data = r.json()\n",
    "    if not data:\n",
    "        print 'DATA RETURNED:', data\n",
    "        print 'TEXT RETURNED:', r.text\n",
    "        return None\n",
    "    if not data[0]:\n",
    "        print 'DATA RETURNED:', data\n",
    "        print 'TEXT RETURNED:', r.text\n",
    "        \n",
    "        return None\n",
    "    s = re.search(r'\\.([\\d]+)[ACGT]', data[0])\n",
    "    if not s:\n",
    "        print 'Could not parse:', r.text\n",
    "        return None\n",
    "    return int(s.group(1))\n",
    "    #print ( data \n",
    "#OUTPUT : [ 'NC 000011 . 1 0 : g .112088901C>T' ]\n",
    "\n",
    "def inverse(s):\n",
    "    \n",
    "    return {\n",
    "        'A': 'T',\n",
    "        'T': 'A',\n",
    "        'C': 'G',\n",
    "        'G': 'C',\n",
    "    }[s]\n",
    "\n",
    "def get_transvar_reference(coord, strand):\n",
    "    s = re.search(r'chr[\\dXY]+:g.[\\d]+([ACGT]+)/', coord)\n",
    "    if s:\n",
    "        reference = s.group(1)\n",
    "    else:\n",
    "        return None\n",
    "\n",
    "    if strand == '-':\n",
    "        return inverse(reference)\n",
    "    elif strand == '+':\n",
    "        return reference\n",
    "    else:\n",
    "        assert False\n",
    "\n",
    "def get_transvar_chromosome_position(coord):\n",
    "    s = re.search(r'chr([\\dXY]+):g.([\\d]+)', coord)\n",
    "    if s:\n",
    "        return s.group(1), int(s.group(2))\n",
    "\n",
    "    if coord == '././.':\n",
    "        return None, None\n",
    "    \n",
    "    print 'coord:', coord\n",
    "    raise Exception('Unknown coord')\n",
    "        \n",
    "    \n",
    "    \n",
    "def parse_transvar_output(output):\n",
    "    ls = output.split('\\n')\n",
    "    ls = [x.split('\\t') for x in ls if x.strip()]\n",
    "    try:\n",
    "        strand_i = ls[0].index('strand')\n",
    "    except Exception as e:\n",
    "        print '====='\n",
    "        print output\n",
    "        print '====='\n",
    "        raise e\n",
    "    coord_i = ls[0].index('coordinates(gDNA/cDNA/protein)')\n",
    "    transcript_i = ls[0].index('transcript')\n",
    "    \n",
    "    try:\n",
    "        ret = {l[transcript_i]:{\n",
    "            'reference': get_transvar_reference(l[coord_i], l[strand_i]),\n",
    "            'position': get_transvar_chromosome_position(l[coord_i]),\n",
    "            } for l in ls[1:]}\n",
    "    except Exception as e:\n",
    "        print output\n",
    "        raise e\n",
    "    return ret    \n",
    "\n",
    "def add_one(d, key):\n",
    "    d[key] = d.get(key, 0) + 1\n",
    "\n",
    "def load_transvar_results(fn):\n",
    "    \n",
    "    stats_3 = {}\n",
    "    vep_tr_transcripts = []\n",
    "    \n",
    "    prob_ab_f = codecs.open('problematic_abstracts.json', 'w', 'utf-8')\n",
    "    \n",
    "    c = 0\n",
    "    with open(fn) as f:\n",
    "        for l in f:\n",
    "            c += 1\n",
    "            \n",
    "            if c % 10000 == 0:\n",
    "                print 'Records:', c\n",
    "            \n",
    "            data = json.loads(l)\n",
    "            \n",
    "            hgvs_from_record_ret = hgvs_from_record(data)\n",
    "            if hgvs_from_record_ret is None:\n",
    "                add_one(stats_3, 'COULD NOT FIND HGVS')\n",
    "                \n",
    "                if data['transcripts']:\n",
    "                    add_one(stats_3, 'NT__ TRANSCRIPTS WITHOUT HGVS')\n",
    "                \n",
    "                continue\n",
    "                                    \n",
    "                \n",
    "            hgvs, hgvs_reference, transvar_key = hgvs_from_record_ret\n",
    "            #print hgvs\n",
    "            #print hgvs_reference\n",
    "            if transvar_key in transvar:\n",
    "                transvar_output = transvar[transvar_key]\n",
    "            else:\n",
    "                transvar_output = ''\n",
    "            \n",
    "            \n",
    "            #transvar_output = data['transvar'] # THERE IS NOT DATA THERE! FIX IT\n",
    "            #print transvar_output\n",
    "            if not transvar_output.strip():\n",
    "                add_one(stats_3, 'transvar_crashed')\n",
    "                transvar_info = None\n",
    "            else:\n",
    "                transvar_info = parse_transvar_output(transvar_output)\n",
    "            #print transvar_info\n",
    "            \n",
    "            vep_info = vep.get(hgvs, None)\n",
    "            #print vep_info\n",
    "            \n",
    "            ### TRANSVAR / VEP FOUND SOUMETHING\n",
    "            NO_VEP = False\n",
    "            YES_VEP = False\n",
    "            NO_TR = False\n",
    "            YES_TR = False\n",
    "            \n",
    "            if vep_info is None:\n",
    "                NO_VEP = True\n",
    "                add_one(stats_3, 'NO_VEP')\n",
    "            else:\n",
    "                YES_VEP = True\n",
    "                add_one(stats_3, 'YES_VEP')\n",
    "            \n",
    "            if transvar_info and all([x['position'][1] is None for x in transvar_info.values()]):\n",
    "                    #add_one(stats_3, 'transvar_found ././.')\n",
    "                    # TODO: CHECK VEP HGVS_REF!!!\n",
    "                    NO_TR = True\n",
    "                    add_one(stats_3, 'NO_TR')\n",
    "            else:\n",
    "                YES_TR = True\n",
    "                add_one(stats_3, 'YES_TR')\n",
    "            \n",
    "            assert NO_VEP != YES_VEP\n",
    "            assert NO_TR != YES_TR\n",
    "            \n",
    "            for x in [('YES_VEP', YES_VEP), ('NO_VEP', NO_VEP)]:\n",
    "                for y in [('YES_TR', YES_TR), ('NO_TR', NO_TR)]:\n",
    "                    if x[1] and y[1]:\n",
    "                        add_one(stats_3, x[0] + '_' + y[0])\n",
    "            \n",
    "            #continue\n",
    "            \n",
    "            #if YES_VEP and NO_TR:\n",
    "            #    print data\n",
    "            #    print '====='\n",
    "            #    print vep_info\n",
    "            #    #assert False\n",
    "            \n",
    "            #if NO_TR:\n",
    "            #    continue\n",
    "            \n",
    "            # END OF TRANSVAR / VEP FOUND SOMETHING\n",
    "            \n",
    "            # CHECK tr_ <--> HGVS\n",
    "            if data['transcripts']:\n",
    "                add_one(stats_3, 'NT__ TRANSCRIPTS WITH HGVS')\n",
    "                if any(['.' in x for x in data['transcripts']]):\n",
    "                    add_one(stats_3, 'NT__ TRANSCRIPTS WITH HGVS AND VERSION')\n",
    "                    \n",
    "                if len(data['transcripts'])>1:\n",
    "                    add_one(stats_3, 'NT__ TRANSCRIPTS MORE THAN ONE')\n",
    "                else:\n",
    "                    print hgvs\n",
    "                    print hgvs_reference\n",
    "                    print transvar_key\n",
    "                    print data['transcripts']\n",
    "                    tr_hgvs = data['transcripts'][0] + ':' + hgvs.split(':')[1]\n",
    "                    print tr_hgvs\n",
    "                    tr_position = mutalyzer(tr_hgvs)\n",
    "                    print tr_position\n",
    "                    #assert False\n",
    "                    if not tr_position:\n",
    "                        add_one(stats_3, 'MUTALYZER FAILED')\n",
    "                    elif vep_info:\n",
    "                        if vep_info['start'] == tr_position:\n",
    "                            add_one(stats_3, 'MUTALYZER_OK')\n",
    "                        else:\n",
    "                            print vep_info\n",
    "                            print data['abstract']\n",
    "                            prob_ab_f.write(data['abstract'] + '\\n')\n",
    "                            add_one(stats_3, 'MUTALYZER MISMATCHED POSITION')\n",
    "            #END OF tr_ <--> HGVS Check\n",
    "\n",
    "            \n",
    "            #### CHECK VEP // HGVS REFERENCE\n",
    "            hgvs_vep_concordance = False\n",
    "            if vep_info:\n",
    "                if hgvs_reference in vep_info['allele_string']:\n",
    "                    if vep_info['allele_string'].index(hgvs_reference) == 0:\n",
    "                        add_one(stats_3, 'hgvs_reference_SAME_VEP')\n",
    "                        hgvs_vep_concordance = True\n",
    "                    else:\n",
    "                        add_one(stats_3, 'hgvs_reference_NOT_SAME_VEP')\n",
    "                else:\n",
    "                    add_one(stats_3, 'hgvs_reference_NOT_PRESENT_IN_VEP')\n",
    "            \n",
    "            ### END OF VEP // HGVS REFERENCE CONCORDANC CHECK\n",
    "            \n",
    "            ### CHECK TRANSVAR CONCORDANCE\n",
    "            if NO_TR:\n",
    "                continue\n",
    "            \n",
    "            \n",
    "            more_than_one = False\n",
    "            reference_found = False\n",
    "            reference_not_found = False\n",
    "            more_than_one_marked = False\n",
    "            vep_concordant = False\n",
    "            vep_discordant = False\n",
    "            transcript_counter = 0\n",
    "            \n",
    "            for k,v in transvar_info.iteritems():\n",
    "                if more_than_one:\n",
    "                    if not more_than_one_marked:\n",
    "                        add_one(stats_3, 'more_than_one')\n",
    "                        more_than_one_marked = True\n",
    "                \n",
    "                if v['reference'] == hgvs_reference:\n",
    "                    reference_found = True\n",
    "                    \n",
    "                if v['reference'] != hgvs_reference:\n",
    "                    reference_not_found = True\n",
    "                \n",
    "                if vep_info:\n",
    "                    if v['reference'] in vep_info['allele_string']:\n",
    "                        if vep_info['allele_string'].index(v['reference']) == 0:\n",
    "                            if vep_info['start'] == v['position'][1]:\n",
    "                                vep_concordant = True\n",
    "                            else:\n",
    "                                vep_discordant = True\n",
    "                        else:\n",
    "                            vep_discordant = True\n",
    "                    else:\n",
    "                        vep_discordant = True\n",
    "                \n",
    "                more_than_one = True\n",
    "            \n",
    "                transcript_counter += 1\n",
    "            \n",
    "            if not more_than_one:\n",
    "                # Transvar found nada\n",
    "                add_one(stats_3, 'transvar_not_found')\n",
    "            else:\n",
    "                add_one(stats_3, 'transvar_found_something')\n",
    "            \n",
    "            if reference_found:\n",
    "                add_one(stats_3, 'reference_found')\n",
    "            else:\n",
    "                add_one(stats_3, 'reference_not_found')\n",
    "                \n",
    "            if reference_found and reference_not_found: # I know..\n",
    "                add_one(stats_3, 'reference_found_and_reference_not_found')\n",
    "                \n",
    "            if reference_found and not reference_not_found:\n",
    "                add_one(stats_3, 'reference_found_only')\n",
    "                \n",
    "            if not reference_found and reference_not_found:\n",
    "                add_one(stats_3, 'not_reference_found_only')\n",
    "                \n",
    "            if not reference_found and not reference_not_found:\n",
    "                assert False\n",
    "            \n",
    "            add_one(stats_3, 'TRANSCRIPT_COUNTER_{}'.format(transcript_counter))\n",
    "            \n",
    "            if vep_info:\n",
    "                if vep_concordant:\n",
    "                    add_one(stats_3, 'vep_concordant')\n",
    "\n",
    "                if vep_discordant:\n",
    "                    add_one(stats_3, 'vep_discordant')\n",
    "\n",
    "                if vep_concordant and vep_discordant: # Can happen\n",
    "                    add_one(stats_3, 'vep_concordant_and_discordant')\n",
    "\n",
    "                if vep_concordant and not vep_discordant:\n",
    "                    add_one(stats_3, 'vep_concordant_and_not_discordant')\n",
    "\n",
    "                if not vep_concordant and vep_discordant:\n",
    "                    add_one(stats_3, 'not_vep_concordant_and_discordant')\n",
    "\n",
    "                if not vep_concordant and not vep_discordant:\n",
    "                    add_one(stats_3, 'not_vep_concordant_not_discordant')\n",
    "\n",
    "                vep_tr_transcripts.append((vep_info['tr'], transcript_counter))\n",
    "                    \n",
    "            if hgvs_vep_concordance and reference_found and not reference_not_found and vep_concordant and not vep_discordant:\n",
    "                add_one(stats_3, 'ALL_PERFECT')\n",
    "                \n",
    "                if data['transcripts']:\n",
    "                    add_one(stats_3, 'NT__ TRANSCRIPTS ALL PERFECT')\n",
    "                    if any(['.' in x for x in data['transcripts']]):\n",
    "                        add_one(stats_3, 'NT__ TRANSCRIPTS ALL PERFECT AND VERSION')\n",
    "                    #print data['abstract']\n",
    "                \n",
    "                    \n",
    "            \n",
    "            #print (stats_3)\n",
    "            #assert False\n",
    "    prob_ab_f.close()\n",
    "    print 'Created problematic_abstracts.json'\n",
    "    print json.dumps(stats_3, indent=4)\n",
    "    print 'Length of vep_tr_transcripts:', len(vep_tr_transcripts)\n",
    "    \n",
    "    return vep_tr_transcripts\n",
    "\n",
    "#load_transvar_results('parsed_abstracts_transvar.json')\n",
    "plot_data = load_transvar_results('parsed_abstracts.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "def vep_parse_post(hgvs):\n",
    "\n",
    " \n",
    "    server = \"https://grch37.rest.ensembl.org\"\n",
    "    ext = \"/vep/human/hgvs\"\n",
    "    headers={ \"Content-Type\" : \"application/json\", \"Accept\" : \"application/json\"}\n",
    "    \n",
    "    assert type(hgvs) is list\n",
    "    #data = '[' + ','.join([rept(x) for x in hgvs]) + ']'\n",
    "    data = json.dumps({\"hgvs_notations\": hgvs})\n",
    "    print data\n",
    "    \n",
    "    r = requests.post(server+ext, headers=headers, data=data)\n",
    " \n",
    "    if not r.ok:\n",
    "        print ('ERROR:')\n",
    "        print (r.text)\n",
    "        return None\n",
    " \n",
    "    decoded = r.json()\n",
    "    return decoded"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{\"hgvs_notations\": [\"TGFBR2:c.1381T>C\"]}\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[{u'allele_string': u'T/C',\n",
       "  u'assembly_name': u'GRCh37',\n",
       "  u'end': 30715648,\n",
       "  u'id': u'TGFBR2:c.1381T>C',\n",
       "  u'input': u'TGFBR2:c.1381T>C',\n",
       "  u'most_severe_consequence': u'synonymous_variant',\n",
       "  u'seq_region_name': u'3',\n",
       "  u'start': 30715648,\n",
       "  u'strand': 1,\n",
       "  u'transcript_consequences': [{u'amino_acids': u'L',\n",
       "    u'biotype': u'protein_coding',\n",
       "    u'cdna_end': 1688,\n",
       "    u'cdna_start': 1688,\n",
       "    u'cds_end': 1306,\n",
       "    u'cds_start': 1306,\n",
       "    u'codons': u'Ttg/Ctg',\n",
       "    u'consequence_terms': [u'synonymous_variant'],\n",
       "    u'gene_id': u'ENSG00000163513',\n",
       "    u'gene_symbol': u'TGFBR2',\n",
       "    u'gene_symbol_source': u'HGNC',\n",
       "    u'hgnc_id': 11773,\n",
       "    u'impact': u'LOW',\n",
       "    u'protein_end': 436,\n",
       "    u'protein_start': 436,\n",
       "    u'strand': 1,\n",
       "    u'transcript_id': u'ENST00000295754',\n",
       "    u'variant_allele': u'C'},\n",
       "   {u'amino_acids': u'L',\n",
       "    u'biotype': u'protein_coding',\n",
       "    u'cdna_end': 1664,\n",
       "    u'cdna_start': 1664,\n",
       "    u'cds_end': 1381,\n",
       "    u'cds_start': 1381,\n",
       "    u'codons': u'Ttg/Ctg',\n",
       "    u'consequence_terms': [u'synonymous_variant'],\n",
       "    u'gene_id': u'ENSG00000163513',\n",
       "    u'gene_symbol': u'TGFBR2',\n",
       "    u'gene_symbol_source': u'HGNC',\n",
       "    u'hgnc_id': 11773,\n",
       "    u'impact': u'LOW',\n",
       "    u'protein_end': 461,\n",
       "    u'protein_start': 461,\n",
       "    u'strand': 1,\n",
       "    u'transcript_id': u'ENST00000359013',\n",
       "    u'variant_allele': u'C'}]}]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vep_parse_post(['TGFBR2:c.1381T>C'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: matplotlib in /home/user/mutationinfo/localpython/lib/python2.7/site-packages\n",
      "Requirement already satisfied: python-dateutil>=2.1 in /home/user/mutationinfo/localpython/lib/python2.7/site-packages (from matplotlib)\n",
      "Requirement already satisfied: subprocess32 in /home/user/mutationinfo/localpython/lib/python2.7/site-packages (from matplotlib)\n",
      "Requirement already satisfied: cycler>=0.10 in /home/user/mutationinfo/localpython/lib/python2.7/site-packages (from matplotlib)\n",
      "Requirement already satisfied: six>=1.10 in /home/user/mutationinfo/localpython/lib/python2.7/site-packages/six-1.11.0-py2.7.egg (from matplotlib)\n",
      "Requirement already satisfied: backports.functools-lru-cache in /home/user/mutationinfo/localpython/lib/python2.7/site-packages (from matplotlib)\n",
      "Requirement already satisfied: pytz in /home/user/mutationinfo/localpython/lib/python2.7/site-packages (from matplotlib)\n",
      "Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.1 in /home/user/mutationinfo/localpython/lib/python2.7/site-packages (from matplotlib)\n",
      "Requirement already satisfied: numpy>=1.7.1 in /home/user/mutationinfo/localpython/lib/python2.7/site-packages (from matplotlib)\n",
      "Requirement already satisfied: kiwisolver>=1.0.1 in /home/user/mutationinfo/localpython/lib/python2.7/site-packages (from matplotlib)\n",
      "Requirement already satisfied: setuptools in /home/user/mutationinfo/localpython/lib/python2.7/site-packages (from kiwisolver>=1.0.1->matplotlib)\n"
     ]
    }
   ],
   "source": [
    "!pip install matplotlib"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting py27-backports-functools_lru_cache\n",
      "\u001b[31m  Could not find a version that satisfies the requirement py27-backports-functools_lru_cache (from versions: )\u001b[0m\n",
      "\u001b[31mNo matching distribution found for py27-backports-functools_lru_cache\u001b[0m\n",
      "\u001b[33mYou are using pip version 9.0.1, however version 9.0.2 is available.\n",
      "You should consider upgrading via the 'pip install --upgrade pip' command.\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!pip install py27-backports-functools_lru_cache"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAX4AAAD8CAYAAABw1c+bAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDIuMi4yLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvhp/UCwAAIABJREFUeJzsvWuMZOl53/d7z/2cqu6q6svcZ3Z2SWothpYoZEDHiAzIN5oiFMsJAkVEEFCOjLUC60OABImMBFEg54OAwEkMKLHM2IQkwKYVIFYswIIkRkDAOJBlLSVRpqQluUvu7tz7VvdzP+fNh1NVW11T1V3Tl+3b8wMGU3Xq3Lpn6v8+53mf9/8orTWCIAjC1cE46xsQBEEQPlxE+AVBEK4YIvyCIAhXDBF+QRCEK4YIvyAIwhVDhF8QBOGKIcIvCIJwxRDhFwRBuGKI8AuCIFwxrLO+gXlsbGzo+/fvn/VtCIIgXBi++tWv7mitN5fZ91DhV0rdBX4JuA5o4Ata67+nlFoDfhm4D7wL/IjWuj3n+M8D/+3o7f+gtf7Fw655//593nzzzWXuXxAEQQCUUu8tu+8yqZ4c+C+01h8H/h3gbymlPg78FPBbWuuPAb81ej97I2vATwN/BvgU8NNKqdayNycIgiCcPIcKv9b6qdb690av+8CfALeBHwbG0fsvAn9tzuF/Bfiy1npv9DTwZeAzJ3HjgiAIwtF4qcldpdR94PuA3wGua62fjj56RpUKmuU28HDq/aPRNkEQBOGMWFr4lVJ14P8E/nOtdW/6M115Ox/L31kp9YZS6k2l1Jvb29vHOZUgCIJwAEsJv1LKphL9f6y1/mejzc+VUjdHn98EtuYc+hi4O/X+zmjbC2itv6C1fqC1frC5udTEtCAIgnAEDhV+pZQC/hHwJ1rr/2nqo18FPj96/Xngn885/DeATyulWqNJ3U+PtgmCIAhnxDIR/78L/CfAX1BK/cHoz2eBnwX+slLqW8BfGr1HKfVAKfUPAbTWe8DfAX539OdnRtsuLWleMkxy0rw861sRBEGYizqPrRcfPHigL2Idf5qXPG6HaEABt1sBjiWLowVBOH2UUl/VWj9YZl9RpRMkK0o0UHMt9Oi9IAjCeUOE/wSxTQMFDJMcBWiNpH0EQTh3nEuvnouKYxncbgVV5K9hux9L2kcQhHOHKNEJ41gGNddCKSTtIwjCuUSE/5SYTfvYpvyqBUE4H0iq55SYTvvYpiFpHkEQzg0i/KeIY4ngC4Jw/hBVEgRBuGKI8AuCIFwxJNVzRqR5Kfl/QRDOBBH+M0CsHQRBOEtEbc4AsXYQBOEsEeE/A6TGXxCEs0RSPWeA1PgLgnCWiPCfEVLjLwjCWSHKIwiCcMUQ4RcEQbhiiPALgiBcMQ7N8Sulvgj8ELCltf7EaNsvA6+PdmkCHa31J+cc+y7QBwogX7YtmCAIgnB6LDO5+wvAzwG/NN6gtf6Pxq+VUn8X6B5w/J/XWu8c9QYFQRCEk+VQ4ddaf0UpdX/eZ0opBfwI8BdO9rYEQRCE0+K4Of4/BzzXWn9rweca+E2l1FeVUm8c81qCIAjCCXDcOv7PAV864PPv11o/VkpdA76slHpLa/2VeTuOBoY3AO7du3fM2xIEQRAWceSIXyllAf8B8MuL9tFaPx79vQX8CvCpA/b9gtb6gdb6webm5lFvSxAEQTiE46R6/hLwltb60bwPlVI1pdTK+DXwaeDrx7ieIAiCcAIcKvxKqS8Bvw28rpR6pJT68dFHP8pMmkcpdUsp9Wujt9eBf6mU+hrwr4F/obX+9ZO7dUEQBOEoLFPV87kF239szrYnwGdHr78NfO8x708QBEE4YWTlriAIwhVDhF8QBOGKIcIvCIJwxRDhFwRBuGJII5YrSpqX0gFMEK4oIvxXkDQvedwO0YACbrcCEX9BuELIt/0KkhUlGqi5Fnr0XhCEq4MI/xXENg0UMExy1Oi9IAhXB0n1XEEcy+B2K5AcvyBcUUT4ryiOJYIvCFcVEf5zymlU3RzlnFL9IwiXDxH+c8hpVN0c5ZxS/SMIlxP5Fp9DTqPq5ijnlOofQbiciPCfQ06j6uYo55TqH0G4nEiq5xxyGlU3RzmnVP8IwuVEhP+cchpVN0c5p1T/CMLlQ4T/JZEqF0EQLjoi/C+BVLkIgnAZENV6CaTKRRCEy8Ayzda/qJTaUkp9fWrbf6+UeqyU+oPRn88uOPYzSqlvKKXeVkr91Ene+FkgVS6CIFwGllGuXwA+M2f7/6y1/uToz6/NfqiUMoH/FfhB4OPA55RSHz/OzZ414yqX66uepHkEQbiwHKpcWuuvAHtHOPengLe11t/WWqfAPwV++AjnOVc4lkHNtUT0BUG4sBxHvX5SKfWHo1RQa87nt4GHU+8fjbZdStK8ZJjkpLnk/QVBON8cVfj/PvAR4JPAU+DvHvdGlFJvKKXeVEq9ub29fdzTfaiMq32e92Iet0MRf0EQzjVHEn6t9XOtdaG1LoH/nSqtM8tj4O7U+zujbYvO+QWt9QOt9YPNzc2j3NaZIdU+giBcJI4k/Eqpm1Nv/33g63N2+13gY0qpV5VSDvCjwK8e5XrnHan2EQThInHoAi6l1JeAHwA2lFKPgJ8GfkAp9UlAA+8Cf3O07y3gH2qtP6u1zpVSPwn8BmACX9Ra/9Gp/BRnjHjaCIJwkVBa67O+hxd48OCBfvPNN8/6NgRBEC4MSqmvaq0fLLOvhKaCIAhXDBF+QRCEK4YIvyAIwhVDhF8QBOGKIcIvCIJwxRDhFwRBuGKI8AuCIFwxpAPXJUbaRAqCMA8R/kuKtIkUBGERogQjPixb5Q/rOmIcJwjCIiTi5/Si49lUy4cZhYtxnCAIixDhZ390PExysqI8tiDPE/nTuM4ixDhOEIRFiPBzOtHxPJH/sKNwxxLBFwThRUT4OZ3oeJ7ISxQuCMJ5QIR/xElHx/NEXsorBUE4D4jwH4FlBXx6MJHySkEQzgsi/C/JUQX8w5zYFQRBOIgrozwnVT9/1Pp4Ka8UBOG8sEzP3S8CPwRsaa0/Mdr2PwL/HpAC7wB/XWvdmXPsu0AfKIB82bZgJ81JplmOKuAysSsIwnlhGfX5BeAzM9u+DHxCa/09wDeBv33A8X9ea/3JsxJ9ONlVrGMBv77qvfQA4lgGNdcS0RcE4Uw5VIG01l8B9ma2/abWOh+9/VfAnVO4txPjpNMsIuCCIFxkTmJy9z8FfnnBZxr4TaWUBv6B1voLJ3C9l0bSLIIgCB9wLOFXSv03QA784wW7fL/W+rFS6hrwZaXUW6MniHnnegN4A+DevXvHua25yCpWQRCEiiMroVLqx6gmff9jrbWet4/W+vHo7y3gV4BPLTqf1voLWusHWusHm5ubR70tQRAE4RCOJPxKqc8A/xXwV7XW4YJ9akqplfFr4NPA1496o4IgCMLJcKjwK6W+BPw28LpS6pFS6seBnwNWqNI3f6CU+vnRvreUUr82OvQ68C+VUl8D/jXwL7TWv34qP4UgCIKwNIfm+LXWn5uz+R8t2PcJ8NnR628D33usuxMEQRBOHJntFARBuGKI8AuCIFwxxKTtFBD7ZUEQzjMi/CeM2C8LgnDeEUU6YU7SF0gQBOE0uPTC/zJ2zCdh3TzrC6Q1J2IHLQiCcFJc6lTPYWmX6Vw8cCIpmmlfIK1hux/PPafMAwiCcFZcauE/qOvV7KDQDJwT65A19gUaJvncc8o8gCAIZ8mlVpuD7Jhnc/HAiXfIWnR9mQcQBOEsudQR/0F2zLOiXHMtaq51oumXRdeXNoyCIJwll1b4p3PoNffFH3ORKJ90ymWeHbT0BxAE4Sy5lMK/bA79LD36pT+AIAhnxaVUHsmhC4IgLOZSRvzL5NClnFIQhKvKpRT+w3LoUk4pCMJV5tKqnWNVk7rzBP2ip4JOYoWxIAhXl0sZ8R/GRS6nXPZpZTw4AJMBcHal8nh1sVJIyksQrhBXUvgvcjnlQauRx6R5ybs7A551Y0Bzo+FzqxlM7CPy0ROOBra6MdcaHo5pSMpLEK4IV1L44WKVU44jda0r4c+L8sCnlawoyQqN75ijVJYmTD+wj9jpJwAEjkmJxrGMyblnnwwuyu9IEITlWepbrZT6olJqSyn19alta0qpLyulvjX6u7Xg2M+P9vmWUurzJ3XjV4VxaudRO+T339tjqx8D0AqchRG6bRrYpiJKC+I0xzYVgWNN0lu2qbBNRVqUGCjSvJwMIuPrPe/FPG6HMo8gCJeQZSP+XwB+DvilqW0/BfyW1vpnlVI/NXr/X08fpJRaA34aeECVWfiqUupXtdbt4974VWGc2nFMY190ftATi2MZ3N+os7niAR/k+B0r2JfjHyY5rcCZrG4+yFhOEITLw1LfaK31V4C9mc0/DPzi6PUvAn9tzqF/Bfiy1npvJPZfBj5zxHs915xWpc14InpedH4QjmXQqjm0as4+O4rpSqdOmDJIcjph+sL1LuLEtyAIy3GcHP91rfXT0etnwPU5+9wGHk69fzTadqk4zXUB0xPRt5vBiVXgLJokvsgT34IgLMeJTO5qrbVSSh++52KUUm8AbwDcu3fvJG7rQ2OZSpuXZXaCdZnzvcyk7HRknxclaV5imx+I/6ISURkQBOHic5xv73Ol1E2A0d9bc/Z5DNyden9ntO0FtNZf0Fo/0Fo/2NzcPMZtfficdHrkKBOsL3vMOLJvBQ4A7TA98DiZ9BWEy8NxFOpXgXGVzueBfz5nn98APq2Uao2qfj492napGIvo9VXvpdI8i+YFjrKyeN4xh807jCN7axTBh2kxWfQ17/xpUaJUNd9w0VY7C4LwAUulepRSXwJ+ANhQSj2iqtT5WeD/UEr9OPAe8COjfR8AP6G1/hta6z2l1N8Bfnd0qp/RWs9OEn8oHJSmOE4KY7rGXqmXO27RvMBRniDmNXlfZt7BNg3youTRXghobFPNtbrQulrsVaIxUNxuBsv/sIIgnCuWEn6t9ecWfPQX5+z7JvA3pt5/Efjike7uhDhIZI8zMTs+Ni3Kl14Be9C8wFEmWGePWXbewbEMNlc8skKz6tujxV8v7qsUk59vHPkLgnAxuRIrd+eJ4Hh7mr/cxOz008GiGvtlJncPi+qPsrJ49phlnxpqrkXgmGTF4lJR2zRwTGPy80qZpyBcXK6E8B+UBhn71iwjkNMRflFobjb8F2rslxXF0y6bfJnzL7OvlHkKwuXhSgj/YWmQVlAtcpoWtHl5//EEZy/OCZMMgHvrNZRiX409jK0RDhfc0xTQlzn/MvteJH8jQRAWcyWEf5qxoE8bnc1OZg7inLe3+mgNNdfk/kZ9MjAUhSZMMgLHwjQVSrGvmbs0eREE4bxzJYR/3iSsojI6mxX9NC95Z6vPN5518WyLumexueLhWNVTwb31GgCmqeamdaafJjphSidMaQaOiL8gCOeGKyH8k0lYazQJO56knJO6GO/r2hZpWTKMi30163XP4t56jTDNCZwXyx7H8wmdMGWrWzlpDpNcIn9BEM4Nl1KJZhcuTYzO8tEk7CHVK4Fj4tsGcZLjWYpOmE7OleYl2/2YQZKz3Y9fWBw1vSL2WsOjGTgXsr2jIAiXl0sX8S/KsS9rdDa2NK65FturCRsr7mReAKpIPi1KmoGzsPzTsYzJ5+JyKQjCeePSCf9BOfaXMTpr+A5pXk5skMcloON5Aji4nn28MGpRSkgQBOGsuHTCf1COHTiwDn32aWFzxZs8HYwHlMCxqHs2ddei4TuTJ4F5NhDjHrfDJMex9q8WPuw+zmO9/Hm9L0EQXo5LJ/zjSHtnENOquQSORS/K6ITppLvUojLL2fr+2VLNytNmCCgcSzFMcqzRQDM+31gc07yq+Z/d1o2qpxB/1Apx9j7Oaznoeb0vQRBenksn/ONIOy1KtvoRz/sRtlEZkXmOeWBuXmuI06qKZzaNM+1p0whsumE2er3fBmIsjlFSsNWLMEyFgaLlO3zjWY8oK+mECQ9eXZ9M+k7fx8t4+89G4Isi8pOI1E+j58BpIU8mgnAwl074xwLVDByitCAvNDeaHsMkpyj0wsnW8YBhmoqi0Gw2vRdEY+xpUzUtqVzKps83LY7dMMN1LNbrld99mOWUaK6turTDhL1BynrdeeE+lnXmnJeWGqeWZp9ATiJSvygtGeXJRBAO51IJ/3gydrwq1zbVRKwc02Cz6S2s6JkeMMZpnnk0R41Lximg2chyPL/QDhPQ8LxbcKPh0/AcDEK6Ucrmisu99WDuwq5lPXFmI/Awnd8k/biR+nT0fBG8ei7Sk4kgnBWXRvinI70s1zgWoBWeY0wi+LpnvXDMWMgOi2gHcc77u8PJit3xit9pURmL9rh5+Tjy31zxaNUcvu+VtUmVT92zJusNZoV0mQqk2fsNHGtu+ehxIvV50fP0nMei3+VZCu1FeTIRhLPk0gj/ONKzTYNHwyG9xKQf5Xz0Wh3bfLFJyjxRu90K5nagGsQ5bz3r0YlSGr7Dqmcd6G8/fmpI85LAMbFNY1TZY1D3vIXXfxnBnPdk4FgvRuTHcdVcNno+T+kVcREVhMO5NMI/jvR6UUZWaoo053k3JitKPna9fqCnzljUbNOgE6ZoqnRNNZlb8v7ukE6U0humxFmBYwQHRpLT4pNm5b4nhbEoHiaqy0TQ8544Fg1Gp5nXP2/pFXERFYSDuTTCPxbbYZITpTnvbA8IXAVKz82lzxO1Klde4NsmnSijF2VYpsHOMMExDaK8xDJNLGM5v32A93aGPOlFuKZBqzauKKoatiwS1bOOoOfl9ccNZqZ/tjGSXhGEi8WRhV8p9Trwy1ObXgP+O631/zK1zw9QNWH/zmjTP9Na/8xRr3kYVaTncG+9xl6YUnMtikJjm/tLHaESqZprYY/y9QDb/ZhnvYjdfkLds/Asg9utGkUBtgmbdY+PXquDWq7LVlaUlBrCOGc3y+nHGQomNfzTC8SmzzUegA5qhXhazBt0bNM4cCCS9IogXCyOLPxa628AnwRQSpnAY+BX5uz6/2qtf+io1zkKzcDh/nqNrKiah08LVz4S0t1BCmhuNHxqbp2sKLFMg7trAXlRcqcV8KwX895uiGlCYJusBy4olopqJxVGZUngWbQCB9c2AbVwgdj4uO1+zO4gYXcQc6Phn2gEfVgKaVGbysNSOZJeuVycl8l64XQ4qVTPXwTe0Vq/d0LnOxZjo7V5Hbd2+glRWuI75mgBlZ7sN/bkCWwLyzTYqLlomBi1zevUNY/pqNk2FZs1F88xGc8vH5QSGQ9Ar12rTyqCTuqLt0wKaVHaRlI5V4ezTjUKp89JCf+PAl9a8NmfVUp9DXgC/Jda6z86oWseyGwEOhYu21TgGJOIv+nbEyGfdfDUmon18rxOXbNMWzOMBxpg34ABB/sFTVtIB465sHxy9prLDEbLOovOS9tIKufqcN4m64WT59jCr5RygL8K/O05H/8e8IrWeqCU+izwfwEfW3CeN4A3AO7du3fc29rH9MQvVOJ6q1mlMKbFfDxYjMX0ZXLX01HSbAP32QHjoPNMX3M8oZrm5dy5gGUjs+kOZI/bIWFSUHPNA51FF6VyFq09EC4PMll/+TmJiP8Hgd/TWj+f/UBr3Zt6/WtKqf9NKbWhtd6Zs+8XgC8APHjwQB/3puZN5m734xdM1Wb3H0f5yyxammaZBu7L3Ou06M62i5wuB513zenIbPqc086iaE1R6rnXXeZ3KimAy8+yAY/MA1xcTkL4P8eCNI9S6gbwXGutlVKfour4tXsC1zyQedF3Vmh2BwmvXatPotas+CD9Mt4/TgtMUx2YDhlfYzxQjNNC4yhpfM2DSiDn3etYTOGDhi+Otb9d5PT9LIrM5tpLw6g81Zx4F72/O5zMPSwj4uNqo0Zg73sqEk6XsxDYwybrJQi42BxL+JVSNeAvA39zattPAGitfx74D4H/TCmVAxHwo1rrY0fzhzE7mQuw6tvsDmK6YYZtqkn0nxclNdea5L6zotxn5qY1L6Q2xv/ph2nO03bEzZZPzbFo+g5hVq0j2OrH+yL1RaWbs1H7MMknor/VjWnVnUm7yEWOobPNXubZS49TXbapSPPqZzRNtXQe94Nqo6ri6EbDkxTAh8B5FViZB7jYHEv4tdZDYH1m289Pvf454OeOc42jMB0Jj100s6LkRsNnc6WyTNjqx+RlycN2yFrgMogzgH1mbvPSPo5VLfTqRhm7w4S9MMEwFSuuTS/K0MBWL+F2y68i9VFefLx6tyg099ZrE9+g2agdmJjFQTUxfH+9vm/QmJeWmm72Mu9JYLzGoeZa+45dNo9bfbk1d1o1oqw40WojYTHnVWBlHuBic2lW7k4zm6McN0Cpu1YV3edVNB1mOd0w49X1OoFr0gqcfat8x41bZmvat/sxW72YndFTw7NuzNDLcS0D3zZph8nI/9+YRNclmmFcECbVAPPR6yuTx+npewUmTWMc03hh1fF0BLgoLXVQjnb6EX6et888xtH+s15EUcDtpr/UvIdwfM6rwMqivYvNpf32Tov3k3bE7jAGFDcaHpsrHtcaVd77YTukF2c4pvFCb9xFtg5Zobm/WQMFvm0QZSWvrNd4byckygrurtcIbJNbI4HUGt7fHRImGYFjYZpqX+Q2W020KC0E+yPA2bTUbBrosC/jsouuxtf0HYt+lJKX5eH/AAuQCcGX4zwLrCzau7hcWuEfR8ZhWvC4E1H3zFGFSzXFMJ4svbFaddUyR3n/6d6486LxcZ4bFNdW3Em0bZsGt5s+eVl1+pqN1u+t1wAmZm3zGsEsk8udHowO6zFwUthmZW2d5QVro8Voy8wJzIrVec1Xn3dEYIWT5tIK/zhKbQQ2z3oRg7jAszWBYwIf+OSkeUl75OszL4c6/aUb99h9dXOF592IKCvx80rcWoHDndYHzVlmK3rqnsVHr68sjNymI/lOWPXlPU6jlpPEsYxDB65pFgn8ec1XC8JV49IK/3RkvFFz2ai72JYx6o6V7jMgG+fUD8uhjs8ZpjntMMVQYBmKVd9+ISqbJ3wHRW7jc3fClK1uDHxgDT1vAdiHLZiHDVzTLBL485qvFoSrxqUV/nGp47hWPS1KArfy4JkWpJprvVQE3QwcwjTnZtMnykrCJKPmWPtELCvKSQ3+dL37QfntcSQ/7t4VOBbf2e6TFdVTynlIiyw74CwS+POcrxaEq8SlFX6oFlZ5I7+bsV1DXpSj2n7NMK66ZI3tmWcXW00L9XRJZj6aVPVtg5rtcW+9tk/EtIatbkyJxkBxuxksld92rA+6d/WiDFAXcrHUslVFgiCcDZdO+Jfpo5sVJVv9iGe9KqXS8Gx8x9znk58V5b5FXlFa0IlSbMskSnPWAhddaG41q3UB04u8lGKycCstKp+dZfPbY9GcXmx1UmmRD7OiRgReEM4vl0r4F/XRnfassUyDRmCw1UtQKPbClL1eSuCZfO+91sTKIC1KtnoJH71WJyv0xOtmb5igMGj4No/aQwD6UbbPS8c2jUnV0HgidBjntAcpYZq/kBqaZXax1UkItVTUCIIw5lIJ/yS3Poq0xzn86dRNnBaUaCwTBlHOs07I5orL7jDn/d2QwDGxTDWawE14e6tasBQ4JpqqTNMyDKKsABSBa9KN0n1eOrPzBmle8vXHHZKyRIeaB6+sLyW6x4maZ6N7qagRBGHMpRL+ebl1+MCL/mknwjQVBorvu7vGzjAm1wXNwOZZJ6EoS/JCEaUl/TjjZtNn1bO51fL3WR2EaU6Y5FB30Jq5XjrTot0JU5KypOHbdKOM7IgLoF7Ge39e+0SpqBEEAS6Z8CsFdd+iLMEwqveDOOedrT6PO2El5o2g6mVbViLeCBwGcUHTNzENg91BzF6YVZYLtsWq98HEb5qVvL83ZGeYYBuK9ZrHtRWP282AMMkJs5zOMKVZc0jzkjDNsY3q2Kd7EU+IsUyw77686L5MqmZedP+y1UuCIFxeLpXwp1nJ288GlJQYGNxuBOwMkmoBV5KTZiXvbPe53fRRVBU/r67XedwZEmclvThlb5jxqBPRjzI8yyROM/pRQd03edaJiNISrTQbdZcoy8nykjDJ+Z1vb9ONClzb4Luur9IexHSTjN1Bxseu1wDNxooLWhNmObX84G5es7xMquagckoRfEEQLpXwZ2VJa8Wh4dl044xunJLkJbah2OolWAY4tsWtZkCUFfR6GVFS0okSXNsgSnNMw8BQkOQFtqn4N4+7DLOCVc8hzgo822C7n9CLMm43A9qDhE6c8YcPu6z6FiuezdNOyLN+xO4g5VkvJkoL6q6B1tCNMrb7CWlevtQE66yYz7OLHnMS9fLiqSMIl5dLJfyBY+EaBmGaY1AJYz/OiTLNvTWfm82A93cH/MmTLlmpqbkmRQHrdYd2nKOMKv9/u+HTG2ZYhkIpxarnEqUFGs1GzSHNC263fF5Zr/Gt5z00ClNBL8pRGrxrdbK0ZBAXFAV0o4QosVj1ckCz6tkvNFU5jGkxX2QXPbv/onMfJupSASQIl5tLJfyOZfD6zdWJqA2SnI0Vj6ftmCTPaQ9T+lFOnBeseA6ubZJoTTfKedaJuLnqUfctXtusc2fN52E75Fk3Js5yVn2b9ZrFe+2IvIRn3ZiG57I7TAmTgrgoafk2N5oe99drtIcpW8OYFc/kzprPes3l1Y0VdoYJUVYQOIt73h708437ARy1QmdRx6/pbmJp/mJ11HgfeQIQhIvPpRH+WUFr+A6dMKUbZvSTlBXPpj3IuLsesDtMeW83JEoLbjVdPraximeZNGs2eVGyUfdoeA6+bfHxG03e3wu52fQYxjlJUnCj6TNMc0xTcWPVpz8aOJqBRanh6487rAY2n7q/wTBOud4MSLIS01TcbQUT/x1YnK45iGUqdGZbQy4q65zt+HWt4ZEXJc97MeboCeha3WOrd/ATxlH/zWQwEYQPn0sj/GNBcyyDbpgRppVFQ1Fq0LCx4tKLM4oCPMug6dvcXfexDYNGUJVsag21ugka3t8b8rgTUhRQ80zW6y5P2xFhXrDVj6m7Nqu+xe4wphdnZKUmLkpWTANDGdRsi9LWXFvxsS1F6UKcldxuBLRGVT8HpVMWieIynv3jc89r1L6o45fJWlmqAAAgAElEQVRjGpOOYXmhWau5H9hFlCe/BkDSSYJwdlwa4bdH1grf2RlQFBCNJmJXfItelBAmBXdbAY5psDtMsQwDzzbZHaa8vzdkreZSFJrNusf7e0P++EmHdpTj2AZ3TI9elFHzLT5xs0E3zLmzEbBR83j7eZ9SaW40PFzHpO4Y1H2LVs3FUNWAs9WP6cYpWV7wtGtS86xJ43LfMQnTYtQ6sWq3OIjziS/QWLDHJm+HieV4zcLYJC4pS/JCAx8saJvt0zve36CyiLBNhW2aaM2kQc346eCk1gCc5oIyeZIQhIM5tvArpd4F+kAB5FrrBzOfK+DvAZ8FQuDHtNa/d9zrzuJYBk3f4d2dkLpnsjdIiLMc1zZH/jsuaPjaozalhifdmGSnwLUM4rTgRsNHm1TtGOOMdpQRJTm2slmv+2yuuNim4nk3olGzWfUsUHB3rU7dtQmzglXf5iObdQLHmkTjaV7ytBOxN0zwHLtqwZjkbPdjHu8Ned6PubbqYY8an6d5ydcfddgdxrRqLut1dyKK88QSPsi9A/si/bpv0e5Xbp+uYUzM4qb79E7bWtxuBpP7nj5vJZ4vNnU/Dqe1oEyeJAThcE4q4v/zWuudBZ/9IPCx0Z8/A/z90d8njm0ZeHa1ejYvS0qt8G2TrKy6bj3cC9kbJtRdmyTLQcFazaYT5zzrxqzXHKKkoD1MGMY5hmFgmyaGquYMAsdi2jETwLEUdc9mxbd5Za2GM7r+OEIP05yNmodnmSRlFVUDpEVJgSYvwBrZPYx9gt7ZGTBMC7rhfsvnabHMiyqy3+kneE41uNVci7QoaQYOWVFiKMVHr6/QrFX3u8gsbtbvf8y01cW8pu7zWDbaPi2L5qtqTSFPOcLL8GGken4Y+CWttQb+lVKqqZS6qbV+etIXqrkWNxpVK8VrKx7dOMNzLEgrAXAdA8s0edKJwVAElsn7nYhVz8EyFE3foR2lrLoWjmnhWgbKBNcy2e7HbK5UkXk3zEbpkOoLplRl9/ykG2KZ1RPERt1lmFbVN4Mkm6R+bjZ8sqKyeN7pp0RZTpRUNs9Q5dvrjoVrG7jK5GbDf6EV5DDJedIJ+aMnXdrDlJsNn7woWKt79KOMrChpD1JadYdBnBG45j47iWUj7bGYpPlyYvqy0fZpLCi7itYU8pQjvCwnIfwa+E2llAb+gdb6CzOf3wYeTr1/NNq2T/iVUm8AbwDcu3fvSDfiWAb3N+qTapYnnZCs0DR9m4Zf+dyveCZFYdMPSxxb4ecmn7q/RuBa2FbVWzYtNfev1wgskzwvcez9ZY1jxtFlVmraw4y8H7Pi2TzpRPi2Sd21+K6bq1xreLSC6olhux8TpgXoKtKvuzZaa242Kj+gwDGp+xZFUZnDNWvOCz9jVhhAlRqK0pytQYJjKu77lb20O+p21QycSVvI6TaOmyse3SidpKLGvQqmI//peQYFZLkmSot9A94s5yHavorNXs7D7124WJyE8H+/1vqxUuoa8GWl1Fta66+87ElGA8YXAB48eKCPejPTUeR4EBgLQOWzr7ndhD942KYVWChyskJPUiU3Gz69OAOq3HgnzXnai3CNSkArW+eqL243SmkPUwZpNaC0hyXtYYZnV7bNSV6lawLHnKRfxn2AUeCZBq2mx+aKi2Mbk4Frc6Xy+F+Ugql6DaiqGYxlo/IczzR4tDfkRsNno+6x3Y/nRr1pXvKkM16fUBDnJa6l8O3qaen+Rh2A93eHbA9ibMvENirx952D/7ucl2j7qllTnJffu3BxOLbwa60fj/7eUkr9CvApYFr4HwN3p97fGW07dWYFYBxRd8IM21A0AxfTMKh75kRsO1FKq+bg2SbNwGaQ5JNJ1+m+uE/2QspRiqcoNTdWfFY9h16SsTdIybXmdivgVtOf1OyneUk+enKwDIXrGuhSz3H1dCZplvG2acZdwz5yrRrYOmFGI7AJk4LNFY+6ZwFVVB+N+gOPU1UAWaExTUWnm9INcxqBSX3NJiv05JrmKLJ/uDsksCwC3+RG0z+wG9hRo23JTx+Pq/iUIxyPYwm/UqoGGFrr/uj1p4GfmdntV4GfVEr9U6pJ3e5p5PfnMS+NcbsV0AxyHEuR5po4KyhHFgjNwEHDKEWS0/Ar22WtmdS8j9MkpmlQlCUrnk3gWFxbdQkciyedEH9USfTKen2y0nbczQug7lq8slFV//SijM0V74Va/emc7XTNfpqX/P57e5RoilKzFrj04pRelLFe/6AcdJxS2h0k3GlZfGdnwPNewopXrRiuJoh1NfFbVOWkm3UXravUgaKywGgFLq+sBzzvRZOnl8ObyCwvPJKfPhmu2lOOcDyOG/FfB36lqtjEAv6J1vrXlVI/AaC1/nng16hKOd+mKuf868e85kIGcT6xQkbB21s93t8LsZXBvfUat1o+Y4XZrHuVL76qxL8XZTij0s6x8GVFSc2xCLOcvWFV666AXpzx9naf3X5VcvnJu2sAdKOUYZrjWmaV5olSOmFKlBZ044zXb1R2EllRNYQBCEY9gcf3Pi4FHedsO2HKO1t9rFF6RyloDzNurnm0BxlRXnC94dOLMrJC0w5T4rTANKvqo2e9iLe3enzjaY/rDY80K/lzr1/jk3dbvL87nAjuvfXaZA5inGe7NWpAY5kGNxr+aHL7xd7Ex+Gw/PSH8TQgTxwfIL+Lq8GxhF9r/W3ge+ds//mp1xr4W8e5zjIM4pzff2+PMM140o25seLy5rtt+mmBoTTf3u6xvurRHlRtFm+u+hiG4u3nA+I8BwzuNH2uNzx8u+rC9fZWn0Gc8byX4DomNdfEUYqs0HSiFK0MeknG1x+3edwZMkwy3t4asF7zSYqcT9xq8bwbcb3hsTNI8G2TbphiKhPTgkZhc2+9xjDO+frjziSKf/36KlGaE6UFvSilHVYe/4M4YxBVP9/DvSF31n2gJM0L0ryk4TvEaUEnTCk1hF6OoaFUilxrelFOO8z41taA773drMpINdRcE9sw2BnEDJOCRmCPJnKNffMkwCQ6z0erhxfNQyzLbIlqlVIrl16wdhhiSLc88ru4OlyalbthmpMWBd244K2nPd7dsXjreYeiqD57tupT2xpimAZ5lnGjWWOY5jzthtxo1qAsebgXcqvl4xhqsrp2q5vQjTNuNnzivOD6qkOclDxqR5RliWObdLwUvQ1Jpnnei4hWNcpQPPFD3nreJStXKcuSh7t9klyR6xLfMnmyV33JDBRJWbJRc/nGsx6+HRJlOb5j8s72kN4wYaPhYZkGjmPxqVc3eNaN+ej1GqZR9ffdHSS89bRHXJY83BnyPXeblF1YrzncbPi8tztgq5eiVcm7W0N2+wnNmk3Ds4nznLee9rAMg+f9iPXApeZb2KYCXT0ZTT+JOJbBo72qYipwzKUEYlaAp72ExhPfnbAa5Dph+oETKS9frTJ97sNcTM9LRcx5iLTPy+9COH0ujfAHjkWSl3z9UZf3d4a4Nuz2ErRSlIWmFyYUjoVhKvZ6KTtRBiiiuKAX9vAdA9sw6McpWV6y4ljYtkmcFRgGmEpRdyxavsvXtts870Vs9xNsU3FvLcC1TSzLouZZlKokL+Bpd8gwzXl3Z0CUVyWce/2UdpRhKthcdQnTnLtrNYqyZKuoBGrVtxlmOdv9lD951CEpNM97Kf/2qy0cQxFlOa3A5uZqUD15AChFq+awF6aAJkwLtvsxnSjlWTfi9mqAYyaYJnTCnM6wJM5ztroxlmmy04u4tebzaDci15rX66sM05z33tvFsQwMFJ+43UQB3TADdNXJrFg82Qv72156TrWeYHNlVHWU5jxtR2yseBRaV9VQgTMRnaNUq0xHreOUV3PqnLP3eR4qYs5LpP2yv4vzMFgJR+PSCH/ds/jY9RV++51tar5FmOQUhaagivyiTJPmObosUYYmSTSlMnBthUZjKEWUFkTDAtNWlGhWtCYHWq5DzVGsr3qVr06eM0gyhkk68sGBFdfmznoNx1B4jknDsVAortV9nnUjHndjFJqtXjyaOK0mSX//YZdOmPPdN1d4dTOgM7TpRRm9MGO3H7E9TDCUYmsQ85GbNb5nvUnTd7Ctah6j6TtkZYlrGjzcHdALU5Qy2BskKG3w+o0V3tkaYNmKjbpDXGiSrKreyTKNMko+er3Odj/mOzshu2HKim/ycC/kdsNHGYqG77DVSwizfLKAzDYVWVG+kJ6ZJs1L3t0Z8K3nfXaHKffXa1xveKOns5L2MOX5IGF7kFJzTGyrqiQaVzkdpVplOmrNimph3EFCdpYVMS+7QO60eZnfxXkZrISjcWmEH6qo//7mCnmu+L29XQoUutCkFJS6xHMs4sKALKcwSnxb0fBc6o4FGp6XETotSRIIo5zQL/jTt5t89HqdbFSKuTfIoFQ87yeEmYYcOnGKaSjyvDJes02D9iDBMUxMsxpENAVhXFBojWMYhGnBziBhfcXh/kaAOxoI+klGlJb0wpS0qNxFXddk0zVpeDag6EQZe4OYQVKg0NwZ+f+XKAZJwfWGx4ptYFgGT7sxgyjH3TQxDIOPrQXsDlK6UTYxtgscm9c2AtpRSmAZbKz42Lbi2qrLu7tD/vBhB0XJWq1aCNeqOROR2u7H+9Iz01/+rCgJ04I4L0iygnd3BjR8G7tuMIiq35UJPB9E3Fr18W2b+shEbnq18lEtqx3TYLO52MV0zFlUxEwLZz6aLD8PdfjL/i4kLXSxuVzCb1u0fItcl6y4DnWvZBClZGFBnBX04wKDgrQAx1To0sB3SshztC7xXJtelJMVBYFnV92zhhl/9KhP3TPRWlNqsEywlGLVNujmJWUOeQkog1LnhElOJ85xrBxDKTSj8kjbJtdQswzqnoWBoiwV7+4M+dOeDYBpKNZqNo/bEUrB2opLlpdsrLgorchzTYnmUTvmvd0QQ2lM00BrxYpr8ZHrdTzbpCw0Gk2YFNR9C8syyJOCwLHYuO7xtBtNVuVurnjcavh883mPrX6CVhpLKZJRRO9ZBtow6MQZb2/1eWW9NpnUtUxj4Zd/LMJ5oVlbcbGUInBNtgcxrmNghHCn5ZMXJTca1RqDw0zgDksvXJSa9lnhbI1WVp/ne57mPKTIhKNzaYQ/zUu2BzGWaVF3LW40HPaGGUPDxDJzihKKPMfxTBzTJPBMklH+vjdM6ScZ63UXx1J4jkWpDaI0Q1Oiy5J+XPK4HbHTj1Ea8rJkxbOI8pK6ZxEXBZ2o8t4ZZCVZrqnZCtcwGOYaSgPD1Ly2FlCgWXVslKloBh6ubbLi29imQZIXPGpHtMME24Q/++oacV6w5rvcaAWA5mk75nff2yXKC3ShubbqcH01wDYsykJj2GCY1QI1VqAzzKpBQ5ds9RJqbs699do+J853+wPSvMRSmmFWYloG7WGGa1v4Nmz1Y6Ik5/24KkNt+Ha1voDFX37HMvjItZXq36co6QwSdgYpYZrzXddXsU2DumuxVncBNbGDWNScZtn0wjhqHa/jOGsxnTdYTS8GLAqNvWJMFvpdBC7KACvM5+L8TzuEYZLz7s6wqt3fHZIUBTcbHjcaLtt9l3aYstuv8tqGkdMb5BSGYqcXE2XVo3aaa9YCh7ptUipFVNhYymB7GJOmGttUlFoROCarZY5nWzT8KmqLS41rGNRqDkkO2i6JUo0yNHlZkpQlnmHgOhaeZbIXJwyGBaVW1FyD7W7MWs1lLXCpuw7/1s0G/+ZRB9Oo7BRWAwffMXl7q08/irENA8832enH9MKST96tegAzcuB82onYGySgqnr8Zs2m/zSnH2fshQmbK96kIUwnTAnTghW/spLYDYdoNHu7MZs1l+uNgEfdyuYhyUtagYXvmGQjJ1DYby8xLXR1z+K7bzXY7lcNa7TW7PZTdoKEZlANHpWVxv4qnHnloi+TXljU0+DDZtFgNbYQGd/jdj8+0PX0PCKLxi4ul0b4AXYHCe/uhCR5yd4wo+U7WIaBa4BhQuAqSm3gWDCIC1o1iygqaQQWtjKJ85LAtfA8B6XAKQoGcWWr/CgPUUqRlQVhBpZh4VoK23IoUVwLDCxTYZomm77JxopHkhds9RK60YA4TrEDl26UMjAUgW0TU9BPErpDi6dOzCfuNbHM6Rr62qhHb847W33+6FmXMMlp+Q6uWXnovLJe5xO3VwncqqLINg3e3RkAEBdVc3g1WnbciRKMuKoK+sjIk2cstHvDBDTEWQ5aEaeaMC3xWibXVl2KcpX2MKUTp7SHOZ6ToaiM5sY+R7BY6ALHwjYUlmmwUbe5tuq+kMsf9xK2R75Ds+WiB6UXpgcb+MBrKHCsSfXRaYrUohTUQYOVUuCNFvBJnlz4MLk0wm+bBmWh2eonDKKUUms6YcZa4BB4NjdNi2daVT77qirPXPc8BkZelXKaBumg5PXrq3STlDzX2JZJZxjytBuhtcKq2dxv+Zi2ySBOyXKoWybbw4S1wEMrg8A2KYoSW0GmIUwysjwjKTRuXpBmBpsrDtthQhyXpGnB3TVI8oK3nna51QiwDIMwzfnmsx67YUZZam40XFzLoGZ71ByL77u/im/bGAp81wR0Fb0PU97dCbFMRT9O8CyDQVLQizPqjkknzomzgq896vDKeo1elPHatTq3WwF1t5rk/p1vb/PO9oC8gPf3qqeLMC1oRwm7/Zz6ho2CiUj3omxkjWFNunk1g6rncSdMaQbOyDLbZ5hU9tEN33lB5MbC3huV2k5aP44EcTa9ANUgMFuv3wwcTFMRuDZhsr+nwWlwUArqoMFK8uTCWXFphF8pWKtVUXinqGwYdvoxUZKxWnMwR7X4llKYyqDmKxzbZNM22Fj1CVwTSw141g95vBtVdfkG9NOcXpzjGFVUeq/pc6MV8J2dAXuDlEGWg9a0fAfHMgnjjCCweXurRyfMGSYZYaqxTEWhqxuteQ5P+wlYEGYFRVE1eM/zyuK5F8X4buWhYxsGe1FMO1K80qpR96p0zEebdTbrHt04xTYq2+jH7Yg/ftqhE2XUXbNqND/MCTwLy1I0/aoBjWUYdMMU1mrEWc7TTjSxb7ZNg7vrdfaGKTuDjF6UsNVT2IbFaxurNP20qnIalWM+aoeVRQYae9RQ5nE7pB9lI5fTDzp93WoGB6Y2pvsN2GbVBnIsiNN2HI5t7GsO048yylFf5TCt9lOAbxsUhUnDt0/t/910q8t56wUOyoUv+kzq44XT5tIIf5qVvL0T0gsztocJOi+IXUV7qDF7BigTU4FlGYRxTs0x+fZODxPFW9t9lFYM44S1usfuIKFVc+mnOUWRE6WQZNBLcpShGWYF7+/0GWYFvajANOH3H+6htEIZCsMwaPdjzJF1cl6CaYBtZ6R5wSDNaEcpNcsgzgve2w3xHYvXNuv4jsF3thJyIt7ZGlCUirjIaQY2vqW40fBYr7vYhsHXHrYxjOpnb9YcnrZD/vhJD4XCtmCt5vC0E8MAykJxbdVGKcWqb/P+zhAMTc2y2eonrDpV7b/vVE8zW4OE9iCh0IrAtTAo6RcZnmmyO0hJ0py6b1MUJa9u1InSgmFSELhV7j/KStCawLEmi7ymUxudkWNo4FiTNNE4z+9YxsRLKXCsiTFdUpa0+ynffXt1sn5i1bfZHcTkGtrDGJTidisgHxnwDdOcd7b7DNOc+xv1ExXS2ab2wD6n1TEH5cJnP5P6eOHD4NII/7hX7pNOVNXLl2BmmlyDokRRkgMmldP+qp2hDLANKKqKTjRgxhlxotkjpdQlSQZ5ARkQJZqneyGPiYiSgqSARIPKoZPkBAajShnop+AYmqiEcSsVM4MwStmmWm0bOlW7yN1Bwjef9skKRSdMMVSVCy+VgWMqDGXQSwv+77e2ee1azN2Wi+/YPB8ktAKbx52It7YG/OHDduU7lMMrmzWaI3fROC0xFeyFUGaadpSyO0jpjWrd86Kg4Tlcb+VkWUk3yhhGOQ3PZjfMqsEDsEyFbSjQippjcq3h0Q5hb5BgGQbtMGG7D7v9lNVrNtu9jGfdiI0p189xy8j3dgbEWYlpGNxuVk9cGtjqxrTqDu1ByrWGN2r1aFBS5fuf5FnlYRRnDOOcJC8JHIOP326y26/WUzQDh51+Aih8xyTPNcOkOPEc+jh/P57gnm14c5xznve8vzyVXGwujfBneck7z/v0o5y4rER8YjM5RTH6u5fNP8+gX+3Rj0qs0SnGx2Ql7AyrL2Y+dcz4MvGoQVdaVMdEo/c5UAK2hny0s6sgjsEKysoeOi+JkoxeZGAbihW/yv8mec7TbowXVvMOpgHv7fT5ruurgOI7g4R3tvuEccE3nnardpEY1H2LwK3+DOIYw4JrKzWetCMe7yUkecqgq+lHKWu+TVYMeS0ucCwoCs3TXgy6ZNV38G2TKC1Y9RxsEzxbVdYQO5XxW91xyIyCVs3BMU0etYf8f9/aRqNAKV5Zq03SMlXkr2gPs6r81jBIioLXNuo0gqoZvUG16M0ZpY7G8zffaQ8ZhAWP2xGOqeinKf2wpGvArVaKbSosU1WCmZf045SHeyGubZCXHlqvvMT/qMOZXSx2XNGfPeeivP9Zi648lVx8Lo3w25ZB07NRJujy8P3HjCcp523L+cCHHyrxZmbbLOXUfrPHTY818eiiRgRFEVMUOcoAjebeekB7WFk7GJYBZtUTYJAVfHOrR2DbWKbJ6zfr7IUZ272Yx52YXpxjm7DiumSFJslLrq16bK66uEaV5trpVYPJIKkqlixD0ag7pFnJKxs+7UHGXhoDJWFSgsoo9oa4hkGc5TR8i2urPtcbLms1l7yIiPOCEk17WNCquXiOiR1XLSQHcUY3TiszOcvg3e2YJMt561mPrCzxbZPNFRdFJShj0S8LTTeq/P8Dx+LVa3U812T1Tou9MGann7DTS2jVvInB261mgAJcy2BrENFPCtJC8z13GviuOaluOilOo5b9sHOelOgeZ/C4KE8lwmIujfAHtkXNt3GMyvR/Xu/GeSI/bz895/VYMw4aU15ivAGqtJNtgW8r1kZe93FcsjdMubXm49oWnmVQtxWduCBKS8oCVmtVW8f3tkNSrdmLMjJdUrOrwcWzIC+rdQdhUlADvuvmCt983icrIcsLTFV94ctCE6Uldctk1THoW2BiEKca11ZYRjVJ7DsmRgYbda+q47ctdgcJvShnkObcbQUURpXKKYuSKM1xbQPftSd5+p1+wrNuRJKV7A4Tao6J6VjcavncW6sRZjn1zTqoKj2lddXr90mnErooLehHITuDhEKXxGmJu6pQKDzLxLEMOsOM9jDlWTdmve6SpAVRXllNn0b0fBq17Aedc5hUVhfLGORNM1vuepzBQ6qRLj6XRvjDNEdrjWsbqKRcKOjzxH8ZNOCpKqf/sgK/iAIYZmAYmqLQGGh2ooSEklXf5XbDpcAgygu6ccidps/WICLLCpRlMkwzdvspoLCUQS3wKFHcWfMrY7q0YLOueNoNGcQF3SihWXPwOhbdOKMZ2FyreeRFSZhl/D/f2sVWsBLYNGo2WVqQ66pZTeBamIaBacDOIONOyyYv4E/dWmGrX80BFKWuWlECq57Dum9z/1rVaazhV2JWdx2SImGj5nKz6bO+4nKvVWN7EPOsG5GVmjQtsJ2xNxGgDDZWXKK0YG+Y4poGGVD3HVp1j7pj0g5T9gYZ7+0OWVuxeH97iGsZtOoONxset5ovituy0fOHlVpZpnfAdj9md5CwO4i50fBf2rF0XO56nIhdVu2eDh9mCu/yCH+WU2iFUupAYXapBHdBiv9AopcYMZYdYCzAsQ08z6QT5ax6NjWn6uDVT0rurTnsDAxc26TuVGWTjm3wSqvO2zsDXNfg+qpHnhds1DzitEBZBp6heN6PGYYZ/SznT90y0Sj2+tWktWcbKAXP+zFpqRkMUzJKdKnxHBvbMbhWc7nT9NnuJwzjnDgvMQyNVvDqeo29UTrKtlRV0qkVj/Zi6q7JnYaPYRiAoh2m1SIvxwKlyYuCFc9mPXC52wqwLYNhvyAvIclK0rLE0ybP+zFN7/9v70x/K8vTu/55zn7u4uu9XFW9VKd70oMyQpnQJIqCQkQURCCKeMGLIJEXvAlCCCUsQpA3CP6ACKQI0CgDSkQIgiwgoYASKZGSSBA0M8xkZroHZnpmuqurq8rrXc/223jxu9fldtkuu1zTruV8JKtc9rnnPOfa/v6e8/yeJSaJfJpkJwlRJuLuqGRWa/ppSCcOSKKQJA4pak03C3llrU+tHavdhI2+rxk4ifOELD6uePZ5rqOMJQoDvmuzx6h4eGTn0XMdFZDj98n8Gpfx2Nuq3SfLx71v8twIfyeOGE4rxjNz5nGPI/rh/HUXeVo479NFFMFKGrGap9Shpp+nGGd5b29K0WjuDmfszBq0Nigd8cp6l++5OWBSGbRzjKqG93YLlnop60sZ46Li/qRBR0I63xxNwoB3t6es9TL6ScjLyxkEAcNCMSoaAmB31qCcAevodyyfXFoGK9zdL6mdHyh/c5DziRt9VOP4xvbYzyIOHL00YSmPuTbIeOfuGG0NezNFL42wOK4v58xqzf3dKdE8iycOA7J5Ve3OpOLusOBbO1O0cyxlMThhUiqSUFghYWVeBNZo32p5b1ax0svoxKGfIhaHFKXGOkdZK64vZdxa7zGrDB8OS+JQHkrnPE/I4nHi2Y/juZ3nOgt7G20PR3aedO3jAnL8PrtpdNi2uvXYnw4+7n2TxxZ+EXkZ+BX83F0HfMY59y+PHfMjwH8FvjX/0m865/75417zbINgo58Sx0B9+mGP4+kvlpKLhIii+Uf1iONSgamyfHN3wno35eVOTNEounHEzrikqDSVE15e9t7d6xs9bix3mJSaRvmq3pVuyssrGX9y+4Bh5atosyhEGz8hLIkCgkCI8R5+YGA2Lw7rZAm5sTgMzjpEfGprL4u4MywYV5puGhI4IVkLGE0buouwT+irbJW2JKFvWXFzObPWmcQAABupSURBVMc4R2MM1/o5HxzMGBUKZSzjombSaD7cK7i51mHZ+fRXZRwr3YQsGdAYizUOEaG/mgMwrjQ3eDDj19c7hGSJv8dSzdtli/DmVp8bgw6zRjOrDf/v/piXVjuHvX+S6IH3f56QxeMMJ3kcz+081zmPvScJSDeNTnxdK/gX5zsVjvm4900u4/Fr4B84574gIn3g8yLyu865t48d94fOuZ+4xHXOhdKW3WlzoYye03jcfYCjhJy9yITzDxdCnkSsdlJeW+/73PhZSCAwLIWlbkLPgjOQZgEr3YRKWXbHBW/fnTIuNfuTkkmpmCkLAt04prGWJAoZZBFJEqGVxSCIBKyvJHQbx0YvYaYUb384JgxDtLUMsogsDtDWsJRGNNqLsDKOQRqy2U95db3L/lQTBkISQznPkd8eV+yMSyBgViv6aXI4pF0Zy/60xhiFb8gg3JtUIAFxACvdlIOp4vZB4cNBzjGcKeJQ2Ohn5JGfV7BIC82TkOVOws6komw029OS1zd6VIsc/9qQp8GRfvcnp/Q8KmRx0Xj243pu573Oo+w9TUDa0Mzl+U6GYz7ufZPHFn7n3F3g7vzziYi8A9wEjgv/x4IyFmMt5rKKzeVFH8586AD8U4QBmgpqVREF4sv+uzHW+FbPvsdNwLd3Z8xqi9GGb+/MOCgbvv7hiA/GDcYaOmHovXXr2B83BIH44rQsplCOfhbQ7UXsFRXDiWJUViwnMcoYqrpBHKRh4BeRqmEzzMAIWRKQNkIsQn85ZrWX8v5BQWMAcby80qFoNNPGUO8XjMqGr+9MeW21w0GhuL6c86euD3yl7sy3NfDVvAF7k5owEjaWEgRfEDauFJVRjGeKJImpa03UickT4f29GZXWjAtNL49IQr9H8d6eb+Y2LjWV0nwwLPnSnSHT0rDWT9jqJcRRyFIWPVbb44t6eJcdX/i4f/BHz/O0brxedf3BZflOh2M+zsX5icT4ReQW8Gngj0/49g+KyJeAD4F/6Jz76pO45sNGwN6sZqoffejTRmFAGUNTNXzhWwckQcjWwPI9NwaMKk2/k7AznnC/Unz+9hBnHaPKhzastWwt5WzPSirtDv+onAhb3ZShMmwsJdwfNWyPSnbHM8alD+cksR8q42Q+mwDAeg/5K/dGGGMJQkGsZZ2M//7le/Q7/pd+Pc94eQXiQMijkK/dG1M0mtt7M/Ymfi5BURu+uTvjtfUO39rx3U23JwX9NCLJY6pa886dCUkiVLWhbBTWCbuzmvUgYKkb4xBq47i9P+ZPPrC8tzcjjQOu9TOur+QMi5o4jFDaUCnH/kzxwX5JpRV39meoGwOur3Qf69H5pNbOwBMbBHN8CtfxNtTn5SRP9Pgid9WZSad5y0/SrvOc6zLXe57SWC8t/CLSA34D+Dnn3PjYt78AvOqcm4rIXwb+C/CJU87zM8DPALzyyisXtqOoDXcOphd+3dOCNYbboxpjK5w4vnQH/uCbO6gGZmVFYSAQ3zbZGJg1EAXQSSAJYFIbiloRRSFVabDieB+otGEya5gpzd6kZto8qDouFKAgwvh01RBUA5NQ0Wjns6Ocb1mxVyi0hiwRvt3NuNZLeWMzJ4gi7o0LDsqG8bTxc3THNXEgrGQx2jm+cX9EIH7YzLDQGOtotGNcabYnFUt5Qq00B2VDVVkmSrPSNSzlKcu5n0MwnTUkYUieRExqxZ1hyajwbTfWen6v4N7YP3XcmxQUlUE7X8+QJRF/+qXBR2L8iyEt8GCWwPFc93e3J9wf1/SzkH7mw0qL1tHGOF5Z69LLHhbYxXkfJQwLDxIH37g/ZW/WsNZNLtxT6CxPdGHPzqQ6bKH9pEIUx0X0rFDISTbCyfUEjyPO5wnDXDZU8zylsV5K+EUkxov+rzrnfvP4948uBM653xaRfyUi68653ROO/QzwGYC33nrrwtGWu8MZ6kkl2F8BBzND0UyYNb6RmTUGZWCqH4SeuhE0+sHegbIQ1PDu7pRQvGfcKO+NBAHMakUsMI2FunRMmgcb1UdZVCg75xcTY/yxR9/Och67UqUjjhR1HrFTaF7byLixkpOmIb/39j2i0G/6ltpwf1yhrMNahwSKr+9MiKOA60spO7rxTwtpRNUY7k9qZk0DImx0U1a7GVtLHbJIGFW+r9C00tw+KJgUNVEkJFHISpZxUGqcNjgRDqp63rLBsdlNCSTg3Z0JfhAlvLHZp5NEfDj0g2XAsTXIubHcYWdS0RifNTToxOxN/SyEnWlFP63ZqDMmpaKfRT5sBbxxrX8oANPKz03Ynfk3a62b8sZm/6HFYcFi5vE37k/55u6U6ypnUvnN89NSNU87z0me6ELoisawN614baOPMnYuvJcb83jS0wpw6gJ01EZt/ID5WfVwMRo8XnHZ8YXlpHt8EqGa52Wv5DJZPQJ8FnjHOfcLpxyzBdx3zjkR+X4gAPYe95pnEYUhwZMIzl8RykAvE8LQ57k3BopjYavZCWGsmYOihBRHkkBjIcb/W2j/x2PmoncWAb65XJb6TJkAR8LDWUkVUDaa/UJxf1zxZ26tUDaK8bQhwA+iiXCsZBEr3QScsF83KONIAuFaL6NUGm0cLo6opxXbzlHXFmUd3Sxm0ii+tTNlXDaMCt/2eq+ouTHo8MmtPrvTmP1Ko7QhTQKs1cyUIwuFotZsdBLCyFc9p1FIL4tIQ+Fr98bsThpurOREgW/g5vsHOYpG0xjLuFQUjWZcKbR19NOIolJU2l/v/rhCmZTNQUYYyqF4NNry/t6Me+OSydzjN8an0h5dHI6SRH4K196s4XqT4bDcH2k2+zWNtucWvdM80YXQLRaxcek3y5+E978499GhOXHoN9BPCoUsbFw8fWxPKj7cL7ACe9OKtW5Go/2C0BifJdZcoDL5+MJy0j0+T6Gay3IZj/+HgJ8GviwiX5x/7eeBVwCcc/8G+GvA3xYRDZTATznnviPy3E1D7DMs/LWFSeN8po8Dc8pexaKm4CiO+Way9l66g3nmDPQD3zxu0UPotIeiNIRGgXGGXhaQR/7NVPrB9QS/qAw6MUkobI9r/ue7e0QCk1qz1olIIkEpx0Y/xYmbzzf2A1FqbZkphTbQjYVOFLHXaNIwZOo0eRhQKz8opxHYnsBwVrOURyRByKAb4+Y9iEIRjItZ7aZM64BRXXFnVDFTmk4c8Uo/ZmupQxA5rAvYnlR084TGaO6PKq4t+UpgcHRj/2dQNYZx2RDHEUHg5yprI8yUIbEh+4XyWVJx4DfTeSAeyljCUOjnCfdHFRBwYxA9tDgsWk8v5h1304i1buIXm6mim4g/9gKiByd7okfz/rcG2aFXflA0l96gPG1ozllD430xWXAoyME8/beoDdpaDoqGstHcH1eEgRAg3FzunPv+F4tfo+2J9/g8hWouy2Wyev6Is/uV4Zz7ReAXH/caF6HWjjwNvII+g8QCWnnBa8zpAn1aeZrD/zAXnUAXX5va87WYqObXDDRElSUMIRK/AC0ykEL8foIyDm0Nw7Lm69tgFOyXJdr6Bnmb/Yw0jimVweCYlg1xFLCUx2wuZyQi3B9XaOsf6JW2aAfrKynOBkxLRRwG1LWhVAZlHKXW3Gw6bC4lrPYSPhzWBKFjZ1TTzSM2XOJnJvdTemmECFhxqMZxbTkEIuIgIA5CsiTglTW/4au0ZVg2HBQNRT2fKSBC3fjCszQKCedD4MeForaGqBGG1tJNQ4BDT1WAtW5CcmMATuh3osP+/IuNYuvgYFazOfCD6jf6vp3Ecifh3e0p39qd8uUPhmz003OL3mmcJHSLwTGX9XqPevDxvCOqMQ7y01+zeJ+08V59UWl2pzWC0J9nXZWNYbWbHi4kF2mstxD3ODz9Hp+XUM1leW4qd7XWjGbPpuiDbwcRuY+2e74w7oG3D/j5AIEXb4X31mv8D/1ojYFvBOGxwFBDrKGbecG3cDjLIElhkEdYGzApGz+nV1uUNSjlaIyl0pY8iegmwmaSM8iFlY7vrT+a+oKvOORwRGYUClkspFHIvWHFQaXRU8u1XsLrG32uL2fcOyh5eS2nn/h5vZNyl+2JYVgpUq1ZzlOyULGeJ0waRREY7h4UfH17wsurHT5xrcefvbXGai/zIYn5DceRz/UfV5px7buIvrzSQWnnR3QGQicO6WUxo7LBIZTGcb0XA35+wt1hSWO88L+y1uW1+TzjkzaKo9B3HxWBe8OKxVzh5Y6vfn51vUvdGLI4Oox5X4bjQvckvV5/bj+17f29GRbHV+4M2RxkDw24P7onoIylbAzWwbv3p6z2EqpaE4cBcSjEYXhYof048YHWs380z43wEwSEEae7xM8Ap4l+woMBMifd3mKw4PTYH4mykAb+sWwR5kmBLAFt/cZxPD/n8b8vBUyrB60qQqCb+rj1nWFFFIC2Gd+9tUzZaPaKhv2mIpi7aEXZUDV+GtlqJ6E2mlHV0EsjKmu4NuiwP6lxOPpZyua8p87upEHEobQhEl8I9uU7Nbvjiv2yYTVPWBtk3N6r2J/VTCtFJw0xypKnQpKEhM5inWO/aBCEPI6I4pCt5Q7XBznv7c744u0hYQhbSxnaOMZF7cMXYtHG0U3Dw8KzlU5CUWt2JhHjWnHvoCSPhEEe8/5+wf+9N6ZsDEkoaOP41EvL9LLo0LtW2nJvXLI3bZjVhq2lhKL27/piYxMgDoW6MdwbVwzyhA8PysdK7zzOSbUC5znnebNrFpPVRGBvWh/OUViEXYrG/2Y385CLMs57+XlEaQxxGNLL48NBNov9kuMjOi+S7dN69mfz3Aj/cid+cm0znzIsD2L2R1nE+8+qEBbnM3wS63P2wxBCgSTxIh4GPl0THt7IPboQWWB/ntkTAOu5Ty0dlzUbvYxBHpMHIQdVQ9VoLMJ6HtFPI9IoQFvHuKzJY2Fn5DtxVtqQRTHd5YBBLyEOhLsHJZUxvl20rgldRD+LyeMunTQijsW3pxbrs5ispUOICwNuLmUUjaGDUCjN/qxi3Gje3Z1gnWP0mmK9l7I9qSmUQhpBdf1+xJ1hgbVCEsFyHtOZ58Ev5ukezBocjo1uSiC+78+gk1AOSz+2s9YUwG5R8/7ejOuDnK/cGWJxjIuGnWlDrSzTRnNrPWejn86bztl5ZTGHHUQr5Qv59mYVN+r8Iymol0lzvEitwNGxkqelri7wU97M4RCdw81ZZQ/fg1oZEEijEGN9A71KWbYPKma5xpiUNzb6LJrKLUZ0Pir186I860VkT4rnRviVts90OudxQrw3rniwYbv4pQ95kIVTm5M99hCf4x84mCp/ntpAYuYnFj8LoNE+/LMIzR5tVyGL6/AghBTgFwRlwSjL3VGJMY40jlBoGqV9DpFziAtYyhLGlWJvWnFQaJz4Xvqzxm/QBjgsjp1Rg8FhxdFPIoZlQ6UCeklAUVucsUxrTS8OiWPDZKpQyuC0xc6FZ1obRtPar2xhQCwhm/2ATpzw5rUe8XyjNQxBVECl5zHgwHc/FYRpqbg7KunnMdNSEwSCdo5po6mVxRhLFga8tzdjrdHcG1ZMa0WlNHEYMkj9hu6oarA41noZ08qQJwFZHCL4+c2jQvHKWhc1z0A57GCaRvSyaJ7q+eCnepYQnyVmi+ybJAr4YL84DC09SjjVPA6/yHICTsxOWrSKDkMBA5+6uUwS+32NYfHgPbhzMKMThaz2MpyDmyu+lUcvi6kbjRG4Oyrpzu/9ePbNaamYFxHydnLYA54b4f9wWBI84QlLHzcLDz7De+bW+XnARwnwm66RQOgexOCPC78BRs3D1zCAtX4PXFk/XwC8mPdC6OVCWTvsfG/AzJ8Gjvcv8m0ehFACXzymHfuTBu0sjXGEIkwqxd1xgTWWQhm6aUjRaLI49K2YK+P3IIww6EYobemkCWtZTGkszjkyCdha8xO/3t8tfVhp3FA7S5oGGJcQx4JWMKsbGus3BM285UQeRWix3J/P4B3kybzVhEGA1zf7hx7lpNLszxoqpdlazlHGUjSGPI0YZDHLWUzRWKa14v60YtJY0kBY7yd8360VtscVvSwhCQMGWUJAwd60Ik8CNpf63B3WOGC1lxCGgogX5Gie3TObx7m3BpnvjZTHHxlEf5IQw9me8EJER4XiaGjpUdk8cRhg5mmunTT+SHbSUY7OHZ7VmiQODm3uJBEBwt7Ut+8utWF3WhEg3FrvstHPGBWKHWPoHbnGoqncohDu6H0cXQwuKuTt5LAHPDfCv9pNyBKYPIMtGwD6EfTziGGpySIIw5AASBpDPY/lWCCPoZOG4AwuCKCxRMrn7S+eChx+gTD4H/BC/xN8aGe1H2GMeDGYC3s3D+hkIS8vdYhjv2m5Pap9ZoyyhAFQQy/1fXVurXXJ0hilLIU2BCJIIMQSYqxlkMds9lNWOwmIUNuColJ005hXVjscFA3GVlwf5JSqoZkYnIN+FpCnIS/NWzf3ujG3Nnqs9zKsFRptKLVmEHRxgeVg2hCLIIllWnvRsBaW+77XjwSCw3vbyhiSKODWeu+hTJf1bkrVzOilIZPa8O3dGW9sLbHaSzEW+lmE1o6dyZT9QlFUGnGwvtrF4cjjiDc2+x8JpXz61VUvnIn//6trfiM4S8LDbB/4aG983zK595AXe5oQw+lFU/Bw9o2ab0I/KpsniXzmE3DYsuKk15yVG9/LosP3AHwa6eL9Xix6Z11jWDS4+b83VzoPbdguqqjPK+RtHv8Dnhvh/96X1/jz332d3/nq3VMHqT+NhMCNpYhPXuuzVxvypGYpTwgsEEBjDEWpkDgglIAoChmkAbiQpTzk3qTCGGEpC+fhlIZS+RGNYYjvsGk0YRCQxhFpErLRTRmWil4aMJ131lzrZHzXtR5bg5ybqzmzUvPuzoRRpbm9XzLoxIxnNdeWfFXtd2/12Bp0mNaKWhm2JxU7owjjLEVj6WYRS1nM1kpOEoZEAso5Xt/ssdFPmVYN25OGa4OMO/sFq92USlk2BylFZRiVGuMseRLy2lqXPI7YnVSEYcCnbq4AXvCqFQ1OmDWaQhnSOGBWWl6/3iUW2Js1DNIEJ4A88CiPi+Prm320cRxUDdeWAqa1phOHDPKYjX6GiA8VOHF0c8U4j7BO2OxnJJGcGDvvZdFH4uKbSxnLneQhUT9Py+SzRPI87ZyTKLlwD/5eFvHGtf6lehMt3oNF6wjn+Ijtp13jtPbSJz3NnFfI22yfBzw3wr/SS/h7f/GTvLHZ4/ao4MP9kqJpmJaGKApZymMQxwf7M0YzhQWWsoBr/Zy704ZmXk0aC/TzGBcEFE3DaOIDHHEIvQ7kQcxrG316aYgEAcO6oW4slbXsHsxIs4SqbrjW61A5SycMqJRDWT/4+2BWUyvoxNDJY75na4kf+MQ6b2wM2B6Vvq1yGs43XB3KOO6NSwIRZkpjjdDPY/YmJVYC+mnCxkrCIEtoGsPt/YJJ1RDOww2dNObaUswgj1ntZljr90LevjtiWjZo4xh0Em5t9nhpucOwaFjv+5THV9d7vLc3YbUT001ikkh4aa3D6+s93twagPjHeaUtX7s/RimLcT7w5PC5729eWwJgVDWMCkWWhAQI672Ur94ZMtOG64MOr1/zU6WuD/zA9GLeeiIKA97Y9LHlG8s5ylo6cYSy89huEPD+/ow7w5Jx0dDNQsTCS+s+pv/hsGR3UhMFAUvz+PFJ9LKIN68vHWaTbMFDYh6Hlm7iz7HeSbg+yA/j2ZcptDpvBsppInleMXucTJfzvOa8x5xm51nFZ5edT/A4tr4IyHeokPZSvPXWW+5zn/vcY712WmmKRhMHfkNod+oLhda6KZ0k4v39Kbf3Z6SR3+CKo4C9ac1w1pDGfoGIQqET+2KS9/cmVNqy1ktYyjKWOhHXBx2Utoyqhk4cUSjNuFTkcQjiB4QvxhEWysdtO3HE3XHJtFJEgeAEltKE68s5nSQ6rOSEj+Z/Lyo9lbEo7fvtLwqPRlXjh6kc6cS4OC6eV0/ifK768UZkjfKvBz9AfZFNcbSqtNGWUdk8SDecX+ukrJDjDc5O69B49OtHf1ZJHHzk2qed4yQW3uTR+z5+D3EYsDyvKn3Uuc66bpsV8vHRvtcXQ0Q+75x761zHPm/C39LS0vIichHhb5fRlpaWlheMVvhbWlpaXjBa4W9paWl5wWiFv6WlpeUFoxX+lpaWlheMVvhbWlpaXjCeynROEdkB3rtqO46xDjw0K/gp51mz+VmzF1qbPw6eNXvhamx+1Tm3cZ4Dn0rhfxoRkc+dN0f2aeFZs/lZsxdamz8OnjV74em3uQ31tLS0tLxgtMLf0tLS8oLRCv/5+cxVG/AYPGs2P2v2Qmvzx8GzZi885Ta3Mf6WlpaWF4zW429paWl5wWiF/xGIyL8VkW0R+cpV23IeRORlEfl9EXlbRL4qIj971TY9ChHJROR/i8iX5jb/s6u26TyISCgi/0dE/ttV23IeROTbIvJlEfmiiDwT7W9FZFlEfl1EviYi74jID161TachIm/O39vFx1hEfu6q7TqJNtTzCETkh4Ep8CvOuU9dtT2PQkSuA9edc18QkT7weeCvOufevmLTTkVEBOg656YiEgN/BPysc+5/XbFpZyIifx94C1hyzv3EVdvzKETk28BbzrlnJideRH4Z+EPn3C+JSAJ0nHPDq7brUYhICNwBfsA597TVJLUe/6Nwzv0BsH/VdpwX59xd59wX5p9PgHeAm1dr1dk4z3T+33j+8VR7JCLyEvBXgF+6alueV0RkAPww8FkA51zzLIj+nB8F3n0aRR9a4X+uEZFbwKeBP75aSx7NPGzyRWAb+F3n3NNu878A/hFgr9qQC+CA3xGRz4vIz1y1MefgNWAH+HfzkNoviUj3qo06Jz8F/NpVG3EarfA/p4hID/gN4Oecc+OrtudROOeMc+57gZeA7xeRpzasJiI/AWw75z5/1bZckD/nnPs+4MeBvzMPYz7NRMD3Af/aOfdpYAb846s16dHMQ1I/Cfznq7blNFrhfw6Zx8l/A/hV59xvXrU9F2H+KP/7wF+6alvO4IeAn5zHzP8j8BdE5N9frUmPxjl3Z/7vNvBbwPdfrUWP5APggyNPf7+OXwiedn4c+IJz7v5VG3IarfA/Z8w3Sj8LvOOc+4Wrtuc8iMiGiCzPP8+BHwO+drVWnY5z7p84515yzt3CP9L/nnPub1yxWWciIt35Zj/zcMlfBJ7qTDXn3D3gtoi8Of/SjwJPbZLCEf46T3GYB/yjVMsZiMivAT8CrIvIB8A/dc599mqtOpMfAn4a+PI8Zg7w8865375Cmx7FdeCX55kQAfCfnHPPRIrkM8Q14Le8X0AE/Afn3P+4WpPOxd8FfnUePvkm8Dev2J4zmS+qPwb8rau25SzadM6WlpaWF4w21NPS0tLygtEKf0tLS8sLRiv8LS0tLS8YrfC3tLS0vGC0wt/S0tLygtEKf0tLS8sLRiv8LS0tLS8YrfC3tLS0vGD8f+f1KqvmJhY1AAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import random\n",
    "plt.plot([x[0] + (random.random()-0.5) for x in plot_data if x[1]>1], [x[1] + (random.random()-0.5) for x in plot_data if x[1]>1], '.', alpha=0.1)\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{\"record\": 1, \"abstract\": \"The aim of this study was to characterise a novel mutation in the gene encoding RhAG in order to elucidate a molecular mechanism for Rh antigen expression and spherocytosis.Rhesus-associated glycoprotein (RhAG) is critical for maintaining the structure and stability of erythrocytes. Single missense mutations in the gene encoding RhAG are sufficient to induce spherocytosis and deficiencies in Rh complex formation. We report a novel missense mutation that incompletely disrupts Rh antigen expression and selectively knocks out RhD antigen expression.Blood samples were taken from a 38-year-old male, his brother, his wife and his daughter in Xi'an, China. To detect the proband's RhAG and D antigen expression, the RBC were stained with anti-D and anti-RhAG and analysed by flow cytometry. Red blood cell morphology was detected with atomic force microscopy (AFM). Genomic DNA was isolated from whole blood samples, and the RHD, RHCE and RHAG alleles were sequenced and analysed. The mutation was mapped onto a predicted crystal structure of RhAG by the I-TASSER server and visualised using PyMOL.Morphological testing by AFM found clear evidence of spherocytosis in the proband's erythrocytes. RHAG gene sequencing identified the mutation at sequence 236G\\u2009>\\u2009A, resulting in a serine to asparagine substitution at residue 79 (S79N). Family survey indicated that inheriting this allele is necessary and sufficient to cause the condition. Mapping the mutation onto a predicted crystal structure of RhAG revealed the proximity of the mutation to the critical structural elements of the protein.A novel RHAG mutation significantly lowers RhAG antigen expression and antigen-mediated agglutination intensity.\", \"gene\": \"RHAG\", \"transcripts\": [], \"hgvs\": \"236G>A\"}\r\n",
      "{\"record\": 2, \"abstract\": \"We aimed to identify mutations associated with osteochondromatosis in a litter of American Staffordshire Terrier puppies.We hypothesized that the associated mutation would be located in a gene that causes osteochondromatosis in humans.A litter of 9 American Staffordshire puppies, their sire and dam, 3 of 4 grandparents, 26 healthy unrelated American Staffordshire Terriers, and 154 dogs of 27 different breeds.Whole genome sequencing was performed on the proband, and variants were compared against polymorphisms derived from 154 additional dogs across 27 breeds, as well as single nucleotide polymorphism database 146. One variant was selected for follow-up sequencing. Parentage and genetic mosaicism were evaluated across the litter.We found 56,301 genetic variants unique to the proband. Eleven variants were located in or near the gene exostosin 2 (EXT2), which is strongly associated with osteochondromatosis in humans. One heterozygous variant (c.969C\\u2009>\\u2009A) is predicted to result in a stop codon in exon 5 of the gene. Sanger sequencing identified the identical mutation in all affected offspring. The mutation was absent in the unaffected offspring, both parents, all available grandparents, and 26 healthy unrelated American Staffordshire Terriers.These findings represent the first reported mutation associated with osteochondromatosis in dogs. Because this mutation arose de novo, the identical mutation is unlikely to be the cause of osteochondromatosis in other dogs. However, de novo mutations in EXT2 are common in humans with osteochondromatosis, and by extension, it is possible that dogs with osteochondromatosis could be identified by sequencing the entire EXT2 gene.\", \"gene\": \"EXT2\", \"transcripts\": [], \"hgvs\": \"c.969C>A\"}\r\n",
      "{\"record\": 3, \"abstract\": \"Dystrophic epidermolysis bullosa (DEB), pretibial, a rare subtype of epidermolysis bullosa (EB), is characterized by recurrent blisters and erosions predominantly on the pretibial region. We report the case of a 60-year-old Japanese woman with persistent blistering eruptions and scar formation on the pretibial region and elbows. Mutational analysis revealed a previously reported c.5797C>T mutation in exon 70 (p.R1933X) and a novel c.6348+1G>A mutation in intron 76 of COL7A1. Reverse transcription polymerase chain reaction revealed that the c.6348+1G>A mutation resulted in the skipping of exon 76 (69 bp) and the retention of intron 76 (75 bp), and both transcripts were in-frame. From these results, we diagnosed the patient as having recessive DEB, pretibial. A review of previously reported mutations in DEB, pretibial, revealed that one-third of DEB, pretibial, cases showed a recessive inheritance pattern, and no case had a combination of premature termination codon (PTC)/PTC mutations. The DEB, pretibial, case described herein is the first reported case of a compound heterozygote with PTC/in-frame mutations. Although no special characteristic features of the mutations were identified, a high diversity of COL7A1 mutations was shown even in DEB, pretibial.\", \"gene\": \"COL7A1\", \"transcripts\": [], \"hgvs\": \"c.5797C>T\"}\r\n",
      "{\"record\": 4, \"abstract\": \"Dystrophic epidermolysis bullosa (DEB), pretibial, a rare subtype of epidermolysis bullosa (EB), is characterized by recurrent blisters and erosions predominantly on the pretibial region. We report the case of a 60-year-old Japanese woman with persistent blistering eruptions and scar formation on the pretibial region and elbows. Mutational analysis revealed a previously reported c.5797C>T mutation in exon 70 (p.R1933X) and a novel c.6348+1G>A mutation in intron 76 of COL7A1. Reverse transcription polymerase chain reaction revealed that the c.6348+1G>A mutation resulted in the skipping of exon 76 (69 bp) and the retention of intron 76 (75 bp), and both transcripts were in-frame. From these results, we diagnosed the patient as having recessive DEB, pretibial. A review of previously reported mutations in DEB, pretibial, revealed that one-third of DEB, pretibial, cases showed a recessive inheritance pattern, and no case had a combination of premature termination codon (PTC)/PTC mutations. The DEB, pretibial, case described herein is the first reported case of a compound heterozygote with PTC/in-frame mutations. Although no special characteristic features of the mutations were identified, a high diversity of COL7A1 mutations was shown even in DEB, pretibial.\", \"gene\": \"COL7A1\", \"transcripts\": [], \"hgvs\": \"c.6348+1G>A\"}\r\n",
      "{\"record\": 5, \"abstract\": \"Dystrophic epidermolysis bullosa (DEB), pretibial, a rare subtype of epidermolysis bullosa (EB), is characterized by recurrent blisters and erosions predominantly on the pretibial region. We report the case of a 60-year-old Japanese woman with persistent blistering eruptions and scar formation on the pretibial region and elbows. Mutational analysis revealed a previously reported c.5797C>T mutation in exon 70 (p.R1933X) and a novel c.6348+1G>A mutation in intron 76 of COL7A1. Reverse transcription polymerase chain reaction revealed that the c.6348+1G>A mutation resulted in the skipping of exon 76 (69 bp) and the retention of intron 76 (75 bp), and both transcripts were in-frame. From these results, we diagnosed the patient as having recessive DEB, pretibial. A review of previously reported mutations in DEB, pretibial, revealed that one-third of DEB, pretibial, cases showed a recessive inheritance pattern, and no case had a combination of premature termination codon (PTC)/PTC mutations. The DEB, pretibial, case described herein is the first reported case of a compound heterozygote with PTC/in-frame mutations. Although no special characteristic features of the mutations were identified, a high diversity of COL7A1 mutations was shown even in DEB, pretibial.\", \"gene\": \"COL7A1\", \"transcripts\": [], \"hgvs\": \"c.6348+1G>A\"}\r\n",
      "{\"record\": 6, \"abstract\": \"To identify the pathogenic mutation underlying microcephaly primary hereditary (MCPH) in a large consanguineous Pakistani family.A five-generation family with an autosomal recessive transmission of MCPH was recruited. Targeted next-generation DNA sequencing was carried out to analyze the genomic DNA sample from the proband with MCPH using a previously designed panel targeting 46 known microcephaly-causing genes. Sanger sequencing was performed to verify all identified variants.We found a novel homozygous nonsense mutation, c.7543C>T, in the ASPM gene. This mutation led to the substitution of an arginine with a stop codon at amino acid residue 2515 (p.Arg2515Ter). The mutation cosegregated with the MCPH phenotype in all affected and obligate carrier family members, but was not present in public databases (dbSNP147, Exome Variant Server, the 1000 Genomes Project, Exome Aggregation Consortium, Human Gene Mutation Database, and ClinVar) or 200 control individuals. The c.7543C>T mutation in ASPM may activate nonsense-mediated mRNA decay pathways and could underlie the pathogenesis of MCPH through a loss-of-function mechanism.The c.7543C>T (p.Arg2515Ter) mutation in ASPM is a novel pathogenic mutation for the typical MCPH phenotype in this family.\", \"gene\": \"ASPM\", \"transcripts\": [], \"hgvs\": \"c.7543C>T\"}\r\n",
      "{\"record\": 7, \"abstract\": \"To identify the pathogenic mutation underlying microcephaly primary hereditary (MCPH) in a large consanguineous Pakistani family.A five-generation family with an autosomal recessive transmission of MCPH was recruited. Targeted next-generation DNA sequencing was carried out to analyze the genomic DNA sample from the proband with MCPH using a previously designed panel targeting 46 known microcephaly-causing genes. Sanger sequencing was performed to verify all identified variants.We found a novel homozygous nonsense mutation, c.7543C>T, in the ASPM gene. This mutation led to the substitution of an arginine with a stop codon at amino acid residue 2515 (p.Arg2515Ter). The mutation cosegregated with the MCPH phenotype in all affected and obligate carrier family members, but was not present in public databases (dbSNP147, Exome Variant Server, the 1000 Genomes Project, Exome Aggregation Consortium, Human Gene Mutation Database, and ClinVar) or 200 control individuals. The c.7543C>T mutation in ASPM may activate nonsense-mediated mRNA decay pathways and could underlie the pathogenesis of MCPH through a loss-of-function mechanism.The c.7543C>T (p.Arg2515Ter) mutation in ASPM is a novel pathogenic mutation for the typical MCPH phenotype in this family.\", \"gene\": \"ASPM\", \"transcripts\": [], \"hgvs\": \"c.7543C>T\"}\r\n",
      "{\"record\": 8, \"abstract\": \"To identify the pathogenic mutation underlying microcephaly primary hereditary (MCPH) in a large consanguineous Pakistani family.A five-generation family with an autosomal recessive transmission of MCPH was recruited. Targeted next-generation DNA sequencing was carried out to analyze the genomic DNA sample from the proband with MCPH using a previously designed panel targeting 46 known microcephaly-causing genes. Sanger sequencing was performed to verify all identified variants.We found a novel homozygous nonsense mutation, c.7543C>T, in the ASPM gene. This mutation led to the substitution of an arginine with a stop codon at amino acid residue 2515 (p.Arg2515Ter). The mutation cosegregated with the MCPH phenotype in all affected and obligate carrier family members, but was not present in public databases (dbSNP147, Exome Variant Server, the 1000 Genomes Project, Exome Aggregation Consortium, Human Gene Mutation Database, and ClinVar) or 200 control individuals. The c.7543C>T mutation in ASPM may activate nonsense-mediated mRNA decay pathways and could underlie the pathogenesis of MCPH through a loss-of-function mechanism.The c.7543C>T (p.Arg2515Ter) mutation in ASPM is a novel pathogenic mutation for the typical MCPH phenotype in this family.\", \"gene\": \"ASPM\", \"transcripts\": [], \"hgvs\": \"c.7543C>T\"}\r\n",
      "{\"record\": 9, \"abstract\": \"Autosomal recessive woolly hair is a relatively rare hereditary hair disorder characterized by sparse, short, curly hair. This condition is known to be caused by mutations in the LIPH gene, LPAR6 gene or KRT25 gene. In the Japanese population, most patients with autosomal recessive woolly hair carry one of two founder mutations in the LIPH gene, c.736T>A (p.Cys246Ser) or c.742C>A (p.His248Asn). However, occasionally, individuals with this condition carry compound heterozygous mutations, typically one founder mutation and another mutation. In this study, we describe a patient with a compound heterozygous mutation in the LIPH gene at c.736T>A and c.1095-3C>G. The latter mutation created a novel splice site. This was the fourth splice site mutation to be described in the LIPH gene. Furthermore, we performed an in vitro transcription assay in cultured cells, and demonstrated that the c.1095-3C>G mutation led to a frame-shift, which created a premature termination codon at the protein level (p.Glu366Ilefs*7). Finally, we summarized the mutations previously reported for the LIPH gene. Our findings provide further clues as to the molecular basis of autosomal recessive woolly hair.\", \"gene\": \"LIPH\", \"transcripts\": [], \"hgvs\": \"c.736T>A\"}\r\n",
      "{\"record\": 10, \"abstract\": \"Autosomal recessive woolly hair is a relatively rare hereditary hair disorder characterized by sparse, short, curly hair. This condition is known to be caused by mutations in the LIPH gene, LPAR6 gene or KRT25 gene. In the Japanese population, most patients with autosomal recessive woolly hair carry one of two founder mutations in the LIPH gene, c.736T>A (p.Cys246Ser) or c.742C>A (p.His248Asn). However, occasionally, individuals with this condition carry compound heterozygous mutations, typically one founder mutation and another mutation. In this study, we describe a patient with a compound heterozygous mutation in the LIPH gene at c.736T>A and c.1095-3C>G. The latter mutation created a novel splice site. This was the fourth splice site mutation to be described in the LIPH gene. Furthermore, we performed an in vitro transcription assay in cultured cells, and demonstrated that the c.1095-3C>G mutation led to a frame-shift, which created a premature termination codon at the protein level (p.Glu366Ilefs*7). Finally, we summarized the mutations previously reported for the LIPH gene. Our findings provide further clues as to the molecular basis of autosomal recessive woolly hair.\", \"gene\": \"LIPH\", \"transcripts\": [], \"hgvs\": \"c.742C>A\"}\r\n"
     ]
    }
   ],
   "source": [
    "!head parsed_abstracts.json"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
