{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def load_gene_names():\n",
    "    ret = set()\n",
    "    syns = {}\n",
    "\n",
    "    with open('genenames.txt') as f:\n",
    "        c = 0\n",
    "        for l in f:\n",
    "            c += 1\n",
    "            if c == 1:\n",
    "                header = l.split('\\t')\n",
    "                syn_index = header.index('Synonyms')\n",
    "                symb_index = header.index('Previous Symbols')\n",
    "                continue\n",
    "                \n",
    "            ls = l.split('\\t')\n",
    "            gene = ls[1]\n",
    "            ret.add(gene)\n",
    "            this_syn = [x.strip() for x in ls[syn_index].split(',')]\n",
    "            for x in this_syn:\n",
    "                syns[x] = gene\n",
    "            \n",
    "            symbols = [x.strip() for x in ls[symb_index].split(',')]\n",
    "            for x in symbols:\n",
    "                syns[x] = gene\n",
    "            \n",
    "            \n",
    "    return {x: None for x in ret}, syns\n",
    "\n",
    "stop = {\n",
    "    '2': None,\n",
    "    'T': None,\n",
    "    'B': None,\n",
    "}\n",
    "\n",
    "def check_gene_name(x):\n",
    "    \n",
    "    if x in stop:\n",
    "        return None\n",
    "    \n",
    "    if x in genenames:\n",
    "        if not 'realnames' in stats:\n",
    "            stats['realnames'] = 0\n",
    "        stats['realnames'] += 1\n",
    "        \n",
    "        return x\n",
    "    \n",
    "    if x in genesyns:\n",
    "        \n",
    "        if not 'syns' in stats:\n",
    "            stats['syns'] = 0\n",
    "        stats['syns'] += 1\n",
    "        \n",
    "        return genesyns[x]\n",
    "    \n",
    "    extra_synonyms = {\n",
    "        # Found : Real\n",
    "        'FXII': 'F12',\n",
    "    }\n",
    "    \n",
    "    if x in extra_synonyms:\n",
    "        return extra_synonyms[x]\n",
    "    \n",
    "    if x.lower() in ['mitochondrial', 'mtdna']:\n",
    "        \n",
    "        if not 'mitochondrial' in stats:\n",
    "            stats['mitochondrial'] = 0\n",
    "        stats['mitochondrial'] += 1\n",
    "        \n",
    "        return 'mtdna'\n",
    "    \n",
    "    return None\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "genenames, genesyns = load_gene_names()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "genenames['BRCA2']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'HPRT'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-10-511b85dce355>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mgenesyns\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'HPRT'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m: 'HPRT'"
     ]
    }
   ],
   "source": [
    "genesyns['HPRT']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Record: 1\n",
      "Record: 2\n",
      "Record: 3\n",
      "Record: 4\n",
      "Record: 5\n",
      "Record: 6\n",
      "Record: 7\n",
      "Record: 8\n",
      "Record: 9\n",
      "Record: 10\n",
      "Record: 11\n",
      "Record: 12\n",
      "Record: 13\n",
      "Record: 14\n",
      "Record: 15\n",
      "Record: 16\n",
      "Record: 17\n",
      "Record: 18\n",
      "Record: 19\n",
      "Record: 20\n",
      "Record: 21\n",
      "Record: 22\n",
      "Record: 23\n",
      "Record: 24\n",
      "Record: 25\n",
      "Record: 26\n",
      "Record: 27\n",
      "Record: 28\n",
      "Record: 29\n",
      "Record: 30\n",
      "Record: 31\n",
      "Record: 32\n",
      "Record: 33\n",
      "Record: 34\n",
      "Record: 35\n",
      "Record: 36\n",
      "Record: 37\n",
      "Record: 38\n",
      "Record: 39\n",
      "Record: 40\n",
      "Record: 41\n",
      "Record: 42\n",
      "Record: 43\n",
      "Record: 44\n",
      "Record: 45\n",
      "Record: 46\n",
      "Record: 47\n",
      "Record: 48\n",
      "Record: 49\n",
      "Record: 50\n",
      "Record: 51\n",
      "Record: 52\n",
      "Record: 53\n",
      "Record: 54\n",
      "Record: 55\n",
      "Record: 56\n",
      "Record: 57\n",
      "Record: 58\n",
      "Record: 59\n",
      "Record: 60\n",
      "Record: 61\n",
      "Record: 62\n",
      "Record: 63\n",
      "Record: 64\n",
      "Record: 65\n",
      "Record: 66\n",
      "Record: 67\n",
      "('Abstract counter:', 100)\n",
      "Record: 68\n",
      "Record: 69\n",
      "Record: 70\n",
      "Record: 71\n",
      "Record: 72\n",
      "Record: 73\n",
      "Record: 74\n",
      "Record: 75\n",
      "Record: 76\n",
      "Record: 77\n",
      "Record: 78\n",
      "Record: 79\n",
      "Record: 80\n",
      "Record: 81\n",
      "Record: 82\n",
      "Record: 83\n",
      "Record: 84\n",
      "Record: 85\n",
      "Record: 86\n",
      "Record: 87\n",
      "Record: 88\n",
      "Record: 89\n",
      "Record: 90\n",
      "Record: 91\n",
      "Record: 92\n",
      "Record: 93\n",
      "Record: 94\n",
      "Record: 95\n",
      "Record: 96\n",
      "Record: 97\n",
      "Record: 98\n",
      "Record: 99\n",
      "('Abstract counter:', 200)\n",
      "('HGVS FOUND:', 100)\n",
      "Record: 100\n",
      "Generated: parsed_abstracts.json\n",
      "Generated: stats.json\n"
     ]
    }
   ],
   "source": [
    "import re\n",
    "import json\n",
    "import codecs\n",
    "\n",
    "stats = {}\n",
    "\n",
    "def gen_abstracts(filename, per=100):\n",
    "    abstract_counter = 0\n",
    "    hgvs_counter = 0\n",
    "    with open(filename) as f:\n",
    "        for l in f:\n",
    "                        \n",
    "            data = json.loads(l)\n",
    "            for result in data['resultList']['result']:\n",
    "                \n",
    "                if not 'pubmedcounter' in stats:\n",
    "                    stats['pubmedcounter'] = 0\n",
    "                stats['pubmedcounter'] += 1\n",
    "                \n",
    "                if 'abstractText'in result:\n",
    "                    abstract = result['abstractText']\n",
    "                    abstract_counter += 1\n",
    "                    if abstract_counter % per == 0:\n",
    "                        print ('Abstract counter:', abstract_counter)\n",
    "                        \n",
    "                    if not 'abstractcounter' in stats:\n",
    "                        stats['abstractcounter'] = 0\n",
    "                    stats['abstractcounter'] += 1\n",
    "                        \n",
    "                    #if 'NM_' in abstract:\n",
    "                    #    print (abstract)\n",
    "                    #    assert False\n",
    "                    yield abstract\n",
    "                else:\n",
    "                    if not 'noabstract' in stats:\n",
    "                        stats['noabstract'] = 0\n",
    "                    stats['noabstract'] += 1\n",
    "\n",
    "def gen_hgvs(filename):\n",
    "    hgvs_counter = 0\n",
    "    \n",
    "    for abstract in gen_abstracts(filename):\n",
    "        \n",
    "        transcripts = get_transcripts(abstract)\n",
    "        \n",
    "        for hgvs in gen_hgvs_2(abstract):\n",
    "            hgvs_counter += 1\n",
    "\n",
    "            if hgvs_counter % 100 == 0:\n",
    "                print ('HGVS FOUND:', hgvs_counter)\n",
    "            \n",
    "            gene = get_gene_names(abstract, hgvs[1])\n",
    "            \n",
    "            yield abstract, gene, ''.join(hgvs), transcripts\n",
    "\n",
    "def get_gene_names(abstract, hgvs):\n",
    "    pos = abstract.index(hgvs)\n",
    "    \n",
    "    before = abstract[:pos]\n",
    "    after = abstract[pos+len(hgvs):]\n",
    "    \n",
    "    #Check before\n",
    "    before = before.split()[::-1]\n",
    "    before = [noparenthesis(x) for x in before]\n",
    "    \n",
    "    for word in before:\n",
    "        name = check_gene_name(word)\n",
    "        if name:\n",
    "            return name\n",
    "    \n",
    "    #Check after\n",
    "    after = [noparenthesis(x) for x in after.split()]\n",
    "    for word in after:\n",
    "        name = check_gene_name(word)\n",
    "        if name:\n",
    "            return name\n",
    "    \n",
    "    if not 'no_gene' in stats:\n",
    "        stats['no_gene'] = 0\n",
    "        \n",
    "    stats['no_gene'] += 1\n",
    "    \n",
    "    #print abstract\n",
    "    #print hgvs\n",
    "    \n",
    "    #if stats['no_gene'] > 1:\n",
    "    #    assert False\n",
    "    return None\n",
    "\n",
    "def gen_hgvs_2(abstract):\n",
    "    #245 A>\n",
    "    #found = re.findall(r'(.{1,100})([\\d]+[\\s]*[ACGT]+[\\s]*>)(.{1,100})',abstract, re.UNICODE)\n",
    "    found = re.findall(r'[\\d]+[\\s]*[ACGT]+[\\s]*>',abstract, re.UNICODE)\n",
    "    \n",
    "    if not found:\n",
    "        if not 'nohgvs' in stats:\n",
    "            stats['nohgvs'] = 0\n",
    "        stats['nohgvs'] += 1\n",
    "    \n",
    "    for f in found:\n",
    "        start = abstract.index(f)\n",
    "        before = abstract[:start]\n",
    "        after = abstract[start+len(f):]\n",
    "        \n",
    "        # We do not want any HGVS before and after the one that we found\n",
    "        before = before.replace('>','')\n",
    "        after = after.replace('>', '')\n",
    "\n",
    "        yield before, f, after\n",
    "    \n",
    "\n",
    "def get_transcripts(text):\n",
    "    return re.findall(r'[A-Z][A-Z]_[\\d]{4,15}\\.?[\\d]*', text, re.UNICODE)\n",
    "        \n",
    "def nowhite(p):\n",
    "    return re.sub(r'[\\s]', '', p, flags=re.UNICODE)\n",
    "\n",
    "def noparenthesis(p):\n",
    "    return re.sub(r'[\\(\\),\\.\\:]', '', p, flags=re.UNICODE)\n",
    "\n",
    "def post_process(hgvs):\n",
    "\n",
    "    s = None\n",
    "    \n",
    "    if not s:\n",
    "        # c.6348+1G>A\n",
    "        s = re.search(r'[CcGgpP][\\s]*\\.[\\s]*[\\d]+[\\s]*[\\+\\-][\\s]*[\\d]+[\\s]*[ACGT]+[\\s]*>[\\s]*[ACGT]+', hgvs, re.UNICODE)\n",
    "    \n",
    "    if not s:\n",
    "        # c.969C > A\n",
    "        s = re.search(r'[CcGgpPm][\\s]*\\.[\\s]*[\\d]+[\\s]*[ACGT][\\s]*>[\\s]*[ACGTS]+', hgvs, re.UNICODE)\n",
    "    \n",
    "    if not s:\n",
    "        # 236G > A\n",
    "        s = re.search(r'[\\d]+[\\s]*[ACGT]+[\\s]*>[\\s]*[ACGT]+', hgvs, re.UNICODE)\n",
    "    \n",
    "    if not s:\n",
    "        # c.743C > del \n",
    "        s = re.search(r'[cCgGpP]\\.[\\d]+[ACGT]+[\\s]*>[\\s]*del', hgvs, re.UNICODE)\n",
    "    \n",
    "    #print ('=====')\n",
    "    if s:\n",
    "        result = nowhite(s.group(0))\n",
    "        #print (hgvs)\n",
    "        #print (result)\n",
    "        return result\n",
    "    else:\n",
    "        #print (hgvs)\n",
    "        #assert False\n",
    "        \n",
    "        if not 'noparsehgvs' in stats:\n",
    "            stats['noparsehgvs'] = 0\n",
    "        stats['noparsehgvs'] += 1\n",
    "        \n",
    "        return None\n",
    "        \n",
    "        \n",
    "def do_1():\n",
    "    output_fn = 'parsed_abstracts.json'\n",
    "    output_f = codecs.open(output_fn, 'w', 'utf-8')\n",
    "    records = 0\n",
    "    for abstract, gene, hgvs, transcripts in gen_hgvs('abstracts.json'):\n",
    "        hgvs = post_process(hgvs)\n",
    "        if hgvs is None:\n",
    "            continue\n",
    "\n",
    "        \n",
    "        records += 1\n",
    "        \n",
    "        #if records <8:\n",
    "        #    continue\n",
    "        \n",
    "        if records > 100:\n",
    "            break\n",
    "        \n",
    "        to_save = {\n",
    "            'abstract': abstract,\n",
    "            'gene': gene,\n",
    "            'hgvs': hgvs,\n",
    "            'transcripts': transcripts,\n",
    "            'record': records\n",
    "        }\n",
    "        \n",
    "        #print gene, hgvs, transcripts\n",
    "        if False:\n",
    "            output_f.write('=========== {} ================\\n'.format(records))\n",
    "            output_f.write(abstract + '\\n')\n",
    "            output_f.write('GENE:' + gene + '\\n')\n",
    "            output_f.write('HGVS:'+ hgvs + '\\n')\n",
    "            output_f.write('Transcripts:' + str(transcripts) + '\\n')\n",
    "        if True:\n",
    "            print 'Record:', records\n",
    "            json.dump(to_save, output_f)\n",
    "            output_f.write('\\n')\n",
    "            output_f.flush()\n",
    "               \n",
    "    output_f.close()\n",
    "    print 'Generated:', output_fn\n",
    "    \n",
    "    with open('stats.json', 'w') as f:\n",
    "        json.dump(stats, f)\n",
    "    print 'Generated:', 'stats.json'\n",
    "\n",
    "def do_2():\n",
    "    counter={}\n",
    "    for abstract in gen_abstracts('abstracts.json', per=10000):\n",
    "        for tr in ['NT_', 'NG_', 'NC_']:\n",
    "            if tr in abstract:\n",
    "                #print '=====  {}  ======'.format(tr)\n",
    "                if not tr in counter:\n",
    "                    counter[tr] = []\n",
    "                    \n",
    "                counter[tr].append(abstract)\n",
    "                #print abstract\n",
    "                #assert False\n",
    "                \n",
    "    #print (counter)\n",
    "    return counter\n",
    "        \n",
    "do_1()\n",
    "#do_2()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input\ttranscript\tgene\tstrand\tcoordinates(gDNA/cDNA/protein)\tregion\tinfo\r\n",
      "SERPINF1:c.1067\tCCDS11012 (protein_coding)\tSERPINF1\t+\tchr17:g.1680550T/c.1067T/.\tinside_[cds_in_exon_7]\tsource=CCDS\r\n"
     ]
    }
   ],
   "source": [
    "!/home/user/mutationinfo/localpython/bin/transvar canno --ccds -i 'SERPINF1:c.1067' --refversion hg19"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'LIS1'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-36-6a6307ca66fe>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mgenenames\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'LIS1'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m: 'LIS1'"
     ]
    }
   ],
   "source": [
    "genenames['LIS1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('Abstract counter:', 10000)\n",
      "('Abstract counter:', 20000)\n",
      "('Abstract counter:', 30000)\n",
      "('Abstract counter:', 40000)\n",
      "('Abstract counter:', 50000)\n",
      "('Abstract counter:', 60000)\n",
      "('Abstract counter:', 70000)\n",
      "('Abstract counter:', 80000)\n",
      "('Abstract counter:', 90000)\n",
      "('Abstract counter:', 100000)\n",
      "('Abstract counter:', 110000)\n",
      "('Abstract counter:', 120000)\n",
      "('Abstract counter:', 130000)\n",
      "('Abstract counter:', 140000)\n",
      "('Abstract counter:', 150000)\n",
      "('Abstract counter:', 160000)\n",
      "('Abstract counter:', 170000)\n",
      "('Abstract counter:', 180000)\n",
      "('Abstract counter:', 190000)\n",
      "('Abstract counter:', 200000)\n",
      "('Abstract counter:', 210000)\n",
      "('Abstract counter:', 220000)\n",
      "('Abstract counter:', 230000)\n",
      "('Abstract counter:', 240000)\n",
      "('Abstract counter:', 250000)\n",
      "('Abstract counter:', 260000)\n",
      "('Abstract counter:', 270000)\n",
      "('Abstract counter:', 280000)\n",
      "('Abstract counter:', 290000)\n",
      "('Abstract counter:', 300000)\n",
      "('Abstract counter:', 310000)\n",
      "('Abstract counter:', 320000)\n",
      "('Abstract counter:', 330000)\n",
      "('Abstract counter:', 340000)\n",
      "('Abstract counter:', 350000)\n",
      "('Abstract counter:', 360000)\n",
      "('Abstract counter:', 370000)\n",
      "('Abstract counter:', 380000)\n",
      "('Abstract counter:', 390000)\n",
      "('Abstract counter:', 400000)\n",
      "('Abstract counter:', 410000)\n",
      "('Abstract counter:', 420000)\n",
      "('Abstract counter:', 430000)\n",
      "('Abstract counter:', 440000)\n",
      "('Abstract counter:', 450000)\n",
      "('Abstract counter:', 460000)\n",
      "('Abstract counter:', 470000)\n",
      "('Abstract counter:', 480000)\n",
      "('Abstract counter:', 490000)\n",
      "('Abstract counter:', 500000)\n",
      "('Abstract counter:', 510000)\n",
      "('Abstract counter:', 520000)\n",
      "('Abstract counter:', 530000)\n",
      "('Abstract counter:', 540000)\n",
      "('Abstract counter:', 550000)\n",
      "('Abstract counter:', 560000)\n",
      "('Abstract counter:', 570000)\n",
      "('Abstract counter:', 580000)\n",
      "('Abstract counter:', 590000)\n",
      "('Abstract counter:', 600000)\n",
      "('Abstract counter:', 610000)\n",
      "('Abstract counter:', 620000)\n",
      "('Abstract counter:', 630000)\n",
      "('Abstract counter:', 640000)\n",
      "('Abstract counter:', 650000)\n",
      "('Abstract counter:', 660000)\n",
      "('Abstract counter:', 670000)\n",
      "('Abstract counter:', 680000)\n",
      "('Abstract counter:', 690000)\n",
      "('Abstract counter:', 700000)\n",
      "('Abstract counter:', 710000)\n",
      "('Abstract counter:', 720000)\n",
      "('Abstract counter:', 730000)\n",
      "('Abstract counter:', 740000)\n",
      "('Abstract counter:', 750000)\n",
      "('Abstract counter:', 760000)\n",
      "('Abstract counter:', 770000)\n",
      "('Abstract counter:', 780000)\n",
      "('Abstract counter:', 790000)\n",
      "('Abstract counter:', 800000)\n",
      "('Abstract counter:', 810000)\n",
      "('Abstract counter:', 820000)\n",
      "('Abstract counter:', 830000)\n",
      "('Abstract counter:', 840000)\n",
      "('Abstract counter:', 850000)\n",
      "('Abstract counter:', 860000)\n",
      "('Abstract counter:', 870000)\n",
      "('Abstract counter:', 880000)\n",
      "('Abstract counter:', 890000)\n",
      "('Abstract counter:', 900000)\n",
      "('Abstract counter:', 910000)\n"
     ]
    }
   ],
   "source": [
    "counter = do_2()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('nt_ng_abstracts.json', 'w') as f:\n",
    "    json.dump(counter, f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NG_ 52\n",
      "NT_ 24\n",
      "NC_ 68\n"
     ]
    }
   ],
   "source": [
    "for x,y in counter.iteritems():\n",
    "    print x, len(y)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/user/mutationinfo/localpython/lib/python2.7/site-packages/psycopg2-2.7.4-py2.7-linux-x86_64.egg/psycopg2/__init__.py:144: UserWarning: The psycopg2 wheel package will be renamed from release 2.8; in order to keep installing from binary please use \"pip install psycopg2-binary\" instead. For details see: <http://initd.org/psycopg/docs/install.html#binary-install-from-pypi>.\n",
      "  \"\"\")\n"
     ]
    }
   ],
   "source": [
    "from MutationInfo import MutationInfo as MI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING dbfile.open_index: Falling back to hash index: unable to import bsddb\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING:pygr-log:Falling back to hash index: unable to import bsddb\n"
     ]
    }
   ],
   "source": [
    "mi = MI()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# RUN_1\n",
    "* Run with:  ```/home/user/mutationinfo/localpython/bin/python run_1.py```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Overwriting run_1.py\n"
     ]
    }
   ],
   "source": [
    "%%writefile run_1.py\n",
    "# -*- coding: utf-8 -*-\n",
    "\n",
    "def load_gene_names():\n",
    "    ret = set()\n",
    "    syns = {}\n",
    "\n",
    "    with open('genenames.txt') as f:\n",
    "        c = 0\n",
    "        for l in f:\n",
    "            c += 1\n",
    "            if c == 1:\n",
    "                header = l.split('\\t')\n",
    "                syn_index = header.index('Synonyms')\n",
    "                symb_index = header.index('Previous Symbols')\n",
    "                continue\n",
    "                \n",
    "            ls = l.split('\\t')\n",
    "            gene = ls[1]\n",
    "            ret.add(gene)\n",
    "            this_syn = [x.strip() for x in ls[syn_index].split(',')]\n",
    "            for x in this_syn:\n",
    "                syns[x] = gene\n",
    "            \n",
    "            symbols = [x.strip() for x in ls[symb_index].split(',')]\n",
    "            for x in symbols:\n",
    "                syns[x] = gene\n",
    "            \n",
    "            \n",
    "    return {x: None for x in ret}, syns\n",
    "\n",
    "stop = {\n",
    "    '2': None,\n",
    "    'T': None,\n",
    "    'B': None,\n",
    "}\n",
    "\n",
    "def check_gene_name(x):\n",
    "    \n",
    "    if x in stop:\n",
    "        return None\n",
    "    \n",
    "    if x in genenames:\n",
    "        if not 'realnames' in stats:\n",
    "            stats['realnames'] = 0\n",
    "        stats['realnames'] += 1\n",
    "        \n",
    "        return x\n",
    "    \n",
    "    if x in genesyns:\n",
    "        \n",
    "        if not 'syns' in stats:\n",
    "            stats['syns'] = 0\n",
    "        stats['syns'] += 1\n",
    "        \n",
    "        return genesyns[x]\n",
    "    \n",
    "    extra_synonyms = {\n",
    "        # Found : Real\n",
    "        'FXII': 'F12',\n",
    "    }\n",
    "    \n",
    "    if x in extra_synonyms:\n",
    "        return extra_synonyms[x]\n",
    "    \n",
    "    if x.lower() in ['mitochondrial', 'mtdna']:\n",
    "        \n",
    "        if not 'mitochondrial' in stats:\n",
    "            stats['mitochondrial'] = 0\n",
    "        stats['mitochondrial'] += 1\n",
    "        \n",
    "        return 'mtdna'\n",
    "    \n",
    "    return None\n",
    "\n",
    "###################################\n",
    "genenames, genesyns = load_gene_names()\n",
    "\n",
    "import re\n",
    "import gzip\n",
    "import json\n",
    "import codecs\n",
    "\n",
    "stats = {}\n",
    "\n",
    "def gen_abstracts(filename, per=100):\n",
    "    abstract_counter = 0\n",
    "    hgvs_counter = 0\n",
    "    with gzip.open(filename, 'rt') as f:\n",
    "        for l in f:\n",
    "                        \n",
    "            data = json.loads(l)\n",
    "            for result in data['resultList']['result']:\n",
    "                \n",
    "                if not 'pubmedcounter' in stats:\n",
    "                    stats['pubmedcounter'] = 0\n",
    "                stats['pubmedcounter'] += 1\n",
    "                \n",
    "                if 'abstractText'in result:\n",
    "                    abstract = result['abstractText']\n",
    "                    abstract_counter += 1\n",
    "                    if abstract_counter % per == 0:\n",
    "                        print ('Abstract counter:', abstract_counter)\n",
    "                        \n",
    "                    if not 'abstractcounter' in stats:\n",
    "                        stats['abstractcounter'] = 0\n",
    "                    stats['abstractcounter'] += 1\n",
    "                        \n",
    "                    #if 'NM_' in abstract:\n",
    "                    #    print (abstract)\n",
    "                    #    assert False\n",
    "                    yield abstract\n",
    "                else:\n",
    "                    if not 'noabstract' in stats:\n",
    "                        stats['noabstract'] = 0\n",
    "                    stats['noabstract'] += 1\n",
    "\n",
    "def gen_hgvs(filename):\n",
    "    hgvs_counter = 0\n",
    "    \n",
    "    for abstract in gen_abstracts(filename):\n",
    "        \n",
    "        transcripts = get_transcripts(abstract)\n",
    "        \n",
    "        for hgvs in gen_hgvs_2(abstract):\n",
    "            hgvs_counter += 1\n",
    "\n",
    "            if hgvs_counter % 100 == 0:\n",
    "                print ('HGVS FOUND:', hgvs_counter)\n",
    "            \n",
    "            gene = get_gene_names(abstract, hgvs[1])\n",
    "            \n",
    "            yield abstract, gene, ''.join(hgvs), transcripts\n",
    "\n",
    "def get_gene_names(abstract, hgvs):\n",
    "    pos = abstract.index(hgvs)\n",
    "    \n",
    "    before = abstract[:pos]\n",
    "    after = abstract[pos+len(hgvs):]\n",
    "    \n",
    "    #Check before\n",
    "    before = before.split()[::-1]\n",
    "    before = [noparenthesis(x) for x in before]\n",
    "    \n",
    "    for word in before:\n",
    "        name = check_gene_name(word)\n",
    "        if name:\n",
    "            return name\n",
    "    \n",
    "    #Check after\n",
    "    after = [noparenthesis(x) for x in after.split()]\n",
    "    for word in after:\n",
    "        name = check_gene_name(word)\n",
    "        if name:\n",
    "            return name\n",
    "    \n",
    "    if not 'no_gene' in stats:\n",
    "        stats['no_gene'] = 0\n",
    "        \n",
    "    stats['no_gene'] += 1\n",
    "    \n",
    "    #print abstract\n",
    "    #print hgvs\n",
    "    \n",
    "    #if stats['no_gene'] > 1:\n",
    "    #    assert False\n",
    "    return None\n",
    "\n",
    "def gen_hgvs_2(abstract):\n",
    "    #245 A>\n",
    "    #found = re.findall(r'(.{1,100})([\\d]+[\\s]*[ACGT]+[\\s]*>)(.{1,100})',abstract, re.UNICODE)\n",
    "    found = re.findall(r'[\\d]+[\\s]*[ACGT]+[\\s]*>',abstract, re.UNICODE)\n",
    "    \n",
    "    if not found:\n",
    "        if not 'nohgvs' in stats:\n",
    "            stats['nohgvs'] = 0\n",
    "        stats['nohgvs'] += 1\n",
    "    \n",
    "    for f in found:\n",
    "        start = abstract.index(f)\n",
    "        before = abstract[:start]\n",
    "        after = abstract[start+len(f):]\n",
    "        \n",
    "        # We do not want any HGVS before and after the one that we found\n",
    "        before = before.replace('>','')\n",
    "        after = after.replace('>', '')\n",
    "\n",
    "        yield before, f, after\n",
    "    \n",
    "\n",
    "def get_transcripts(text):\n",
    "    return re.findall(r'[A-Z][A-Z]_[\\d]{4,15}\\.?[\\d]*', text, re.UNICODE)\n",
    "        \n",
    "def nowhite(p):\n",
    "    return re.sub(r'[\\s]', '', p, flags=re.UNICODE)\n",
    "\n",
    "def noparenthesis(p):\n",
    "    return re.sub(r'[\\(\\),\\.\\:]', '', p, flags=re.UNICODE)\n",
    "\n",
    "def post_process(hgvs):\n",
    "\n",
    "    s = None\n",
    "    \n",
    "    if not s:\n",
    "        # c.6348+1G>A\n",
    "        s = re.search(r'[CcGgpP][\\s]*\\.[\\s]*[\\d]+[\\s]*[\\+\\-][\\s]*[\\d]+[\\s]*[ACGT]+[\\s]*>[\\s]*[ACGT]+', hgvs, re.UNICODE)\n",
    "    \n",
    "    if not s:\n",
    "        # c.969C > A\n",
    "        s = re.search(r'[CcGgpPm][\\s]*\\.[\\s]*[\\d]+[\\s]*[ACGT][\\s]*>[\\s]*[ACGTS]+', hgvs, re.UNICODE)\n",
    "    \n",
    "    if not s:\n",
    "        # 236G > A\n",
    "        s = re.search(r'[\\d]+[\\s]*[ACGT]+[\\s]*>[\\s]*[ACGT]+', hgvs, re.UNICODE)\n",
    "    \n",
    "    if not s:\n",
    "        # c.743C > del \n",
    "        s = re.search(r'[cCgGpP]\\.[\\d]+[ACGT]+[\\s]*>[\\s]*del', hgvs, re.UNICODE)\n",
    "    \n",
    "    #print ('=====')\n",
    "    if s:\n",
    "        result = nowhite(s.group(0))\n",
    "        #print (hgvs)\n",
    "        #print (result)\n",
    "        return result\n",
    "    else:\n",
    "        #print (hgvs)\n",
    "        #assert False\n",
    "        \n",
    "        if not 'noparsehgvs' in stats:\n",
    "            stats['noparsehgvs'] = 0\n",
    "        stats['noparsehgvs'] += 1\n",
    "        \n",
    "        return None\n",
    "        \n",
    "        \n",
    "def do_1():\n",
    "    output_fn = 'parsed_abstracts.json'\n",
    "    output_f = codecs.open(output_fn, 'w', 'utf-8')\n",
    "    records = 0\n",
    "    for abstract, gene, hgvs, transcripts in gen_hgvs('abstracts.json.gz'):\n",
    "        hgvs = post_process(hgvs)\n",
    "        if hgvs is None:\n",
    "            continue\n",
    "\n",
    "        \n",
    "        records += 1\n",
    "        if records % 100 == 0:\n",
    "            print 'Records:', records\n",
    "        \n",
    "        #if records <8:\n",
    "        #    continue\n",
    "        \n",
    "        #if records > 100:\n",
    "        #    break\n",
    "        \n",
    "        to_save = {\n",
    "            'abstract': abstract,\n",
    "            'gene': gene,\n",
    "            'hgvs': hgvs,\n",
    "            'transcripts': transcripts,\n",
    "            'record': records\n",
    "        }\n",
    "        \n",
    "        #print gene, hgvs, transcripts\n",
    "        if False:\n",
    "            output_f.write('=========== {} ================\\n'.format(records))\n",
    "            output_f.write(abstract + '\\n')\n",
    "            output_f.write('GENE:' + gene + '\\n')\n",
    "            output_f.write('HGVS:'+ hgvs + '\\n')\n",
    "            output_f.write('Transcripts:' + str(transcripts) + '\\n')\n",
    "        if True:\n",
    "            json.dump(to_save, output_f)\n",
    "            output_f.write('\\n')\n",
    "            output_f.flush()\n",
    "               \n",
    "    output_f.close()\n",
    "    print 'Generated:', output_fn\n",
    "    \n",
    "    with open('stats.json', 'w') as f:\n",
    "        json.dump(stats, f)\n",
    "    print 'Generated:', 'stats.json'\n",
    "\n",
    "def do_2():\n",
    "    counter={}\n",
    "    for abstract in gen_abstracts('abstracts.json.gz', per=10000):\n",
    "        for tr in ['NT_', 'NG_', 'NC_']:\n",
    "            if tr in abstract:\n",
    "                #print '=====  {}  ======'.format(tr)\n",
    "                if not tr in counter:\n",
    "                    counter[tr] = []\n",
    "                    \n",
    "                counter[tr].append(abstract)\n",
    "                #print abstract\n",
    "                #assert False\n",
    "                \n",
    "    #print (counter)\n",
    "    return counter\n",
    "        \n",
    "do_1()\n",
    "#do_2()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# RUN_2\n",
    "\n",
    "* Run with ```/home/user/mutationinfo/localpython/bin/python run_2.py```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Overwriting run_2.py\n"
     ]
    }
   ],
   "source": [
    "%%writefile run_2.py\n",
    "\n",
    "#assert False # THIS IS ON PURPOSE. DO NOT RUN THIS\n",
    "\n",
    "import re\n",
    "import json\n",
    "import subprocess\n",
    "\n",
    "#import pandas as pd\n",
    "\n",
    "from StringIO import StringIO\n",
    "\n",
    "def execute(command):\n",
    "    process = subprocess.Popen(command, shell=True,\n",
    "                           stdout=subprocess.PIPE, \n",
    "                           stderr=subprocess.PIPE)\n",
    "\n",
    "    # wait for the process to terminate\n",
    "    out, err = process.communicate()\n",
    "    errcode = process.returncode\n",
    "    \n",
    "    out = out.decode('utf-8')\n",
    "    err = err.decode('utf-8')\n",
    "    \n",
    "    #print ('STDOUT:')\n",
    "    #print (out)\n",
    "    #print ('ERR:')\n",
    "    #print (err)\n",
    "    #print ('RETURN CODE:', errcode)\n",
    "    \n",
    "    return out\n",
    "\n",
    "def inverse(s):\n",
    "    \n",
    "    return {\n",
    "        'A': 'T',\n",
    "        'T': 'A',\n",
    "        'C': 'G',\n",
    "        'G': 'C',\n",
    "    }[s]\n",
    "\n",
    "stats_2 = {}\n",
    "\n",
    "def get_transvar_reference(coord, strand):\n",
    "    s = re.search(r'chr[\\dXY]+:g.[\\d]+([ACGT]+)/', coord)\n",
    "    if s:\n",
    "        reference = s.group(1)\n",
    "    else:\n",
    "        return None\n",
    "\n",
    "    if strand == '-':\n",
    "        return inverse(reference)\n",
    "    elif strand == '+':\n",
    "        return reference\n",
    "    else:\n",
    "        assert False\n",
    "\n",
    "\n",
    "def parse_transvar_output(output,command=None, hgvs=None, abstract=None):\n",
    "    ls = output.split('\\n')\n",
    "    ls = [x.split('\\t') for x in ls if x.strip()] \n",
    "    strand_i = ls[0].index('strand')\n",
    "    coord_i = ls[0].index('coordinates(gDNA/cDNA/protein)')\n",
    "    transcript_i = ls[0].index('transcript')\n",
    "    \n",
    "    ret = {l[transcript_i]:get_transvar_reference(l[coord_i], l[strand_i])  for l in ls[1:]}\n",
    "    return ret    \n",
    "\n",
    "def load_parsed_abstracts():\n",
    "    with open('parsed_abstracts.json') as f:\n",
    "        for l in f:\n",
    "            data = json.loads(l)\n",
    "            yield data\n",
    "            \n",
    "def apply_transvar():\n",
    "    \n",
    "    transvar_cmd_p = \"/home/user/mutationinfo/localpython/bin/transvar canno --ccds -i '{hgvs}' --refversion hg19\"\n",
    "    fn = 'parsed_abstracts_transvar.json'\n",
    "    f = open(fn, 'w')\n",
    "    counter = 0\n",
    "    input_transvar_fn = 'genes_hgvs.txt'\n",
    "    input_transvar_f = open(input_transvar_fn, 'w')\n",
    "    for parsed in load_parsed_abstracts():\n",
    "        counter += 1\n",
    "        \n",
    "        if counter < 0:\n",
    "            continue\n",
    "            \n",
    "        #if counter > 100:\n",
    "        #    break\n",
    "        \n",
    "        print 'Counter:', counter\n",
    "        \n",
    "        abstract = parsed['abstract']\n",
    "        \n",
    "        if 'dog' in abstract:\n",
    "            stats_2['nothumnan'] = stats_2.get('nothumnan', 0) + 1\n",
    "            continue\n",
    "\n",
    "        \n",
    "        #print parsed\n",
    "        hgvs = parsed['hgvs']\n",
    "        \n",
    "        #Get hgvs reference\n",
    "        s = re.search(r'([ACGT]+)>', hgvs)\n",
    "        if not s:\n",
    "            continue\n",
    "            #assert False\n",
    "            \n",
    "        hgvs_reference = s.group(1)\n",
    "                \n",
    "        #If it starts with a number assuming coding position\n",
    "        if re.search(r'^[\\d]+', hgvs, re.UNICODE):\n",
    "            hgvs = 'c.' + hgvs\n",
    "            \n",
    "            \n",
    "        #Get everything except the reference\n",
    "        s = re.search(r'^([^ACGT]+)[ACGT]+>', hgvs)\n",
    "        if not s:\n",
    "            continue\n",
    "            #assert False\n",
    "        except_reference = s.group(1)\n",
    "\n",
    "        \n",
    "        gene = parsed['gene']\n",
    "        if not gene:\n",
    "            continue\n",
    "            #assert False\n",
    "\n",
    "        transcripts = parsed['transcripts']\n",
    "        if transcripts:\n",
    "            pass\n",
    "            #assert False\n",
    "        \n",
    "        hgvs = gene + ':' + hgvs\n",
    "        hgvs_no_reference = gene + ':' + except_reference\n",
    "        #transvar_cmd = transvar_cmd_p.format(hgvs=hgvs)\n",
    "        transvar_cmd = transvar_cmd_p.format(hgvs=hgvs_no_reference)\n",
    "        #print transvar_cmd\n",
    "        #print abstract\n",
    "        #transvar_output = execute(transvar_cmd) # DO NOT EXECUTE IT\n",
    "        input_transvar_f.write('{}\\t{}\\n'.format(gene, except_reference))\n",
    "        continue\n",
    "        \n",
    "        parsed['transvar'] = transvar_output\n",
    "        f.write(json.dumps(parsed) + '\\n')\n",
    "        continue\n",
    "        \n",
    "        #print transvar_output\n",
    "        #transvar_output_f = StringIO(transvar_output)\n",
    "        #df = pd.read_csv(transvar_output_f, sep='\\t')\n",
    "        #print df\n",
    "        \n",
    "        transvar_references = parse_transvar_output(transvar_output, transvar_cmd, hgvs, abstract)\n",
    "        if len(transvar_references) == 0:\n",
    "            assert False\n",
    "            \n",
    "        if len(transvar_references) == 1:\n",
    "            if transvar_references.values()[0] == hgvs_reference:\n",
    "                stats_2['transvarok'] = stats_2.get('transvarok', 0) + 1\n",
    "            else:                \n",
    "                print transvar_cmd\n",
    "                print 'HGVS:', hgvs\n",
    "                print 'transvar_reference:', transvar_references.values()[0], 'Hgvs_reference:', hgvs_reference\n",
    "                assert transvar_references.values()[0]\n",
    "                assert hgvs_reference\n",
    "                stats_2['1_reference_not_compatible'] = stats_2.get('1_reference_not_compatible', 0) + 1\n",
    "        else:\n",
    "            # Take all references\n",
    "            all_references = list(set(transvar_references.values()))\n",
    "            if len(all_references) == 1:\n",
    "                if all_references[0] == hgvs_reference:\n",
    "                    stats_2['1_reference_many_transcripts'] = stats_2.get('1_reference_many_transcripts', 0) + 1\n",
    "                else:\n",
    "                    assert False\n",
    "            else:\n",
    "                assert hgvs_reference # Not null\n",
    "                assert [x for x in transvar_references.values() if x] # Not empty\n",
    "                if hgvs_reference in all_references:\n",
    "                    stats_2['many_references_exists'] = stats_2.get('many_references_exists', 0) + 1\n",
    "                else:\n",
    "                    stats_2['many_references_NOT_exists'] = stats_2.get('many_references_NOT_exists', 0) + 1\n",
    "\n",
    "    f.close()\n",
    "    input_transvar_f.close()\n",
    "    print 'Created_file:', input_transvar_fn\n",
    "    with open('stats_2.json', 'w') as f:\n",
    "        json.dump(stats_2, f)\n",
    "    \n",
    "    print 'created file:', fn\n",
    "    print 'Created file: stats_2.json'\n",
    "\n",
    "def do_3():\n",
    "    apply_transvar()\n",
    "    \n",
    "do_3()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### RUN:\n",
    "\n",
    "```\n",
    "/home/user/mutationinfo/localpython/bin/transvar canno --ccds -l genes_hgvs.txt -g 1 -m 2 --refversion hg19 --oneline > transvar_output.txt\n",
    "```\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input\ttranscript\tgene\tstrand\tcoordinates(gDNA/cDNA/protein)\tregion\tinfo\r\n",
      "RHAG:c.236G>A\tCCDS4927 (protein_coding)\tRHAG\t-\tchr6:g.49586997C>T/c.236G>A/p.S79N\tinside_[cds_in_exon_2]\tCSQN=Missense;reference_codon=AGT;alternative_codon=AAT;source=CCDS\r\n"
     ]
    }
   ],
   "source": [
    "!/home/user/mutationinfo/localpython/bin/transvar canno --ccds -i 'RHAG:c.236G>A' --refversion hg19"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input\ttranscript\tgene\tstrand\tcoordinates(gDNA/cDNA/protein)\tregion\tinfo\n",
      "EXT2:c.969C>A\t.\t.\t.\t././.\t.\tno_valid_transcript_found\n"
     ]
    }
   ],
   "source": [
    "!/home/user/mutationinfo/localpython/bin/transvar canno --ccds -i 'EXT2:c.969C>A' --refversion hg19"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input\ttranscript\tgene\tstrand\tcoordinates(gDNA/cDNA/protein)\tregion\tinfo\n",
      "EGFR:c.2576T>G\t.\t.\t.\t././.\t.\tno_valid_transcript_found\n"
     ]
    }
   ],
   "source": [
    "!/home/user/mutationinfo/localpython/bin/transvar canno --ccds -i 'EGFR:c.2576T>G' --refversion hg19"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input\ttranscript\tgene\tstrand\tcoordinates(gDNA/cDNA/protein)\tregion\tinfo\n",
      "EGFR:c.2576\tCCDS5514 (protein_coding)\tEGFR\t+\tchr7:g.55259518C/c.2576C/.\tinside_[cds_in_exon_21]\tsource=CCDS\n"
     ]
    }
   ],
   "source": [
    "!/home/user/mutationinfo/localpython/bin/transvar canno --ccds -i 'EGFR:c.2576' --refversion hg19"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "3 cases of exon 21 2576T>G mutation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input\ttranscript\tgene\tstrand\tcoordinates(gDNA/cDNA/protein)\tregion\tinfo\n",
      "EGFR:c.2576\tCCDS5514 (protein_coding)\tEGFR\t+\tchr7:g.55259518C/c.2576C/.\tinside_[cds_in_exon_21]\tsource=CCDS\n"
     ]
    }
   ],
   "source": [
    "!/home/user/mutationinfo/localpython/bin/transvar canno --ccds -i 'EGFR:c.2576' --refversion hg19"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input\ttranscript\tgene\tstrand\tcoordinates(gDNA/cDNA/protein)\tregion\tinfo\n",
      "BRCA1:c.5251\tCCDS11453 (protein_coding)\tBRCA1\t-\tchr17:g.41209095G/c.5251C/.\tinside_[cds_in_exon_18]\tsource=CCDS\n",
      "BRCA1:c.5251\tCCDS11456 (protein_coding)\tBRCA1\t-\tchr17:g.41215355T/c.5251A/.\tinside_[cds_in_exon_18]\tsource=CCDS\n",
      "BRCA1:c.5251\tCCDS11459 (protein_coding)\tBRCA1\t-\tchr17:g.41201152A/c.5251T/.\tinside_[cds_in_exon_18]\tsource=CCDS\n"
     ]
    }
   ],
   "source": [
    "!/home/user/mutationinfo/localpython/bin/transvar canno --ccds -i 'BRCA1:c.5251' --refversion hg19"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input\ttranscript\tgene\tstrand\tcoordinates(gDNA/cDNA/protein)\tregion\tinfo\n",
      "BRCA1:c.5251C>T\tCCDS11453 (protein_coding)\tBRCA1\t-\tchr17:g.41209095G>A/c.5251C>T/p.R1751*\tinside_[cds_in_exon_18]\tCSQN=Nonsense;reference_codon=CGA;alternative_codon=TGA;source=CCDS\n"
     ]
    }
   ],
   "source": [
    "!/home/user/mutationinfo/localpython/bin/transvar canno --ccds -i 'BRCA1:c.5251C>T' --refversion hg19"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input\ttranscript\tgene\tstrand\tcoordinates(gDNA/cDNA/protein)\tregion\tinfo\n",
      "BRCA1:c.5251A>T\tCCDS11456 (protein_coding)\tBRCA1\t-\tchr17:g.41215355T>A/c.5251A>T/p.N1751Y\tinside_[cds_in_exon_18]\tCSQN=Missense;reference_codon=AAT;alternative_codon=TAT;source=CCDS\n"
     ]
    }
   ],
   "source": [
    "!/home/user/mutationinfo/localpython/bin/transvar canno --ccds -i 'BRCA1:c.5251A>T' --refversion hg19"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input\ttranscript\tgene\tstrand\tcoordinates(gDNA/cDNA/protein)\tregion\tinfo\r\n",
      "IGSF1:c.2066\tCCDS14629 (protein_coding)\tIGSF1\t-\tchrX:g.130412084G/c.2066C/.\tinside_[cds_in_exon_12]\tsource=CCDS\r\n",
      "IGSF1:c.2066\tCCDS55490 (protein_coding)\tIGSF1\t-\tchrX:g.130412057T/c.2066A/.\tinside_[cds_in_exon_11]\tsource=CCDS\r\n",
      "IGSF1:c.2066\tCCDS55491 (protein_coding)\tIGSF1\t-\tchrX:g.130412099G/c.2066C/.\tinside_[cds_in_exon_12]\tsource=CCDS\r\n"
     ]
    }
   ],
   "source": [
    "!/home/user/mutationinfo/localpython/bin/transvar canno --ccds -i 'IGSF1:c.2066' --refversion hg19"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input\ttranscript\tgene\tstrand\tcoordinates(gDNA/cDNA/protein)\tregion\tinfo\n",
      "IGSF1:c.2065+1G>A\t.\t.\t.\t././.\t.\tno_valid_transcript_found\n"
     ]
    }
   ],
   "source": [
    "!/home/user/mutationinfo/localpython/bin/transvar canno --ccds -i 'IGSF1:c.2065+1G>A' --refversion hg19"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input\ttranscript\tgene\tstrand\tcoordinates(gDNA/cDNA/protein)\tregion\tinfo\r\n",
      "FAH:c.1062+5\tCCDS10314 (protein_coding)\tFAH\t+\tchr15:g.80472572G/c.1062+5G/.\tinside_[intron_between_exon_12_and_13]\tsource=CCDS\r\n"
     ]
    }
   ],
   "source": [
    "!/home/user/mutationinfo/localpython/bin/transvar canno --ccds -i 'FAH:c.1062+5' --refversion hg19"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input\ttranscript\tgene\tstrand\tcoordinates(gDNA/cDNA/protein)\tregion\tinfo\n",
      "[wrap_exception] warning: invalid_gene_PCD\n",
      "PCD:c.710\t.\t.\t.\t././.\t.\tError_invalid_gene_PCD\n"
     ]
    }
   ],
   "source": [
    "!/home/user/mutationinfo/localpython/bin/transvar canno --ccds -i 'PCD:c.710' --refversion hg19 "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Run 3\n",
    "Run with ```/home/user/mutationinfo/localpython/bin/python run_3.py```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing run_3.py\n"
     ]
    }
   ],
   "source": [
    "%%writefile run_3.py\n",
    "import re\n",
    "import json\n",
    "import time\n",
    "import requests\n",
    "\n",
    "def load_parsed_abstracts():\n",
    "    with open('parsed_abstracts.json') as f:\n",
    "        for l in f:\n",
    "            data = json.loads(l)\n",
    "            yield data\n",
    "            \n",
    "def hgvs_from_record(parsed):\n",
    "        \n",
    "    abstract = parsed['abstract']\n",
    "\n",
    "    if 'dog' in abstract:\n",
    "        #stats_2['nothumnan'] = stats_2.get('nothumnan', 0) + 1\n",
    "        return None\n",
    "\n",
    "\n",
    "    #print parsed\n",
    "    hgvs = parsed['hgvs']\n",
    "\n",
    "    #Get hgvs reference\n",
    "    s = re.search(r'([ACGT]+)>', hgvs)\n",
    "    if not s:\n",
    "        return None\n",
    "        #assert False\n",
    "\n",
    "    hgvs_reference = s.group(1)\n",
    "\n",
    "    #If it starts with a number assuming coding position\n",
    "    if re.search(r'^[\\d]+', hgvs, re.UNICODE):\n",
    "        hgvs = 'c.' + hgvs\n",
    "\n",
    "\n",
    "    #Get everything except the reference\n",
    "    s = re.search(r'^([^ACGT]+)[ACGT]+>', hgvs)\n",
    "    if not s:\n",
    "        return None\n",
    "        #continue\n",
    "        #assert False\n",
    "    except_reference = s.group(1)\n",
    "\n",
    "\n",
    "    gene = parsed['gene']\n",
    "    if not gene:\n",
    "        return None\n",
    "        #assert False\n",
    "\n",
    "    transcripts = parsed['transcripts']\n",
    "    if transcripts:\n",
    "        pass\n",
    "        #assert False\n",
    "\n",
    "    hgvs = gene + ':' + hgvs\n",
    "    hgvs_no_reference = gene + ':' + except_reference\n",
    "    #transvar_cmd = transvar_cmd_p.format(hgvs=hgvs)\n",
    "    return hgvs\n",
    "\n",
    "def vep_parse(hgvs):\n",
    "    headers={ \"Content-Type\" : \"application/json\"}\n",
    "    vep_url = \"https://rest.ensembl.org/vep/human/hgvs/{var}?\"\n",
    "\n",
    "    url = vep_url.format(var=hgvs)\n",
    "    responce = requests.get(url, headers=headers)\n",
    "    if responce.ok:\n",
    "        return responce.json()\n",
    "    else:\n",
    "        print ('Error:')\n",
    "        print (responce.text)\n",
    "\n",
    "def vep_parse_post(hgvs):\n",
    "\n",
    " \n",
    "    server = \"https://rest.ensembl.org\"\n",
    "    ext = \"/vep/human/hgvs\"\n",
    "    headers={ \"Content-Type\" : \"application/json\", \"Accept\" : \"application/json\"}\n",
    "    \n",
    "    assert type(hgvs) is list\n",
    "    #data = '[' + ','.join([rept(x) for x in hgvs]) + ']'\n",
    "    data = json.dumps({\"hgvs_notations\": hgvs})\n",
    "    print data\n",
    "    \n",
    "    r = requests.post(server+ext, headers=headers, data=data)\n",
    " \n",
    "    if not r.ok:\n",
    "        print ('ERROR:')\n",
    "        print (r.text)\n",
    "        return None\n",
    " \n",
    "    decoded = r.json()\n",
    "    return decoded\n",
    "\n",
    "def vep_run():\n",
    "    \n",
    "    count = 0\n",
    "    \n",
    "    f = open('vep_results.json', 'w')\n",
    "    \n",
    "    all_hgvs = set()\n",
    "    for parsed in load_parsed_abstracts():\n",
    "        count += 1\n",
    "        \n",
    "        print ('count:', count)\n",
    "        \n",
    "        if count < 1:\n",
    "            continue\n",
    "            \n",
    "        #if count > 10:\n",
    "        #    break\n",
    "        \n",
    "        hgvs = hgvs_from_record(parsed)\n",
    "        print hgvs\n",
    "        \n",
    "        if hgvs is None:\n",
    "            continue\n",
    "        \n",
    "        all_hgvs.add(hgvs)\n",
    "        if len(all_hgvs) <200:\n",
    "            continue\n",
    "        \n",
    "        # Accessing VEP!\n",
    "        vep = vep_parse_post(list(all_hgvs))\n",
    "        \n",
    "        # Saving\n",
    "        f.write(json.dumps(vep) + '\\n')\n",
    "        \n",
    "        #Emptying set\n",
    "        all_hgvs = set()\n",
    "        time.sleep(10) # Be nice\n",
    "        \n",
    "        #vep = vep_parse(hgvs)\n",
    "        #vep = vep_parse_post([hgvs])\n",
    "        #print vep\n",
    "        #assert False\n",
    "\n",
    "    # Last batch\n",
    "    vep = vep_parse_post(list(all_hgvs))    \n",
    "    f.write(json.dumps(vep) + '\\n')\n",
    "\n",
    "        \n",
    "    f.close()\n",
    "    print ('Created: vep_results.json')\n",
    "        \n",
    "vep_run()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/user/mutationinfo/localpython/lib/python2.7/site-packages/psycopg2-2.7.4-py2.7-linux-x86_64.egg/psycopg2/__init__.py:144: UserWarning: The psycopg2 wheel package will be renamed from release 2.8; in order to keep installing from binary please use \"pip install psycopg2-binary\" instead. For details see: <http://initd.org/psycopg/docs/install.html#binary-install-from-pypi>.\n",
      "  \"\"\")\n"
     ]
    }
   ],
   "source": [
    "import hgvs.parser\n",
    "import hgvs.dataproviders.uta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "hdp = hgvs.dataproviders.uta.connect()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "hgvs_parser = hgvs.parser.Parser().parse_hgvs_variant"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    " p = hgvs_parser('RHAG:c.236G>A')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "import hgvs.variantmapper\n",
    "import hgvs.assemblymapper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "am = hgvs.assemblymapper.AssemblyMapper(hdp, alt_aln_method=u'genewise',)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "ename": "HGVSDataNotAvailableError",
     "evalue": "No alignments for RHAG in GRCh38 using genewise",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mHGVSDataNotAvailableError\u001b[0m                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-19-6f5829b3df9d>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mam\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mc_to_g\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mp\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/home/user/mutationinfo/localpython/lib/python2.7/site-packages/hgvs-1.1.1-py2.7.egg/hgvs/assemblymapper.pyc\u001b[0m in \u001b[0;36mc_to_g\u001b[0;34m(self, var_c)\u001b[0m\n\u001b[1;32m     92\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     93\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mc_to_g\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvar_c\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 94\u001b[0;31m         \u001b[0malt_ac\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_alt_ac_for_tx_ac\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvar_c\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mac\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     95\u001b[0m         \u001b[0mvar_out\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msuper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mAssemblyMapper\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mc_to_g\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvar_c\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0malt_ac\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0malt_aln_method\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0malt_aln_method\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     96\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_maybe_normalize\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvar_out\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/home/user/mutationinfo/localpython/lib/python2.7/site-packages/hgvs-1.1.1-py2.7.egg/hgvs/assemblymapper.pyc\u001b[0m in \u001b[0;36m_alt_ac_for_tx_ac\u001b[0;34m(self, tx_ac)\u001b[0m\n\u001b[1;32m    138\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0malt_acs\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    139\u001b[0m             raise HGVSDataNotAvailableError(\"No alignments for {tx_ac} in {an} using {am}\".format(\n\u001b[0;32m--> 140\u001b[0;31m                 tx_ac=tx_ac, an=self.assembly_name, am=self.alt_aln_method))\n\u001b[0m\u001b[1;32m    141\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    142\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0malt_acs\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mHGVSDataNotAvailableError\u001b[0m: No alignments for RHAG in GRCh38 using genewise"
     ]
    }
   ],
   "source": [
    "am.c_to_g(p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'variantmapper' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-12-ce4f3ed11a66>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mvariantmapper\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'variantmapper' is not defined"
     ]
    }
   ],
   "source": [
    "variantmapper"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# VEP but for hg19\n",
    "Run with ```/home/user/mutationinfo/localpython/bin/python run_4.py```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Overwriting run_4.py\n"
     ]
    }
   ],
   "source": [
    "%%writefile run_4.py\n",
    "import re\n",
    "import json\n",
    "import time\n",
    "import requests\n",
    "\n",
    "def load_parsed_abstracts():\n",
    "    with open('parsed_abstracts.json') as f:\n",
    "        for l in f:\n",
    "            data = json.loads(l)\n",
    "            yield data\n",
    "            \n",
    "def hgvs_from_record(parsed):\n",
    "        \n",
    "    abstract = parsed['abstract']\n",
    "\n",
    "    if 'dog' in abstract:\n",
    "        #stats_2['nothumnan'] = stats_2.get('nothumnan', 0) + 1\n",
    "        return None\n",
    "\n",
    "\n",
    "    #print parsed\n",
    "    hgvs = parsed['hgvs']\n",
    "\n",
    "    #Get hgvs reference\n",
    "    s = re.search(r'([ACGT]+)>', hgvs)\n",
    "    if not s:\n",
    "        return None\n",
    "        #assert False\n",
    "\n",
    "    hgvs_reference = s.group(1)\n",
    "\n",
    "    #If it starts with a number assuming coding position\n",
    "    if re.search(r'^[\\d]+', hgvs, re.UNICODE):\n",
    "        hgvs = 'c.' + hgvs\n",
    "\n",
    "\n",
    "    #Get everything except the reference\n",
    "    s = re.search(r'^([^ACGT]+)[ACGT]+>', hgvs)\n",
    "    if not s:\n",
    "        return None\n",
    "        #continue\n",
    "        #assert False\n",
    "    except_reference = s.group(1)\n",
    "\n",
    "\n",
    "    gene = parsed['gene']\n",
    "    if not gene:\n",
    "        return None\n",
    "        #assert False\n",
    "\n",
    "    transcripts = parsed['transcripts']\n",
    "    if transcripts:\n",
    "        pass\n",
    "        #assert False\n",
    "\n",
    "    hgvs = gene + ':' + hgvs\n",
    "    hgvs_no_reference = gene + ':' + except_reference\n",
    "    #transvar_cmd = transvar_cmd_p.format(hgvs=hgvs)\n",
    "    return hgvs\n",
    "\n",
    "def vep_parse(hgvs):\n",
    "    headers={ \"Content-Type\" : \"application/json\"}\n",
    "    vep_url = \"https://grch37.rest.ensembl.org/vep/human/hgvs/{var}?\"\n",
    "\n",
    "    url = vep_url.format(var=hgvs)\n",
    "    responce = requests.get(url, headers=headers)\n",
    "    if responce.ok:\n",
    "        return responce.json()\n",
    "    else:\n",
    "        print ('Error:')\n",
    "        print (responce.text)\n",
    "\n",
    "def vep_parse_post(hgvs):\n",
    "\n",
    " \n",
    "    server = \"https://grch37.rest.ensembl.org\"\n",
    "    ext = \"/vep/human/hgvs\"\n",
    "    headers={ \"Content-Type\" : \"application/json\", \"Accept\" : \"application/json\"}\n",
    "    \n",
    "    assert type(hgvs) is list\n",
    "    #data = '[' + ','.join([rept(x) for x in hgvs]) + ']'\n",
    "    data = json.dumps({\"hgvs_notations\": hgvs})\n",
    "    print data\n",
    "    \n",
    "    r = requests.post(server+ext, headers=headers, data=data)\n",
    " \n",
    "    if not r.ok:\n",
    "        print ('ERROR:')\n",
    "        print (r.text)\n",
    "        return None\n",
    " \n",
    "    decoded = r.json()\n",
    "    return decoded\n",
    "\n",
    "def vep_run():\n",
    "    \n",
    "    count = 0\n",
    "    \n",
    "    f = open('vep_results_hg19.json', 'w')\n",
    "    \n",
    "    all_hgvs = set()\n",
    "    for parsed in load_parsed_abstracts():\n",
    "        count += 1\n",
    "        \n",
    "        print ('count:', count)\n",
    "        \n",
    "        if count < 1:\n",
    "            continue\n",
    "            \n",
    "        #if count > 10:\n",
    "        #    break\n",
    "        \n",
    "        hgvs = hgvs_from_record(parsed)\n",
    "        print hgvs\n",
    "        \n",
    "        if hgvs is None:\n",
    "            continue\n",
    "        \n",
    "        all_hgvs.add(hgvs)\n",
    "        if len(all_hgvs) <200:\n",
    "            continue\n",
    "        \n",
    "        # Accessing VEP!\n",
    "        vep = vep_parse_post(list(all_hgvs))\n",
    "        \n",
    "        #print vep\n",
    "        #assert False\n",
    "        \n",
    "        # Saving\n",
    "        f.write(json.dumps(vep) + '\\n')\n",
    "        f.flush()\n",
    "        \n",
    "        #Emptying set\n",
    "        all_hgvs = set()\n",
    "        time.sleep(10) # Be nice\n",
    "        \n",
    "        #vep = vep_parse(hgvs)\n",
    "        #vep = vep_parse_post([hgvs])\n",
    "        #print vep\n",
    "        #assert False\n",
    "\n",
    "    # Last batch\n",
    "    vep = vep_parse_post(list(all_hgvs))    \n",
    "    f.write(json.dumps(vep) + '\\n')\n",
    "\n",
    "        \n",
    "    f.close()\n",
    "    print ('Created: vep_results_hg19.json')\n",
    "        \n",
    "vep_run()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create stats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{\n",
      "    \"assembly_name\": \"GRCh37\", \n",
      "    \"end\": 41245861, \n",
      "    \"seq_region_name\": \"17\", \n",
      "    \"id\": \"BRCA1:c.1687C>T\", \n",
      "    \"allele_string\": \"C/T\", \n",
      "    \"start\": 41245861, \n",
      "    \"colocated_variants\": [\n",
      "        {\n",
      "            \"phenotype_or_disease\": 1, \n",
      "            \"end\": 41245861, \n",
      "            \"seq_region_name\": 17, \n",
      "            \"allele_string\": \"HGMD_MUTATION\", \n",
      "            \"start\": 41245861, \n",
      "            \"id\": \"CM950144\", \n",
      "            \"strand\": 1\n",
      "        }, \n",
      "        {\n",
      "            \"pubmed\": [\n",
      "                25741868, \n",
      "                20104584, \n",
      "                26467025, \n",
      "                23199084, \n",
      "                26295337, \n",
      "                15024741, \n",
      "                15383404, \n",
      "                16168118, \n",
      "                7837387, \n",
      "                15876480, \n",
      "                11504767, \n",
      "                12566964, \n",
      "                19208665, \n",
      "                26745875\n",
      "            ], \n",
      "            \"gnomad_amr_maf\": 0, \n",
      "            \"gnomad_fin_allele\": \"A\", \n",
      "            \"gnomad_nfe_maf\": 5.396e-05, \n",
      "            \"id\": \"rs80356898\", \n",
      "            \"gnomad_oth_allele\": \"A\", \n",
      "            \"gnomad_asj_maf\": 0, \n",
      "            \"phenotype_or_disease\": 1, \n",
      "            \"end\": 41245861, \n",
      "            \"gnomad_eas_allele\": \"A\", \n",
      "            \"gnomad_sas_maf\": 0, \n",
      "            \"start\": 41245861, \n",
      "            \"clin_sig\": [\n",
      "                \"pathogenic\"\n",
      "            ], \n",
      "            \"gnomad_allele\": \"A\", \n",
      "            \"gnomad_oth_maf\": 0, \n",
      "            \"gnomad_afr_allele\": \"A\", \n",
      "            \"seq_region_name\": 17, \n",
      "            \"gnomad_amr_allele\": \"A\", \n",
      "            \"gnomad_asj_allele\": \"A\", \n",
      "            \"gnomad_fin_maf\": 0, \n",
      "            \"strand\": 1, \n",
      "            \"gnomad_afr_maf\": 0, \n",
      "            \"gnomad_eas_maf\": 0, \n",
      "            \"gnomad_sas_allele\": \"A\", \n",
      "            \"allele_string\": \"G/A\", \n",
      "            \"gnomad_nfe_allele\": \"A\", \n",
      "            \"gnomad_maf\": 2.446e-05\n",
      "        }\n",
      "    ], \n",
      "    \"input\": \"BRCA1:c.1687C>T\", \n",
      "    \"transcript_consequences\": [\n",
      "        {\n",
      "            \"cds_start\": 799, \n",
      "            \"protein_start\": 267, \n",
      "            \"protein_end\": 267, \n",
      "            \"impact\": \"HIGH\", \n",
      "            \"cdna_end\": 1827, \n",
      "            \"gene_symbol\": \"BRCA1\", \n",
      "            \"amino_acids\": \"Q/*\", \n",
      "            \"hgnc_id\": 1100, \n",
      "            \"codons\": \"Cag/Tag\", \n",
      "            \"gene_symbol_source\": \"HGNC\", \n",
      "            \"cdna_start\": 1827, \n",
      "            \"transcript_id\": \"ENST00000309486\", \n",
      "            \"gene_id\": \"ENSG00000012048\", \n",
      "            \"consequence_terms\": [\n",
      "                \"stop_gained\"\n",
      "            ], \n",
      "            \"variant_allele\": \"T\", \n",
      "            \"cds_end\": 799, \n",
      "            \"strand\": -1, \n",
      "            \"biotype\": \"protein_coding\"\n",
      "        }, \n",
      "        {\n",
      "            \"cds_start\": 1687, \n",
      "            \"protein_start\": 563, \n",
      "            \"protein_end\": 563, \n",
      "            \"impact\": \"HIGH\", \n",
      "            \"cdna_end\": 1881, \n",
      "            \"gene_symbol\": \"BRCA1\", \n",
      "            \"amino_acids\": \"Q/*\", \n",
      "            \"hgnc_id\": 1100, \n",
      "            \"codons\": \"Cag/Tag\", \n",
      "            \"gene_symbol_source\": \"HGNC\", \n",
      "            \"cdna_start\": 1881, \n",
      "            \"transcript_id\": \"ENST00000346315\", \n",
      "            \"gene_id\": \"ENSG00000012048\", \n",
      "            \"consequence_terms\": [\n",
      "                \"stop_gained\"\n",
      "            ], \n",
      "            \"variant_allele\": \"T\", \n",
      "            \"cds_end\": 1687, \n",
      "            \"strand\": -1, \n",
      "            \"biotype\": \"protein_coding\"\n",
      "        }, \n",
      "        {\n",
      "            \"impact\": \"MODIFIER\", \n",
      "            \"hgnc_id\": 1100, \n",
      "            \"gene_id\": \"ENSG00000012048\", \n",
      "            \"gene_symbol_source\": \"HGNC\", \n",
      "            \"gene_symbol\": \"BRCA1\", \n",
      "            \"transcript_id\": \"ENST00000351666\", \n",
      "            \"biotype\": \"protein_coding\", \n",
      "            \"consequence_terms\": [\n",
      "                \"intron_variant\"\n",
      "            ], \n",
      "            \"variant_allele\": \"T\", \n",
      "            \"strand\": -1\n",
      "        }, \n",
      "        {\n",
      "            \"impact\": \"MODIFIER\", \n",
      "            \"hgnc_id\": 1100, \n",
      "            \"gene_id\": \"ENSG00000012048\", \n",
      "            \"gene_symbol_source\": \"HGNC\", \n",
      "            \"gene_symbol\": \"BRCA1\", \n",
      "            \"transcript_id\": \"ENST00000352993\", \n",
      "            \"biotype\": \"protein_coding\", \n",
      "            \"consequence_terms\": [\n",
      "                \"intron_variant\"\n",
      "            ], \n",
      "            \"variant_allele\": \"T\", \n",
      "            \"strand\": -1\n",
      "        }, \n",
      "        {\n",
      "            \"cds_start\": 1687, \n",
      "            \"protein_start\": 563, \n",
      "            \"protein_end\": 563, \n",
      "            \"impact\": \"HIGH\", \n",
      "            \"cdna_end\": 1919, \n",
      "            \"gene_symbol\": \"BRCA1\", \n",
      "            \"amino_acids\": \"Q/*\", \n",
      "            \"hgnc_id\": 1100, \n",
      "            \"codons\": \"Cag/Tag\", \n",
      "            \"gene_symbol_source\": \"HGNC\", \n",
      "            \"cdna_start\": 1919, \n",
      "            \"transcript_id\": \"ENST00000354071\", \n",
      "            \"gene_id\": \"ENSG00000012048\", \n",
      "            \"consequence_terms\": [\n",
      "                \"stop_gained\"\n",
      "            ], \n",
      "            \"variant_allele\": \"T\", \n",
      "            \"cds_end\": 1687, \n",
      "            \"strand\": -1, \n",
      "            \"biotype\": \"protein_coding\"\n",
      "        }, \n",
      "        {\n",
      "            \"cds_start\": 1687, \n",
      "            \"protein_start\": 563, \n",
      "            \"protein_end\": 563, \n",
      "            \"impact\": \"HIGH\", \n",
      "            \"cdna_end\": 1806, \n",
      "            \"gene_symbol\": \"BRCA1\", \n",
      "            \"amino_acids\": \"Q/*\", \n",
      "            \"hgnc_id\": 1100, \n",
      "            \"codons\": \"Cag/Tag\", \n",
      "            \"gene_symbol_source\": \"HGNC\", \n",
      "            \"cdna_start\": 1806, \n",
      "            \"transcript_id\": \"ENST00000357654\", \n",
      "            \"gene_id\": \"ENSG00000012048\", \n",
      "            \"consequence_terms\": [\n",
      "                \"stop_gained\"\n",
      "            ], \n",
      "            \"variant_allele\": \"T\", \n",
      "            \"cds_end\": 1687, \n",
      "            \"strand\": -1, \n",
      "            \"biotype\": \"protein_coding\"\n",
      "        }, \n",
      "        {\n",
      "            \"cds_start\": 1039, \n",
      "            \"protein_start\": 347, \n",
      "            \"protein_end\": 347, \n",
      "            \"impact\": \"HIGH\", \n",
      "            \"cdna_end\": 1038, \n",
      "            \"gene_symbol\": \"BRCA1\", \n",
      "            \"amino_acids\": \"Q/*\", \n",
      "            \"hgnc_id\": 1100, \n",
      "            \"codons\": \"Cag/Tag\", \n",
      "            \"gene_symbol_source\": \"HGNC\", \n",
      "            \"cdna_start\": 1038, \n",
      "            \"transcript_id\": \"ENST00000412061\", \n",
      "            \"flags\": [\n",
      "                \"cds_start_NF\"\n",
      "            ], \n",
      "            \"gene_id\": \"ENSG00000012048\", \n",
      "            \"consequence_terms\": [\n",
      "                \"stop_gained\"\n",
      "            ], \n",
      "            \"variant_allele\": \"T\", \n",
      "            \"cds_end\": 1039, \n",
      "            \"strand\": -1, \n",
      "            \"biotype\": \"non_stop_decay\"\n",
      "        }, \n",
      "        {\n",
      "            \"impact\": \"MODIFIER\", \n",
      "            \"gene_symbol\": \"BRCA1\", \n",
      "            \"cdna_end\": 1788, \n",
      "            \"hgnc_id\": 1100, \n",
      "            \"gene_id\": \"ENSG00000012048\", \n",
      "            \"gene_symbol_source\": \"HGNC\", \n",
      "            \"cdna_start\": 1788, \n",
      "            \"transcript_id\": \"ENST00000461221\", \n",
      "            \"biotype\": \"nonsense_mediated_decay\", \n",
      "            \"consequence_terms\": [\n",
      "                \"3_prime_UTR_variant\", \n",
      "                \"NMD_transcript_variant\"\n",
      "            ], \n",
      "            \"variant_allele\": \"T\", \n",
      "            \"strand\": -1\n",
      "        }, \n",
      "        {\n",
      "            \"impact\": \"MODIFIER\", \n",
      "            \"distance\": 2020, \n",
      "            \"hgnc_id\": 1100, \n",
      "            \"gene_id\": \"ENSG00000012048\", \n",
      "            \"gene_symbol_source\": \"HGNC\", \n",
      "            \"gene_symbol\": \"BRCA1\", \n",
      "            \"transcript_id\": \"ENST00000461574\", \n",
      "            \"flags\": [\n",
      "                \"cds_start_NF\", \n",
      "                \"cds_end_NF\"\n",
      "            ], \n",
      "            \"biotype\": \"protein_coding\", \n",
      "            \"consequence_terms\": [\n",
      "                \"upstream_gene_variant\"\n",
      "            ], \n",
      "            \"variant_allele\": \"T\", \n",
      "            \"strand\": -1\n",
      "        }, \n",
      "        {\n",
      "            \"impact\": \"MODIFIER\", \n",
      "            \"gene_symbol\": \"BRCA1\", \n",
      "            \"cdna_end\": 1751, \n",
      "            \"hgnc_id\": 1100, \n",
      "            \"gene_id\": \"ENSG00000012048\", \n",
      "            \"gene_symbol_source\": \"HGNC\", \n",
      "            \"cdna_start\": 1751, \n",
      "            \"transcript_id\": \"ENST00000467274\", \n",
      "            \"biotype\": \"retained_intron\", \n",
      "            \"consequence_terms\": [\n",
      "                \"non_coding_transcript_exon_variant\"\n",
      "            ], \n",
      "            \"variant_allele\": \"T\", \n",
      "            \"strand\": -1\n",
      "        }, \n",
      "        {\n",
      "            \"impact\": \"MODIFIER\", \n",
      "            \"hgnc_id\": 1100, \n",
      "            \"gene_id\": \"ENSG00000012048\", \n",
      "            \"gene_symbol_source\": \"HGNC\", \n",
      "            \"gene_symbol\": \"BRCA1\", \n",
      "            \"transcript_id\": \"ENST00000468300\", \n",
      "            \"biotype\": \"protein_coding\", \n",
      "            \"consequence_terms\": [\n",
      "                \"intron_variant\"\n",
      "            ], \n",
      "            \"variant_allele\": \"T\", \n",
      "            \"strand\": -1\n",
      "        }, \n",
      "        {\n",
      "            \"cds_start\": 1687, \n",
      "            \"protein_start\": 563, \n",
      "            \"protein_end\": 563, \n",
      "            \"impact\": \"HIGH\", \n",
      "            \"cdna_end\": 1848, \n",
      "            \"gene_symbol\": \"BRCA1\", \n",
      "            \"amino_acids\": \"Q/*\", \n",
      "            \"hgnc_id\": 1100, \n",
      "            \"codons\": \"Cag/Tag\", \n",
      "            \"gene_symbol_source\": \"HGNC\", \n",
      "            \"cdna_start\": 1848, \n",
      "            \"transcript_id\": \"ENST00000470026\", \n",
      "            \"flags\": [\n",
      "                \"cds_end_NF\"\n",
      "            ], \n",
      "            \"gene_id\": \"ENSG00000012048\", \n",
      "            \"consequence_terms\": [\n",
      "                \"stop_gained\"\n",
      "            ], \n",
      "            \"variant_allele\": \"T\", \n",
      "            \"cds_end\": 1687, \n",
      "            \"strand\": -1, \n",
      "            \"biotype\": \"protein_coding\"\n",
      "        }, \n",
      "        {\n",
      "            \"cds_start\": 1687, \n",
      "            \"protein_start\": 563, \n",
      "            \"protein_end\": 563, \n",
      "            \"impact\": \"HIGH\", \n",
      "            \"cdna_end\": 1919, \n",
      "            \"gene_symbol\": \"BRCA1\", \n",
      "            \"amino_acids\": \"Q/*\", \n",
      "            \"hgnc_id\": 1100, \n",
      "            \"codons\": \"Cag/Tag\", \n",
      "            \"gene_symbol_source\": \"HGNC\", \n",
      "            \"cdna_start\": 1919, \n",
      "            \"transcript_id\": \"ENST00000471181\", \n",
      "            \"gene_id\": \"ENSG00000012048\", \n",
      "            \"consequence_terms\": [\n",
      "                \"stop_gained\"\n",
      "            ], \n",
      "            \"variant_allele\": \"T\", \n",
      "            \"cds_end\": 1687, \n",
      "            \"strand\": -1, \n",
      "            \"biotype\": \"protein_coding\"\n",
      "        }, \n",
      "        {\n",
      "            \"impact\": \"MODIFIER\", \n",
      "            \"distance\": 326, \n",
      "            \"hgnc_id\": 1100, \n",
      "            \"gene_id\": \"ENSG00000012048\", \n",
      "            \"gene_symbol_source\": \"HGNC\", \n",
      "            \"gene_symbol\": \"BRCA1\", \n",
      "            \"transcript_id\": \"ENST00000473961\", \n",
      "            \"flags\": [\n",
      "                \"cds_start_NF\", \n",
      "                \"cds_end_NF\"\n",
      "            ], \n",
      "            \"biotype\": \"protein_coding\", \n",
      "            \"consequence_terms\": [\n",
      "                \"downstream_gene_variant\"\n",
      "            ], \n",
      "            \"variant_allele\": \"T\", \n",
      "            \"strand\": -1\n",
      "        }, \n",
      "        {\n",
      "            \"impact\": \"MODIFIER\", \n",
      "            \"distance\": 2002, \n",
      "            \"hgnc_id\": 1100, \n",
      "            \"gene_id\": \"ENSG00000012048\", \n",
      "            \"gene_symbol_source\": \"HGNC\", \n",
      "            \"gene_symbol\": \"BRCA1\", \n",
      "            \"transcript_id\": \"ENST00000476777\", \n",
      "            \"flags\": [\n",
      "                \"cds_end_NF\"\n",
      "            ], \n",
      "            \"biotype\": \"protein_coding\", \n",
      "            \"consequence_terms\": [\n",
      "                \"downstream_gene_variant\"\n",
      "            ], \n",
      "            \"variant_allele\": \"T\", \n",
      "            \"strand\": -1\n",
      "        }, \n",
      "        {\n",
      "            \"cds_start\": 1609, \n",
      "            \"protein_start\": 537, \n",
      "            \"protein_end\": 537, \n",
      "            \"impact\": \"HIGH\", \n",
      "            \"cdna_end\": 1722, \n",
      "            \"gene_symbol\": \"BRCA1\", \n",
      "            \"amino_acids\": \"Q/*\", \n",
      "            \"hgnc_id\": 1100, \n",
      "            \"codons\": \"Cag/Tag\", \n",
      "            \"gene_symbol_source\": \"HGNC\", \n",
      "            \"cdna_start\": 1722, \n",
      "            \"transcript_id\": \"ENST00000477152\", \n",
      "            \"flags\": [\n",
      "                \"cds_end_NF\"\n",
      "            ], \n",
      "            \"gene_id\": \"ENSG00000012048\", \n",
      "            \"consequence_terms\": [\n",
      "                \"stop_gained\"\n",
      "            ], \n",
      "            \"variant_allele\": \"T\", \n",
      "            \"cds_end\": 1609, \n",
      "            \"strand\": -1, \n",
      "            \"biotype\": \"protein_coding\"\n",
      "        }, \n",
      "        {\n",
      "            \"impact\": \"MODIFIER\", \n",
      "            \"hgnc_id\": 1100, \n",
      "            \"gene_id\": \"ENSG00000012048\", \n",
      "            \"gene_symbol_source\": \"HGNC\", \n",
      "            \"gene_symbol\": \"BRCA1\", \n",
      "            \"transcript_id\": \"ENST00000478531\", \n",
      "            \"flags\": [\n",
      "                \"cds_end_NF\"\n",
      "            ], \n",
      "            \"biotype\": \"protein_coding\", \n",
      "            \"consequence_terms\": [\n",
      "                \"intron_variant\"\n",
      "            ], \n",
      "            \"variant_allele\": \"T\", \n",
      "            \"strand\": -1\n",
      "        }, \n",
      "        {\n",
      "            \"impact\": \"MODIFIER\", \n",
      "            \"hgnc_id\": 1100, \n",
      "            \"gene_id\": \"ENSG00000012048\", \n",
      "            \"gene_symbol_source\": \"HGNC\", \n",
      "            \"gene_symbol\": \"BRCA1\", \n",
      "            \"transcript_id\": \"ENST00000484087\", \n",
      "            \"flags\": [\n",
      "                \"cds_start_NF\", \n",
      "                \"cds_end_NF\"\n",
      "            ], \n",
      "            \"biotype\": \"protein_coding\", \n",
      "            \"consequence_terms\": [\n",
      "                \"intron_variant\"\n",
      "            ], \n",
      "            \"variant_allele\": \"T\", \n",
      "            \"strand\": -1\n",
      "        }, \n",
      "        {\n",
      "            \"impact\": \"MODIFIER\", \n",
      "            \"hgnc_id\": 1100, \n",
      "            \"gene_id\": \"ENSG00000012048\", \n",
      "            \"gene_symbol_source\": \"HGNC\", \n",
      "            \"gene_symbol\": \"BRCA1\", \n",
      "            \"transcript_id\": \"ENST00000487825\", \n",
      "            \"flags\": [\n",
      "                \"cds_start_NF\", \n",
      "                \"cds_end_NF\"\n",
      "            ], \n",
      "            \"biotype\": \"protein_coding\", \n",
      "            \"consequence_terms\": [\n",
      "                \"intron_variant\"\n",
      "            ], \n",
      "            \"variant_allele\": \"T\", \n",
      "            \"strand\": -1\n",
      "        }, \n",
      "        {\n",
      "            \"impact\": \"MODIFIER\", \n",
      "            \"hgnc_id\": 1100, \n",
      "            \"gene_id\": \"ENSG00000012048\", \n",
      "            \"gene_symbol_source\": \"HGNC\", \n",
      "            \"gene_symbol\": \"BRCA1\", \n",
      "            \"transcript_id\": \"ENST00000491747\", \n",
      "            \"biotype\": \"protein_coding\", \n",
      "            \"consequence_terms\": [\n",
      "                \"intron_variant\"\n",
      "            ], \n",
      "            \"variant_allele\": \"T\", \n",
      "            \"strand\": -1\n",
      "        }, \n",
      "        {\n",
      "            \"impact\": \"MODIFIER\", \n",
      "            \"distance\": 268, \n",
      "            \"hgnc_id\": 1100, \n",
      "            \"gene_id\": \"ENSG00000012048\", \n",
      "            \"gene_symbol_source\": \"HGNC\", \n",
      "            \"gene_symbol\": \"BRCA1\", \n",
      "            \"transcript_id\": \"ENST00000492859\", \n",
      "            \"biotype\": \"nonsense_mediated_decay\", \n",
      "            \"consequence_terms\": [\n",
      "                \"downstream_gene_variant\"\n",
      "            ], \n",
      "            \"variant_allele\": \"T\", \n",
      "            \"strand\": -1\n",
      "        }, \n",
      "        {\n",
      "            \"cds_start\": 1546, \n",
      "            \"protein_start\": 516, \n",
      "            \"protein_end\": 516, \n",
      "            \"impact\": \"HIGH\", \n",
      "            \"cdna_end\": 1778, \n",
      "            \"gene_symbol\": \"BRCA1\", \n",
      "            \"amino_acids\": \"Q/*\", \n",
      "            \"hgnc_id\": 1100, \n",
      "            \"codons\": \"Cag/Tag\", \n",
      "            \"gene_symbol_source\": \"HGNC\", \n",
      "            \"cdna_start\": 1778, \n",
      "            \"transcript_id\": \"ENST00000493795\", \n",
      "            \"gene_id\": \"ENSG00000012048\", \n",
      "            \"consequence_terms\": [\n",
      "                \"stop_gained\"\n",
      "            ], \n",
      "            \"variant_allele\": \"T\", \n",
      "            \"cds_end\": 1546, \n",
      "            \"strand\": -1, \n",
      "            \"biotype\": \"protein_coding\"\n",
      "        }, \n",
      "        {\n",
      "            \"impact\": \"MODIFIER\", \n",
      "            \"hgnc_id\": 1100, \n",
      "            \"gene_id\": \"ENSG00000012048\", \n",
      "            \"gene_symbol_source\": \"HGNC\", \n",
      "            \"gene_symbol\": \"BRCA1\", \n",
      "            \"transcript_id\": \"ENST00000493919\", \n",
      "            \"flags\": [\n",
      "                \"cds_end_NF\"\n",
      "            ], \n",
      "            \"biotype\": \"protein_coding\", \n",
      "            \"consequence_terms\": [\n",
      "                \"intron_variant\"\n",
      "            ], \n",
      "            \"variant_allele\": \"T\", \n",
      "            \"strand\": -1\n",
      "        }, \n",
      "        {\n",
      "            \"impact\": \"MODIFIER\", \n",
      "            \"distance\": 268, \n",
      "            \"hgnc_id\": 1100, \n",
      "            \"gene_id\": \"ENSG00000012048\", \n",
      "            \"gene_symbol_source\": \"HGNC\", \n",
      "            \"gene_symbol\": \"BRCA1\", \n",
      "            \"transcript_id\": \"ENST00000494123\", \n",
      "            \"flags\": [\n",
      "                \"cds_end_NF\"\n",
      "            ], \n",
      "            \"biotype\": \"protein_coding\", \n",
      "            \"consequence_terms\": [\n",
      "                \"downstream_gene_variant\"\n",
      "            ], \n",
      "            \"variant_allele\": \"T\", \n",
      "            \"strand\": -1\n",
      "        }, \n",
      "        {\n",
      "            \"impact\": \"MODIFIER\", \n",
      "            \"distance\": 268, \n",
      "            \"hgnc_id\": 1100, \n",
      "            \"gene_id\": \"ENSG00000012048\", \n",
      "            \"gene_symbol_source\": \"HGNC\", \n",
      "            \"gene_symbol\": \"BRCA1\", \n",
      "            \"transcript_id\": \"ENST00000497488\", \n",
      "            \"flags\": [\n",
      "                \"cds_end_NF\"\n",
      "            ], \n",
      "            \"biotype\": \"protein_coding\", \n",
      "            \"consequence_terms\": [\n",
      "                \"downstream_gene_variant\"\n",
      "            ], \n",
      "            \"variant_allele\": \"T\", \n",
      "            \"strand\": -1\n",
      "        }, \n",
      "        {\n",
      "            \"impact\": \"MODIFIER\", \n",
      "            \"hgnc_id\": 1100, \n",
      "            \"gene_id\": \"ENSG00000012048\", \n",
      "            \"gene_symbol_source\": \"HGNC\", \n",
      "            \"gene_symbol\": \"BRCA1\", \n",
      "            \"transcript_id\": \"ENST00000586385\", \n",
      "            \"biotype\": \"protein_coding\", \n",
      "            \"consequence_terms\": [\n",
      "                \"intron_variant\"\n",
      "            ], \n",
      "            \"variant_allele\": \"T\", \n",
      "            \"strand\": -1\n",
      "        }, \n",
      "        {\n",
      "            \"impact\": \"MODIFIER\", \n",
      "            \"hgnc_id\": 1100, \n",
      "            \"gene_id\": \"ENSG00000012048\", \n",
      "            \"gene_symbol_source\": \"HGNC\", \n",
      "            \"gene_symbol\": \"BRCA1\", \n",
      "            \"transcript_id\": \"ENST00000591534\", \n",
      "            \"biotype\": \"protein_coding\", \n",
      "            \"consequence_terms\": [\n",
      "                \"intron_variant\"\n",
      "            ], \n",
      "            \"variant_allele\": \"T\", \n",
      "            \"strand\": -1\n",
      "        }, \n",
      "        {\n",
      "            \"impact\": \"MODIFIER\", \n",
      "            \"hgnc_id\": 1100, \n",
      "            \"gene_id\": \"ENSG00000012048\", \n",
      "            \"gene_symbol_source\": \"HGNC\", \n",
      "            \"gene_symbol\": \"BRCA1\", \n",
      "            \"transcript_id\": \"ENST00000591849\", \n",
      "            \"biotype\": \"protein_coding\", \n",
      "            \"consequence_terms\": [\n",
      "                \"intron_variant\"\n",
      "            ], \n",
      "            \"variant_allele\": \"T\", \n",
      "            \"strand\": -1\n",
      "        }\n",
      "    ], \n",
      "    \"most_severe_consequence\": \"stop_gained\", \n",
      "    \"strand\": -1\n",
      "}\n",
      "[u'ENST00000309486', u'ENST00000346315', u'ENST00000351666', u'ENST00000352993', u'ENST00000354071', u'ENST00000357654', u'ENST00000412061', u'ENST00000461221', u'ENST00000461574', u'ENST00000467274', u'ENST00000468300', u'ENST00000470026', u'ENST00000471181', u'ENST00000473961', u'ENST00000476777', u'ENST00000477152', u'ENST00000478531', u'ENST00000484087', u'ENST00000487825', u'ENST00000491747', u'ENST00000492859', u'ENST00000493795', u'ENST00000493919', u'ENST00000494123', u'ENST00000497488', u'ENST00000586385', u'ENST00000591534', u'ENST00000591849']\n"
     ]
    },
    {
     "ename": "AssertionError",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAssertionError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-17-5a05b29c73b5>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     43\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     44\u001b[0m \u001b[0;31m#vep = load_vep_results('vep_results.json') # Replace this with 'vep_results_hg19.json'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 45\u001b[0;31m \u001b[0mvep\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mload_vep_results\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'vep_results_hg19.json'\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;31m# Replace this with 'vep_results_hg19.json'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m<ipython-input-17-5a05b29c73b5>\u001b[0m in \u001b[0;36mload_vep_results\u001b[0;34m(fn)\u001b[0m\n\u001b[1;32m     36\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;34m'BRCA1'\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mvariant\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'id'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     37\u001b[0m                     \u001b[0;32mcontinue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 38\u001b[0;31m                 \u001b[0mret\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mvariant\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'id'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mvep_info\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvariant\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     39\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     40\u001b[0m     \u001b[0;32mprint\u001b[0m \u001b[0;34m'Lines:'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mc\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-17-5a05b29c73b5>\u001b[0m in \u001b[0;36mvep_info\u001b[0;34m(vep)\u001b[0m\n\u001b[1;32m      8\u001b[0m     \u001b[0;31m#transcripts = set(transcripts)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m     \u001b[0;32mprint\u001b[0m \u001b[0mtranscripts\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 10\u001b[0;31m     \u001b[0;32massert\u001b[0m \u001b[0mFalse\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     11\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     12\u001b[0m     ret = {\n",
      "\u001b[0;31mAssertionError\u001b[0m: "
     ]
    }
   ],
   "source": [
    "import json\n",
    "\n",
    "def vep_info(vep):\n",
    "    # Take only the essential info\n",
    "    \n",
    "    print json.dumps(vep, indent=4)\n",
    "    transcripts = [x['transcript_id'] for x in vep['transcript_consequences']]\n",
    "    #transcripts = set(transcripts)\n",
    "    print transcripts\n",
    "    assert False\n",
    "    \n",
    "    ret = {\n",
    "        'chr': vep['seq_region_name'],\n",
    "        'start' : vep['start'],\n",
    "        'end' : vep['end'],\n",
    "        'allele_string' : vep['allele_string'].split('/'),\n",
    "        'strand' : vep['strand'],\n",
    "        'tr': len(transcripts)\n",
    "    }\n",
    "   \n",
    "    return ret\n",
    "\n",
    "def load_vep_results(fn):    \n",
    "    ret = {}\n",
    "    \n",
    "    with open(fn) as f:\n",
    "        c = 0\n",
    "        for l in f:\n",
    "            c += 1\n",
    "            data = json.loads(l)\n",
    "\n",
    "            #print data[0]\n",
    "            #assert False\n",
    "            \n",
    "            for variant in data:\n",
    "                if not 'BRCA1' in variant['id']:\n",
    "                    continue\n",
    "                ret[variant['id']] = vep_info(variant)\n",
    "            \n",
    "    print 'Lines:', c\n",
    "    print 'Record:', len(ret)\n",
    "    return ret\n",
    "\n",
    "#vep = load_vep_results('vep_results.json') # Replace this with 'vep_results_hg19.json'\n",
    "vep = load_vep_results('vep_results_hg19.json') # Replace this with 'vep_results_hg19.json'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'allele_string': [u'T', u'G'],\n",
       " 'chr': u'12',\n",
       " 'end': 56118130,\n",
       " 'start': 56118130,\n",
       " 'strand': 1}"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vep['RDH5:c.758T>G']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def load_transvar_results(fn):\n",
    "    '''\n",
    "    transvar_output.txt\n",
    "    '''\n",
    "\n",
    "    ret = {}\n",
    "    with open(fn) as f:\n",
    "        c=0\n",
    "        for l in f:\n",
    "            c+=1\n",
    "            ls = l.replace('\\n', '').split('\\t')\n",
    "            if c == 1:\n",
    "                header = ls\n",
    "                header_str = l\n",
    "                continue\n",
    "                \n",
    "            ret[ls[0]] = '\\t'.join(header[1:]) + '\\n' + '\\t'.join(ls[1:]).replace('|||\\t', '\\n')\n",
    "            #break\n",
    "        print ('Records:', c)\n",
    "    return ret"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('Records:', 42115)\n"
     ]
    }
   ],
   "source": [
    "transvar=load_transvar_results('transvar_output.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['RHAG|c.236']"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "transvar.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "transcript\tgene\tstrand\tcoordinates(gDNA/cDNA/protein)\tregion\tinfo\n",
      "CCDS10690 (protein_coding)\tPHKG2\t+\tchr16:g.30764875C/c.553C/.\tinside_[cds_in_exon_5]\tsource=CCDS\t\n",
      "CCDS54002 (protein_coding)\tPHKG2\t+\tchr16:g.30764875C/c.553C/.\tinside_[cds_in_exon_5]\tsource=CCDS\n"
     ]
    }
   ],
   "source": [
    "print transvar['PHKG2|c.553']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Records: 10000\n",
      "Records: 20000\n",
      "Records: 30000\n",
      "Records: 40000\n",
      "{\n",
      "    \"TRANSCRIPT_COUNTER_6\": 194, \n",
      "    \"TRANSCRIPT_COUNTER_7\": 103, \n",
      "    \"TRANSCRIPT_COUNTER_4\": 955, \n",
      "    \"TRANSCRIPT_COUNTER_5\": 255, \n",
      "    \"TRANSCRIPT_COUNTER_2\": 6026, \n",
      "    \"TRANSCRIPT_COUNTER_3\": 2630, \n",
      "    \"YES_VEP\": 22573, \n",
      "    \"reference_found_only\": 19048, \n",
      "    \"reference_found\": 23041, \n",
      "    \"reference_not_found\": 9226, \n",
      "    \"vep_concordant\": 21278, \n",
      "    \"TRANSCRIPT_COUNTER_8\": 90, \n",
      "    \"TRANSCRIPT_COUNTER_9\": 31, \n",
      "    \"not_reference_found_only\": 9226, \n",
      "    \"NO_VEP\": 19541, \n",
      "    \"NO_VEP_NO_TR\": 8777, \n",
      "    \"YES_VEP_YES_TR\": 21503, \n",
      "    \"TRANSCRIPT_COUNTER_1\": 21896, \n",
      "    \"vep_discordant\": 3652, \n",
      "    \"hgvs_reference_SAME_VEP\": 22572, \n",
      "    \"reference_found_and_reference_not_found\": 3993, \n",
      "    \"transvar_found_something\": 32267, \n",
      "    \"more_than_one\": 10371, \n",
      "    \"not_vep_concordant_and_discordant\": 225, \n",
      "    \"COULD NOT FIND HGVS\": 2384, \n",
      "    \"vep_concordant_and_not_discordant\": 17851, \n",
      "    \"TRANSCRIPT_COUNTER_13\": 6, \n",
      "    \"ALL_PERFECT\": 17850, \n",
      "    \"YES_TR\": 32267, \n",
      "    \"TRANSCRIPT_COUNTER_19\": 3, \n",
      "    \"TRANSCRIPT_COUNTER_14\": 4, \n",
      "    \"TRANSCRIPT_COUNTER_15\": 4, \n",
      "    \"TRANSCRIPT_COUNTER_16\": 7, \n",
      "    \"NO_TR\": 9847, \n",
      "    \"TRANSCRIPT_COUNTER_10\": 22, \n",
      "    \"TRANSCRIPT_COUNTER_11\": 26, \n",
      "    \"TRANSCRIPT_COUNTER_12\": 15, \n",
      "    \"YES_VEP_NO_TR\": 1070, \n",
      "    \"NO_VEP_YES_TR\": 10764, \n",
      "    \"hgvs_reference_NOT_PRESENT_IN_VEP\": 1, \n",
      "    \"vep_concordant_and_discordant\": 3427\n",
      "}\n",
      "Length of vep_tr_transcripts: 21503\n"
     ]
    }
   ],
   "source": [
    "import re\n",
    "import json\n",
    "\n",
    "def hgvs_from_record(parsed):\n",
    "        \n",
    "    abstract = parsed['abstract']\n",
    "\n",
    "    if 'dog' in abstract:\n",
    "        #stats_2['nothumnan'] = stats_2.get('nothumnan', 0) + 1\n",
    "        return None\n",
    "\n",
    "\n",
    "    #print parsed\n",
    "    hgvs = parsed['hgvs']\n",
    "\n",
    "    #Get hgvs reference\n",
    "    s = re.search(r'([ACGT]+)>', hgvs)\n",
    "    if not s:\n",
    "        return None\n",
    "        #assert False\n",
    "\n",
    "    hgvs_reference = s.group(1)\n",
    "\n",
    "    #If it starts with a number assuming coding position\n",
    "    if re.search(r'^[\\d]+', hgvs, re.UNICODE):\n",
    "        hgvs = 'c.' + hgvs\n",
    "\n",
    "\n",
    "    #Get everything except the reference\n",
    "    s = re.search(r'^([^ACGT]+)[ACGT]+>', hgvs)\n",
    "    if not s:\n",
    "        return None\n",
    "        #continue\n",
    "        #assert False\n",
    "    except_reference = s.group(1)\n",
    "\n",
    "\n",
    "    gene = parsed['gene']\n",
    "    if not gene:\n",
    "        return None\n",
    "        #assert False\n",
    "\n",
    "    transcripts = parsed['transcripts']\n",
    "    if transcripts:\n",
    "        pass\n",
    "        #assert False\n",
    "\n",
    "    hgvs = gene + ':' + hgvs\n",
    "    hgvs_no_reference = gene + ':' + except_reference\n",
    "    transvar_key = gene + '|' + except_reference\n",
    "    #transvar_cmd = transvar_cmd_p.format(hgvs=hgvs)\n",
    "    return hgvs, hgvs_reference, transvar_key\n",
    "\n",
    "def inverse(s):\n",
    "    \n",
    "    return {\n",
    "        'A': 'T',\n",
    "        'T': 'A',\n",
    "        'C': 'G',\n",
    "        'G': 'C',\n",
    "    }[s]\n",
    "\n",
    "def get_transvar_reference(coord, strand):\n",
    "    s = re.search(r'chr[\\dXY]+:g.[\\d]+([ACGT]+)/', coord)\n",
    "    if s:\n",
    "        reference = s.group(1)\n",
    "    else:\n",
    "        return None\n",
    "\n",
    "    if strand == '-':\n",
    "        return inverse(reference)\n",
    "    elif strand == '+':\n",
    "        return reference\n",
    "    else:\n",
    "        assert False\n",
    "\n",
    "def get_transvar_chromosome_position(coord):\n",
    "    s = re.search(r'chr([\\dXY]+):g.([\\d]+)', coord)\n",
    "    if s:\n",
    "        return s.group(1), int(s.group(2))\n",
    "\n",
    "    if coord == '././.':\n",
    "        return None, None\n",
    "    \n",
    "    print 'coord:', coord\n",
    "    raise Exception('Unknown coord')\n",
    "        \n",
    "    \n",
    "    \n",
    "def parse_transvar_output(output):\n",
    "    ls = output.split('\\n')\n",
    "    ls = [x.split('\\t') for x in ls if x.strip()]\n",
    "    try:\n",
    "        strand_i = ls[0].index('strand')\n",
    "    except Exception as e:\n",
    "        print '====='\n",
    "        print output\n",
    "        print '====='\n",
    "        raise e\n",
    "    coord_i = ls[0].index('coordinates(gDNA/cDNA/protein)')\n",
    "    transcript_i = ls[0].index('transcript')\n",
    "    \n",
    "    try:\n",
    "        ret = {l[transcript_i]:{\n",
    "            'reference': get_transvar_reference(l[coord_i], l[strand_i]),\n",
    "            'position': get_transvar_chromosome_position(l[coord_i]),\n",
    "            } for l in ls[1:]}\n",
    "    except Exception as e:\n",
    "        print output\n",
    "        raise e\n",
    "    return ret    \n",
    "\n",
    "def add_one(d, key):\n",
    "    d[key] = d.get(key, 0) + 1\n",
    "\n",
    "def load_transvar_results(fn):\n",
    "    \n",
    "    stats_3 = {}\n",
    "    vep_tr_transcripts = []\n",
    "    \n",
    "    c = 0\n",
    "    with open(fn) as f:\n",
    "        for l in f:\n",
    "            c += 1\n",
    "            \n",
    "            if c % 10000 == 0:\n",
    "                print 'Records:', c\n",
    "            \n",
    "            data = json.loads(l)\n",
    "            \n",
    "            hgvs_from_record_ret = hgvs_from_record(data)\n",
    "            if hgvs_from_record_ret is None:\n",
    "                add_one(stats_3, 'COULD NOT FIND HGVS')\n",
    "                continue\n",
    "                \n",
    "            hgvs, hgvs_reference, transvar_key = hgvs_from_record_ret\n",
    "            #print hgvs\n",
    "            #print hgvs_reference\n",
    "            if transvar_key in transvar:\n",
    "                transvar_output = transvar[transvar_key]\n",
    "            else:\n",
    "                transvar_output = ''\n",
    "            \n",
    "            #transvar_output = data['transvar'] # THERE IS NOT DATA THERE! FIX IT\n",
    "            #print transvar_output\n",
    "            if not transvar_output.strip():\n",
    "                add_one(stats_3, 'transvar_crashed')\n",
    "                transvar_info = None\n",
    "            else:\n",
    "                transvar_info = parse_transvar_output(transvar_output)\n",
    "            #print transvar_info\n",
    "            \n",
    "            vep_info = vep.get(hgvs, None)\n",
    "            #print vep_info\n",
    "            \n",
    "            ### TRANSVAR / VEP FOUND SOUMETHING\n",
    "            NO_VEP = False\n",
    "            YES_VEP = False\n",
    "            NO_TR = False\n",
    "            YES_TR = False\n",
    "            \n",
    "            if vep_info is None:\n",
    "                NO_VEP = True\n",
    "                add_one(stats_3, 'NO_VEP')\n",
    "            else:\n",
    "                YES_VEP = True\n",
    "                add_one(stats_3, 'YES_VEP')\n",
    "            \n",
    "            if transvar_info and all([x['position'][1] is None for x in transvar_info.values()]):\n",
    "                    #add_one(stats_3, 'transvar_found ././.')\n",
    "                    # TODO: CHECK VEP HGVS_REF!!!\n",
    "                    NO_TR = True\n",
    "                    add_one(stats_3, 'NO_TR')\n",
    "            else:\n",
    "                YES_TR = True\n",
    "                add_one(stats_3, 'YES_TR')\n",
    "            \n",
    "            assert NO_VEP != YES_VEP\n",
    "            assert NO_TR != YES_TR\n",
    "            \n",
    "            for x in [('YES_VEP', YES_VEP), ('NO_VEP', NO_VEP)]:\n",
    "                for y in [('YES_TR', YES_TR), ('NO_TR', NO_TR)]:\n",
    "                    if x[1] and y[1]:\n",
    "                        add_one(stats_3, x[0] + '_' + y[0])\n",
    "            \n",
    "            #continue\n",
    "            \n",
    "            #if YES_VEP and NO_TR:\n",
    "            #    print data\n",
    "            #    print '====='\n",
    "            #    print vep_info\n",
    "            #    #assert False\n",
    "            \n",
    "            #if NO_TR:\n",
    "            #    continue\n",
    "            \n",
    "            # END OF TRANSVAR / VEP FOUND SOMETHING\n",
    "            \n",
    "            #### CHECK VEP // HGVS REFERENCE\n",
    "            hgvs_vep_concordance = False\n",
    "            if vep_info:\n",
    "                if hgvs_reference in vep_info['allele_string']:\n",
    "                    if vep_info['allele_string'].index(hgvs_reference) == 0:\n",
    "                        add_one(stats_3, 'hgvs_reference_SAME_VEP')\n",
    "                        hgvs_vep_concordance = True\n",
    "                    else:\n",
    "                        add_one(stats_3, 'hgvs_reference_NOT_SAME_VEP')\n",
    "                else:\n",
    "                    add_one(stats_3, 'hgvs_reference_NOT_PRESENT_IN_VEP')\n",
    "            \n",
    "            ### END OF VEP // HGVS REFERENCE CONCORDANC CHECK\n",
    "            \n",
    "            ### CHECK TRANSVAR CONCORDANCE\n",
    "            if NO_TR:\n",
    "                continue\n",
    "            \n",
    "            \n",
    "            more_than_one = False\n",
    "            reference_found = False\n",
    "            reference_not_found = False\n",
    "            more_than_one_marked = False\n",
    "            vep_concordant = False\n",
    "            vep_discordant = False\n",
    "            transcript_counter = 0\n",
    "            \n",
    "            for k,v in transvar_info.iteritems():\n",
    "                if more_than_one:\n",
    "                    if not more_than_one_marked:\n",
    "                        add_one(stats_3, 'more_than_one')\n",
    "                        more_than_one_marked = True\n",
    "                \n",
    "                if v['reference'] == hgvs_reference:\n",
    "                    reference_found = True\n",
    "                    \n",
    "                if v['reference'] != hgvs_reference:\n",
    "                    reference_not_found = True\n",
    "                \n",
    "                if vep_info:\n",
    "                    if v['reference'] in vep_info['allele_string']:\n",
    "                        if vep_info['allele_string'].index(v['reference']) == 0:\n",
    "                            if vep_info['start'] == v['position'][1]:\n",
    "                                vep_concordant = True\n",
    "                            else:\n",
    "                                vep_discordant = True\n",
    "                        else:\n",
    "                            vep_discordant = True\n",
    "                    else:\n",
    "                        vep_discordant = True\n",
    "                \n",
    "                more_than_one = True\n",
    "            \n",
    "                transcript_counter += 1\n",
    "            \n",
    "            if not more_than_one:\n",
    "                # Transvar found nada\n",
    "                add_one(stats_3, 'transvar_not_found')\n",
    "            else:\n",
    "                add_one(stats_3, 'transvar_found_something')\n",
    "            \n",
    "            if reference_found:\n",
    "                add_one(stats_3, 'reference_found')\n",
    "            else:\n",
    "                add_one(stats_3, 'reference_not_found')\n",
    "                \n",
    "            if reference_found and reference_not_found: # I know..\n",
    "                add_one(stats_3, 'reference_found_and_reference_not_found')\n",
    "                \n",
    "            if reference_found and not reference_not_found:\n",
    "                add_one(stats_3, 'reference_found_only')\n",
    "                \n",
    "            if not reference_found and reference_not_found:\n",
    "                add_one(stats_3, 'not_reference_found_only')\n",
    "                \n",
    "            if not reference_found and not reference_not_found:\n",
    "                assert False\n",
    "            \n",
    "            add_one(stats_3, 'TRANSCRIPT_COUNTER_{}'.format(transcript_counter))\n",
    "            \n",
    "            if vep_info:\n",
    "                if vep_concordant:\n",
    "                    add_one(stats_3, 'vep_concordant')\n",
    "\n",
    "                if vep_discordant:\n",
    "                    add_one(stats_3, 'vep_discordant')\n",
    "\n",
    "                if vep_concordant and vep_discordant: # Can happen\n",
    "                    add_one(stats_3, 'vep_concordant_and_discordant')\n",
    "\n",
    "                if vep_concordant and not vep_discordant:\n",
    "                    add_one(stats_3, 'vep_concordant_and_not_discordant')\n",
    "\n",
    "                if not vep_concordant and vep_discordant:\n",
    "                    add_one(stats_3, 'not_vep_concordant_and_discordant')\n",
    "\n",
    "                if not vep_concordant and not vep_discordant:\n",
    "                    add_one(stats_3, 'not_vep_concordant_not_discordant')\n",
    "\n",
    "                vep_tr_transcripts.append((vep_info['tr'], transcript_counter))\n",
    "                    \n",
    "            if hgvs_vep_concordance and reference_found and not reference_not_found and vep_concordant and not vep_discordant:\n",
    "                add_one(stats_3, 'ALL_PERFECT')\n",
    "            \n",
    "            #print (stats_3)\n",
    "            #assert False\n",
    "            \n",
    "    print json.dumps(stats_3, indent=4)\n",
    "    print 'Length of vep_tr_transcripts:', len(vep_tr_transcripts)\n",
    "    \n",
    "    return vep_tr_transcripts\n",
    "\n",
    "#load_transvar_results('parsed_abstracts_transvar.json')\n",
    "plot_data = load_transvar_results('parsed_abstracts.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: matplotlib in /home/user/mutationinfo/localpython/lib/python2.7/site-packages\n",
      "Requirement already satisfied: python-dateutil>=2.1 in /home/user/mutationinfo/localpython/lib/python2.7/site-packages (from matplotlib)\n",
      "Requirement already satisfied: subprocess32 in /home/user/mutationinfo/localpython/lib/python2.7/site-packages (from matplotlib)\n",
      "Requirement already satisfied: cycler>=0.10 in /home/user/mutationinfo/localpython/lib/python2.7/site-packages (from matplotlib)\n",
      "Requirement already satisfied: six>=1.10 in /home/user/mutationinfo/localpython/lib/python2.7/site-packages/six-1.11.0-py2.7.egg (from matplotlib)\n",
      "Requirement already satisfied: backports.functools-lru-cache in /home/user/mutationinfo/localpython/lib/python2.7/site-packages (from matplotlib)\n",
      "Requirement already satisfied: pytz in /home/user/mutationinfo/localpython/lib/python2.7/site-packages (from matplotlib)\n",
      "Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.1 in /home/user/mutationinfo/localpython/lib/python2.7/site-packages (from matplotlib)\n",
      "Requirement already satisfied: numpy>=1.7.1 in /home/user/mutationinfo/localpython/lib/python2.7/site-packages (from matplotlib)\n",
      "Requirement already satisfied: kiwisolver>=1.0.1 in /home/user/mutationinfo/localpython/lib/python2.7/site-packages (from matplotlib)\n",
      "Requirement already satisfied: setuptools in /home/user/mutationinfo/localpython/lib/python2.7/site-packages (from kiwisolver>=1.0.1->matplotlib)\n"
     ]
    }
   ],
   "source": [
    "!pip install matplotlib"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting py27-backports-functools_lru_cache\n",
      "\u001b[31m  Could not find a version that satisfies the requirement py27-backports-functools_lru_cache (from versions: )\u001b[0m\n",
      "\u001b[31mNo matching distribution found for py27-backports-functools_lru_cache\u001b[0m\n",
      "\u001b[33mYou are using pip version 9.0.1, however version 9.0.2 is available.\n",
      "You should consider upgrading via the 'pip install --upgrade pip' command.\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!pip install py27-backports-functools_lru_cache"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAX4AAAD8CAYAAABw1c+bAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDIuMi4yLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvhp/UCwAAIABJREFUeJzsvWmsJFl23/e7sWVEZObLfFut3VXVG5tLmxzS5ZEXySAlcjQc0KI32BwbxlCmMZIhAvYHw6RgwDSoLwQMeQFoi2pTA0qAPOIHiyYB0iQHhIWxAEpi94hDznCW7um1trfnGvuN6w+RkZUv672qt1XVW84PKFQukZmR+TL/995zzv0fZYxBEARBuDhYz/sEBEEQhGeLCL8gCMIFQ4RfEAThgiHCLwiCcMEQ4RcEQbhgiPALgiBcMET4BUEQLhhPFH6l1ItKqf9XKfVnSqmvK6X+q8ntS0qpLyml3pn8v7jP4z83OeYdpdTnTvoNCIIgCIdDPWkDl1LqKnDVGPMVpVQbeBv4d4GfBraNMb+klPp5YNEY83Nzj10C3gJuA2by2H/VGLNz4u9EEARBOBDOkw4wxtwH7k8uD5VS3wCuAz8J/PDksL8P/BPg5+Ye/peBLxljtgGUUl8CPg188XGvubKyYm7dunXQ9yAIgnDhefvttzeNMasHOfaJwj+LUuoW8IPAPwcuTwYFgAfA5T0ech34eOb6ncltj+XWrVu89dZbhzk1QRCEC41S6sODHnvg5K5SqgX8X8B/bYwZzN5nqnjRsUx/lFKfV0q9pZR6a2Nj4zhPJQiCIDyGAwm/UsqlEv1/aIz5x5Ob1ybx/zoPsL7HQ+8CL85cf2Fy2yMYY940xtw2xtxeXT3QakUQBEE4Agep6lHA3wO+YYz5n2bu+i2grtL5HPCbezz894BPKaUWJ1U/n5rcJgiCIDwnDjLj/7eA/wz4i0qpP578+wzwS8CPKaXeAX50ch2l1G2l1K8CTJK6fwv4o8m/X6wTvYIgCMLz4YnlnM+D27dvG0nuCoIgHByl1NvGmNsHOVZ27gqCIFwwDlXOKQjnnawoyXWJa1t4jsyLhPOJCL8gTMiKkrs7EQZQwPXFUMRfOJfIt1oQJuS6xADNhoOZXBeE84gIvyBMcG0LBYzTAjW5LgjnEQn1CMIEz7G4vhhKjF8494jwC8IMtdjXYR4Rf+E8IsIvCDPUCd5Ml2htuLHcpOXLz0Q4X8g3WhBmyHVJpksGcU6UFQC8erktM3/hXCHfZkGYwbUttDZEWUHYcLFtJdU9wrlDZvyCMIPnWNxYbgJg2wrPtqS6Rzh3iPALwhwt3+HVy22p7hHOLSL8grAHniOCL5xf5JstCIJwwRDhFwRBuGCI8AuCIFwwRPgFQRAuGCL8giAIFwwRfkEQhAvGE8s5lVJfAH4CWDfGvDG57deB1yeHdIGeMeYTezz2A2AIaKA4aD9IQRAE4elxkDr+XwN+GfgH9Q3GmP+4vqyU+ttA/zGP/xFjzOZRT1AQBEE4WZ4o/MaYLyulbu11n1JKAf8R8BdP9rQEQRCEp8VxY/x/AVgzxryzz/0G+H2l1NtKqc8/7omUUp9XSr2llHprY2PjmKclCIIg7Mdxhf+zwBcfc/+fN8b8EPDjwN9QSv3b+x1ojHnTGHPbGHN7dXX1mKclCIIg7MeRhV8p5QD/PvDr+x1jjLk7+X8d+A3gk0d9PUEQBOFkOM6M/0eBbxpj7ux1p1KqqZRq15eBTwFfO8brCYIgCCfAE4VfKfVF4A+B15VSd5RSPzO566eYC/Mopa4ppX5ncvUy8E+VUl8F/gXw28aY3z25UxcEQRCOwkGqej67z+0/vcdt94DPTC6/B/zAMc9PEARBOGFk564gCMIFQ4RfEAThgiHCLwiCcMEQ4RcEQbhgiPALgiBcMET4BUEQLhgi/IIgCBcMEX5BEIQLhgi/IAjCBUOEXxAE4YJxkA5cgnAhyIqSXJe4toXnyJxIOL+I8AsClejf3YkwgAKuL4Yi/sK5Rb7ZggDkusQAzYaDmVzPipJxWpAV5fM+PUE4UWTGLwiAa1sooBdlaG3I8pL1QSIrAOFcIt9kQQA8x2K17aO1wbYV9/sx46xAAZmuYv+CcF6QGb8gTFAKfM+m2XDYGKTc78V4roWF4no3fN6nJwgnhgi/IEyowz3jtMBScHUxoNlwyIoSpZ732QnCySHCLwgTPMfi+mJYJXoNbAwTjAHPtnBtiYoK5wcRfkGYwXMe1vB7Tih1/cK55CDN1r+glFpXSn1t5rb/QSl1Vyn1x5N/n9nnsZ9WSn1LKfWuUurnT/LEBeFp4zkWzYYjoi+cOw7yjf414NN73P4/G2M+Mfn3O/N3KqVs4H8Dfhz4XuCzSqnvPc7JCoIgCMfnicJvjPkysH2E5/4k8K4x5j1jTAb8I+Anj/A8giA8A2TD2sXhOGvYn1VK/ckkFLS4x/3XgY9nrt+Z3LYnSqnPK6XeUkq9tbGxcYzTEgThsNSWFWuDhLs7kYj/Oeeowv93gFeATwD3gb993BMxxrxpjLltjLm9urp63KcTBOEQ7GVZIZxfjiT8xpg1Y4w2xpTA/0EV1pnnLvDizPUXJrcJgnDKmN3DoCbXhfPLkco5lVJXjTH3J1f/PeBrexz2R8BrSqmXqAT/p4D/5EhnKQjCU2V2D4OUr55/nij8SqkvAj8MrCil7gC/APywUuoTgAE+AP7a5NhrwK8aYz5jjCmUUj8L/B5gA18wxnz9qbwLQRCOTb2HoU7yygBwflHGmOd9Do9w+/Zt89Zbbz3v0xCEC4f0JTi7KKXeNsbcPsix8hcVBGGKJHkvBiL8giBMkSTvxUC8egRBmCJJ3ouBCL8g7MFFbrw+a1QnnE9E+AVhDklwCucd+TYLwhyS4BTOOyL8gjCHJDiF846EegRhDklwCucdEX5B2ANJcArnGRF+QbjAXOTqpYuMCL8gXFCkeuniIn9lQbigSPXSxUVm/IJwDjhMyKY+1hikeumCIsIvCGecw4Rs5o9dbfsohcT4LxjylxaEM85hQjbzx9ain+tS+uxeIGTGLwhnnMNsOJs/1hgkwXsBEeEXhANyWksfD7PhbP7Y2RXAOC3IdXmq3pvwdBDhF4QDcNpLHw+z4Wz+WEnwXjxE+AXhAJz2mfFRVyNiT3ExOUiz9S8APwGsG2PemNz2PwL/DpAB3wH+qjGmt8djPwCGgAaKg/aDFITTxmk2bjvuakTsKS4eB/lr/xrw6bnbvgS8YYz5fuDbwN98zON/xBjzCRF94SxTz4wvL/inLswjG7GEw/LEb68x5svA9txtv2+MKSZX/xnwwlM4N0E4VXiORbPhnCrRh9O9GhFOJyfxDfnPgf9nn/sM8PtKqbeVUp9/3JMopT6vlHpLKfXWxsbGCZyWIFwMTvNqRDidHCu5q5T674AC+If7HPLnjTF3lVKXgC8ppb45WUE8gjHmTeBNgNu3b5vjnJcgXDQkTi8chiN/U5RSP02V9P1PjTF7CrUx5u7k/3XgN4BPHvX1BEEQhJPhSMKvlPo08N8Cf8UYE+1zTFMp1a4vA58CvnbUExXOF1lRMk4LsQkQhOfAE4VfKfVF4A+B15VSd5RSPwP8MtCmCt/8sVLqVybHXlNK/c7koZeBf6qU+irwL4DfNsb87lN5F8KZoi4/XBsk3N2JRPwF4RnzxBi/Meaze9z89/Y59h7wmcnl94AfONbZCeeS074ZShDOO/JrE545Un4oCM8XsWwQnjliEyAIzxcRfuG5IOWHgvD8kF+eIAjCBUOEXxAE4YIhwi8IgnDBkBj/BeS0dpISBOHZIMJ/AZgVepAeq4Jw0RHhP8fUtggbwwRnUjvfDT3ZPCUIFxz5xZ9TaluEe72YB/0Ez7GonfRk85QgXGxkxn8OmI/ZZ0VJL8rIdMlC4LI1SuhHOaFn02w4NBuOxPgF4QIjwn/Gme+3utr22RgmZLpkvZ9wqeNzpROw2vZ3dY8SwReEi4sI/xln3vAsygoMVSwfYDH06IaeCL0gCFNEDc4484ZnoedMr3u2tafoixe+IFxsZMZ/xtnb8MwnygpC79HG4POhISnnFISLhwj/OWDW8CwrSjaGCYbJrN/ZLezihf90kc1xwllAhP+c8SRhFy/8p4espoSzggj/OeNJwi5e+CdPPcvPCllNCWeDA30rlVJfUEqtK6W+NnPbklLqS0qpdyb/L+7z2M9NjnlHKfW5kzpxYW9qYb+84O874/Qca1dpp3B06ln+nZ2Ij7bGDOKMzWFKMWORIQinjYN+M38N+PTcbT8P/IEx5jXgDybXd6GUWgJ+AfhzwCeBX9hvgBBODhH2Z8c4LejHOWu9hPc3x7yzNiLKqtm+VE4Jp5UDhXqMMV9WSt2au/kngR+eXP77wD8Bfm7umL8MfMkYsw2glPoS1QDyxSOdrSCcIupE+t1exLfvD9CAKQ0Nx6LhWoAi9GyJ9QunjuN8Gy8bY+5PLj8ALu9xzHXg45nrdya3CSeM1OY/e3Jd4tgW1xcDMl1iSsP2KOeDzSF3tmIcS2EmxwnCaeJEkrvGGKOUMk8+cn+UUp8HPg9w48aNkzitC4NUkzwf6kQ6RnG1G6CNwaAIGy6uo/hwa8StlZbE+oVTx3GEf00pddUYc18pdRVY3+OYuzwMBwG8QBUSegRjzJvAmwC3b98+1iBy0ZDa/OdDnUjvhgWuZdGPM1w1wnEVS0GDxabHatuXv4Vw6jjON/K3gLpK53PAb+5xzO8Bn1JKLU6Sup+a3CacIPXMsxdlJJnGyLD5zPAci8Wmx82VJs2GQ9u3ub+T8KAfMUxyme0Lp5KDlnN+EfhD4HWl1B2l1M8AvwT8mFLqHeBHJ9dRSt1WSv0qwCSp+7eAP5r8+8U60SucHJ5jsdr20dpg26py55RY/zOhzq3U8f524HG547PcClgMGyj1vM9QEB7loFU9n93nrr+0x7FvAf/FzPUvAF840tkJB0Yp8Cd++ycd7hEbgr2Zza3EWcEgKrizHTFICjBwa7kpM37hVCI7d88JJ23FMEoqi2cMbI5SSgwWihvLTVq+fG3gYW7FcyzubGc4tsVyy+f7XwwAuLYYyEApnErkF3xOOEkrhp1RxlsfbqGNYXOQ0vIdcm1YCFwAXr3cFkHj4WDbj3LAcKXjT3fs1t3OBOE0It/Mc8SsS+dRyYqS9zZGbI9TfNch1SXpOCNOczABZTeUqqEJ9WA7TgtcuwrmX+n4j3Q7E4TThgi/sItxWlBicGybUZqBgeVmg6FtYZQhybXErWeoBltP+hgLZwoRfmFKbUGwPU7RumQp8Lh2PWQQa3ailCQ3bPQjsqsLIm5znMRqSxCeFSL8wpRcl2S6ZJxqtqOU72xkaKqKlbQwrLYbbCc5/TiTBK8gnGHk1ytMcW2LYVLQi3KywpBkmijVGKAfZSjLUBQGZIOYIJxpRPiFXQSuRWkMSV7iuhY745zQt3hptYnv2dhYhFKtIghnGglKClNyXbIQePyF11Z5dbXFK8tNbAuajk1aaCwsgoYlyV1BOOPI1E2YUtelu47Fy5daxJmmxJDmGsY5Ly4F+J4tNgSCcMaRqZswpfb8aTUcbi41WW55pJnmQS8mKTSb4/TQu4KlT4AgnD5kxn9K2csf56Q8c/Z7nrqc0wCFLumGHp5jsRC4BJ5DVpSHshme9bIpdPnIxqasKLm3E9FPcq4uBFzq+Ed+T4IgHBwR/lPIXo1VgGM3W6ln3xvDBGcS1qmfJytKelFGpkuaDYc72xG5NoBhkOSMswILNZ3tH2QQqr1sXNvizvaYXJtpK0KAtz/Y5re/epcSw3Lo8R/cvsG1Z9RERoznhIuMCP9T5igCs1djFeBYzVbqwSTKNFujhJdW2+S6nD733Z2ITFfHBK5DoTULgcsgzllqNuiELllRotTBO37VOYNBnANq+hy5LsmKkm+tDdiIUgLXprcx5k/v7hBl+qkbwe03sMpAIFwURPifIkdtibif02ahSzaHKa6tDl1ZUw8mndBla5QyiHNcW5EVlQgbIPQcqjk+oBRRVkxey8YY8OyqouegHb/mvWyyopy+n6woadgWSVIwinMsW5EXJRujBHi6RnDz5z9Oi+lqR2sjDqTCuUe+3U+Ro7ZErJOsUVYQes40FHMc6sEkK0qudHy6gUcvztiJMorJrD/ONK5lcWO5EuvF0KMbetP3MjsbPqgF9P5eNg4vrbb4vhe7ZJnGtW0cyyZsuNi2eqpGcMZAkunqNerQlS4ZJAVRmgNnw4FUwlXCURHhf4oc1SN/Nsk6Tgs8J5x2eOqERwv1zA8mSsEoK6aD0uJE4OuZuWdb0+Ru/fjZ56pn8od5/fnz9V2LxcCjZ3I6oQOq2kBWryyeBvVna9sKrQ2r3SpZrXuGKM0JPeepDzwnwVFXk4IAIvxPlaN65I/TgiirYux1HP64jVbmB5PVtr/r+epqm8O4TPaiDDP5/7DCM04LCg03lkMWgpyVls9S0+NS29814Jw09SqsG3rVe5/sSVhpNch1ie/ZT3XgOSmOupoUBBDhf+oc1rWxFuitUcrWKOFKJ5iK8FEGkdlk6qxQKMWez3fQ8z2O8FRlnDHrg4TNUYJjK1CGZsN+qqIPj67CjIEPNkfk2uDYiktnxEv/pDuuCReLIwu/Uup14NdnbnoZ+O+NMf/LzDE/DPwm8P7kpn9sjPnFo77mRaAO6bx8qUU/ynfVzR9lEJmto4fdQvG456tLP4E9hfA4wjNOC7bGCd1m1dHrtcstVp6R4M4PoOO04EE/IfBs4kxzrXs2QiYn2XFNuHgcWfiNMd8CPgGglLKBu8Bv7HHo/2eM+Ymjvs5FYzYJe9z2ffOz8sXJbPpJQpEVJR9sjnjQjwHFlY7PrZXWnnH+owpPXhqSoiDRmrDhsNj0dr3+0xS02QFvnAKYieHo2bIdlR4AwlE5qVDPXwK+Y4z58ISe78IyK6jGMK2zn99hexBhnJ+VH3RGXeUVDL7noIBcm0dCOccRZ9e2yNKCb6yNcBS0PIdO4NHynWeetGw2HK50AnJt6Aau9MkVLgQn9S3/KeCL+9z3byilvgrcA/4bY8zXT+g1zzV1rH+vHbYHFcajzspd28K1FUlWAIpO4GIMk3r86jmOI85KQbftcz0raDVcoNozUDV1P3ju4CRWBp5jcWulJSET4UJxbOFXSnnAXwH+5h53fwW4aYwZKaU+A/zfwGv7PM/ngc8D3Lhx47indWaphb0f56wPUq51fZK8slG4tOAfOql6lHBALYar7co7x7WtaUWQoqqIOU5FiWtbLPgOCotRmrPatgk9Z3rfQXIHJ7kykJCJcNE4iRn/jwNfMcaszd9hjBnMXP4dpdT/rpRaMcZs7nHsm8CbALdv3z5bwdYZDjsLrY83hqkdwjjV7IxS7uxEfP1uj0sdn0GS8UlvBc95NtUc9cYrJq81K/RwvF3EnmPx2uUFVtvVQFaHeer7DrJKOegAeJxVgWyQEs4rJyH8n2WfMI9S6gqwZowxSqlPUtlAb53Aa55KHjcL3c9ts/bIWe8nXOr4FIXh4+0RgyRHWYbQt3l5pUmmy0k4xH/m1RxPo3TQcywuLeztxnmQGfhBzumwq4LZvxEc3xRPEE4rxxJ+pVQT+DHgr83c9tcBjDG/AvyHwH+plCqAGPgpY8yZnc0/idlZ6MYw4V4vYqVVlWPuJSL18Z5jUWLwbItC66q0MfAYjhN2ooJxrmlY1jQc8qxDE/MJ5ygryCYC2Y9zNobJoeyaZznqrPpJu4dn3UbrzVpPyhfM/o2OG846bcjqRZjlWMJvjBkDy3O3/crM5V8Gfvk4r3GWqGehG8OEb9wdsNj2+Hgr4vWrC1MR6UUZvSijG3q7Sjct1ERMq2TqArASenwicMnLko7vPXPjsPkwlDHVextnBX/ycY+8LBnFmtcut7i1UjxS8nmQ558V29W2j1IcSpz22j08v5ICnrgbdz50BAf3IzrtHCcfIgPG+URq106QehZ6rxexEHh0fJd+kpPrypWyF2VTIRqnBdcXw+lM+no3nIpeVlRhHdey6MUTYYszmv6z21G6VxhKa4NtK5qNqvxSl4amr/Fde8+SzycxK7a9KOOjrXHV2pGDidN+cf5ZWwZgajZ32NLXw9hXnGaOusta/IDOLyL8J4znWISuw8YwZmOcVCGaFx1W25W4wUOfmMqAzXok7l9X0AzjnNLASrsx9ex5Vj+8aRjKnoShHIuMyrY4oyTwKoFNIsjL8khJ3lmxnR1UDipO+8X5Z2+fN5vbj/2Symdd6Grb7UKXh1697GVfnesnbwAUTj8i/CdMVpRsjlJW2x7KtllteXhu9UOZFfxCPxT4WQ/42bj/Wj8m0SU744QXlprPNNwwDUPpSRhq4ti52q3CMde74S4DuaPYLcznDjaGyaHE6XFifZQE+Hkr65ydsUO18jnM32l2AK2/r/P7SoSziQj/CVInFEsDS63KArlh21MRmxWkrCi524vYHmfkReUNf7UbgIE4K7i7nXFnEHNjqUlRmqduXjbP7LnOhqGA6Wz8JHIOs2Jb20+fhFifNxE/CrOtL+NMA4dbwcx/X3ei7Nwkuy86IvwnxCgp+GhrTIlhZ5yx2GzQ9BxuLDcf8bjxHAtjCu73YrbHKY5jk+Rj1oYJtqVIc80ozRnGOb0oY3GSCH7WzItn7eGTa4Nrq0Mncw/7eqeBs5zcdG2LQpfc2R4DCncSSjus+FfhyGpScx6S3YII/4mQFSUfbY3ZGCV4jo3v2Sw23ceWOCoFVzshvmMzSAp0qek2HLQ2jAvNSsvHtmxcpVhuNqYeMlOb5bwkL0tCz5l63DwNgZp93nkny2pn7/NLgD5tUT7ryU3PsSab5MyufsdHeQ9HDZ8JpxMR/hMg1yW2rVDAO2tDllsencBltb3/Y4wBS8Fis0E7cAHD1jhjFBdoDEWhWAwclls+r1xq7ypTHGfFtFy0YVm8cb07rf45ikDNl23WP+x54Wt6DrNOlnlRcjd6PsL4LET5PDQ7aTYcQs/e1e/4qJzGFZlwNET4T4A6CTZKdBUGsSyyYv/yxtn2f2jDq5eqEaK43yfJNb5lsdTyubEU7ort57ok0yVpVpKVBU0vYBhrNscJtmXtEijgsX76NXWIKtWaBzsxnaZH03N45VK7qt2fvL9BnD/iZOk6FiZ7OsL4pNn8sxDl89DsRGbqwl6I8J8A9ZK6H+fEecFOlKIU3Fpp7nn8fu3/HNui3XBwbIvAtR8RbGNgvZ8Q5ZqtYU6aD3BtxWKzqquf7yr1oJ8AhuWmz7XF4JHnq0NU9/sR6/2Ue4OYhmuzGFYNUl651H4kRnxtLtFbx32LSQLQtY8vwAeZzZ9EK8onieFZEs3HvR+ZqQvziPCfEM2GQ9Nz6IYuruuwFFb9cmsr49kf3l7t/6KsYH0QsxPlKKBz7dGEbpQWOI7ihVZI6Fk86KVc6jQYxlUP3U7gTTcd5doQeDa5Lrnbi0FB6Nm7RLQOUbmOTaQ1llLYGFzLrkRX8UiMWCl2edbXtgkbw4Sdya7k44ZdDjKbP44oH9ba+rSL5lnPRQjPHhH+E8JzLG4sVzP8EoPWhnu9iGDSzGT2x7irTC6vZt2ZLnEsxfdd71Bow7XFYNePd5QUfPWjbe4PElzL4oWlkGajeu7NUcpis4Ex1aBShXoMcaZJ8uqya1eWELWIzm7sCVybS02P1bBatXTCKi7sTmwOQs+ebrKad1qqdspaOLZ17LDLrlwDTzZgO3L3r2OEiU5jlc95yEUIzxYR/hPEcyxWWg3u92NKA1ujjJcveXtWU9SXP9x8WA0ECoWiEziPdILqxxnbcc5yy+deL6YfpZQoNoaabtCgE7j0opxBnNMOXFzb4ruvLpAXJd9+MOB+P8ZCcanlM04T7vdibFtxrxfR9l1eXGpxbTEAeGRD1mrb56OtMbat2BgmeE647wrmoCGfeQE9jG/PcWe4Rw0TndaZ9XnIRQjPFhH+E6IWhSjT9OOcF5ZChklGP8qns+d56lBL6DlEWcFyq8G17qOxeADXsii1YSfO2BwmJJlmoWXTch0Cz+Ld9RFFqbEthe/Z1WMmr7na8aezwY+2q/0CO+OUxbDBxijDdxzyIuOm3ZyGimZRCvzJexhM9hY0G86uhu2HCfnsJaDzs9b5kNL853acGe5Rw0SndWZ9lnIRwulAhP+EqEVhIXDZGiXEmeZKJ2C17e9bVePaFp5tsRC4081e++2GDRsOy+0Gm6MUrUt6ScJ2rFjwXULXxrItfMfiwSBFEeFaUBQGx1HsjCqPoDQrKTG0fIc404ySglKDMYZUl0QT4R6nGqXg1UttWr6DMbA9zFgfJtg2rA3iStRtayruhwn57CWg86uGcVqQFeX0s5tdITyvGe5pnlmfhVyEcHoQ4T8halHIdflEwa85zExNKbi50uTSgs84K9iJM+5vx2SZ5h2leON6h7ws6QYeN1fCaou+qiqHcl2S5CW+a7M+jMFAy3dY8F0wsBOlOJbF/X7M9jgjzguSvKr7fuVSm3u9iPVhwuYo4UrHp6QybzOwS9zrz6AXZXvmA+Y/q1kBnV013OtFfPN+H1Bc6fhc64a7Wj/OupoetbPWUUI2MrMWzgsi/CfE0zAGm53lGlOZuTmWxc2lEGfHYnuY0woctscpm+OUl1daBG4VglGVIzG9KCPNSnzPYrXdwHUUrYZD6DnTGfe769V5jNNK8I1R+G5V2RNlBbk2NH2bUWoT5ZqmqhLF8x73dVnr4/IBj/us6lUDKPxJUjzXhigrHlkhzK4E9qqcehx7rTjq25/0PDKzFs4DIvwnyEFFYa/E5vz1Ol7uTPxWgOmGr3/t1govrSakRYHBoHC52gl45VJ7l4d8VpS8uz7EYFgfJNPQUm0lUfv+v7M2xLYgyw3XF33ioiRwbEJv0gTdpKz1Ewpd0m44/MALizR9Z0+RrPMB8+Ge+fe432dVhXIUSVYAVVOa0HtoCzwbYjnqzH2vctrTmLQVhKeFCP8zYrZUsQ5bFLoKzfTibGp3u9r22RgmRJlma5Tw0mqbONNkRUk3dBkXmigvCF2HN651WRuntD2Hywv+I8nQXJcMJs8dZyUNx2Kl5U/dFjeGCQ/RKpolAAAgAElEQVQGCUlecmu5yf1BTNBwWW7ZXO0G013D1xaDynPfUeSFwXWsfROvtTHYbCP2w9bN31ppTXyAHu469pxH2yweNdk6v+I4rUnbo7Cf/YYgzCLC/wwYJQXfWR9Oxb4WmQe9mEGQE2Waly+1pjNwA3RCl61RyiDOAcNOlLIVpWz0ExYCl3Fa0G263FwMubHcnHab2uWt03DIS8POOCUpNJvDtKrf1yW9cY5nW+SFJsoK3lsfEfo2Vzo+xuy2eahnyGu9BMdWbAwf3VX8OA4rrJXQe4/cPt9m8TjJ1vkVx2lN2h6GvbqmzSbgBaHm2MKvlPoAGAIaKIwxt+fuV8D/CnwGiICfNsZ85bive1YYJQV/erfHexvDyokzzsl01ckq8Bz+zVdXiLLiYdmnZZFkepIk9qczX8e20KVhlOTo0mDZCsuq4uH1rG6c7o6Fu7bFSrNBoUtWOz6OXTVHv9eLGSU5SV5yfSng+1/sUmiDZz1suDIbTtkYJpQGCmO4tdQCdid1Z8M4uS5xbItOuHfFzlGFda/Bo9lwTiTZel6Stnt1TTOGM72CEZ4OJzXj/xFjzOY+9/048Nrk358D/s7k/3PP1K65n9AbF7TCkigtyXWBshTRKGUQF9MqINe2uNeLiLKCojS8fnmBxWa1AcyLLMa6IE4rI7idOCVKCqzFKnRUi+58i72r3YB8UtaJgVFaMEwLVhcC8kLT8lyaDYfNUUKz2UBrw2r3oZ10LSYr7QbDJCPO9K59CbOzTK0NVzvBvhU7j8trPIm9Bo+T3EV7HpK2+3VNO6srGOHp8SxCPT8J/ANjjAH+mVKqq5S6aoy5/wxe+7mS65LSQG5KUl1gRjatwGZrDO2GQyuAbtPh5nLV0GRnnPHxTjSdjQdutREryqvesarh8OrlNmHDZmuU4VoWLy5Xse+6UXleGJqNuu9vQqZLhklGK3AZxTnNwAFtyAtNw7F4aaXJg35ClhdsjBSrLW9qGgdPLlOtHUMHcU6UVTH4G8vNR+LLs8J6lKTs/OABZzMh+zQtH2Y/o9muaWfhcxGeLSch/Ab4faWUAf6uMebNufuvAx/PXL8zue3cC79rW1gKfMdmMfRYaNgsNRsUpcGyIHAcVpq7K2zSXE/LKcdZwZffWacf5ygDyy0P37WIM5vlljed9daNyl3b4s54DAr65Ni2wrMtLFvRnOwOXm37cA36cUHHd3nQTxgnOR9sxeRmxEroc2upRbNRvYdZMakThvPvUU9KLsOGS0l1edZOuq5SAqZVR0dNytbHzYe1zkI441lYPpyHlYvw9DkJ4f/zxpi7SqlLwJeUUt80xnz5sE+ilPo88HmAGzdunMBpPV0Oaut7tRPwnfURnm3huC6twOP7X/BoNZwqDutaD9s2GnAsBbbBdx3SXHN/J8b1LEZxBlbJ91xexHEUqy0ft06+tqsQ0Vo/Ji8rJ83aVG1caKK0oGdl6NIwTqt8Qlkq7vdjHGV4fzNimGgWmx7dsEFelo+8D9h7hj1vTrczyqbduq4vhsBui+grnaCydubRmP+TPtPTsHv3ODzr6qHTaCgnnA6OLfzGmLuT/9eVUr8BfBKYFf67wIsz11+Y3Db/PG8CbwLcvn17nz2fp4NDlSe6Fi+tttgZp6RlFXMNPHtavmkM07aNoedwrduk1bBBwfvrQ3aSjKhXkGuI85K86NH2HaK0oBN6uHY1CMSZxgBlUZVShp7N1U7AR9tjLEvRizMWm9WAc7UTsjNO6cUFbd9DGY02JcMkx7ZMVbs/x6xo9SZePPWsvuVXIaheVIl+3Wfg4caoyiLaTC4rxZ4x/8d9prO5hCTTXJ2Enc5SOONZDlb1ZKJe9R13dSGDyPniWMKvlGoCljFmOLn8KeAX5w77LeBnlVL/iCqp2z/r8f3DzNwqp0sb1/HR2kxnx1FWEHoOSlUbs8KGS5TmVfhGK6JM8/FOTOBZpLmiG7q8utrmTn/MKE55b2PI7ZtLaAx3dyI0EHoOxmKy4asaPOqmLnd2YuK8xLVtFnyHxWaDpNDYCrbjkoXAJskMr15u7+kXNGvHsN5PAKaz+nrmXwv+vLC5tqosJJh07rIf3cD1pM+0ziVsjTM+3hqzNki5tRKeeMP3p8mzqh6a7QEdNlwWfOdYq4vT6koqHJ3jzvgvA79RVWziAP+nMeZ3lVJ/HcAY8yvA71CVcr5LVc75V4/5ms+dw8zcZn/sWV7Sj6vZcjDZjbrarmqtF3yHpmuz0m4wSgsCD8BiKfDoBA2arkVWaorCYNs2UZzx/vYIrWGl1cC1LZJcYxlFO3Cr2XVRbeD62p0e64OMK12fdsPhxmJI03e4tdLkfj/i+laM71kMxjlRqtkZZ4/U6dfvoxdVhm+zs/o6PxF6e5dXzm/IYvLZ1Z/bQTz461zCMM7wHZuWb5Pr/dtbnlbmk9xPYxDY5fqa5jTdvd1hD/N8Zy2fIjyeYwm/MeY94Af2uP1XZi4b4G8c53VOmqP+4GYft9r2p2L3uB9yndiMsoKv3+2TZJpRVnD7pSWiiQumZ1sUZYnlKPpRxtY4ozQlShnWBgme5eAvuHzX5Q4YwzDJWfQ9hnFBw7ZItSYpCmxV2R18vBMxHKfYtmIrykgnHbQWmy4opjtvK998i1xr1jYToqzgKx9tszlOeXFx92y6fh95UYVaNnSChWKcFHzrwYASg4XiB28u7Vox1J/JrLdOPXusrSgMTEtBPdfat+TzxnKTXJdsjlMKXe7aGXzYv+fzCl1kRUlvnBHlBeNEEzTsaSP7KC/o+B6LrUc3rx2GvVxf59/jQWxDzmo+ZZQU099my3ceKS6QQesC7tw96rJ1L8FyJklMz6mSmLP3dwMPVFVjvzXK2Bgl3NmJuLncZHOY8PYHmyS5mXjtwM44Jy9KhmnB5QWPKC+xgXfWRnQCj42xTdPzuNIJSfSIZmm410+4suDz/saYnVFG2/dAVZuy/uzegPVBSsu36QY+NxZDXMtiKXxYDfTB5ogPtsZYKJaaLpe6HqHr4FrWrtn09NiNEWvDhE7Dw3EVN5ZafLQTkWnN5U7I1qgaPCqztd32FLUdRZQVjLOCZsMhmmxUG6Y5cVqFcl691N5lKTH7d2r5Dt9zrbPrRzz/ue/lijprY5Dr6nK9C7gOv80PVk8SiscNHPPCM/uYP7vb519+vE2UaLTRfM+17jTP47kK17L59BvXpuJ/lAHqcSGleR+oWZuQ2b/TSbqhPm1mP6OsKPmXH25PJyJvXO9Wv73tiKIsub4Y8trlhT2/H6fxvT0tLpzwH3XZOvu4zWEKsGt3KjDtU/vO+hBlFI6jSNKCVugS2BajWHN3O6EoDcNYo5RFlBf4lk1RVnH/uCiIMo9UGywqMRxmBZ5nk5cld3sRG8OM9WHM5iCpKnfyqnFJP8l4f3PI2x9uMYgLCq0JXJcXFjUvrTZZbHn4k/r+yq658uZfbHsMowKMIS00wzSflovW7z3KNL2kIMpKijKlU07q/Y0hLUq2RtUKwLWsqRAnmca2Fd3Qoxdl08qlb9wbsNRyKbVhmBbc68U4tkWiNYUuWWo1dj22zhvkutqJ6jkPVwV1WSdU1UNRpukE7nRArwetKNOsD2LsScjIYGj7HnGWE2ea168uTGeHVRVSTG0LPZ9HeNzkYZQUu4RndgU0Tgve2xrx8VbEg0HCxztj3v5gB9+18R2bN15cZCdL2BwnLLa8Y8XW9yrrnG0WVPtAVX/b3aWxj3NDPW3Mf0ZKVUUQjm34eCchaFg4ls0wLSozvl7MtW44tQS5qPmLCyf8R122zj7Otati9lnfec+x2BqmfOvBgP44Z3HBYzl0ud+PSbcjSl3SizPyQtNLMhYDD2MgLgxXuh5JVjBMCnJjGMUJSWGIck2caVJtcFG8vzlGlyXfWBsQJyVrowStSwpK0gR24oJRlpGkBXkJeQGd0PBCt4lnVdfjvMB16mJ8Q2+coYBbKyGXOz5bo4xgLiZcv/eiKMlLzfagJA0Mycc7XO74XOsG02bvSoGJH9bra12VkCaTqqPQs1lse1xdCBhnBUZBJ/AYZzk7o4zAtrjSDch1SZxpCp0CZupxNO9BU5vCvbM2ZGtc5U7qJvP1wHBnO6IwJXd3Em4uVyuHzVGKHmfEhSbJq30Q33utMx3kitIA5XRVAg9tmx83eYiyghLDcsufroBmZ/3GwP1BwrfXRkRJNYFwMou8MGhTsrLg41q7N8d5trWrX/JRqc87bNjc3SnZGqV7up/u54Z6GpmvNhvGOR9uj/jn727iuhbvb434128tkeQa37WwLbXv4y9S/uLCCT8wNTQ7zCxmfvk8Tgq+sznCd2w2hglNz+Hrd3vc6cVsDmOWI58PbWsyA3XoRTlhw6oEb0MT5Zpu4FJS8spSk16r4M4m7MQ5qa7msAq41vVpui4F1Q9VKeiNc0Zxzs4oochzXNvGxjDMS3RZkhTgO6At6IQeC02PhmfTDhz6SY5SikuV5Q5JVnBvJ2WU5eyMMxaa3tSLv/4ReI7FK5fa5KXhvY0hoae50m5QKrjS8cGoXfHUeoD0bIvVbuUGGqearXHC1thQ6pJxVrAxSMi0IdUaz7W5tdQEDP0on8Tvqx9pnGkyyxA27F0eNFWYzaLpOSgUTc/iXi9mwXd3rVY2RymWrRgmOcM0x7EtbiyFDJKCne0Mx4KtcTpNOK8PEt55MEAbw0vLBdc6k0GKh6GQOCvYHmUE3m5LhNBzsFDTFdBsaWyz4bAUOhRaU6LJy5K1XoTnWnTCBkmuubUc0pl8P42pBrp69XC9Gx71Kw88dE590E8qi+/Je2n5Dp6zO5Qzf/20Mjsh09rQDlyudwMcW7HabLA1ytkYplxdDGn7bpX3mHGWPWv5i5PiQgn//LJuP2vhJ9EbZ3x7bcg4zVhq+biOYjiIyTQ0HJu0VGyPMnzXYivKsKyQwLOJ8oK1YYpjKRqeg2UpOk0XLIssKzHKYpyWQElRlCwv+ERpzk6UUCpFnFfeOqM4Jc4NuqxE0bgWrmPh22C7FrrICUObRqG42Q14/VLISyst2r7DKM7x3arbVj/OWR9lvLcxIOxXP4pP3OwSZQVNz9n1I2j5DjeWwqkfUJRp0slsurKWqAaruo4fHg6s47TAcRSXOwH9KMdSkJeVwF9fClDKsNxssLrgT4UVYCfK8ByLtX5MYQx2pCovmsngUseph3FOw7PoNCv7ipVWYzpLd22L5XYDhcJTNt9zrc1Ky5/OEJNc02w46HqwVdXrG2B7mKIci492Itq+Mw07RWnB2iCh0OAkTC036s/pB28u0Y8rS43ZRKkxgLIIHAcbRaENaQYm0qRZTDHZg1HHrpSiWt1MnmN+1/Rh8ZyqKCHXhoXAJdcPn3M+NDRryTF7/bRRv6coK1icWJy3Jt+77XFOlGs2opxuU7PUbEzCPI9Wq52FQe4kuVDCf5xlXT1o9OKMtz/cmfrXjJKcS50Q24JxltOPMtoNm1bDoeM7jHPNne0x7cDjcruqqvHbLjY2l1oujm2TZBmjLCdKc7JCszlMmKgE3ZZLYik8A/f7KVFaGbhZGDwLHNvBcxRgsB1F4NnYlqLhOHQWXRYXfF67usC/cn2RKCumwrwxTLi7E/OtB0PWRxlXOhZtyyHwHELHIWzYlcmXM9MYZpASZQU7o4xcFyw3AwptuLwcsDZIyAoz+Tz8qTlYHTO+uxNhT9w/r3YDFmyLj2xFXlYupS+ttuhOEs+eU+1mTjLN1iglyUuudQNQTJvRZ0XJzqTpe65LVmjg2BbdwGWcFaS6nA4iV9oBd3sxzcDGGKbhmtBzuLXcJNcG11bTiUCzYaOGCstWdBoOGEOSacZ2NSvMyxLbUiw1PdYHKf042xXOqT+zlCq8B1UhQJJpKEuUpaai63uQ5JBNkukKNd05Xc9Go1RPK5iOS7PhEE5CYY+b4c6b79UJ8NOWCJ0vAugGHnG74LuvLPDB5hjftXEtaLgPCxbmuYg2FxdK+I+zrKvjrRvDlFGaE9gW/VST5ZpUV5uwvu/aAmuDmMttn4+2Y6KsoO3buLai23TZHBcshS6jcc5Cw2KYFlzveuxEJZfaAXFhwFIMsoKGregnGa2GR5xW/XPDhs1y6DFOiiqM0HAmG3QsdGmhMPTTjIWgQS/O6CiPKM550K9izRiqPEJhSHKNrRRd3yXLq8TXzW7I1QWfd9eG9JOUj7ci3rjepRdnRJnmwTBGAZvDhNEkqZoWhnbgAFUoZpBkU+H7zvqQfpyR5NWP7fpiSF4YLKpJbdu32R4kNFyLXpTt6gy2MaxCHHe2RqwNMx4MEl5YDHh10mXMtavKnDqkdP1SZUo2OyCM0yrpfW2xGjRqK4t6R+teJaRQ7TvoBh4fbo8ZJDkm0yy3Kr+letDRpeFbDwbTTW31udfflb0KAaJUszZMGSYpjlJQQl6CKSHLIc4Kvrk2JJprOHNQDmojUvc2ftxjp+Z7SUGU5kBlvjdf7fO8BXM+xv/NB32+fm/A1jgjDByWQpcSxUfbMTeWq3agpzVR/Sy5UMJ/mGXd/A+h3kCU5wWubWOMIXQsOoHH+jDlg40hqws+DVdxqeWzFDpooNlwUVblg9Mf5zwYJIySnGGoec1rshC6FBhudJu0Gg73BzHvr40IXJsoK7jWbfDttQxtDOMkxzIwTAs6TYeW5XJrqUmmCwaxZm0y+wwbLlFuGCcFS6FHP8n5+p0ed3diFlsepS7JSnhva8RWVDl3Xu8EvLDcZGucMUhzFp0Gmdb0k2zaGOZuLyJJSxZCj9zAzriyaAZY8D2GcU40SQpWwxD4noMzEelhnBO4DivtygGuGzZY62dkwIdbI7qhNwlFlNMKqQfjjDjPKcqSJPN2hR72+lvODgj14O7aFqFnTw3tSgzjRDOI0mlFz/zy/1Kn8kK614vphO509VP/e+N6l4+2IpYmZZezq8e9CgHGaYGlJjkAZRH4Dn6uWXQcSqvE5JrXry1yvdskyvX0Oed7G+xXhgiHcyudbWqzV/nm1HwvzQk9B9tWe1b7PG8Bnf2s40xzvxfRjzLu7ESUWtMPG3z35TauMiyHjV25q4vMhRD+w7aj26/E62on4G4v4uZSQJ6XdEOPj3sxZQmtwGUnyrm7NeaPP+yjTcmrq21aDYemZ9N0bLpNl7tbIzZGKUYbPlCVe6dR1a7ZwLOJkwKlzHRj12aUYlmqqkbxbLpNH99TtBpeFRKyFOv9DK2r8IGlFJmurBi0MSS5YRAV9OKMXpyyELrkxqCN4bVLLTCGhu3w3VfbQFWV4tgWO+OUpbBBx6/ipllRcr0bEGea7ZEizTWgeGW1xWLYIMkLtscZllXFrm8uhdzrR9ztVZVHjq0wRrE+iDHKkOQlRWHQZckgKljvJay2A7KiqsUvdMmHm2PGcUFalARND9up3uPsrt959hsQZh1GP9oaM4hSRqnGqKrM9NXL7Ue+E3VYpM4pzCdx65DJXk3n5y2k653bb3+4xTgpUKXCs21+6EaHUineXx9jWRajJCec2HE/boU6/x3tht6BRXk+5Dkv6HXS/GonAJj6/ZzGap/Zz7rVcLi7M2YnyugnBYuBAwbisuTuWkqkq9/kcZPk54FzL/xHaUe3Xy7Ac6vdo3V53WLo0Q5c3nOG9OKEzVFMVpQUeUlhDJuDhC1bcXOpRWYM33WpSVYYxrlhoemy1HZp+i4vr7b4s/sD+oOMjXGGbdtoDZdbAR4WhTasDzI2Rxm2VXnqmDInKwxlCcOoCnekhcF1bDwblloNXl5Z4GqnwUqrgbIUvucwSAs6DRvbqWqbL7d9Wn6V6NvcTsEYOkGD5a5XNYJpeTR9Z5eIjdOC78sK7vcSbEtRlNXMdLHpTZO/eVmCUSijKAw4RqEswyApMCqhBIbjDM+xaSiHZqOsZtaTuvJu6HG1E+I7Dh9ujemEDte7Ab0oY5QWuzbRzc9w94rZzt52Y7lZJcVVRsd3sW21p1DuN4jUoag6XDTbuGav16uvQ8HlBZ/lts9O1AcF66OM773W5nrXJ/QclloNwkmu4XEr1PnvKBy8fWRd3VP3RZ4V9EKXuzZ3zfdWOI3VPvVn7doWlxcCbqyGZHlBK3AZRpokMzQa4CqbVsM9dpL8PHDuhb/+gXjOpB2dbU2Xe9P751YCxlSbj6JUV2GZJKPje5WpGlWHozjTNJyCTuiy2vL5rkslO6OMUZwTRQWeZ6GUIc1LHvRjvvGgx4Jr00tzisIwjMAxVcLvo+0xtrJxLMMwqbxobAsWAofNOMVzFVqDb1fP2bAVC36VdNSlphlYKKMIPIuFwOHSQoBlYBCn+A2L11ZbeI3KC6jhObyy0qKfVNYQdc1+P85ZbFZhC8+2udr12RwnfLw9ZrndoBN4DytTYFJ5YlCqKrvMdcn6MKEfZVzuBNzvxQzijH6ckhQlf7YdszaISDLN97ywyBvXFrAdxaJt0w0bDCfCM4ofipjnKJZaHoFXnU/oOYwms9HtUYZjK652551AnyxKLd/h9asL01h/XSnk2nuL//xt9XeqrvI5qJAYA0VZEucZcV5yreuzEHpc6gSMEs0o1QQNPa3jr1+/fs3Z6/OrgWbDmSa6DyvKswPMXjmS2eq3054IDVyHK82AQTcjsF26QYnn2IBLw1E0HOlIBhdA+OsfSFZM2tFNqhnGSRX+MMCHmyN815laGb+3OSLJCu73Y95bjximGZ5t84M3FyetDEseDCLW+jmBY3GnHxG6NtvjjF6UkhWGhuuxOU7ZHKZsRTlpAUaDDdguhI6iwBC4LloXXFoI0QZ8z2IxcIky+PK31xhlxWSDmM1qx6efZQxijTvKSLMC31EUCvqjnLZvszPKubIQ4jgWG4OEQVawNUq5udIkdBwutVx2RhkfbI/YHKYoFLdWKjuHSx2fwLUJXYe339/kKx/36Ec5Tc/hejfk9atNxqkmK2CQ5NgWvLAUstj0UFRVLkUBVxZ8+uOcP/jmA/70Tp+dKGdnmJAbKDV87d4233lhhdVOQNCwuLEc8spKp+o6Nuk7PEhyCm2wLUWcV+GecZpUoaYopSgAVY1CoVdV69zdiRinmqTQUyO62VDLrCDWVtK1fcFOlLExTPa0fNjvO3XYkIdS0G64tHwX17UZpZrVliFNNZuDDGXBDafaoT0bztqvD8Jeq4GD7kKfzx3U73mvHMlZIdclQcPm9WsLWJbFC0sBpoREa1xL4djVfpTTPHA9K8698M/+QK53q/83hgkbo5S7vTGOZfGVD7fxHJfClBhTsjVMcWybrVHMINI4tmKYVnYKVzohWaGxLcXOOKPTdHl3fYTnKO70YpJU0/BskryqiugnOcMUdhWR5TDMK9Gy4xwNDJMxrlvF9ddsC8eyWOvndJuKrITClGBgs5+RG0OgqhkipUUYeGSmCvMUkzaPDwZVOeSq5/PRZsQ4Lljt+myObFbbPqHr4Fg54yxHoegEHi+vtghdhweDmPVhJbzb45Sv3tkhsC3+8AOfl1eadCa7jjNtiDLNctPlpdU2S2GDD9bHvLs+ZJjkbPWyqoolyhhk0/J0ohG89fEWn+68QJyVrA9zml5EmmmMpfj22pBWw8GxLG6uNLEthVKwNcpQCtb6Kde6IWuDmMWmnsbae+OcdzcGxFnJN+/1+cTNRTy72hWbTconX7300Ha6ql6xcCbCeWc7IteG0LN3hY72qpY5yiZA17YoDHR9jx+6scj9nZjFpse7myPWBikNy8K14OZKi0sLVZ6j2XDIdDm1vZjtg3DU2ffjBq6zXNdevy+ATlDt4PZsi9V2+0z1bXgWnHvhh0db9jm2hTEl794fkRaab60Nudap4sdxlqNshyiN2B7lGAxJntOPqrK2O9tRVbWz4PPu+pBxVlUTKFOJe6kqx8qhyfFtGMVzoj+Hnvw/0hAA6JKYklJDbCAZGQIbllsWcVIwTDLyAqIYsrIKSTWyElVWYSnQbI0dhuOMQkNJySAuGCQ572yMuN4NeHm1xHEs7vfiysqZIa9fM4TuEn/y0Tbf3hjx9Ts7vLtRJcrGiSF0YW2Y4NkW7UaDB/2qB4BjVcm/9WHC+iClMCVtv8HGIOaP729zp7d3aeIgNvzJnR1e7IbYyuKPBtukhebl1RYoRTdw6UUpyZqm1bDZHKb0x/9/e2cWIlmW3vffOefuseValbV0V01PLzOD0cxI40FCkpEHZGRZWH4QRsY2ejAIjA0SSBhZL8YCPfhFtkA2ZrCEZZAXYY0WjB88yAJLBkmeRaPp6e7pmemu7loyKyuXWO96Fj+cG5nZ2VndtW8Zf0gi4sbNiHsibvzPd7/z//5fw8ZygpSCvPTeQv00oGlbP769N+XqXkEUSOLA599zY7ixX9BYS9l4ovv4+cH7UiajvAF8YVNea4Z5TRYFB4HC7QzN7qYIMAokL5/p8fqNMcO88usVheGN7QkCX3WaV4bGmIOJqJ+GTIqGRjv2Z9XBOXw/UsoPI/cnPZ1zOxwP8hZkf3ucCuI/ivkPfVI2DLoR3Ujxxs0xV/cKdqcVjbZIVVI1DpyPQEe5ptawPysZVxrjHMs9b3hWNoZeqigqRy+SjPKGyjp0A/kHMf5JxwaUGrSDBk8ssYT1fkg/ChgWNYHyhG+AQIAKoKoNKoBR0aAtVLVXiCSxIK8csQJrHVVjmJQN41nNxZWM80spxljOL2esphE7s5JXt8Zc38t5bXPC7qQ5OBYakNqRV5pu7KthG+vXO6x1XFrrMG7z890kIi+NJ+fbQEioaks3C3l3f0qiFKU2jHoGrKPoWOLW8tobv83Ia8t+WVFrSyAEs8bwxo0xSaRYKWIUjkAKbo68BHNrXLDaiUE4amsRSCr9fs+bpSyiE3ub57zWbI9KGuPXbLpJyKRseOFM96DvwP0UAQ6LmrN9b8AWItjOS0ZlDc4XkAWBQxta/b9jtRuTRsrnpoPkPX0Q7ofQnlZy/zA8q+N60GePtY0AACAASURBVDh1xD+PCjpRwN6sYn/WcKaXMKsb9govsywby/l+yrisuTWtqGsfte8XMMgsFu9x8/xSzM1hgXU+gl7uBFS198W3zi8A2uYuDk5CqkAbaIz/crSFcd6QNxZpDGV7iSDbKqhYKprIYa0jDiVoQ2UdVhtKDWmsiIKAadXQGMGtaUnZWGQgyULFtNasGd/jV2soG8vNUcmobKjd4RVJDWT4CaSbhAgh+ObWxM9OzpPaaieiH/vK2Zn27SJvh0BCEEIvDLheW2oFRdUgpeWVjR79NKLUhlhJNsc+5bbSDdHakiQha92YnVnF2zszBllIXvneBJfXM+JQ8t2XVsiigPVuTKAEX782RErBtGoOFqjnTpzzyt3zS5kvdGvPk91pRRYrJmXNKG/IInXXksajaaJZpbk+LLBCYJ3DOcfetMZZS9WAcIJIBTjnTelWO8mBXHSt6680nsbc+wJPHk4d8YP/US91Is70E9+LthuxdcObal1czhiVGiug1t6ZseaQABWWM1nESi+hbgxnejFBKKm14Z3dkknRMKv9Ii4c5rU/DBJfvVkeea95vlJbqAuDdV4ZksawkiVE0nvz5JVmVDRUtUFbUJ7/kQIEkpVeQhwonAERQj/2EedGP2Z9ELPaC0njgFldE0lBYSzCHh7HHMbBtNI02rDWj7mym7Pei72UtDFcGPjL7Kv7MwySNBKMm5M/AaH9AvvV/RwnffpCa8veuGbtowmXVrp8+d19ZlXN9b2Scd5wZcdwrp/w0kYfJIxyv4ZwbU8wSCUfO9enE0VeYmpsWzEdtU3pBVmsDhRc4N9za1TggPHM+xOt9xNmlT4QAwBstP195/n8O5U0nugN5WBaaKZVw0o/YXNSEagAJR2VsxTaf+p5rXl+tdNWKR9KKY9X3C6wwL3gVBI/zOVxvqioE7XtCGOFs6CkQOGLaaTQhMKSSB+FVxpuTCpe3OhzbrWDdYK8btiZ1FhXHJBKmgAWjIXJHfxWLZ70BX4SiIFAeRVM1YDGp4JCBVEYUBvNciclUopYBTTaUcmGcub3rfCThJIN0jnODxL2C1+s040Vq92IC8sZg9Q3bC/rhq+8MyINJVkgWO4IypHD4CeABIhjvG1BErKURqx0QhRwa1rw6nXJ9rBic1pQ1Ya3b06x8vbT3shCXDWMypqVLKIXQS+J6aYBs8qwOSqQEkInSUPJWDkiIQkC1cpOvS+RNY7GGmQVoi3Eke8+dWtS0099xW3TFmA5x0Gx1TxtM6k014cldaMJAvjBl8+emCc+PGfu7hxz+P8fF43vtGYso1nFjVGOsY5JVWNtQycOUUL4808JwHvzHF9DOFpx+yRYJizwdOLUEn+oJEWtefX6EK29Audj5/rsT2uGhaasG1azmNoaHBVl7fPpgyRgkEQ02iEkfPLigM1xhdNjXm8MZe2Jd1JCBIdh+13AAUL5SUNJfxsAYeivJPqxopOEfPK5VRpt2Rzn1FpTa4W2JcJJqCwq8JYREm/PfHHVV+q+vNFDSsnGwOv9rXC8uTVlc1IQKUnZwEw75lkqhSd/CVjnWM18cZcEboxLlroRkZJcHc64updTNobtSUFTH/7vSSi1ozGOXhL69BSWTuQ9068Pc2a1YVYZwkASCRh04zbtERMGvrVgAKgg4NxSDFIwKw2DLGKtH5HXhlevD5m16Zs1Yi6cOeyWVhtLXRuUhI+sdQlbm4xu8t6CrNt1X/swa4R5odS1vRkgKGrNrGzQzks4t8cVRW1xVmARnOunXFxKCZVirRu9j/QP/PmDQ8fPuyH+J81gbYHHh3smfiHEc8B/wjdcd8DnnXO/emyfHwJ+H3i73fQF59wv3et7PkhEgWSQhjTae7fktUUbx8sbA97anrLZaKa1wTSHdr2ZwpNL1XBtf8aoqHFOIHC8O84RwqGcz4UH0k8Ad7vA69o/BVjhI3yHj/4VQIBv1mjgzc0hUglKbZiWhqqu0AYQFpQnam00V/Zz9nPDUhbx4tkOw7wmr6EbKBonKOqG3Unlr3IKQ20N0kok9iDiD4FzKymXV7tEUcCs1qz1k1ZvL9ie1ryzm3NtmFPVFmMsRXV70pf4xjCN9vntMBTMyoaiibmxn1M2Dm0Mk1qz1olQSrE1qVjvwq1pyTBvCGXAxkoCSHpxyEoSU1mvsJqWDUY7LAJtDWu9xJO1eG8B1uUzXdReQRwpYinf458/x0mma2nko3hvb3Byj9woOLRBHmQh7+7OeGe/YHNY8c3tCUVtwDk6UYDE7ysDyXo34tIJXb+Gs5q3tieEoT/Wu7EeOK2dphY4GfcT8Wvg55xzXxFC9IAvCyG+6Jx77dh+f+yc+7H7eJ+HhiwOWO3HVNpQNiXDWc3+rKasLU4I4tARBgFSGpQyyECQBN5D/9xShzhUvHNrQhgIZpUmiSIaU1MbH6XbOyR9yaHkU3H4pVgLQSpafbek0YYo8DUC611FYR1LsSKOQsalxoqIfgBRKEiVaBVG4BqDyPyCcD8NUYGCquErV4csZyGDLGA3rxBALxWs1yHXhzMkrXoIiCKYFpZx3mCtr0a+uOw/gzQMsM4x7ceksWRnXLE9NQT+wuNEhMBapnhhvYtQjuujyncnq0ZYZ0kCSRIG9JKQLA5al9CKjV6CsQKtHZ1EEU4V/SxkEIVcWE65MSzaxWeHigXGOm4MPUF3Qu/gmIXBgY59KY24/GKXxlpCKQ8cPo9G1c4d2iHMq5TfvjXBp2PEB2r5j/r9yFamenVvimgXd4UDI8Aaw86koNLewnqtlwCHDeqv7Ey5spOzlzec7yuWO/FdWQ+c1k5TC5yMeyZ+59wmsNnenwghXgcuAMeJ/7Fjf1qzOc4Z55pJVR9UqeJgEEq+PSrZHhXMyoZZU7GcpZSl5tpewXCm0Xa+SOt/qKETvH1rzN60YVrW9DshRaUZ1w0qEFjtfMXgHR7fUW6cR9eN8e85mzmS2KCltwLuhAGzRnNtmCOsYGcMgzTEaEusBEb49FA3i2isQOcVuXWUlWFlPUQg2R1XjIuGxjhujguKyrCTV236QtBLBL0owtmakQ9wEQpiBc+t+qrgsjK8szvhm1tjotZiwBoY5wbnQDo1L6w9EQ3Q70Q8t5KxPS2Rrf1DUWkmZcMtbdG1BhSXzqS8sNYnkN5jSEnH9qyiqhsa6736dyYV//etHbqxopsoirpdw8E/v9FPuLaXs5NXOOv45MVlOklwaNfhJDeG+XsUPscbkM/z/Z443UHPgqOa/3kq5XAh+FBbvpxGfPvmhFvTmjQOqBtDmkZkoSCNQ3JtGIiI1zbHxKFitRtzea3bvq7z42pCamuR4oOVPcedO2vtPZQehyroaU4xPc3H/kF4IDl+IcRl4NPAn53w9PcJIb4G3AB+3jn3jQfxnrfDcSfOWan5wlev8tUru7yxNWmrbh29JGZjEHN1ryAvK7Zn/v99Nnh24msXFYyqmpuTPeraE7sFxND3rQ2B6o51PLfH0Qmjad93PvUMq7p95nC6iPZr1jqC0jisAaVgWjTU1lHWzttESMtq23lqWhpGhWZ3UnBzUiGElxUax4ESJpCw1g0ZVw2hBKthWjd8/eo+V/dzPvncgGllePXGGJwjUIpOJDHW2zbvFzVBBOo26R4B5LVlc1RwsZ9yY1hSNQ0OSVU7b0w3a4hCTXGj4cXVLhdXUs71U3ZmFabxY6it48ZeTt4YtHFESnBlb0oShDy3nPLNrRlnugnG+h7GQaAYziqu7uW8sN7lOztTirqhrL33/FInxFhf5Wvde/145jn3Wem7f5mxY39ac2EpY2daeMVO7VNq55ZSPn5uwNzbfm4PfWElIQkkG32fpnlprUcUBqSJZFJ6+WqpDUVj2BoVB2qiUHmlUjdRrHVibxZ4GyK63ZoEcNBT4GGT2NHf4ZPm4X+nuF0zmg/7n3uZKB71BHPfxC+E6AK/A/ysc2587OmvAJecc1MhxI8Cvwe8dJvX+WngpwGef/75ezqWk5w437o14ctXdvjW9ox3hvXhzuOC17eLe3qfoy8Dh3n56p5e7f5RA/sz56cGCan0pOrw5BUGYKxib1KgjaHUjluTkq1RwaRs0BomLTtHjcOYiiCALPKqnUB6W5y6toyNoZqU/OlbFiEts8pSGu3tkxsLQlBWFXkNxnjSj9vXyO2hvFUBzln2ZzWNcfSjgNXM+/QvdyKGRUWtDWkYMcw1m5OSQjtCJXl3d8ZrWxM2RyV5o3ktkHSikG6mSIOAXhyimfLH39qmbixRILiw1KETKVZ6IUkYMqsb/vStW7y+NWZaGnYmFYHybpSNcRhjUVLBGu+xXa61n6yEFLjaUjSW127s843NMWEgCaVgrZewPa6Ylg29JKSXhodr/NZ/LqGE9W7Mpy4tUWjDeidhVhkurvo1mCRU6LYdZBRIzi9lB9LOpczLcbfH5UGv46O4XSOYud3yoyD9+cRT1galxAMrPHuUOGhGUzQH9R0n2XfPcS/rKAfd7Y5Uhz+KyfG+iF8IEeJJ/7ecc184/vzRicA59z+FEP9OCLHmnNs5Yd/PA58H+MxnPnNPYbP3FjeEgThw4sR5C4Xt/cdFy48G8ylMWWjatq3zU8cv+Bb8+ZWaQepz5mVlmFYNtYbiSEheA6GGOBRMywYL5NqfyIGGrb0cax3L3YQohLJxBE7RmIZSQ10bivZqKGxf0zdkgY70ksqi9s3go9atdHta4YQgC3zDFIPjTD+iNw4JpMPg2J3VNNbyiWDQdobynkVFpZnljjqzWBRJT3FhOeUvrw3Z2i/QOIwGbRwvnO3RCQNqq3n1mp/09qY1ea3ZK2rSMGRvXFM5x17eYE3NxeXswHZ5Wmp2piWVtr4BT50zrSq+tZnzxs0Jsr1KeHGtQxoXDPOSM4OMl872DlxQg1CRRJIkVKhSM6mM9z4Sgh94ZZ21TsKNUY42MEhle4Wm3+Mk2mjLm1tjpBJIBJ++tPIe8r9dI5h7SfHcTSQ637fWhxNP00bLT2Ph2UEzmlqTxbe3757jeDewo75Kx3GU8Bvj2J1WB9Xhj2JyvB9VjwB+HXjdOfcrt9lnA7jpnHNCiM/iuWj3Xt/zgzD3SN+dlj5SEhx45msD+f1nYJ4KONr0Ez5tNb81JcxK4wu0agM4ysZHn8fXX2cObOnNh+bRSyQhjbztbVHVbZGTojaGjYFivO8oKsusOUztzOWgaQTL3Yi1bkI3Dpg2hl4UEglBrQ1j4X2G9htLJ/ItCncmNSudCGUF/SQmCwKUlGhruTjIWO8kFPUU5xyNM+SVN1uTA8G7uzN2pqU31msaBIrLIsVqx81Jzd6sYGdaU1WOvKyZau0jeCyl9Qu+nSjknd0pN8cFDkcnDnjtxghnLdvTmhfXu6x0Yq7v5lzdn1FpSyAAIbiynxNJSdEYzvY77E1r36nLwbe3Jr6HrpQIKUiDACcE1/ZzskjxygaEUhEqn3arteXd3Rm3pt4nKW8MkZJsTyte3ugzLupWgvpe6+STGsHcS/rhTiPYk9JL87aY60vJXddA3A0eVpokCnz/DThsRvNhfQ7mrTi3Rz5he5Kv0vyzytue0hdXMsAdVIc/isnxfiL+7wf+IfB1IcRftNt+EXgewDn374GfAP6xEELjg9KfdM49FAqez7ZnByl5Zbiw7Jtyl7VhXN/pMuuzAYn/YmsOHBUOZJnXRpqg3afk5MriGL+Q6wRUbbrGWCga6GUWofzJmcUKUVqcE9RWkEWK2TGfhrTNcQTS2y0HQnK+H7GxlJAqyVevjtAGnBEsdxTT2jAuC29fnAXsjEqyMGBzUvK9H12hMQ6l4PxSQmO9M2ft7X3Y6EZcXE5YSiMCJXhnb8ZoFiCcVzRFkeSNzSHDvGYvN6wkARsrKeeE74YlJSx1Yla7AcYaBALj4Gvv7iOk5OYkJxSKzXHBrDRcGERMGoN1lkprSoeXngaSSApGs5pJVTPIAnpJwKRoGOYNS1mEwTJIQ7YnOcNC00tDysYwLjVJoFg60pRGKUEWeztthOPMUsLmsGB7XJCFwYkS1JMbwRziTsjybpRAx/fttushWeu3NM/zP+jCs4ctU53bd9/JxDKfcIe5zwXfLr01/6z6acju1FuNH68Of9i4H1XPn/Ah5UnOuV8Dfu1e3+Pujge2R75Bt0SQRb4p961xzu709JS5B7T581YjejzOmk8CwVEN6TEYoGza6t92m8WnZ4T1U4mxjqrSaAeTvEJbg3EcSEAPjkdBFkqsc8yMpc5rpo0mEIJAKVazECkgDuBMN2FaGmrTMC69YieQko2llFxrVrOEstZ87dqI0ljWeynnBr4d5ObIXzJvTxqUUnxkPeNm+6OKQ4WQkrdujbmxW9E4r4rJEm/d24tDbo0rCu0j1LPdgIvLHdIgRCAYlw39LGRSGN66OWJWa67u5nxVgRBeyhtKHxGmse9y1u34PHwvVXRjT87v7syYVZq1XsTOuEIKKBrHjVHFJam4Na14biXjVtFwq11zONP1HeP6SUAkBYEShIHkhfUeYeCLvj5swfE47pQs76bvwNF95128vNzV9/R9WFLSRyFTvZt1kSiQB4R/u89t/lk1xj5ywp/jmancFcKnEyQ+vz/XOJfaFxOdFii85l43vuAsPyapmRNzeWShNWCuZvKYp4u8w41HrPz9omlQ0rfrcw6ySLA/bfxC7pHXnFtPpBHEseLWqGJW+VRTkgQsJyF509BLFYH0KadASpLAMQhjqrqgbgxVbbniJkgktyYFRa35zvYEpF9sPr+UstQN+c72lFHV0Dj/3Q+SLqlSXCk0k9EMZSVLWUiSBLgGAgQXlhIGWUgchexMK8rGkAaKa6OKc8uGIBCMZhWTSvPuXs64aKiN8cVn1mCMJ+HaWBrr6CWKQEpWkoiNfkoUSF5c75M3hjduDLkxqphWGocgCSVY0NZhjWVSarqRYrUTs6ULgkCitQHhiXnu0RMq7830TjFFKsE4b8ji9y/wfhDulCzvxpv/6L6zSvPG5tj3kG5VTvfSuOZOcK9Nce4H8/w83L4fwwf1a3gSeh48M8TvnNfrzyP+y6t+ezcO6SYR47LmNMT9qpVh7k0b6hN0lJbDqH+O45+LAaJ5jkhAGkAQQF3CXgUNjqWyYtCNuDBIKSoH+B7AlecqknYht7H43rsSrLOMSsOoanDWUjTeTz9Wio+sdri4lrI1LNHCsiIStHX0Ysf55S4Xl1OWO8qTZqxIAsn1qqBxms19gxOOQEhmVcN3bjbkleabW2P285pJ0YCAymjOiowLKxmXBinf88IKW6OKvWlFGCga47g6ylnpJvTiiG4MaRSgFHzt+gTVRvXj0hAq7+LqnGO1m1DWDbPG0I0tvTggDQMurab0kog0shRhQBIp9mc1o6JGKsW7ewXX9nOCwMEMxrlmXGq0cQQBHL2gnnv06LahjBGw3k3YnZbvy/F/GO6GLO822p0XnIE7qH0J1cMjukdNovNiuq2RT5RuDFIuH6myPtGY7zbHfS/ptweFZ4b4hcA3Um9PvHnE743IQq4d12A+oygtjKd+WTUKfcR/NI8/rwzW7a098ncUhYNMQqTwXvCRo9EOqSF00EkDpICbw5JGN1TG+8hDO7FYWO5Izg4y1rsx394eURlHpUHhG9FHkaSXZvRCSRoF7M8aurE3VrsxrChLjZCAm9JNBGf6PaJA0UtCAgkvn+lwbinjT97cYXtSEUiJsY5z/ZgLywnf3BozqxqmtV+crpShlynOD1JUINgeV+xMSvLGoJzjhbUuSazoxxG7ecVGLyUJBdeHFTutj1E/C3llo8+5pYTvbE+4ul9wc1xiEVhjiMOALA25uJLxuY+f5fxyhnNwY5jz7e0KpSALA67szg4WQQdxzHo/5dJ6h0AKtHOUtUUbzSivaVovKYDrezOSMOTabs6s1PTjk3P8H4SHTZahkl6pBCwN0vcUsx3HcbK7F/J7lB7882K6NFJt7273niume009PWpLjWeG+EPlXRmPOjDOt/djdbDY+axDAzu1N4iTnLx4O0/h3O7zkPiIXwWQxgFSgtFQa33gG5TFAVEQgLM4F1HUFUa3mn/bOoQ2Fov/IQRSMCk01oAMvJWDaSzjvCIdJEzKmigM2a9KxlVFXtXMau0N3IxBCcGbN2e+oMBYXj7fp3aOq7dKNkelX9BWkjO9iCRSvHMrRwhYykK0aVCt1LIovQ6/aZvSzAuuUJLVfsxyFvHymT7aOv+/U8t6J2K81MFqw6AT8okLfQZphJSC1V7MOzs5Qasi09YxKzVpqKjbuoMokAdRoQNGs5q3tscsd2OkEHQixcYgphuHBMorSQTwtatD3rw5RThvIaKN4+ak4MLAVxFX2rK8Et8TQTwsspyr65JIYYzj/NKdKYFO6m72JBZ7OeevusZlQyi9BcfRK6Z7TT09akuNZ4b4bxfFbI0Ldqd30w3l6ce8mEwd2z5X+/Qywax0B528jkf7ncCnjKJQ0YkUw6JBOOvlnJFko5dyYbWDs/CX18dMywonvHKmtofrAw7fXCXuZygZoJQhji2BFH7iUIpYBhSFJlUBSegIQ8Vq2GGYG2a1IW8sla64upszawyrHZ87ffPmjPNLKa9tDTHWu3gWjWWtE5LGCikFHw8HbE9KGjNGSemln9YRKYVxMGzTKkVlGGQB2jjGecNrmyPODhLAYR2sdmOu7OVY6ReEJ2VDoCRvb+fcmrUSRgdKSOpKk4YR2ji2JgXnl1OiIDoowppVmm9UDWuDlEpbX+F7fsD3v7TGWjchVLKVJdcoCeu9mO1JQaSgsQ5j4K2dGb1OxNm+dyl9Uoqi5vYVs8rQiRU19gP9hI6T3f10N3sUmE9qSgk6YcBzKxnr/fc6ud7r1dSjXqt4ZogfbhfFCJIoIEvAlu/PZz+LUBzq7zPlJZkBkEUwyELSMCBRNdPGkFdQu8NJQSrvCBoEktVuxFIaE0iBtY6dWUU/iVkfJHzy4jJbw5Jv7U7IK+G98xVkOO/KaXwErhvLpGpIY4lSvklBoBTr3QTwcjnd+Eh5VNYkoWSQxAyykKLWOCkxxhBHAbuzilmkcE4QF5qXNhTdKMB2BUJUJJHi/FKKUpI4VNzIC9a7Mdp0yeIABCgkEkeqFIMkoJt6X58sijjbT+glAWVj2Rgk3ByWaOfQ2vH8WkrVGLR17E4bLq32OL+coJRgtmbZmRQsdzsIfDezovbN7o9iTv61dqx1YnYnvgnQ91xaZbkbHdnPV+mOy5pxWSMdTApLoRuyKGBSGqracHNUstZJnoiiqHn0Pqs1r98Ys9yLPtRB9DjZ3W13s0eNeSVv0VgqYxgVDev95H373cvV1KNeq3imiP8kvLDW5bueGzCpNbGqsK3BFQKSMGJzWN9TCijCp0rmqsj5WuhJSODA+4V2v1bU4b32pb+V+KKn+eT0AYrL96EXQJIIpJXEkSAOBE3jiKIAYyGNJZ0gYJBFKCHYz2uGs5JpaGiMpp+lxFJihCVRiiiSXFzusNGP2Jp0GE5LpBJ89tIqWSditRuRJor92YCtLPT6eidJYsleXrM3rljup5zthfzwx8/y2tYM3ThkAH/1+SXW+ylv70xprOONrYkniZWUV872GGQRb26N+MaNCdPKe9as9bysMYkk/ThEBQIhJWcHKWdxjLsRl1czojCgaoxf7HTwyeeWub4/ayNPyUoq6WcJ55dS8tq8p+Aob2sQukrRaEegBJdXuoyLhrKOmDYa5XzkXWtLFioGWch3PT/g3Z3Qr1NMK1a7Cd0s4Ew/ed/iXicOWMpCslixMchO9H+JAsmZfsJnozVvFeBge1KxP6sYFZpBJ+Byay3hrygeP0Eejd5XuiHn+qmvNP6AiP8ksrvT7maPAweVvJWfgD+skvdu8SjXKp554j8zSPinn3uFv/byHjeGBUmouDkuyGtLEkmu7uYHhTUCx1v7E6rCNz23tC6S0qsoXGuAdmEl4vJqn9IYAiXpxQGjvGZaW0ZF5fvgYqgaRyjgwkpGUVvK2pCEkuWOzyULIRiVFaudhOtD30xlUhrKxtBPIi4upax2I97emREJQYOjn0VMcm9XcGtcUltDN465tNLlr1zoUtQWKSWDVHF5rcs7ezmZksway86kJFSKM/2I51e7vL07paoM10c5UaB8qsLoVpce8LFzPb770gpfv7rP1rTka+8GdNKIXhjw6edXGBUNqRK8uZWQRYogkCylAcY5Zo1hJY1JQskr5wa8fG6JKFAMkvAgSlrpxswqw0dWuyBgOYvpxIoLyxnPLXfoxLfIK8vOrOSlsz2yMEAbSxwpQiVY7yX84EfXyRtNKCVZHLSNaUoaY3j71ow4EHz0TI9XNvoHRmnzxcbjDpZHJZONsdya+GTZIA25tNJhc1Qc2Cas9xIur3Z4d2+GAz55YYVOotiZ1ARtledHz7zf1+VuIrtu4mWatbbMak0YJPRSQ6gEaRSQcXvVyKPGPHqvtSVS6o4qXeHkQrMnjfDnuNtK3icZ4iEV0t4XPvOZz7gvfelLD/x1jzoG5rXPIYZS0ljfnq+xlpujgmFe040C0jhkUlSEgY/SvD2zJQtDwsC3xsNxQDhb44JA+QYveWXIm4ZIKdLQi+DHhc8Nn+un0DYECaU8uFwYlXW7jyALfXS+lEXMSs2o9D7yYSD9sWsLAvJKo61jtRuz3ksOWgrOzbveM+ZK01jvM3P0ubqxB+QZKsmorMHBWi+hm3ivmLzW5JUmbwxrnZgzg+RAzzw3sMrC4ECBGLZNTw5sj08guePEe5wM5+8bSkkUytvud7vvuW78d3qSkdndnC8fpDh5EKqUuz0WuDcLhoeN4864T9rxPSg8qVbNQogvO+c+c0f7nibiX2CBBRZ4VnE3xP/kTFcLLLDAAgs8EiyIf4EFFljglGFB/AsssMACpwwL4l9ggQUWOGVYEP8CCyywwCnDgvgXWGCBBU4Znkg5pxDiFvDONrfBUwAABBhJREFUHe6+Bryvh+8pwWLspxOLsZ8+3Mm4Lznn1u/kxZ5I4r8bCCG+dKfa1WcNi7Evxn7acFrH/qDHvUj1LLDAAgucMiyIf4EFFljglOFZIP7PP+4DeIxYjP10YjH204cHOu6nPse/wAILLLDA3eFZiPgXWGCBBRa4CzzVxC+E+BEhxDeFEN8WQvzC4z6ehwkhxG8IIbaFEK8e2bYihPiiEOJb7e3y4zzGhwEhxHNCiD8SQrwmhPiGEOJn2u2nYeyJEOLPhRBfa8f+L9vtHxFC/Fl73v83IUT0Ya/1tEIIoYQQXxVC/I/28akYuxDiihDi60KIvxBCfKnd9sDO+aeW+IUQCvi3wN8EPgH8PSHEJx7vUT1U/EfgR45t+wXgD51zLwF/2D5+1qCBn3POfQL4XuCftN/zaRh7BXzOOfdJ4FPAjwghvhf4V8C/ds69COwD/+gxHuPDxs8Arx95fJrG/tedc586IuN8YOf8U0v8wGeBbzvn3nLO1cB/BX78MR/TQ4Nz7v8Ae8c2/zjwm+393wT+ziM9qEcA59ymc+4r7f0JngQucDrG7pxz0/Zh2P454HPAf2+3P5NjBxBCXAT+FvAf2seCUzL22+CBnfNPM/FfAK4eeXyt3XaacNY5t9ne3wLOPs6DedgQQlwGPg38Gadk7G2q4y+AbeCLwHeAoXNu3pr5WT7v/w3wzzhsPb3K6Rm7A/6XEOLLQoifbrc9sHP+yWjYucB9wznnhBDPrERLCNEFfgf4WefcWBzp4v0sj905Z4BPCSGWgN8FPvaYD+mRQAjxY8C2c+7LQogfetzH8xjwA86560KIM8AXhRBvHH3yfs/5pznivw48d+TxxXbbacJNIcQ5gPZ2+zEfz0OBECLEk/5vOee+0G4+FWOfwzk3BP4I+D5gSQgxD9qe1fP++4G/LYS4gk/jfg74VU7H2HHOXW9vt/ET/md5gOf800z8/w94qV3lj4CfBP7gMR/To8YfAD/V3v8p4Pcf47E8FLR53V8HXnfO/cqRp07D2NfbSB8hRAr8MH6N44+An2h3eybH7pz75865i865y/jf9v92zv19TsHYhRAdIURvfh/4G8CrPMBz/qku4BJC/Cg+D6iA33DO/fJjPqSHBiHEfwF+CO/SdxP4F8DvAb8NPI93M/27zrnjC8BPNYQQPwD8MfB1DnO9v4jP8z/rY/8u/CKewgdpv+2c+yUhxAv4KHgF+CrwD5xz1eM70oeLNtXz8865HzsNY2/H+LvtwwD4z865XxZCrPKAzvmnmvgXWGCBBRa4ezzNqZ4FFlhggQXuAQviX2CBBRY4ZVgQ/wILLLDAKcOC+BdYYIEFThkWxL/AAgsscMqwIP4FFlhggVOGBfEvsMACC5wyLIh/gQUWWOCU4f8DtAKc/4EdvZIAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import random\n",
    "plt.plot([x[0] + (random.random()-0.5) for x in plot_data if x[1]>1], [x[1] + (random.random()-0.5) for x in plot_data if x[1]>1], '.', alpha=0.1)\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[(2, 1),\n",
       " (6, 1),\n",
       " (8, 1),\n",
       " (8, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (22, 2),\n",
       " (7, 1),\n",
       " (6, 3),\n",
       " (5, 1),\n",
       " (6, 1),\n",
       " (6, 1),\n",
       " (9, 1),\n",
       " (14, 1),\n",
       " (14, 1),\n",
       " (2, 1),\n",
       " (28, 4),\n",
       " (28, 4),\n",
       " (28, 4),\n",
       " (28, 4),\n",
       " (15, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (6, 2),\n",
       " (4, 3),\n",
       " (6, 2),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (19, 2),\n",
       " (19, 2),\n",
       " (19, 2),\n",
       " (19, 2),\n",
       " (3, 2),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (11, 1),\n",
       " (4, 1),\n",
       " (6, 1),\n",
       " (6, 1),\n",
       " (6, 1),\n",
       " (6, 1),\n",
       " (6, 1),\n",
       " (6, 1),\n",
       " (6, 1),\n",
       " (8, 1),\n",
       " (8, 1),\n",
       " (4, 2),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (9, 1),\n",
       " (4, 1),\n",
       " (32, 1),\n",
       " (1, 1),\n",
       " (1, 1),\n",
       " (5, 1),\n",
       " (9, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (9, 1),\n",
       " (23, 2),\n",
       " (6, 2),\n",
       " (11, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (1, 1),\n",
       " (1, 1),\n",
       " (14, 1),\n",
       " (1, 1),\n",
       " (4, 1),\n",
       " (14, 1),\n",
       " (4, 2),\n",
       " (14, 1),\n",
       " (8, 1),\n",
       " (5, 1),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (13, 1),\n",
       " (5, 1),\n",
       " (6, 1),\n",
       " (4, 2),\n",
       " (1, 1),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (13, 2),\n",
       " (13, 2),\n",
       " (6, 1),\n",
       " (1, 1),\n",
       " (21, 2),\n",
       " (10, 8),\n",
       " (6, 1),\n",
       " (21, 1),\n",
       " (21, 4),\n",
       " (20, 1),\n",
       " (5, 1),\n",
       " (11, 1),\n",
       " (5, 1),\n",
       " (1, 1),\n",
       " (10, 1),\n",
       " (47, 2),\n",
       " (6, 3),\n",
       " (10, 1),\n",
       " (5, 1),\n",
       " (3, 2),\n",
       " (17, 3),\n",
       " (8, 1),\n",
       " (8, 1),\n",
       " (8, 1),\n",
       " (8, 1),\n",
       " (8, 1),\n",
       " (17, 1),\n",
       " (17, 1),\n",
       " (17, 1),\n",
       " (9, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (6, 2),\n",
       " (6, 2),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (7, 1),\n",
       " (2, 1),\n",
       " (1, 1),\n",
       " (21, 3),\n",
       " (19, 3),\n",
       " (1, 1),\n",
       " (1, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (6, 2),\n",
       " (7, 3),\n",
       " (6, 2),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (3, 2),\n",
       " (3, 2),\n",
       " (3, 2),\n",
       " (3, 2),\n",
       " (5, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (3, 1),\n",
       " (8, 3),\n",
       " (17, 1),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (14, 3),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (1, 1),\n",
       " (7, 1),\n",
       " (12, 2),\n",
       " (1, 1),\n",
       " (5, 2),\n",
       " (6, 1),\n",
       " (6, 1),\n",
       " (8, 2),\n",
       " (1, 1),\n",
       " (13, 1),\n",
       " (17, 2),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (10, 1),\n",
       " (8, 1),\n",
       " (22, 1),\n",
       " (22, 1),\n",
       " (22, 1),\n",
       " (11, 1),\n",
       " (11, 1),\n",
       " (11, 1),\n",
       " (14, 1),\n",
       " (7, 1),\n",
       " (9, 2),\n",
       " (10, 2),\n",
       " (10, 2),\n",
       " (10, 2),\n",
       " (28, 3),\n",
       " (6, 1),\n",
       " (6, 1),\n",
       " (6, 1),\n",
       " (7, 1),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (7, 4),\n",
       " (19, 2),\n",
       " (7, 1),\n",
       " (3, 1),\n",
       " (7, 1),\n",
       " (2, 1),\n",
       " (10, 2),\n",
       " (10, 2),\n",
       " (8, 2),\n",
       " (17, 3),\n",
       " (11, 3),\n",
       " (6, 3),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (1, 1),\n",
       " (6, 3),\n",
       " (5, 1),\n",
       " (10, 3),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (5, 1),\n",
       " (2, 1),\n",
       " (7, 1),\n",
       " (10, 1),\n",
       " (5, 2),\n",
       " (2, 1),\n",
       " (2, 2),\n",
       " (10, 2),\n",
       " (6, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (9, 1),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (9, 1),\n",
       " (9, 1),\n",
       " (9, 7),\n",
       " (9, 7),\n",
       " (10, 3),\n",
       " (5, 1),\n",
       " (2, 1),\n",
       " (5, 1),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (10, 1),\n",
       " (10, 1),\n",
       " (10, 2),\n",
       " (10, 2),\n",
       " (10, 1),\n",
       " (12, 1),\n",
       " (18, 1),\n",
       " (1, 1),\n",
       " (3, 1),\n",
       " (8, 1),\n",
       " (20, 2),\n",
       " (10, 1),\n",
       " (10, 1),\n",
       " (5, 1),\n",
       " (7, 1),\n",
       " (7, 3),\n",
       " (4, 1),\n",
       " (1, 1),\n",
       " (1, 1),\n",
       " (10, 1),\n",
       " (12, 1),\n",
       " (12, 1),\n",
       " (2, 1),\n",
       " (4, 1),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (6, 1),\n",
       " (10, 1),\n",
       " (5, 1),\n",
       " (6, 1),\n",
       " (3, 1),\n",
       " (7, 3),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (20, 4),\n",
       " (5, 2),\n",
       " (10, 3),\n",
       " (10, 3),\n",
       " (7, 1),\n",
       " (7, 1),\n",
       " (9, 1),\n",
       " (25, 2),\n",
       " (17, 2),\n",
       " (11, 3),\n",
       " (4, 2),\n",
       " (13, 1),\n",
       " (1, 1),\n",
       " (30, 1),\n",
       " (6, 2),\n",
       " (6, 2),\n",
       " (2, 1),\n",
       " (14, 3),\n",
       " (9, 1),\n",
       " (9, 1),\n",
       " (7, 1),\n",
       " (17, 1),\n",
       " (9, 1),\n",
       " (17, 1),\n",
       " (9, 1),\n",
       " (2, 1),\n",
       " (3, 1),\n",
       " (1, 1),\n",
       " (21, 1),\n",
       " (4, 1),\n",
       " (2, 1),\n",
       " (5, 2),\n",
       " (3, 1),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (1, 1),\n",
       " (4, 1),\n",
       " (6, 2),\n",
       " (8, 6),\n",
       " (7, 2),\n",
       " (9, 4),\n",
       " (5, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (4, 1),\n",
       " (5, 1),\n",
       " (7, 1),\n",
       " (11, 1),\n",
       " (3, 2),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (22, 2),\n",
       " (10, 2),\n",
       " (14, 1),\n",
       " (14, 1),\n",
       " (10, 2),\n",
       " (10, 2),\n",
       " (13, 3),\n",
       " (9, 1),\n",
       " (7, 1),\n",
       " (7, 1),\n",
       " (7, 1),\n",
       " (7, 1),\n",
       " (14, 1),\n",
       " (5, 1),\n",
       " (2, 1),\n",
       " (23, 8),\n",
       " (4, 1),\n",
       " (5, 1),\n",
       " (13, 2),\n",
       " (6, 1),\n",
       " (6, 1),\n",
       " (6, 1),\n",
       " (6, 1),\n",
       " (6, 1),\n",
       " (5, 1),\n",
       " (6, 1),\n",
       " (6, 1),\n",
       " (6, 1),\n",
       " (6, 2),\n",
       " (6, 2),\n",
       " (3, 1),\n",
       " (5, 1),\n",
       " (3, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (11, 1),\n",
       " (20, 2),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (5, 1),\n",
       " (1, 1),\n",
       " (1, 1),\n",
       " (9, 4),\n",
       " (6, 2),\n",
       " (14, 1),\n",
       " (6, 1),\n",
       " (3, 1),\n",
       " (11, 1),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (1, 1),\n",
       " (1, 1),\n",
       " (10, 1),\n",
       " (10, 1),\n",
       " (20, 3),\n",
       " (7, 2),\n",
       " (7, 2),\n",
       " (12, 1),\n",
       " (17, 3),\n",
       " (10, 2),\n",
       " (5, 1),\n",
       " (14, 3),\n",
       " (14, 3),\n",
       " (3, 1),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (17, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (9, 1),\n",
       " (5, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (9, 2),\n",
       " (9, 2),\n",
       " (9, 2),\n",
       " (13, 1),\n",
       " (11, 4),\n",
       " (14, 2),\n",
       " (14, 2),\n",
       " (14, 2),\n",
       " (9, 1),\n",
       " (2, 1),\n",
       " (3, 1),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (41, 1),\n",
       " (3, 1),\n",
       " (11, 1),\n",
       " (17, 3),\n",
       " (2, 1),\n",
       " (7, 1),\n",
       " (10, 2),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (7, 5),\n",
       " (7, 5),\n",
       " (6, 1),\n",
       " (6, 1),\n",
       " (7, 1),\n",
       " (7, 1),\n",
       " (7, 1),\n",
       " (30, 1),\n",
       " (9, 2),\n",
       " (10, 1),\n",
       " (8, 1),\n",
       " (8, 1),\n",
       " (6, 1),\n",
       " (3, 2),\n",
       " (3, 2),\n",
       " (7, 1),\n",
       " (4, 1),\n",
       " (8, 1),\n",
       " (4, 1),\n",
       " (6, 2),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (5, 1),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (3, 1),\n",
       " (4, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (2, 1),\n",
       " (1, 1),\n",
       " (2, 1),\n",
       " (1, 1),\n",
       " (9, 1),\n",
       " (3, 1),\n",
       " (12, 4),\n",
       " (2, 1),\n",
       " (1, 1),\n",
       " (1, 1),\n",
       " (6, 2),\n",
       " (7, 1),\n",
       " (2, 1),\n",
       " (9, 1),\n",
       " (8, 1),\n",
       " (5, 1),\n",
       " (9, 1),\n",
       " (9, 1),\n",
       " (10, 2),\n",
       " (4, 1),\n",
       " (2, 1),\n",
       " (7, 1),\n",
       " (8, 1),\n",
       " (12, 3),\n",
       " (5, 1),\n",
       " (4, 2),\n",
       " (4, 2),\n",
       " (4, 2),\n",
       " (2, 1),\n",
       " (4, 1),\n",
       " (3, 1),\n",
       " (1, 1),\n",
       " (10, 1),\n",
       " (9, 1),\n",
       " (9, 1),\n",
       " (9, 1),\n",
       " (11, 3),\n",
       " (4, 2),\n",
       " (4, 1),\n",
       " (4, 2),\n",
       " (4, 1),\n",
       " (3, 1),\n",
       " (9, 1),\n",
       " (8, 1),\n",
       " (5, 1),\n",
       " (3, 1),\n",
       " (4, 2),\n",
       " (17, 1),\n",
       " (17, 1),\n",
       " (17, 1),\n",
       " (17, 1),\n",
       " (17, 1),\n",
       " (14, 2),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (8, 1),\n",
       " (17, 1),\n",
       " (17, 1),\n",
       " (3, 1),\n",
       " (15, 2),\n",
       " (15, 2),\n",
       " (30, 1),\n",
       " (30, 1),\n",
       " (9, 1),\n",
       " (7, 1),\n",
       " (3, 2),\n",
       " (9, 8),\n",
       " (12, 6),\n",
       " (9, 8),\n",
       " (12, 6),\n",
       " (11, 10),\n",
       " (12, 1),\n",
       " (10, 2),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (6, 1),\n",
       " (13, 2),\n",
       " (1, 1),\n",
       " (4, 2),\n",
       " (4, 2),\n",
       " (7, 1),\n",
       " (9, 1),\n",
       " (7, 1),\n",
       " (4, 2),\n",
       " (4, 2),\n",
       " (13, 2),\n",
       " (11, 2),\n",
       " (13, 2),\n",
       " (9, 2),\n",
       " (9, 2),\n",
       " (11, 2),\n",
       " (9, 2),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (11, 1),\n",
       " (11, 1),\n",
       " (16, 1),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (5, 1),\n",
       " (7, 1),\n",
       " (6, 2),\n",
       " (5, 1),\n",
       " (8, 1),\n",
       " (8, 1),\n",
       " (8, 1),\n",
       " (8, 1),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (4, 1),\n",
       " (16, 1),\n",
       " (16, 1),\n",
       " (16, 1),\n",
       " (12, 1),\n",
       " (12, 1),\n",
       " (12, 1),\n",
       " (12, 1),\n",
       " (4, 2),\n",
       " (6, 1),\n",
       " (10, 1),\n",
       " (7, 1),\n",
       " (7, 1),\n",
       " (4, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (7, 1),\n",
       " (2, 1),\n",
       " (2, 2),\n",
       " (7, 4),\n",
       " (22, 2),\n",
       " (18, 2),\n",
       " (22, 2),\n",
       " (22, 2),\n",
       " (4, 1),\n",
       " (3, 2),\n",
       " (5, 2),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (9, 2),\n",
       " (8, 4),\n",
       " (6, 1),\n",
       " (6, 1),\n",
       " (8, 3),\n",
       " (5, 1),\n",
       " (8, 1),\n",
       " (8, 1),\n",
       " (5, 1),\n",
       " (8, 1),\n",
       " (8, 1),\n",
       " (8, 1),\n",
       " (5, 1),\n",
       " (8, 1),\n",
       " (15, 2),\n",
       " (15, 2),\n",
       " (9, 1),\n",
       " (15, 2),\n",
       " (18, 1),\n",
       " (18, 1),\n",
       " (16, 1),\n",
       " (6, 1),\n",
       " (5, 1),\n",
       " (4, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (13, 3),\n",
       " (12, 3),\n",
       " (4, 1),\n",
       " (2, 1),\n",
       " (3, 1),\n",
       " (2, 1),\n",
       " (7, 2),\n",
       " (1, 1),\n",
       " (1, 1),\n",
       " (1, 1),\n",
       " (8, 1),\n",
       " (7, 4),\n",
       " (1, 1),\n",
       " (12, 1),\n",
       " (17, 3),\n",
       " (4, 1),\n",
       " (2, 1),\n",
       " (7, 1),\n",
       " (2, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (30, 2),\n",
       " (6, 1),\n",
       " (14, 2),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (6, 1),\n",
       " (6, 1),\n",
       " (6, 1),\n",
       " (11, 1),\n",
       " (6, 1),\n",
       " (6, 1),\n",
       " (6, 1),\n",
       " (6, 1),\n",
       " (6, 1),\n",
       " (10, 2),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (6, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (8, 1),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (5, 1),\n",
       " (8, 1),\n",
       " (5, 1),\n",
       " (8, 1),\n",
       " (10, 1),\n",
       " (11, 1),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (28, 3),\n",
       " (4, 3),\n",
       " (15, 7),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (1, 1),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (9, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (2, 2),\n",
       " (2, 2),\n",
       " (18, 8),\n",
       " (9, 1),\n",
       " (5, 1),\n",
       " (4, 1),\n",
       " (5, 1),\n",
       " (9, 2),\n",
       " (6, 1),\n",
       " (6, 1),\n",
       " (6, 1),\n",
       " (6, 1),\n",
       " (5, 1),\n",
       " (7, 3),\n",
       " (7, 1),\n",
       " (6, 3),\n",
       " (7, 3),\n",
       " (7, 3),\n",
       " (7, 1),\n",
       " (10, 2),\n",
       " (5, 1),\n",
       " (4, 3),\n",
       " (2, 2),\n",
       " (4, 1),\n",
       " (11, 2),\n",
       " (21, 2),\n",
       " (4, 1),\n",
       " (5, 2),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (9, 2),\n",
       " (22, 1),\n",
       " (5, 1),\n",
       " (6, 1),\n",
       " (5, 1),\n",
       " (2, 1),\n",
       " (17, 2),\n",
       " (10, 1),\n",
       " (3, 1),\n",
       " (4, 3),\n",
       " (28, 3),\n",
       " (7, 2),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (9, 1),\n",
       " (5, 1),\n",
       " (9, 1),\n",
       " (5, 1),\n",
       " (6, 4),\n",
       " (6, 3),\n",
       " (22, 1),\n",
       " (6, 1),\n",
       " (6, 1),\n",
       " (4, 2),\n",
       " (4, 2),\n",
       " (4, 2),\n",
       " (4, 2),\n",
       " (4, 1),\n",
       " (3, 1),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (7, 1),\n",
       " (3, 2),\n",
       " (3, 2),\n",
       " (7, 1),\n",
       " (7, 1),\n",
       " (5, 1),\n",
       " (2, 1),\n",
       " (3, 1),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (5, 2),\n",
       " (14, 3),\n",
       " (19, 8),\n",
       " (1, 1),\n",
       " (1, 1),\n",
       " (1, 1),\n",
       " (1, 1),\n",
       " (1, 1),\n",
       " (8, 1),\n",
       " (8, 1),\n",
       " (23, 1),\n",
       " (6, 3),\n",
       " (6, 3),\n",
       " (10, 5),\n",
       " (11, 3),\n",
       " (11, 3),\n",
       " (10, 3),\n",
       " (10, 3),\n",
       " (11, 3),\n",
       " (10, 3),\n",
       " (11, 3),\n",
       " (10, 3),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (2, 1),\n",
       " (4, 2),\n",
       " (17, 3),\n",
       " (18, 1),\n",
       " (5, 1),\n",
       " (11, 1),\n",
       " (4, 3),\n",
       " (8, 2),\n",
       " (4, 3),\n",
       " (8, 2),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (7, 1),\n",
       " (7, 1),\n",
       " (7, 1),\n",
       " (7, 1),\n",
       " (7, 1),\n",
       " (7, 1),\n",
       " (7, 1),\n",
       " (7, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (10, 2),\n",
       " (10, 2),\n",
       " (10, 2),\n",
       " (10, 2),\n",
       " (16, 1),\n",
       " (17, 3),\n",
       " (17, 3),\n",
       " (17, 3),\n",
       " (17, 3),\n",
       " (17, 3),\n",
       " (11, 3),\n",
       " (11, 3),\n",
       " (11, 3),\n",
       " (10, 3),\n",
       " (5, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (11, 1),\n",
       " (3, 1),\n",
       " (7, 1),\n",
       " (7, 1),\n",
       " (7, 1),\n",
       " (7, 1),\n",
       " (5, 2),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (8, 1),\n",
       " (12, 2),\n",
       " (12, 2),\n",
       " (4, 1),\n",
       " (10, 6),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (1, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (1, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (1, 1),\n",
       " (6, 1),\n",
       " (30, 1),\n",
       " (6, 1),\n",
       " (6, 1),\n",
       " (7, 1),\n",
       " (6, 1),\n",
       " (7, 1),\n",
       " (4, 1),\n",
       " (3, 1),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (3, 1),\n",
       " (4, 1),\n",
       " (3, 1),\n",
       " (3, 1),\n",
       " (17, 1),\n",
       " (3, 1),\n",
       " (4, 2),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (10, 6),\n",
       " (6, 1),\n",
       " (1, 1),\n",
       " (1, 1),\n",
       " (1, 1),\n",
       " (1, 1),\n",
       " (9, 1),\n",
       " (5, 2),\n",
       " (5, 2),\n",
       " (7, 1),\n",
       " (3, 2),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (16, 2),\n",
       " (16, 2),\n",
       " (16, 2),\n",
       " (3, 1),\n",
       " (9, 3),\n",
       " (15, 2),\n",
       " (5, 1),\n",
       " (10, 6),\n",
       " (4, 1),\n",
       " (7, 1),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (2, 1),\n",
       " (31, 1),\n",
       " (17, 3),\n",
       " (3, 1),\n",
       " (7, 2),\n",
       " (17, 3),\n",
       " (12, 1),\n",
       " (5, 1),\n",
       " (4, 1),\n",
       " (9, 2),\n",
       " (7, 1),\n",
       " (7, 1),\n",
       " (7, 1),\n",
       " (7, 1),\n",
       " (8, 2),\n",
       " (9, 2),\n",
       " (9, 2),\n",
       " (17, 3),\n",
       " (5, 1),\n",
       " (2, 1),\n",
       " (5, 1),\n",
       " (2, 1),\n",
       " (5, 1),\n",
       " (2, 1),\n",
       " (1, 1),\n",
       " (7, 1),\n",
       " (7, 1),\n",
       " (15, 1),\n",
       " (18, 1),\n",
       " (8, 2),\n",
       " (2, 1),\n",
       " (4, 1),\n",
       " (4, 1),\n",
       " (7, 1),\n",
       " (7, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (7, 1),\n",
       " (7, 2),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (4, 1),\n",
       " (5, 1),\n",
       " (5, 1),\n",
       " (7, 1),\n",
       " (7, 1),\n",
       " (7, 1),\n",
       " (7, 1),\n",
       " (7, 1),\n",
       " ...]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "plot_data"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
